{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## OSU DS4S Data Science Club\n",
    "### RNASeq Workflow\n",
    "\n",
    "1. Setup Environment\n",
    "1. Download SRA Files\n",
    "1. Convert SRA to fastq\n",
    "1. Index Genome\n",
    "1. Run fastQC\n",
    "1. Run trim_galore\n",
    "1. Align Reads\n",
    "1. Run FeatureCounts on each file\n",
    "1. Combine counts\n",
    "1. Run multiqc\n",
    "1. Display html links for QC"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 37,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "import subprocess\n",
    "import os\n",
    "import glob\n",
    "from multiprocessing import Process, current_process, Pool\n",
    "\n",
    "\n",
    "# reusable function to run cmd, pipe stdout to console\n",
    "# and return std out back for error checking\n",
    "\n",
    "def run_cmd(cmd):\n",
    "    p = subprocess.Popen(cmd,\n",
    "                     stdout=subprocess.PIPE,\n",
    "                     stderr=subprocess.STDOUT,shell=True)\n",
    "    output = \"\"\n",
    "    while True:\n",
    "        line = p.stdout.readline().decode(encoding='UTF-8')\n",
    "        if not line:\n",
    "            break\n",
    "        print (line.strip())\n",
    "        output = output + line\n",
    "    return output"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 64,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "MAXCORES = '24'\n",
    "ENVCMD = \"source activate rnaseq\"\n",
    "FASTQS = 'fastqs'\n",
    "TFASTQS = 'trimmed_fastqs'\n",
    "PFASTQS = 'paired_fastqs'\n",
    "TPFASTQS = 'trimmed_paired_fastqs'\n",
    "FASTQCOUT = 'fastQC_output'\n",
    "TFASTQCOUT = 'trimmed_fastQC_output'\n",
    "STARDIR = './star_output'\n",
    "FCDIR = './featureCount_output'\n",
    "\n",
    "# MAX CPUS For Owens should be 24 (28 possible but reserve a few for interactivity)\n",
    "os.environ['MAXCORES'] = '24'\n",
    "\n",
    "# Set Working Directory\n",
    "os.environ['WORKDIR'] = os.getcwd()\n",
    "\n",
    "# Set sra_toolkit cfg location.  Needed to overide default download directory\n",
    "# Will default to data directory in the current working directory.\n",
    "os.environ['VDB_CONFIG'] = os.path.expandvars(\"$WORKDIR/sra_toolkit.kfg\")\n",
    "\n",
    "# The data download directories\n",
    "os.environ['DATA'] =  os.path.expandvars(\"$WORKDIR/data/sra\")\n",
    "\n",
    "# The results directories\n",
    "os.environ['FASTQS'] = 'fastqs'\n",
    "os.environ['TFASTQS'] = 'trimmed_fastqs'\n",
    "os.environ['PFASTQS'] = 'paired_fastqs'\n",
    "os.environ['TPFASTQS'] = 'trimmed_paired_fastqs'\n",
    "os.environ['FASTQCOUT'] = 'fastQC_output'\n",
    "os.environ['TFASTQCOUT'] = 'trimmed_fastQC_output'\n",
    "os.environ['STARDIR'] = './star_output'\n",
    "os.environ['FCDIR'] = './featureCount_output'\n",
    "\n",
    "# The binary directories\n",
    "os.environ['BASEDIR'] = '/fs/project/PAS1307'\n",
    "os.environ['BINDIR'] = os.path.expandvars(\"$BASEDIR/bin\")\n",
    "os.environ['SRABINDIR'] = os.path.expandvars(\"$BASEDIR/bin/sratoolkit.2.8.2-1-ubuntu64/bin\")\n",
    "os.environ['TGBINDIR'] = os.path.expandvars(\"$BASEDIR/bin/trim_galore\")\n",
    "\n",
    "# The Genome directories\n",
    "os.environ['GENOMEDIR'] = os.path.expandvars(\"$BASEDIR/genomes\")\n",
    "#os.environ['GENOME'] = 'Drosophila_melanogaster/Ensembl/BDGP5.25/'         \n",
    "os.environ['GENOME'] = 'Mus_musculus/Ensembl/GRCm38' \n",
    "os.environ['WHOLEGENOMEDIR'] = os.path.expandvars(\"$GENOMEDIR/$GENOME/Sequence/WholeGenomeFasta\")\n",
    "os.environ['GENEANNOTATIONDIR'] = os.path.expandvars(\"$GENOMEDIR/$GENOME/Annotation/Genes\")\n",
    "os.environ['STARGENOMEDIR'] = os.path.expandvars(\"$GENOMEDIR/$GENOME/Sequence/StarIndex\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "cmd = '''\n",
    "mkdir -p $FASTQS\n",
    "mkdir -p $PFASTQS\n",
    "mkdir -p $FASTQCOUT\n",
    "mkdir -p $STARDIR\n",
    "mkdir -p $FCDIR\n",
    "mkdir -p $TFASTQS\n",
    "mkdir -p $TPFASTQS\n",
    "mkdir -p $TFASTQCOUT\n",
    "'''\n",
    "\n",
    "out = run_cmd(cmd)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {
    "collapsed": false,
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "2018-01-26T13:32:32 prefetch.2.8.2: 1) 'ERR899429' is found locally\n",
      "\n",
      "2018-01-26T13:32:32 prefetch.2.8.2: 2) 'ERR899433' is found locally\n",
      "\n",
      "2018-01-26T13:32:33 prefetch.2.8.2: 3) 'ERR899427' is found locally\n",
      "\n",
      "2018-01-26T13:32:33 prefetch.2.8.2: 4) 'ERR899421' is found locally\n",
      "\n",
      "2018-01-26T13:32:33 prefetch.2.8.2: 5) 'ERR899425' is found locally\n",
      "\n",
      "2018-01-26T13:32:33 prefetch.2.8.2: 6) 'ERR899435' is found locally\n",
      "\n",
      "2018-01-26T13:32:33 prefetch.2.8.2: 7) 'ERR899430' is found locally\n",
      "\n",
      "2018-01-26T13:32:33 prefetch.2.8.2: 8) 'ERR899432' is found locally\n",
      "\n",
      "2018-01-26T13:32:34 prefetch.2.8.2: 9) 'ERR899431' is found locally\n",
      "\n",
      "2018-01-26T13:32:34 prefetch.2.8.2: 10) 'ERR899434' is found locally\n",
      "\n",
      "2018-01-26T13:32:34 prefetch.2.8.2: 11) 'ERR899438' is found locally\n",
      "\n",
      "2018-01-26T13:32:34 prefetch.2.8.2: 12) 'ERR899422' is found locally\n",
      "\n",
      "2018-01-26T13:32:34 prefetch.2.8.2: 13) 'ERR899423' is found locally\n",
      "\n",
      "2018-01-26T13:32:34 prefetch.2.8.2: 14) 'ERR899424' is found locally\n",
      "\n",
      "2018-01-26T13:32:35 prefetch.2.8.2: 15) 'ERR899428' is found locally\n",
      "\n",
      "2018-01-26T13:32:35 prefetch.2.8.2: 16) 'ERR899426' is found locally\n",
      "\n",
      "2018-01-26T13:32:35 prefetch.2.8.2: 17) 'ERR899437' is found locally\n",
      "\n",
      "2018-01-26T13:32:35 prefetch.2.8.2: 18) 'ERR899436' is found locally\n"
     ]
    }
   ],
   "source": [
    "cmd = '''\n",
    "$SRABINDIR/prefetch --option-file $WORKDIR/SraAccList.txt\n",
    "'''\n",
    "out = run_cmd(cmd)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "cmd = '''\n",
    "## Make STAR index if not exists\n",
    "if [ ! -d \"$STARGENOMEDIR/\" ]; then\n",
    "    echo \"STAR index does not exist, building STAR index\"\n",
    "\n",
    "    mkdir $STARGENOMEDIR\n",
    "\n",
    "    $BINDIR/STAR \\\n",
    "        --runThreadN $MAXCORES \\\n",
    "        --runMode genomeGenerate \\\n",
    "        --genomeDir $STARGENOMEDIR \\\n",
    "        --genomeFastaFiles $WHOLEGENOMEDIR/genome.fa \\\n",
    "        --sjdbGTFfile $GENEANNOTATIONDIR/genes.gtf \\\n",
    "        --sjdbOverhang 100\n",
    "fi\n",
    "'''\n",
    "out = run_cmd(cmd)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {
    "collapsed": false,
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Dumping .sra files to .fastq.gz files\n",
      "/users/PAS1307/osu8805/rnaseq_2/data/sra/ERR899421.sra\n",
      "/fs/project/PAS1307/bin/sratoolkit.2.8.2-1-ubuntu64/bin\n",
      "/users/PAS1307/osu8805/rnaseq_2/data/sra/ERR899422.sra\n",
      "/fs/project/PAS1307/bin/sratoolkit.2.8.2-1-ubuntu64/bin\n",
      "/users/PAS1307/osu8805/rnaseq_2/data/sra/ERR899423.sra\n",
      "/fs/project/PAS1307/bin/sratoolkit.2.8.2-1-ubuntu64/bin\n",
      "/users/PAS1307/osu8805/rnaseq_2/data/sra/ERR899424.sra\n",
      "/fs/project/PAS1307/bin/sratoolkit.2.8.2-1-ubuntu64/bin\n",
      "/users/PAS1307/osu8805/rnaseq_2/data/sra/ERR899425.sra\n",
      "/fs/project/PAS1307/bin/sratoolkit.2.8.2-1-ubuntu64/bin\n",
      "/users/PAS1307/osu8805/rnaseq_2/data/sra/ERR899426.sra\n",
      "/fs/project/PAS1307/bin/sratoolkit.2.8.2-1-ubuntu64/bin\n",
      "/users/PAS1307/osu8805/rnaseq_2/data/sra/ERR899427.sra\n",
      "/fs/project/PAS1307/bin/sratoolkit.2.8.2-1-ubuntu64/bin\n",
      "/users/PAS1307/osu8805/rnaseq_2/data/sra/ERR899428.sra\n",
      "/fs/project/PAS1307/bin/sratoolkit.2.8.2-1-ubuntu64/bin\n",
      "/users/PAS1307/osu8805/rnaseq_2/data/sra/ERR899429.sra\n",
      "/fs/project/PAS1307/bin/sratoolkit.2.8.2-1-ubuntu64/bin\n",
      "/users/PAS1307/osu8805/rnaseq_2/data/sra/ERR899430.sra\n",
      "/fs/project/PAS1307/bin/sratoolkit.2.8.2-1-ubuntu64/bin\n",
      "/users/PAS1307/osu8805/rnaseq_2/data/sra/ERR899431.sra\n",
      "/fs/project/PAS1307/bin/sratoolkit.2.8.2-1-ubuntu64/bin\n",
      "/users/PAS1307/osu8805/rnaseq_2/data/sra/ERR899432.sra\n",
      "/fs/project/PAS1307/bin/sratoolkit.2.8.2-1-ubuntu64/bin\n",
      "/users/PAS1307/osu8805/rnaseq_2/data/sra/ERR899433.sra\n",
      "/fs/project/PAS1307/bin/sratoolkit.2.8.2-1-ubuntu64/bin\n",
      "/users/PAS1307/osu8805/rnaseq_2/data/sra/ERR899434.sra\n",
      "/fs/project/PAS1307/bin/sratoolkit.2.8.2-1-ubuntu64/bin\n",
      "/users/PAS1307/osu8805/rnaseq_2/data/sra/ERR899435.sra\n",
      "/fs/project/PAS1307/bin/sratoolkit.2.8.2-1-ubuntu64/bin\n",
      "/users/PAS1307/osu8805/rnaseq_2/data/sra/ERR899436.sra\n",
      "/fs/project/PAS1307/bin/sratoolkit.2.8.2-1-ubuntu64/bin\n",
      "/users/PAS1307/osu8805/rnaseq_2/data/sra/ERR899437.sra\n",
      "/fs/project/PAS1307/bin/sratoolkit.2.8.2-1-ubuntu64/bin\n",
      "/users/PAS1307/osu8805/rnaseq_2/data/sra/ERR899438.sra\n",
      "/fs/project/PAS1307/bin/sratoolkit.2.8.2-1-ubuntu64/bin\n",
      "/users/PAS1307/osu8805/rnaseq_2/data/sra/ERR899426.sra is detected as single-end sequencing reads\n",
      "/users/PAS1307/osu8805/rnaseq_2/data/sra/ERR899427.sra is detected as single-end sequencing reads\n",
      "/users/PAS1307/osu8805/rnaseq_2/data/sra/ERR899438.sra is detected as single-end sequencing reads\n",
      "/users/PAS1307/osu8805/rnaseq_2/data/sra/ERR899437.sra is detected as single-end sequencing reads\n",
      "/users/PAS1307/osu8805/rnaseq_2/data/sra/ERR899421.sra is detected as single-end sequencing reads\n",
      "/users/PAS1307/osu8805/rnaseq_2/data/sra/ERR899428.sra is detected as single-end sequencing reads\n",
      "/users/PAS1307/osu8805/rnaseq_2/data/sra/ERR899434.sra is detected as single-end sequencing reads\n",
      "/users/PAS1307/osu8805/rnaseq_2/data/sra/ERR899430.sra is detected as single-end sequencing reads\n",
      "/users/PAS1307/osu8805/rnaseq_2/data/sra/ERR899433.sra is detected as single-end sequencing reads\n",
      "/users/PAS1307/osu8805/rnaseq_2/data/sra/ERR899425.sra is detected as single-end sequencing reads\n",
      "/users/PAS1307/osu8805/rnaseq_2/data/sra/ERR899422.sra is detected as single-end sequencing reads\n",
      "/users/PAS1307/osu8805/rnaseq_2/data/sra/ERR899435.sra is detected as single-end sequencing reads\n",
      "/users/PAS1307/osu8805/rnaseq_2/data/sra/ERR899436.sra is detected as single-end sequencing reads\n",
      "/users/PAS1307/osu8805/rnaseq_2/data/sra/ERR899432.sra is detected as single-end sequencing reads\n",
      "/users/PAS1307/osu8805/rnaseq_2/data/sra/ERR899431.sra is detected as single-end sequencing reads\n",
      "/users/PAS1307/osu8805/rnaseq_2/data/sra/ERR899423.sra is detected as single-end sequencing reads\n",
      "/users/PAS1307/osu8805/rnaseq_2/data/sra/ERR899424.sra is detected as single-end sequencing reads\n",
      "/users/PAS1307/osu8805/rnaseq_2/data/sra/ERR899429.sra is detected as single-end sequencing reads\n",
      "Read 17183069 spots for /users/PAS1307/osu8805/rnaseq_2/data/sra/ERR899427.sra\n",
      "Written 17183069 spots for /users/PAS1307/osu8805/rnaseq_2/data/sra/ERR899427.sra\n",
      "Read 17513754 spots for /users/PAS1307/osu8805/rnaseq_2/data/sra/ERR899428.sra\n",
      "Written 17513754 spots for /users/PAS1307/osu8805/rnaseq_2/data/sra/ERR899428.sra\n",
      "Read 17899859 spots for /users/PAS1307/osu8805/rnaseq_2/data/sra/ERR899435.sra\n",
      "Written 17899859 spots for /users/PAS1307/osu8805/rnaseq_2/data/sra/ERR899435.sra\n",
      "Read 18934244 spots for /users/PAS1307/osu8805/rnaseq_2/data/sra/ERR899430.sra\n",
      "Written 18934244 spots for /users/PAS1307/osu8805/rnaseq_2/data/sra/ERR899430.sra\n",
      "Read 17979693 spots for /users/PAS1307/osu8805/rnaseq_2/data/sra/ERR899421.sra\n",
      "Written 17979693 spots for /users/PAS1307/osu8805/rnaseq_2/data/sra/ERR899421.sra\n",
      "Read 18211995 spots for /users/PAS1307/osu8805/rnaseq_2/data/sra/ERR899432.sra\n",
      "Written 18211995 spots for /users/PAS1307/osu8805/rnaseq_2/data/sra/ERR899432.sra\n",
      "Read 18176427 spots for /users/PAS1307/osu8805/rnaseq_2/data/sra/ERR899436.sra\n",
      "Written 18176427 spots for /users/PAS1307/osu8805/rnaseq_2/data/sra/ERR899436.sra\n",
      "Read 18421390 spots for /users/PAS1307/osu8805/rnaseq_2/data/sra/ERR899434.sra\n",
      "Written 18421390 spots for /users/PAS1307/osu8805/rnaseq_2/data/sra/ERR899434.sra\n",
      "Read 18424646 spots for /users/PAS1307/osu8805/rnaseq_2/data/sra/ERR899423.sra\n",
      "Written 18424646 spots for /users/PAS1307/osu8805/rnaseq_2/data/sra/ERR899423.sra\n",
      "Read 18655823 spots for /users/PAS1307/osu8805/rnaseq_2/data/sra/ERR899437.sra\n",
      "Written 18655823 spots for /users/PAS1307/osu8805/rnaseq_2/data/sra/ERR899437.sra\n",
      "Read 18487849 spots for /users/PAS1307/osu8805/rnaseq_2/data/sra/ERR899424.sra\n",
      "Written 18487849 spots for /users/PAS1307/osu8805/rnaseq_2/data/sra/ERR899424.sra\n",
      "Read 19410496 spots for /users/PAS1307/osu8805/rnaseq_2/data/sra/ERR899426.sra\n",
      "Written 19410496 spots for /users/PAS1307/osu8805/rnaseq_2/data/sra/ERR899426.sra\n",
      "Read 18794828 spots for /users/PAS1307/osu8805/rnaseq_2/data/sra/ERR899422.sra\n",
      "Written 18794828 spots for /users/PAS1307/osu8805/rnaseq_2/data/sra/ERR899422.sra\n",
      "Read 19419534 spots for /users/PAS1307/osu8805/rnaseq_2/data/sra/ERR899431.sra\n",
      "Written 19419534 spots for /users/PAS1307/osu8805/rnaseq_2/data/sra/ERR899431.sra\n",
      "Read 19286135 spots for /users/PAS1307/osu8805/rnaseq_2/data/sra/ERR899429.sra\n",
      "Written 19286135 spots for /users/PAS1307/osu8805/rnaseq_2/data/sra/ERR899429.sra\n",
      "Read 19836280 spots for /users/PAS1307/osu8805/rnaseq_2/data/sra/ERR899433.sra\n",
      "Written 19836280 spots for /users/PAS1307/osu8805/rnaseq_2/data/sra/ERR899433.sra\n",
      "Read 20649441 spots for /users/PAS1307/osu8805/rnaseq_2/data/sra/ERR899425.sra\n",
      "Written 20649441 spots for /users/PAS1307/osu8805/rnaseq_2/data/sra/ERR899425.sra\n",
      "Read 21276620 spots for /users/PAS1307/osu8805/rnaseq_2/data/sra/ERR899438.sra\n",
      "Written 21276620 spots for /users/PAS1307/osu8805/rnaseq_2/data/sra/ERR899438.sra\n"
     ]
    }
   ],
   "source": [
    "cmd = '''\n",
    "is_paired() {\n",
    "    # function to examine whether a .sra file is paired-end sequencing\n",
    "    # ref: https://www.biostars.org/p/139422/\n",
    "    local SRA=\"$1\"\n",
    "    local x=$(\n",
    "        $SRABINDIR/fastq-dump -I -X 1 -Z --split-spot \"$SRA\" 2>/dev/null \\\n",
    "        | awk '{if(NR % 2 == 1) print substr($1,length($1),1)}' \\\n",
    "        | uniq \\\n",
    "        | wc -l\n",
    "    )\n",
    "    # echo $SRA $x\n",
    "    # $x should be 2 if paired-end, 1 if single-end\n",
    "    if [ $x == 2 ]; then\n",
    "        return 0 # true\n",
    "    else\n",
    "        return 1 # false\n",
    "    fi\n",
    "}\n",
    "\n",
    "dump_sra() {\n",
    "    # function to dump sra to fastq file with auto-detecting\n",
    "    # whether reads are from single-end or paired-end\n",
    "    local sra=\"$1\"\n",
    "    echo $sra\n",
    "    echo $SRABINDIR\n",
    "    if is_paired $sra; then\n",
    "        echo \"$sra is detected as paired-end sequencing reads\"\n",
    "        # Note that paired-end sequencing reads should be dumped into two fastq files\n",
    "        $SRABINDIR/fastq-dump --gzip -I --split-files -O paired_fastqs \"$sra\"\n",
    "    else\n",
    "        echo \"$sra is detected as single-end sequencing reads\"\n",
    "        $SRABINDIR/fastq-dump --gzip -O fastqs \"$sra\"\n",
    "    fi\n",
    "}\n",
    "\n",
    "\n",
    "## Dump .sra to .fastq\n",
    "echo \"Dumping .sra files to .fastq.gz files\"\n",
    "\n",
    "## Dump in parallel using xargs\n",
    "export -f is_paired\n",
    "export -f dump_sra\n",
    "\n",
    "find $DATA/*.sra | xargs --max-args=1 --max-procs=$MAXCORES -I {} bash -c 'dump_sra \"$@\"' _ {}\n",
    "'''\n",
    "out = run_cmd(cmd)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 50,
   "metadata": {
    "collapsed": false,
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "fastqs\n",
      "['source activate rnaseq && $TGBINDIR/trim_galore -q 20 -o trimmed_fastqs fastqs/ERR899426.fastq.gz', 'source activate rnaseq && $TGBINDIR/trim_galore -q 20 -o trimmed_fastqs fastqs/ERR899430.fastq.gz', 'source activate rnaseq && $TGBINDIR/trim_galore -q 20 -o trimmed_fastqs fastqs/ERR899435.fastq.gz', 'source activate rnaseq && $TGBINDIR/trim_galore -q 20 -o trimmed_fastqs fastqs/ERR899431.fastq.gz', 'source activate rnaseq && $TGBINDIR/trim_galore -q 20 -o trimmed_fastqs fastqs/ERR899428.fastq.gz', 'source activate rnaseq && $TGBINDIR/trim_galore -q 20 -o trimmed_fastqs fastqs/ERR899436.fastq.gz', 'source activate rnaseq && $TGBINDIR/trim_galore -q 20 -o trimmed_fastqs fastqs/ERR899425.fastq.gz', 'source activate rnaseq && $TGBINDIR/trim_galore -q 20 -o trimmed_fastqs fastqs/ERR899427.fastq.gz', 'source activate rnaseq && $TGBINDIR/trim_galore -q 20 -o trimmed_fastqs fastqs/ERR899433.fastq.gz', 'source activate rnaseq && $TGBINDIR/trim_galore -q 20 -o trimmed_fastqs fastqs/ERR899438.fastq.gz', 'source activate rnaseq && $TGBINDIR/trim_galore -q 20 -o trimmed_fastqs fastqs/ERR899437.fastq.gz', 'source activate rnaseq && $TGBINDIR/trim_galore -q 20 -o trimmed_fastqs fastqs/ERR899422.fastq.gz', 'source activate rnaseq && $TGBINDIR/trim_galore -q 20 -o trimmed_fastqs fastqs/ERR899432.fastq.gz', 'source activate rnaseq && $TGBINDIR/trim_galore -q 20 -o trimmed_fastqs fastqs/ERR899434.fastq.gz', 'source activate rnaseq && $TGBINDIR/trim_galore -q 20 -o trimmed_fastqs fastqs/ERR899421.fastq.gz', 'source activate rnaseq && $TGBINDIR/trim_galore -q 20 -o trimmed_fastqs fastqs/ERR899423.fastq.gz', 'source activate rnaseq && $TGBINDIR/trim_galore -q 20 -o trimmed_fastqs fastqs/ERR899424.fastq.gz', 'source activate rnaseq && $TGBINDIR/trim_galore -q 20 -o trimmed_fastqs fastqs/ERR899429.fastq.gz']\n",
      "No quality encoding type selected. Assuming that the data provided uses Sanger encoded Phred scores (default)\n",
      "\n",
      "No quality encoding type selected. Assuming that the data provided uses Sanger encoded Phred scores (default)\n",
      "\n",
      "No quality encoding type selected. Assuming that the data provided uses Sanger encoded Phred scores (default)\n",
      "\n",
      "No quality encoding type selected. Assuming that the data provided uses Sanger encoded Phred scores (default)\n",
      "\n",
      "No quality encoding type selected. Assuming that the data provided uses Sanger encoded Phred scores (default)\n",
      "\n",
      "No quality encoding type selected. Assuming that the data provided uses Sanger encoded Phred scores (default)\n",
      "\n",
      "No quality encoding type selected. Assuming that the data provided uses Sanger encoded Phred scores (default)\n",
      "\n",
      "No quality encoding type selected. Assuming that the data provided uses Sanger encoded Phred scores (default)\n",
      "No quality encoding type selected. Assuming that the data provided uses Sanger encoded Phred scores (default)\n",
      "\n",
      "\n",
      "No quality encoding type selected. Assuming that the data provided uses Sanger encoded Phred scores (default)\n",
      "\n",
      "No quality encoding type selected. Assuming that the data provided uses Sanger encoded Phred scores (default)\n",
      "\n",
      "No quality encoding type selected. Assuming that the data provided uses Sanger encoded Phred scores (default)\n",
      "\n",
      "No quality encoding type selected. Assuming that the data provided uses Sanger encoded Phred scores (default)\n",
      "\n",
      "No quality encoding type selected. Assuming that the data provided uses Sanger encoded Phred scores (default)\n",
      "\n",
      "No quality encoding type selected. Assuming that the data provided uses Sanger encoded Phred scores (default)\n",
      "\n",
      "No quality encoding type selected. Assuming that the data provided uses Sanger encoded Phred scores (default)\n",
      "\n",
      "No quality encoding type selected. Assuming that the data provided uses Sanger encoded Phred scores (default)\n",
      "\n",
      "No quality encoding type selected. Assuming that the data provided uses Sanger encoded Phred scores (default)\n",
      "\n",
      "Path to Cutadapt set as: 'cutadapt' (default)\n",
      "Path to Cutadapt set as: 'cutadapt' (default)\n",
      "Path to Cutadapt set as: 'cutadapt' (default)\n",
      "Path to Cutadapt set as: 'cutadapt' (default)\n",
      "Path to Cutadapt set as: 'cutadapt' (default)\n",
      "Path to Cutadapt set as: 'cutadapt' (default)\n",
      "Path to Cutadapt set as: 'cutadapt' (default)\n",
      "Path to Cutadapt set as: 'cutadapt' (default)\n",
      "Path to Cutadapt set as: 'cutadapt' (default)\n",
      "Path to Cutadapt set as: 'cutadapt' (default)\n",
      "Path to Cutadapt set as: 'cutadapt' (default)\n",
      "Path to Cutadapt set as: 'cutadapt' (default)\n",
      "Path to Cutadapt set as: 'cutadapt' (default)\n",
      "Path to Cutadapt set as: 'cutadapt' (default)\n",
      "Path to Cutadapt set as: 'cutadapt' (default)\n",
      "Path to Cutadapt set as: 'cutadapt' (default)\n",
      "Path to Cutadapt set as: 'cutadapt' (default)\n",
      "Path to Cutadapt set as: 'cutadapt' (default)\n",
      "1.15\n",
      "Cutadapt seems to be working fine (tested command 'cutadapt --version')\n",
      "1.15\n",
      "Cutadapt seems to be working fine (tested command 'cutadapt --version')\n",
      "1.15\n",
      "1.15\n",
      "Cutadapt seems to be working fine (tested command 'cutadapt --version')\n",
      "Cutadapt seems to be working fine (tested command 'cutadapt --version')\n",
      "1.15\n",
      "1.15\n",
      "1.15\n",
      "1.15\n",
      "1.15\n",
      "1.15\n",
      "Cutadapt seems to be working fine (tested command 'cutadapt --version')\n",
      "1.15\n",
      "1.15\n",
      "Cutadapt seems to be working fine (tested command 'cutadapt --version')\n",
      "1.15\n",
      "1.15\n",
      "Cutadapt seems to be working fine (tested command 'cutadapt --version')\n",
      "1.15\n",
      "Cutadapt seems to be working fine (tested command 'cutadapt --version')\n",
      "1.15\n",
      "Cutadapt seems to be working fine (tested command 'cutadapt --version')\n",
      "Cutadapt seems to be working fine (tested command 'cutadapt --version')\n",
      "1.15\n",
      "Cutadapt seems to be working fine (tested command 'cutadapt --version')\n",
      "Cutadapt seems to be working fine (tested command 'cutadapt --version')\n",
      "1.15\n",
      "Cutadapt seems to be working fine (tested command 'cutadapt --version')\n",
      "Cutadapt seems to be working fine (tested command 'cutadapt --version')\n",
      "Cutadapt seems to be working fine (tested command 'cutadapt --version')\n",
      "Cutadapt seems to be working fine (tested command 'cutadapt --version')\n",
      "Cutadapt seems to be working fine (tested command 'cutadapt --version')\n",
      "Cutadapt seems to be working fine (tested command 'cutadapt --version')\n",
      "\n",
      "\n",
      "AUTO-DETECTING ADAPTER TYPE\n",
      "===========================\n",
      "Attempting to auto-detect adapter type from the first 1 million sequences of the first file (>> fastqs/ERR899429.fastq.gz <<)\n",
      "\n",
      "\n",
      "\n",
      "AUTO-DETECTING ADAPTER TYPE\n",
      "===========================\n",
      "\n",
      "Attempting to auto-detect adapter type from the first 1 million sequences of the first file (>> fastqs/ERR899425.fastq.gz <<)\n",
      "\n",
      "\n",
      "AUTO-DETECTING ADAPTER TYPE\n",
      "===========================\n",
      "Attempting to auto-detect adapter type from the first 1 million sequences of the first file (>> fastqs/ERR899431.fastq.gz <<)\n",
      "\n",
      "\n",
      "\n",
      "AUTO-DETECTING ADAPTER TYPE\n",
      "===========================\n",
      "Attempting to auto-detect adapter type from the first 1 million sequences of the first file (>> fastqs/ERR899421.fastq.gz <<)\n",
      "\n",
      "\n",
      "\n",
      "AUTO-DETECTING ADAPTER TYPE\n",
      "===========================\n",
      "Attempting to auto-detect adapter type from the first 1 million sequences of the first file (>> fastqs/ERR899423.fastq.gz <<)\n",
      "\n",
      "\n",
      "\n",
      "AUTO-DETECTING ADAPTER TYPE\n",
      "===========================\n",
      "Attempting to auto-detect adapter type from the first 1 million sequences of the first file (>> fastqs/ERR899430.fastq.gz <<)\n",
      "\n",
      "\n",
      "\n",
      "AUTO-DETECTING ADAPTER TYPE\n",
      "===========================\n",
      "Attempting to auto-detect adapter type from the first 1 million sequences of the first file (>> fastqs/ERR899438.fastq.gz <<)\n",
      "\n",
      "\n",
      "\n",
      "AUTO-DETECTING ADAPTER TYPE\n",
      "===========================\n",
      "Attempting to auto-detect adapter type from the first 1 million sequences of the first file (>> fastqs/ERR899426.fastq.gz <<)\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "AUTO-DETECTING ADAPTER TYPE\n",
      "AUTO-DETECTING ADAPTER TYPE\n",
      "===========================\n",
      "Attempting to auto-detect adapter type from the first 1 million sequences of the first file (>> fastqs/ERR899428.fastq.gz <<)\n",
      "===========================\n",
      "\n",
      "Attempting to auto-detect adapter type from the first 1 million sequences of the first file (>> fastqs/ERR899427.fastq.gz <<)\n",
      "\n",
      "\n",
      "\n",
      "AUTO-DETECTING ADAPTER TYPE\n",
      "===========================\n",
      "Attempting to auto-detect adapter type from the first 1 million sequences of the first file (>> fastqs/ERR899433.fastq.gz <<)\n",
      "\n",
      "\n",
      "\n",
      "AUTO-DETECTING ADAPTER TYPE\n",
      "===========================\n",
      "Attempting to auto-detect adapter type from the first 1 million sequences of the first file (>> fastqs/ERR899437.fastq.gz <<)\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "AUTO-DETECTING ADAPTER TYPE\n",
      "\n",
      "\n",
      "===========================\n",
      "AUTO-DETECTING ADAPTER TYPE\n",
      "AUTO-DETECTING ADAPTER TYPE\n",
      "===========================\n",
      "===========================\n",
      "Attempting to auto-detect adapter type from the first 1 million sequences of the first file (>> fastqs/ERR899422.fastq.gz <<)\n",
      "Attempting to auto-detect adapter type from the first 1 million sequences of the first file (>> fastqs/ERR899435.fastq.gz <<)\n",
      "\n",
      "\n",
      "Attempting to auto-detect adapter type from the first 1 million sequences of the first file (>> fastqs/ERR899434.fastq.gz <<)\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "AUTO-DETECTING ADAPTER TYPE\n",
      "\n",
      "===========================\n",
      "AUTO-DETECTING ADAPTER TYPE\n",
      "Attempting to auto-detect adapter type from the first 1 million sequences of the first file (>> fastqs/ERR899436.fastq.gz <<)\n",
      "===========================\n",
      "\n",
      "\n",
      "AUTO-DETECTING ADAPTER TYPE\n",
      "\n",
      "===========================\n",
      "Attempting to auto-detect adapter type from the first 1 million sequences of the first file (>> fastqs/ERR899424.fastq.gz <<)\n",
      "Attempting to auto-detect adapter type from the first 1 million sequences of the first file (>> fastqs/ERR899432.fastq.gz <<)\n",
      "\n",
      "\n",
      "Found perfect matches for the following adapter sequences:\n",
      "Adapter type\tCount\tSequence\tSequences analysed\tPercentage\n",
      "Illumina\t11195\tAGATCGGAAGAGC\t1000000\t1.12\n",
      "Nextera\t36\tCTGTCTCTTATA\t1000000\t0.00\n",
      "smallRNA\t27\tTGGAATTCTCGG\t1000000\t0.00\n",
      "Using Illumina adapter for trimming (count: 11195). Second best hit was Nextera (count: 36)\n",
      "\n",
      "Found perfect matches for the following adapter sequences:\n",
      "Adapter type\tCount\tSequence\tSequences analysed\tPercentage\n",
      "Illumina\t11454\tAGATCGGAAGAGC\t1000000\t1.15\n",
      "Writing report to 'trimmed_fastqs/ERR899429.fastq.gz_trimming_report.txt'\n",
      "smallRNA\t16\tTGGAATTCTCGG\t1000000\t0.00\n",
      "\n",
      "Nextera\t14\tCTGTCTCTTATA\t1000000\t0.00\n",
      "SUMMARISING RUN PARAMETERS\n",
      "Using Illumina adapter for trimming (count: 11454). Second best hit was smallRNA (count: 16)\n",
      "==========================\n",
      "\n",
      "Input filename: fastqs/ERR899429.fastq.gz\n",
      "Writing report to 'trimmed_fastqs/ERR899426.fastq.gz_trimming_report.txt'\n",
      "\n",
      "SUMMARISING RUN PARAMETERS\n",
      "==========================\n",
      "Input filename: fastqs/ERR899426.fastq.gz\n",
      "Trimming mode: single-end\n",
      "Trim Galore version: 0.4.5_dev\n",
      "Found perfect matches for the following adapter sequences:\n",
      "Cutadapt version: 1.15\n",
      "Adapter type\tCount\tSequence\tSequences analysed\tPercentage\n",
      "Quality Phred score cutoff: 20\n",
      "Trimming mode: single-end\n",
      "Trim Galore version: 0.4.5_dev\n",
      "Quality encoding type selected: ASCII+33\n",
      "Found perfect matches for the following adapter sequences:\n",
      "Illumina\t5849\tAGATCGGAAGAGC\t1000000\t0.58\n",
      "Nextera\t32\tCTGTCTCTTATA\t1000000\t0.00\n",
      "Adapter type\tCount\tSequence\tSequences analysed\tPercentage\n",
      "smallRNA\t26\tTGGAATTCTCGG\t1000000\t0.00\n",
      "Adapter sequence: 'AGATCGGAAGAGC' (Illumina TruSeq, Sanger iPCR; auto-detected)\n",
      "Using Illumina adapter for trimming (count: 5849). Second best hit was Nextera (count: 32)\n",
      "\n",
      "Maximum trimming error rate: 0.1 (default)\n",
      "Minimum required adapter overlap (stringency): 1 bp\n",
      "Writing report to 'trimmed_fastqs/ERR899433.fastq.gz_trimming_report.txt'\n",
      "\n",
      "SUMMARISING RUN PARAMETERS\n",
      "==========================\n",
      "Input filename: fastqs/ERR899433.fastq.gz\n",
      "Minimum required sequence length before a sequence gets removed: 20 bp\n",
      "Trimming mode: single-end\n",
      "Output file(s) will be GZIP compressed\n",
      "Trim Galore version: 0.4.5_dev\n",
      "\n",
      "Cutadapt version: 1.15\n",
      "Quality Phred score cutoff: 20\n",
      "Quality encoding type selected: ASCII+33\n",
      "Cutadapt version: 1.15\n",
      "Adapter sequence: 'AGATCGGAAGAGC' (Illumina TruSeq, Sanger iPCR; auto-detected)\n",
      "Quality Phred score cutoff: 20\n",
      "Quality encoding type selected: ASCII+33\n",
      "Adapter sequence: 'AGATCGGAAGAGC' (Illumina TruSeq, Sanger iPCR; auto-detected)\n",
      "Maximum trimming error rate: 0.1 (default)\n",
      "Maximum trimming error rate: 0.1 (default)\n",
      "Minimum required adapter overlap (stringency): 1 bp\n",
      "Minimum required adapter overlap (stringency): 1 bp\n",
      "Minimum required sequence length before a sequence gets removed: 20 bp\n",
      "Minimum required sequence length before a sequence gets removed: 20 bp\n",
      "Output file(s) will be GZIP compressed\n",
      "\n",
      "Output file(s) will be GZIP compressed\n",
      "Illumina\t5460\tAGATCGGAAGAGC\t1000000\t0.55\n",
      "smallRNA\t19\tTGGAATTCTCGG\t1000000\t0.00\n",
      "Nextera\t12\tCTGTCTCTTATA\t1000000\t0.00\n",
      "Using Illumina adapter for trimming (count: 5460). Second best hit was smallRNA (count: 19)\n",
      "\n",
      "Found perfect matches for the following adapter sequences:\n",
      "\n",
      "Adapter type\tCount\tSequence\tSequences analysed\tPercentage\n",
      "Writing report to 'trimmed_fastqs/ERR899434.fastq.gz_trimming_report.txt'\n",
      "Illumina\t5633\tAGATCGGAAGAGC\t1000000\t0.56\n",
      "\n",
      "smallRNA\t9\tTGGAATTCTCGG\t1000000\t0.00\n",
      "SUMMARISING RUN PARAMETERS\n",
      "Nextera\t8\tCTGTCTCTTATA\t1000000\t0.00\n",
      "==========================\n",
      "Using Illumina adapter for trimming (count: 5633). Second best hit was smallRNA (count: 9)\n",
      "Input filename: fastqs/ERR899434.fastq.gz\n",
      "\n",
      "Trimming mode: single-end\n",
      "Writing report to 'trimmed_fastqs/ERR899430.fastq.gz_trimming_report.txt'\n",
      "Trim Galore version: 0.4.5_dev\n",
      "\n",
      "Cutadapt version: 1.15\n",
      "SUMMARISING RUN PARAMETERS\n",
      "Quality Phred score cutoff: 20\n",
      "Quality encoding type selected: ASCII+33\n",
      "Adapter sequence: 'AGATCGGAAGAGC' (Illumina TruSeq, Sanger iPCR; auto-detected)\n",
      "Maximum trimming error rate: 0.1 (default)\n",
      "Minimum required adapter overlap (stringency): 1 bp\n",
      "Minimum required sequence length before a sequence gets removed: 20 bp\n",
      "Output file(s) will be GZIP compressed\n",
      "Found perfect matches for the following adapter sequences:\n",
      "\n",
      "Adapter type\tCount\tSequence\tSequences analysed\tPercentage\n",
      "==========================\n",
      "Illumina\t5684\tAGATCGGAAGAGC\t1000000\t0.57\n",
      "Input filename: fastqs/ERR899430.fastq.gz\n",
      "Trimming mode: single-end\n",
      "Nextera\t16\tCTGTCTCTTATA\t1000000\t0.00\n",
      "Trim Galore version: 0.4.5_dev\n",
      "Cutadapt version: 1.15\n",
      "Quality Phred score cutoff: 20\n",
      "Quality encoding type selected: ASCII+33\n",
      "Adapter sequence: 'AGATCGGAAGAGC' (Illumina TruSeq, Sanger iPCR; auto-detected)\n",
      "smallRNA\t12\tTGGAATTCTCGG\t1000000\t0.00\n",
      "Maximum trimming error rate: 0.1 (default)\n",
      "Using Illumina adapter for trimming (count: 5684). Second best hit was Nextera (count: 16)\n",
      "Minimum required adapter overlap (stringency): 1 bp\n",
      "\n",
      "Minimum required sequence length before a sequence gets removed: 20 bp\n",
      "Writing report to 'trimmed_fastqs/ERR899428.fastq.gz_trimming_report.txt'\n",
      "Output file(s) will be GZIP compressed\n",
      "\n",
      "\n",
      "SUMMARISING RUN PARAMETERS\n",
      "==========================\n",
      "Input filename: fastqs/ERR899428.fastq.gz\n",
      "Trimming mode: single-end\n",
      "Trim Galore version: 0.4.5_dev\n",
      "Cutadapt version: 1.15\n",
      "Quality Phred score cutoff: 20\n",
      "Quality encoding type selected: ASCII+33\n",
      "Adapter sequence: 'AGATCGGAAGAGC' (Illumina TruSeq, Sanger iPCR; auto-detected)\n",
      "Maximum trimming error rate: 0.1 (default)\n",
      "Minimum required adapter overlap (stringency): 1 bp\n",
      "Minimum required sequence length before a sequence gets removed: 20 bp\n",
      "Output file(s) will be GZIP compressed\n",
      "Found perfect matches for the following adapter sequences:\n",
      "Adapter type\tCount\tSequence\tSequences analysed\tPercentage\n",
      "\n",
      "Illumina\t7172\tAGATCGGAAGAGC\t1000000\t0.72\n",
      "Nextera\t35\tCTGTCTCTTATA\t1000000\t0.00\n",
      "smallRNA\t21\tTGGAATTCTCGG\t1000000\t0.00\n",
      "Using Illumina adapter for trimming (count: 7172). Second best hit was Nextera (count: 35)\n",
      "\n",
      "Writing report to 'trimmed_fastqs/ERR899431.fastq.gz_trimming_report.txt'\n",
      "\n",
      "SUMMARISING RUN PARAMETERS\n",
      "Found perfect matches for the following adapter sequences:\n",
      "==========================\n",
      "Adapter type\tCount\tSequence\tSequences analysed\tPercentage\n",
      "Input filename: fastqs/ERR899431.fastq.gz\n",
      "Trimming mode: single-end\n",
      "Trim Galore version: 0.4.5_dev\n",
      "Cutadapt version: 1.15\n",
      "Quality Phred score cutoff: 20\n",
      "Illumina\t6043\tAGATCGGAAGAGC\t1000000\t0.60\n",
      "Nextera\t20\tCTGTCTCTTATA\t1000000\t0.00\n",
      "Quality encoding type selected: ASCII+33\n",
      "smallRNA\t8\tTGGAATTCTCGG\t1000000\t0.00\n",
      "Adapter sequence: 'AGATCGGAAGAGC' (Illumina TruSeq, Sanger iPCR; auto-detected)\n",
      "Using Illumina adapter for trimming (count: 6043). Second best hit was Nextera (count: 20)\n",
      "Maximum trimming error rate: 0.1 (default)\n",
      "\n",
      "Minimum required adapter overlap (stringency): 1 bp\n",
      "Writing report to 'trimmed_fastqs/ERR899435.fastq.gz_trimming_report.txt'\n",
      "Minimum required sequence length before a sequence gets removed: 20 bp\n",
      "\n",
      "Output file(s) will be GZIP compressed\n",
      "SUMMARISING RUN PARAMETERS\n",
      "\n",
      "==========================\n",
      "Input filename: fastqs/ERR899435.fastq.gz\n",
      "Trimming mode: single-end\n",
      "Trim Galore version: 0.4.5_dev\n",
      "Cutadapt version: 1.15\n",
      "Quality Phred score cutoff: 20\n",
      "Quality encoding type selected: ASCII+33\n",
      "Adapter sequence: 'AGATCGGAAGAGC' (Illumina TruSeq, Sanger iPCR; auto-detected)\n",
      "Maximum trimming error rate: 0.1 (default)\n",
      "Minimum required adapter overlap (stringency): 1 bp\n",
      "Found perfect matches for the following adapter sequences:\n",
      "Minimum required sequence length before a sequence gets removed: 20 bp\n",
      "Adapter type\tCount\tSequence\tSequences analysed\tPercentage\n",
      "Output file(s) will be GZIP compressed\n",
      "Illumina\t7575\tAGATCGGAAGAGC\t1000000\t0.76\n",
      "\n",
      "smallRNA\t14\tTGGAATTCTCGG\t1000000\t0.00\n",
      "Nextera\t9\tCTGTCTCTTATA\t1000000\t0.00\n",
      "Using Illumina adapter for trimming (count: 7575). Second best hit was smallRNA (count: 14)\n",
      "\n",
      "Writing report to 'trimmed_fastqs/ERR899422.fastq.gz_trimming_report.txt'\n",
      "\n",
      "SUMMARISING RUN PARAMETERS\n",
      "==========================\n",
      "Input filename: fastqs/ERR899422.fastq.gz\n",
      "Trimming mode: single-end\n",
      "Trim Galore version: 0.4.5_dev\n",
      "Cutadapt version: 1.15\n",
      "Quality Phred score cutoff: 20\n",
      "Quality encoding type selected: ASCII+33\n",
      "Adapter sequence: 'AGATCGGAAGAGC' (Illumina TruSeq, Sanger iPCR; auto-detected)\n",
      "Maximum trimming error rate: 0.1 (default)\n",
      "Minimum required adapter overlap (stringency): 1 bp\n",
      "Minimum required sequence length before a sequence gets removed: 20 bp\n",
      "Output file(s) will be GZIP compressed\n",
      "\n",
      "Found perfect matches for the following adapter sequences:\n",
      "Adapter type\tCount\tSequence\tSequences analysed\tPercentage\n",
      "Illumina\t7784\tAGATCGGAAGAGC\t1000000\t0.78\n",
      "smallRNA\t9\tTGGAATTCTCGG\t1000000\t0.00\n",
      "Nextera\t6\tCTGTCTCTTATA\t1000000\t0.00\n",
      "Using Illumina adapter for trimming (count: 7784). Second best hit was smallRNA (count: 9)\n",
      "\n",
      "Writing report to 'trimmed_fastqs/ERR899425.fastq.gz_trimming_report.txt'\n",
      "\n",
      "SUMMARISING RUN PARAMETERS\n",
      "==========================\n",
      "Input filename: fastqs/ERR899425.fastq.gz\n",
      "Trimming mode: single-end\n",
      "Trim Galore version: 0.4.5_dev\n",
      "Cutadapt version: 1.15\n",
      "Quality Phred score cutoff: 20\n",
      "Quality encoding type selected: ASCII+33\n",
      "Adapter sequence: 'AGATCGGAAGAGC' (Illumina TruSeq, Sanger iPCR; auto-detected)\n",
      "Maximum trimming error rate: 0.1 (default)\n",
      "Minimum required adapter overlap (stringency): 1 bp\n",
      "Minimum required sequence length before a sequence gets removed: 20 bp\n",
      "Output file(s) will be GZIP compressed\n",
      "\n",
      "Found perfect matches for the following adapter sequences:\n",
      "Adapter type\tCount\tSequence\tSequences analysed\tPercentage\n",
      "Illumina\t8060\tAGATCGGAAGAGC\t1000000\t0.81\n",
      "Nextera\t19\tCTGTCTCTTATA\t1000000\t0.00\n",
      "smallRNA\t16\tTGGAATTCTCGG\t1000000\t0.00\n",
      "Using Illumina adapter for trimming (count: 8060). Second best hit was Nextera (count: 19)\n",
      "\n",
      "Writing report to 'trimmed_fastqs/ERR899421.fastq.gz_trimming_report.txt'\n",
      "\n",
      "SUMMARISING RUN PARAMETERS\n",
      "==========================\n",
      "Input filename: fastqs/ERR899421.fastq.gz\n",
      "Trimming mode: single-end\n",
      "Trim Galore version: 0.4.5_dev\n",
      "Cutadapt version: 1.15\n",
      "Quality Phred score cutoff: 20\n",
      "Maximum trimming error rate: 0.1 (default)\n",
      "Quality encoding type selected: ASCII+33\n",
      "Adapter sequence: 'AGATCGGAAGAGC' (Illumina TruSeq, Sanger iPCR; auto-detected)\n",
      "Minimum required adapter overlap (stringency): 1 bp\n",
      "Minimum required sequence length before a sequence gets removed: 20 bp\n",
      "Output file(s) will be GZIP compressed\n",
      "\n",
      "Found perfect matches for the following adapter sequences:\n",
      "Adapter type\tCount\tSequence\tSequences analysed\tPercentage\n",
      "Illumina\t5567\tAGATCGGAAGAGC\t1000000\t0.56\n",
      "Nextera\t39\tCTGTCTCTTATA\t1000000\t0.00\n",
      "smallRNA\t22\tTGGAATTCTCGG\t1000000\t0.00\n",
      "Using Illumina adapter for trimming (count: 5567). Second best hit was Nextera (count: 39)\n",
      "\n",
      "Writing report to 'trimmed_fastqs/ERR899432.fastq.gz_trimming_report.txt'\n",
      "\n",
      "SUMMARISING RUN PARAMETERS\n",
      "==========================\n",
      "Input filename: fastqs/ERR899432.fastq.gz\n",
      "Trimming mode: single-end\n",
      "Trim Galore version: 0.4.5_dev\n",
      "Cutadapt version: 1.15\n",
      "Quality Phred score cutoff: 20\n",
      "Quality encoding type selected: ASCII+33\n",
      "Adapter sequence: 'AGATCGGAAGAGC' (Illumina TruSeq, Sanger iPCR; auto-detected)\n",
      "Maximum trimming error rate: 0.1 (default)\n",
      "Minimum required adapter overlap (stringency): 1 bp\n",
      "Minimum required sequence length before a sequence gets removed: 20 bp\n",
      "Output file(s) will be GZIP compressed\n",
      "\n",
      "Found perfect matches for the following adapter sequences:\n",
      "Adapter type\tCount\tSequence\tSequences analysed\tPercentage\n",
      "Illumina\t7627\tAGATCGGAAGAGC\t1000000\t0.76\n",
      "smallRNA\t17\tTGGAATTCTCGG\t1000000\t0.00\n",
      "Nextera\t13\tCTGTCTCTTATA\t1000000\t0.00\n",
      "Using Illumina adapter for trimming (count: 7627). Second best hit was smallRNA (count: 17)\n",
      "\n",
      "Writing report to 'trimmed_fastqs/ERR899423.fastq.gz_trimming_report.txt'\n",
      "Found perfect matches for the following adapter sequences:\n",
      "\n",
      "Adapter type\tCount\tSequence\tSequences analysed\tPercentage\n",
      "SUMMARISING RUN PARAMETERS\n",
      "Illumina\t5778\tAGATCGGAAGAGC\t1000000\t0.58\n",
      "==========================\n",
      "Nextera\t31\tCTGTCTCTTATA\t1000000\t0.00\n",
      "Input filename: fastqs/ERR899423.fastq.gz\n",
      "smallRNA\t29\tTGGAATTCTCGG\t1000000\t0.00\n",
      "Trimming mode: single-end\n",
      "Using Illumina adapter for trimming (count: 5778). Second best hit was Nextera (count: 31)\n",
      "Trim Galore version: 0.4.5_dev\n",
      "\n",
      "Cutadapt version: 1.15\n",
      "Writing report to 'trimmed_fastqs/ERR899427.fastq.gz_trimming_report.txt'\n",
      "\n",
      "Quality Phred score cutoff: 20\n",
      "SUMMARISING RUN PARAMETERS\n",
      "Quality encoding type selected: ASCII+33\n",
      "==========================\n",
      "Adapter sequence: 'AGATCGGAAGAGC' (Illumina TruSeq, Sanger iPCR; auto-detected)\n",
      "Input filename: fastqs/ERR899427.fastq.gz\n",
      "Maximum trimming error rate: 0.1 (default)\n",
      "Trimming mode: single-end\n",
      "Minimum required adapter overlap (stringency): 1 bp\n",
      "Trim Galore version: 0.4.5_dev\n",
      "Minimum required sequence length before a sequence gets removed: 20 bp\n",
      "Cutadapt version: 1.15\n",
      "Output file(s) will be GZIP compressed\n",
      "Quality Phred score cutoff: 20\n",
      "\n",
      "Quality encoding type selected: ASCII+33\n",
      "Adapter sequence: 'AGATCGGAAGAGC' (Illumina TruSeq, Sanger iPCR; auto-detected)\n",
      "Maximum trimming error rate: 0.1 (default)\n",
      "Minimum required adapter overlap (stringency): 1 bp\n",
      "Minimum required sequence length before a sequence gets removed: 20 bp\n",
      "Output file(s) will be GZIP compressed\n",
      "\n",
      "Found perfect matches for the following adapter sequences:\n",
      "Adapter type\tCount\tSequence\tSequences analysed\tPercentage\n",
      "Illumina\t6028\tAGATCGGAAGAGC\t1000000\t0.60\n",
      "Nextera\t13\tCTGTCTCTTATA\t1000000\t0.00\n",
      "smallRNA\t9\tTGGAATTCTCGG\t1000000\t0.00\n",
      "Using Illumina adapter for trimming (count: 6028). Second best hit was Nextera (count: 13)\n",
      "\n",
      "Writing report to 'trimmed_fastqs/ERR899424.fastq.gz_trimming_report.txt'\n",
      "\n",
      "SUMMARISING RUN PARAMETERS\n",
      "==========================\n",
      "Input filename: fastqs/ERR899424.fastq.gz\n",
      "Trimming mode: single-end\n",
      "Trim Galore version: 0.4.5_dev\n",
      "Cutadapt version: 1.15\n",
      "Quality Phred score cutoff: 20\n",
      "Found perfect matches for the following adapter sequences:\n",
      "Quality encoding type selected: ASCII+33\n",
      "Adapter type\tCount\tSequence\tSequences analysed\tPercentage\n",
      "Adapter sequence: 'AGATCGGAAGAGC' (Illumina TruSeq, Sanger iPCR; auto-detected)\n",
      "Maximum trimming error rate: 0.1 (default)\n",
      "Illumina\t6287\tAGATCGGAAGAGC\t1000000\t0.63\n",
      "Minimum required adapter overlap (stringency): 1 bp\n",
      "smallRNA\t16\tTGGAATTCTCGG\t1000000\t0.00\n",
      "Found perfect matches for the following adapter sequences:\n",
      "Minimum required sequence length before a sequence gets removed: 20 bp\n",
      "Nextera\t15\tCTGTCTCTTATA\t1000000\t0.00\n",
      "Adapter type\tCount\tSequence\tSequences analysed\tPercentage\n",
      "Output file(s) will be GZIP compressed\n",
      "Using Illumina adapter for trimming (count: 6287). Second best hit was smallRNA (count: 16)\n",
      "Illumina\t7006\tAGATCGGAAGAGC\t1000000\t0.70\n",
      "\n",
      "\n",
      "Nextera\t27\tCTGTCTCTTATA\t1000000\t0.00\n",
      "Writing report to 'trimmed_fastqs/ERR899436.fastq.gz_trimming_report.txt'\n",
      "smallRNA\t20\tTGGAATTCTCGG\t1000000\t0.00\n",
      "\n",
      "Using Illumina adapter for trimming (count: 7006). Second best hit was Nextera (count: 27)\n",
      "SUMMARISING RUN PARAMETERS\n",
      "\n",
      "==========================\n",
      "Writing report to 'trimmed_fastqs/ERR899438.fastq.gz_trimming_report.txt'\n",
      "Input filename: fastqs/ERR899436.fastq.gz\n",
      "\n",
      "Trimming mode: single-end\n",
      "SUMMARISING RUN PARAMETERS\n",
      "Trim Galore version: 0.4.5_dev\n",
      "==========================\n",
      "Cutadapt version: 1.15\n",
      "Input filename: fastqs/ERR899438.fastq.gz\n",
      "Quality Phred score cutoff: 20\n",
      "Trimming mode: single-end\n",
      "Quality encoding type selected: ASCII+33\n",
      "Trim Galore version: 0.4.5_dev\n",
      "Adapter sequence: 'AGATCGGAAGAGC' (Illumina TruSeq, Sanger iPCR; auto-detected)\n",
      "Cutadapt version: 1.15\n",
      "Quality Phred score cutoff: 20\n",
      "Maximum trimming error rate: 0.1 (default)\n",
      "Quality encoding type selected: ASCII+33\n",
      "Adapter sequence: 'AGATCGGAAGAGC' (Illumina TruSeq, Sanger iPCR; auto-detected)\n",
      "Minimum required adapter overlap (stringency): 1 bp\n",
      "Maximum trimming error rate: 0.1 (default)\n",
      "Minimum required adapter overlap (stringency): 1 bp\n",
      "Minimum required sequence length before a sequence gets removed: 20 bp\n",
      "Minimum required sequence length before a sequence gets removed: 20 bp\n",
      "Output file(s) will be GZIP compressed\n",
      "Output file(s) will be GZIP compressed\n",
      "\n",
      "\n",
      "Found perfect matches for the following adapter sequences:\n",
      "Adapter type\tCount\tSequence\tSequences analysed\tPercentage\n",
      "Illumina\t9320\tAGATCGGAAGAGC\t1000000\t0.93\n",
      "smallRNA\t18\tTGGAATTCTCGG\t1000000\t0.00\n",
      "Nextera\t17\tCTGTCTCTTATA\t1000000\t0.00\n",
      "Using Illumina adapter for trimming (count: 9320). Second best hit was smallRNA (count: 18)\n",
      "\n",
      "Writing report to 'trimmed_fastqs/ERR899437.fastq.gz_trimming_report.txt'\n",
      "\n",
      "SUMMARISING RUN PARAMETERS\n",
      "==========================\n",
      "Input filename: fastqs/ERR899437.fastq.gz\n",
      "Trimming mode: single-end\n",
      "Trim Galore version: 0.4.5_dev\n",
      "Cutadapt version: 1.15\n",
      "Quality Phred score cutoff: 20\n",
      "Quality encoding type selected: ASCII+33\n",
      "Adapter sequence: 'AGATCGGAAGAGC' (Illumina TruSeq, Sanger iPCR; auto-detected)\n",
      "Maximum trimming error rate: 0.1 (default)\n",
      "Minimum required adapter overlap (stringency): 1 bp\n",
      "Minimum required sequence length before a sequence gets removed: 20 bp\n",
      "Output file(s) will be GZIP compressed\n",
      "\n",
      "Writing final adapter and quality trimmed output to ERR899429_trimmed.fq.gz\n",
      "\n",
      "\n",
      ">>> Now performing quality (cutoff 20) and adapter trimming in a single pass for the adapter sequence: 'AGATCGGAAGAGC' from file fastqs/ERR899429.fastq.gz <<<\n",
      "Writing final adapter and quality trimmed output to ERR899426_trimmed.fq.gz\n",
      "\n",
      "\n",
      ">>> Now performing quality (cutoff 20) and adapter trimming in a single pass for the adapter sequence: 'AGATCGGAAGAGC' from file fastqs/ERR899426.fastq.gz <<<\n",
      "Writing final adapter and quality trimmed output to ERR899433_trimmed.fq.gz\n",
      "\n",
      "\n",
      ">>> Now performing quality (cutoff 20) and adapter trimming in a single pass for the adapter sequence: 'AGATCGGAAGAGC' from file fastqs/ERR899433.fastq.gz <<<\n",
      "Writing final adapter and quality trimmed output to ERR899434_trimmed.fq.gz\n",
      "\n",
      "\n",
      ">>> Now performing quality (cutoff 20) and adapter trimming in a single pass for the adapter sequence: 'AGATCGGAAGAGC' from file fastqs/ERR899434.fastq.gz <<<\n",
      "Writing final adapter and quality trimmed output to ERR899430_trimmed.fq.gz\n",
      "\n",
      "\n",
      ">>> Now performing quality (cutoff 20) and adapter trimming in a single pass for the adapter sequence: 'AGATCGGAAGAGC' from file fastqs/ERR899430.fastq.gz <<<\n",
      "Writing final adapter and quality trimmed output to ERR899428_trimmed.fq.gz\n",
      "\n",
      "\n",
      ">>> Now performing quality (cutoff 20) and adapter trimming in a single pass for the adapter sequence: 'AGATCGGAAGAGC' from file fastqs/ERR899428.fastq.gz <<<\n",
      "Writing final adapter and quality trimmed output to ERR899431_trimmed.fq.gz\n",
      "\n",
      "\n",
      ">>> Now performing quality (cutoff 20) and adapter trimming in a single pass for the adapter sequence: 'AGATCGGAAGAGC' from file fastqs/ERR899431.fastq.gz <<<\n",
      "Writing final adapter and quality trimmed output to ERR899435_trimmed.fq.gz\n",
      "\n",
      "\n",
      ">>> Now performing quality (cutoff 20) and adapter trimming in a single pass for the adapter sequence: 'AGATCGGAAGAGC' from file fastqs/ERR899435.fastq.gz <<<\n",
      "Writing final adapter and quality trimmed output to ERR899422_trimmed.fq.gz\n",
      "\n",
      "\n",
      ">>> Now performing quality (cutoff 20) and adapter trimming in a single pass for the adapter sequence: 'AGATCGGAAGAGC' from file fastqs/ERR899422.fastq.gz <<<\n",
      "Writing final adapter and quality trimmed output to ERR899425_trimmed.fq.gz\n",
      "\n",
      "\n",
      ">>> Now performing quality (cutoff 20) and adapter trimming in a single pass for the adapter sequence: 'AGATCGGAAGAGC' from file fastqs/ERR899425.fastq.gz <<<\n",
      "Writing final adapter and quality trimmed output to ERR899421_trimmed.fq.gz\n",
      "\n",
      "\n",
      ">>> Now performing quality (cutoff 20) and adapter trimming in a single pass for the adapter sequence: 'AGATCGGAAGAGC' from file fastqs/ERR899421.fastq.gz <<<\n",
      "Writing final adapter and quality trimmed output to ERR899432_trimmed.fq.gz\n",
      "\n",
      "\n",
      ">>> Now performing quality (cutoff 20) and adapter trimming in a single pass for the adapter sequence: 'AGATCGGAAGAGC' from file fastqs/ERR899432.fastq.gz <<<\n",
      "Writing final adapter and quality trimmed output to ERR899423_trimmed.fq.gz\n",
      "\n",
      "\n",
      ">>> Now performing quality (cutoff 20) and adapter trimming in a single pass for the adapter sequence: 'AGATCGGAAGAGC' from file fastqs/ERR899423.fastq.gz <<<\n",
      "Writing final adapter and quality trimmed output to ERR899427_trimmed.fq.gz\n",
      "\n",
      "\n",
      ">>> Now performing quality (cutoff 20) and adapter trimming in a single pass for the adapter sequence: 'AGATCGGAAGAGC' from file fastqs/ERR899427.fastq.gz <<<\n",
      "Writing final adapter and quality trimmed output to ERR899424_trimmed.fq.gz\n",
      "\n",
      "\n",
      ">>> Now performing quality (cutoff 20) and adapter trimming in a single pass for the adapter sequence: 'AGATCGGAAGAGC' from file fastqs/ERR899424.fastq.gz <<<\n",
      "Writing final adapter and quality trimmed output to ERR899436_trimmed.fq.gz\n",
      "\n",
      "\n",
      ">>> Now performing quality (cutoff 20) and adapter trimming in a single pass for the adapter sequence: 'AGATCGGAAGAGC' from file fastqs/ERR899436.fastq.gz <<<\n",
      "Writing final adapter and quality trimmed output to ERR899437_trimmed.fq.gz\n",
      "Writing final adapter and quality trimmed output to ERR899438_trimmed.fq.gz\n",
      "\n",
      "\n",
      "\n",
      "\n",
      ">>> Now performing quality (cutoff 20) and adapter trimming in a single pass for the adapter sequence: 'AGATCGGAAGAGC' from file fastqs/ERR899437.fastq.gz <<<\n",
      ">>> Now performing quality (cutoff 20) and adapter trimming in a single pass for the adapter sequence: 'AGATCGGAAGAGC' from file fastqs/ERR899438.fastq.gz <<<\n",
      "10000000 sequences processed\n",
      "10000000 sequences processed\n",
      "10000000 sequences processed\n",
      "10000000 sequences processed\n",
      "10000000 sequences processed\n",
      "10000000 sequences processed\n",
      "10000000 sequences processed\n",
      "10000000 sequences processed\n",
      "10000000 sequences processed\n",
      "10000000 sequences processed\n",
      "10000000 sequences processed\n",
      "10000000 sequences processed\n",
      "10000000 sequences processed\n",
      "10000000 sequences processed\n",
      "10000000 sequences processed\n",
      "10000000 sequences processed\n",
      "10000000 sequences processed\n",
      "10000000 sequences processed\n",
      "This is cutadapt 1.15 with Python 3.6.4\n",
      "Command line parameters: -f fastq -e 0.1 -q 20 -O 1 -a AGATCGGAAGAGC fastqs/ERR899427.fastq.gz\n",
      "Running on 1 core\n",
      "Trimming 1 adapter with at most 10.0% errors in single-end mode ...\n",
      "Finished in 606.45 s (35 us/read; 1.70 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:              17,183,069\n",
      "Reads with adapters:                 5,715,645 (33.3%)\n",
      "Reads written (passing filters):    17,183,069 (100.0%)\n",
      "\n",
      "Total basepairs processed: 1,718,306,900 bp\n",
      "Quality-trimmed:              18,179,078 bp (1.1%)\n",
      "Total written (filtered):  1,689,055,644 bp (98.3%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: AGATCGGAAGAGC; Type: regular 3'; Length: 13; Trimmed: 5715645 times.\n",
      "\n",
      "No. of allowed errors:\n",
      "0-9 bp: 0; 10-13 bp: 1\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "A: 28.5%\n",
      "C: 33.9%\n",
      "G: 19.3%\n",
      "T: 18.2%\n",
      "none/other: 0.0%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "1\t3833726\t4295767.2\t0\t3833726\n",
      "2\t1214028\t1073941.8\t0\t1214028\n",
      "3\t318026\t268485.5\t0\t318026\n",
      "4\t89957\t67121.4\t0\t89957\n",
      "5\t39843\t16780.3\t0\t39843\n",
      "6\t22968\t4195.1\t0\t22968\n",
      "7\t19607\t1048.8\t0\t19607\n",
      "8\t17274\t262.2\t0\t17274\n",
      "9\t15642\t65.5\t0\t15387 255\n",
      "10\t14580\t16.4\t1\t13681 899\n",
      "11\t12800\t4.1\t1\t12166 634\n",
      "12\t11858\t1.0\t1\t11372 486\n",
      "13\t10587\t0.3\t1\t10126 461\n",
      "14\t9903\t0.3\t1\t9481 422\n",
      "15\t9084\t0.3\t1\t8685 399\n",
      "16\t8249\t0.3\t1\t7884 365\n",
      "17\t7503\t0.3\t1\t7167 336\n",
      "18\t6685\t0.3\t1\t6423 262\n",
      "19\t5491\t0.3\t1\t5258 233\n",
      "20\t4942\t0.3\t1\t4710 232\n",
      "21\t4365\t0.3\t1\t4185 180\n",
      "22\t3934\t0.3\t1\t3739 195\n",
      "23\t3530\t0.3\t1\t3374 156\n",
      "24\t3289\t0.3\t1\t3128 161\n",
      "25\t2962\t0.3\t1\t2833 129\n",
      "26\t2685\t0.3\t1\t2579 106\n",
      "27\t2445\t0.3\t1\t2328 117\n",
      "28\t2111\t0.3\t1\t2004 107\n",
      "29\t1813\t0.3\t1\t1722 91\n",
      "30\t1667\t0.3\t1\t1573 94\n",
      "31\t1315\t0.3\t1\t1261 54\n",
      "32\t1202\t0.3\t1\t1148 54\n",
      "33\t1062\t0.3\t1\t1000 62\n",
      "34\t961\t0.3\t1\t905 56\n",
      "35\t967\t0.3\t1\t914 53\n",
      "36\t764\t0.3\t1\t718 46\n",
      "37\t727\t0.3\t1\t682 45\n",
      "38\t628\t0.3\t1\t596 32\n",
      "39\t533\t0.3\t1\t505 28\n",
      "40\t459\t0.3\t1\t417 42\n",
      "41\t413\t0.3\t1\t384 29\n",
      "42\t338\t0.3\t1\t317 21\n",
      "43\t326\t0.3\t1\t313 13\n",
      "44\t269\t0.3\t1\t242 27\n",
      "45\t257\t0.3\t1\t234 23\n",
      "46\t243\t0.3\t1\t219 24\n",
      "47\t261\t0.3\t1\t225 36\n",
      "48\t227\t0.3\t1\t204 23\n",
      "49\t168\t0.3\t1\t153 15\n",
      "50\t149\t0.3\t1\t125 24\n",
      "51\t123\t0.3\t1\t115 8\n",
      "52\t92\t0.3\t1\t76 16\n",
      "53\t103\t0.3\t1\t86 17\n",
      "54\t100\t0.3\t1\t87 13\n",
      "55\t74\t0.3\t1\t67 7\n",
      "56\t90\t0.3\t1\t49 41\n",
      "57\t72\t0.3\t1\t50 22\n",
      "58\t95\t0.3\t1\t55 40\n",
      "59\t56\t0.3\t1\t29 27\n",
      "60\t45\t0.3\t1\t27 18\n",
      "61\t47\t0.3\t1\t28 19\n",
      "62\t40\t0.3\t1\t19 21\n",
      "63\t45\t0.3\t1\t21 24\n",
      "64\t53\t0.3\t1\t19 34\n",
      "65\t67\t0.3\t1\t27 40\n",
      "66\t61\t0.3\t1\t27 34\n",
      "67\t44\t0.3\t1\t15 29\n",
      "68\t58\t0.3\t1\t16 42\n",
      "69\t81\t0.3\t1\t10 71\n",
      "70\t128\t0.3\t1\t23 105\n",
      "71\t441\t0.3\t1\t27 414\n",
      "72\t249\t0.3\t1\t107 142\n",
      "73\t86\t0.3\t1\t36 50\n",
      "74\t108\t0.3\t1\t23 85\n",
      "75\t13\t0.3\t1\t2 11\n",
      "76\t17\t0.3\t1\t3 14\n",
      "77\t24\t0.3\t1\t3 21\n",
      "78\t21\t0.3\t1\t2 19\n",
      "79\t18\t0.3\t1\t2 16\n",
      "80\t11\t0.3\t1\t1 10\n",
      "81\t19\t0.3\t1\t1 18\n",
      "82\t29\t0.3\t1\t0 29\n",
      "83\t40\t0.3\t1\t0 40\n",
      "84\t9\t0.3\t1\t1 8\n",
      "85\t19\t0.3\t1\t0 19\n",
      "86\t20\t0.3\t1\t0 20\n",
      "87\t16\t0.3\t1\t0 16\n",
      "88\t19\t0.3\t1\t1 18\n",
      "89\t16\t0.3\t1\t0 16\n",
      "90\t12\t0.3\t1\t0 12\n",
      "91\t18\t0.3\t1\t0 18\n",
      "92\t5\t0.3\t1\t0 5\n",
      "93\t6\t0.3\t1\t0 6\n",
      "94\t11\t0.3\t1\t0 11\n",
      "95\t13\t0.3\t1\t1 12\n",
      "96\t22\t0.3\t1\t0 22\n",
      "97\t7\t0.3\t1\t1 6\n",
      "98\t5\t0.3\t1\t0 5\n",
      "99\t9\t0.3\t1\t0 9\n",
      "100\t65\t0.3\t1\t1 64\n",
      "\n",
      "\n",
      "RUN STATISTICS FOR INPUT FILE: fastqs/ERR899427.fastq.gz\n",
      "=============================================\n",
      "17183069 sequences processed in total\n",
      "Sequences removed because they became shorter than the length cutoff of 20 bp:\t4146 (0.0%)\n",
      "\n",
      "This is cutadapt 1.15 with Python 3.6.4\n",
      "Command line parameters: -f fastq -e 0.1 -q 20 -O 1 -a AGATCGGAAGAGC fastqs/ERR899428.fastq.gz\n",
      "Running on 1 core\n",
      "Trimming 1 adapter with at most 10.0% errors in single-end mode ...\n",
      "Finished in 614.42 s (35 us/read; 1.71 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:              17,513,754\n",
      "Reads with adapters:                 5,912,940 (33.8%)\n",
      "Reads written (passing filters):    17,513,754 (100.0%)\n",
      "\n",
      "Total basepairs processed: 1,751,375,400 bp\n",
      "Quality-trimmed:              18,659,765 bp (1.1%)\n",
      "Total written (filtered):  1,721,197,884 bp (98.3%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: AGATCGGAAGAGC; Type: regular 3'; Length: 13; Trimmed: 5912940 times.\n",
      "\n",
      "No. of allowed errors:\n",
      "0-9 bp: 0; 10-13 bp: 1\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "A: 28.9%\n",
      "C: 33.5%\n",
      "G: 19.2%\n",
      "T: 18.4%\n",
      "none/other: 0.0%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "1\t3975796\t4378438.5\t0\t3975796\n",
      "2\t1245174\t1094609.6\t0\t1245174\n",
      "3\t323946\t273652.4\t0\t323946\n",
      "4\t94193\t68413.1\t0\t94193\n",
      "5\t41397\t17103.3\t0\t41397\n",
      "6\t23770\t4275.8\t0\t23770\n",
      "7\t19936\t1069.0\t0\t19936\n",
      "8\t17233\t267.2\t0\t17233\n",
      "9\t16542\t66.8\t0\t16318 224\n",
      "10\t15454\t16.7\t1\t13627 1827\n",
      "11\t13559\t4.2\t1\t12089 1470\n",
      "12\t12738\t1.0\t1\t11527 1211\n",
      "13\t11129\t0.3\t1\t10091 1038\n",
      "14\t10870\t0.3\t1\t9869 1001\n",
      "15\t10015\t0.3\t1\t9089 926\n",
      "16\t9043\t0.3\t1\t8178 865\n",
      "17\t8121\t0.3\t1\t7285 836\n",
      "18\t7148\t0.3\t1\t6440 708\n",
      "19\t5871\t0.3\t1\t5283 588\n",
      "20\t5312\t0.3\t1\t4749 563\n",
      "21\t4516\t0.3\t1\t4061 455\n",
      "22\t4109\t0.3\t1\t3711 398\n",
      "23\t3823\t0.3\t1\t3413 410\n",
      "24\t3458\t0.3\t1\t3085 373\n",
      "25\t3275\t0.3\t1\t2933 342\n",
      "26\t2860\t0.3\t1\t2579 281\n",
      "27\t2578\t0.3\t1\t2312 266\n",
      "28\t2370\t0.3\t1\t2137 233\n",
      "29\t1942\t0.3\t1\t1746 196\n",
      "30\t1700\t0.3\t1\t1501 199\n",
      "31\t1406\t0.3\t1\t1278 128\n",
      "32\t1333\t0.3\t1\t1194 139\n",
      "33\t1090\t0.3\t1\t978 112\n",
      "34\t1084\t0.3\t1\t974 110\n",
      "35\t941\t0.3\t1\t859 82\n",
      "36\t936\t0.3\t1\t840 96\n",
      "37\t819\t0.3\t1\t707 112\n",
      "38\t716\t0.3\t1\t637 79\n",
      "39\t650\t0.3\t1\t580 70\n",
      "40\t506\t0.3\t1\t430 76\n",
      "41\t423\t0.3\t1\t362 61\n",
      "42\t375\t0.3\t1\t326 49\n",
      "43\t314\t0.3\t1\t284 30\n",
      "44\t294\t0.3\t1\t267 27\n",
      "45\t313\t0.3\t1\t282 31\n",
      "46\t276\t0.3\t1\t242 34\n",
      "47\t286\t0.3\t1\t246 40\n",
      "48\t231\t0.3\t1\t197 34\n",
      "49\t202\t0.3\t1\t166 36\n",
      "50\t178\t0.3\t1\t150 28\n",
      "51\t159\t0.3\t1\t140 19\n",
      "52\t127\t0.3\t1\t94 33\n",
      "53\t97\t0.3\t1\t74 23\n",
      "54\t79\t0.3\t1\t68 11\n",
      "55\t105\t0.3\t1\t86 19\n",
      "56\t102\t0.3\t1\t55 47\n",
      "57\t70\t0.3\t1\t52 18\n",
      "58\t107\t0.3\t1\t54 53\n",
      "59\t62\t0.3\t1\t41 21\n",
      "60\t63\t0.3\t1\t40 23\n",
      "61\t40\t0.3\t1\t30 10\n",
      "62\t36\t0.3\t1\t19 17\n",
      "63\t36\t0.3\t1\t22 14\n",
      "64\t46\t0.3\t1\t16 30\n",
      "65\t57\t0.3\t1\t21 36\n",
      "66\t49\t0.3\t1\t30 19\n",
      "67\t26\t0.3\t1\t16 10\n",
      "68\t49\t0.3\t1\t19 30\n",
      "69\t71\t0.3\t1\t12 59\n",
      "70\t122\t0.3\t1\t22 100\n",
      "71\t372\t0.3\t1\t49 323\n",
      "72\t206\t0.3\t1\t112 94\n",
      "73\t76\t0.3\t1\t40 36\n",
      "74\t115\t0.3\t1\t16 99\n",
      "75\t25\t0.3\t1\t11 14\n",
      "76\t19\t0.3\t1\t2 17\n",
      "77\t19\t0.3\t1\t3 16\n",
      "78\t17\t0.3\t1\t0 17\n",
      "79\t15\t0.3\t1\t3 12\n",
      "80\t14\t0.3\t1\t1 13\n",
      "81\t10\t0.3\t1\t1 9\n",
      "82\t18\t0.3\t1\t0 18\n",
      "83\t26\t0.3\t1\t0 26\n",
      "84\t13\t0.3\t1\t1 12\n",
      "85\t12\t0.3\t1\t0 12\n",
      "86\t16\t0.3\t1\t0 16\n",
      "87\t17\t0.3\t1\t1 16\n",
      "88\t21\t0.3\t1\t0 21\n",
      "89\t21\t0.3\t1\t0 21\n",
      "90\t12\t0.3\t1\t0 12\n",
      "91\t12\t0.3\t1\t0 12\n",
      "92\t20\t0.3\t1\t0 20\n",
      "93\t10\t0.3\t1\t2 8\n",
      "94\t6\t0.3\t1\t0 6\n",
      "95\t4\t0.3\t1\t1 3\n",
      "96\t20\t0.3\t1\t0 20\n",
      "97\t12\t0.3\t1\t2 10\n",
      "98\t13\t0.3\t1\t1 12\n",
      "99\t5\t0.3\t1\t1 4\n",
      "100\t40\t0.3\t1\t1 39\n",
      "\n",
      "\n",
      "RUN STATISTICS FOR INPUT FILE: fastqs/ERR899428.fastq.gz\n",
      "=============================================\n",
      "17513754 sequences processed in total\n",
      "Sequences removed because they became shorter than the length cutoff of 20 bp:\t3937 (0.0%)\n",
      "\n",
      "This is cutadapt 1.15 with Python 3.6.4\n",
      "Command line parameters: -f fastq -e 0.1 -q 20 -O 1 -a AGATCGGAAGAGC fastqs/ERR899435.fastq.gz\n",
      "Running on 1 core\n",
      "Trimming 1 adapter with at most 10.0% errors in single-end mode ...\n",
      "Finished in 626.45 s (35 us/read; 1.71 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:              17,899,859\n",
      "Reads with adapters:                 5,992,885 (33.5%)\n",
      "Reads written (passing filters):    17,899,859 (100.0%)\n",
      "\n",
      "Total basepairs processed: 1,789,985,900 bp\n",
      "Quality-trimmed:              18,555,538 bp (1.0%)\n",
      "Total written (filtered):  1,759,628,077 bp (98.3%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: AGATCGGAAGAGC; Type: regular 3'; Length: 13; Trimmed: 5992885 times.\n",
      "\n",
      "No. of allowed errors:\n",
      "0-9 bp: 0; 10-13 bp: 1\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "A: 28.1%\n",
      "C: 34.7%\n",
      "G: 19.4%\n",
      "T: 17.8%\n",
      "none/other: 0.0%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "1\t3999291\t4474964.8\t0\t3999291\n",
      "2\t1282222\t1118741.2\t0\t1282222\n",
      "3\t332866\t279685.3\t0\t332866\n",
      "4\t95522\t69921.3\t0\t95522\n",
      "5\t42887\t17480.3\t0\t42887\n",
      "6\t24950\t4370.1\t0\t24950\n",
      "7\t21206\t1092.5\t0\t21206\n",
      "8\t18488\t273.1\t0\t18488\n",
      "9\t16994\t68.3\t0\t16704 290\n",
      "10\t15911\t17.1\t1\t14897 1014\n",
      "11\t13870\t4.3\t1\t13019 851\n",
      "12\t12783\t1.1\t1\t12181 602\n",
      "13\t11300\t0.3\t1\t10861 439\n",
      "14\t10643\t0.3\t1\t10200 443\n",
      "15\t9754\t0.3\t1\t9345 409\n",
      "16\t8937\t0.3\t1\t8523 414\n",
      "17\t8092\t0.3\t1\t7702 390\n",
      "18\t7299\t0.3\t1\t6942 357\n",
      "19\t6007\t0.3\t1\t5744 263\n",
      "20\t5351\t0.3\t1\t5084 267\n",
      "21\t4724\t0.3\t1\t4475 249\n",
      "22\t4345\t0.3\t1\t4148 197\n",
      "23\t3747\t0.3\t1\t3547 200\n",
      "24\t3638\t0.3\t1\t3474 164\n",
      "25\t3380\t0.3\t1\t3195 185\n",
      "26\t3070\t0.3\t1\t2934 136\n",
      "27\t2821\t0.3\t1\t2697 124\n",
      "28\t2347\t0.3\t1\t2241 106\n",
      "29\t2048\t0.3\t1\t1962 86\n",
      "30\t1819\t0.3\t1\t1704 115\n",
      "31\t1481\t0.3\t1\t1409 72\n",
      "32\t1341\t0.3\t1\t1286 55\n",
      "33\t1156\t0.3\t1\t1102 54\n",
      "34\t1176\t0.3\t1\t1102 74\n",
      "35\t1070\t0.3\t1\t997 73\n",
      "36\t1003\t0.3\t1\t943 60\n",
      "37\t835\t0.3\t1\t780 55\n",
      "38\t776\t0.3\t1\t732 44\n",
      "39\t682\t0.3\t1\t643 39\n",
      "40\t552\t0.3\t1\t507 45\n",
      "41\t470\t0.3\t1\t446 24\n",
      "42\t430\t0.3\t1\t400 30\n",
      "43\t355\t0.3\t1\t335 20\n",
      "44\t400\t0.3\t1\t364 36\n",
      "45\t327\t0.3\t1\t300 27\n",
      "46\t281\t0.3\t1\t253 28\n",
      "47\t296\t0.3\t1\t277 19\n",
      "48\t252\t0.3\t1\t227 25\n",
      "49\t194\t0.3\t1\t169 25\n",
      "50\t195\t0.3\t1\t171 24\n",
      "51\t162\t0.3\t1\t147 15\n",
      "52\t142\t0.3\t1\t123 19\n",
      "53\t115\t0.3\t1\t101 14\n",
      "54\t93\t0.3\t1\t77 16\n",
      "55\t100\t0.3\t1\t90 10\n",
      "56\t105\t0.3\t1\t75 30\n",
      "57\t78\t0.3\t1\t67 11\n",
      "58\t97\t0.3\t1\t61 36\n",
      "59\t89\t0.3\t1\t62 27\n",
      "60\t83\t0.3\t1\t52 31\n",
      "61\t62\t0.3\t1\t45 17\n",
      "62\t53\t0.3\t1\t32 21\n",
      "63\t53\t0.3\t1\t27 26\n",
      "64\t51\t0.3\t1\t28 23\n",
      "65\t74\t0.3\t1\t26 48\n",
      "66\t50\t0.3\t1\t29 21\n",
      "67\t49\t0.3\t1\t17 32\n",
      "68\t66\t0.3\t1\t14 52\n",
      "69\t122\t0.3\t1\t23 99\n",
      "70\t153\t0.3\t1\t27 126\n",
      "71\t541\t0.3\t1\t52 489\n",
      "72\t247\t0.3\t1\t134 113\n",
      "73\t88\t0.3\t1\t40 48\n",
      "74\t124\t0.3\t1\t12 112\n",
      "75\t22\t0.3\t1\t14 8\n",
      "76\t16\t0.3\t1\t4 12\n",
      "77\t25\t0.3\t1\t2 23\n",
      "78\t31\t0.3\t1\t2 29\n",
      "79\t15\t0.3\t1\t1 14\n",
      "80\t18\t0.3\t1\t1 17\n",
      "81\t14\t0.3\t1\t1 13\n",
      "82\t18\t0.3\t1\t1 17\n",
      "83\t28\t0.3\t1\t0 28\n",
      "84\t12\t0.3\t1\t1 11\n",
      "85\t25\t0.3\t1\t0 25\n",
      "86\t33\t0.3\t1\t0 33\n",
      "87\t11\t0.3\t1\t1 10\n",
      "88\t37\t0.3\t1\t1 36\n",
      "89\t15\t0.3\t1\t1 14\n",
      "90\t10\t0.3\t1\t0 10\n",
      "91\t16\t0.3\t1\t1 15\n",
      "92\t13\t0.3\t1\t0 13\n",
      "93\t8\t0.3\t1\t1 7\n",
      "94\t11\t0.3\t1\t0 11\n",
      "95\t14\t0.3\t1\t2 12\n",
      "96\t19\t0.3\t1\t1 18\n",
      "97\t12\t0.3\t1\t1 11\n",
      "98\t8\t0.3\t1\t1 7\n",
      "99\t8\t0.3\t1\t0 8\n",
      "100\t65\t0.3\t1\t0 65\n",
      "\n",
      "This is cutadapt 1.15 with Python 3.6.4\n",
      "Command line parameters: -f fastq -e 0.1 -q 20 -O 1 -a AGATCGGAAGAGC fastqs/ERR899421.fastq.gz\n",
      "Running on 1 core\n",
      "Trimming 1 adapter with at most 10.0% errors in single-end mode ...\n",
      "Finished in 632.70 s (35 us/read; 1.71 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:              17,979,693\n",
      "Reads with adapters:                 6,092,945 (33.9%)\n",
      "Reads written (passing filters):    17,979,693 (100.0%)\n",
      "\n",
      "Total basepairs processed: 1,797,969,300 bp\n",
      "Quality-trimmed:              18,790,294 bp (1.0%)\n",
      "Total written (filtered):  1,766,392,323 bp (98.2%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: AGATCGGAAGAGC; Type: regular 3'; Length: 13; Trimmed: 6092945 times.\n",
      "\n",
      "No. of allowed errors:\n",
      "0-9 bp: 0; 10-13 bp: 1\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "A: 28.1%\n",
      "C: 34.2%\n",
      "G: 19.8%\n",
      "T: 17.9%\n",
      "none/other: 0.0%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "1\t4021207\t4494923.2\t0\t4021207\n",
      "2\t1286435\t1123730.8\t0\t1286435\n",
      "3\t339826\t280932.7\t0\t339826\n",
      "4\t99285\t70233.2\t0\t99285\n",
      "5\t46599\t17558.3\t0\t46599\n",
      "6\t28672\t4389.6\t0\t28672\n",
      "7\t25114\t1097.4\t0\t25114\n",
      "8\t21936\t274.3\t0\t21936\n",
      "9\t20640\t68.6\t0\t20359 281\n",
      "10\t18936\t17.1\t1\t18187 749\n",
      "11\t16885\t4.3\t1\t16276 609\n",
      "12\t15739\t1.1\t1\t15370 369\n",
      "13\t14562\t0.3\t1\t14229 333\n",
      "14\t13666\t0.3\t1\t13374 292\n",
      "15\t12729\t0.3\t1\t12470 259\n",
      "16\t11661\t0.3\t1\t11383 278\n",
      "17\t10694\t0.3\t1\t10457 237\n",
      "18\t9592\t0.3\t1\t9353 239\n",
      "19\t7682\t0.3\t1\t7513 169\n",
      "20\t7275\t0.3\t1\t7111 164\n",
      "21\t6392\t0.3\t1\t6253 139\n",
      "22\t5927\t0.3\t1\t5793 134\n",
      "23\t5326\t0.3\t1\t5214 112\n",
      "24\t5141\t0.3\t1\t5007 134\n",
      "25\t4658\t0.3\t1\t4539 119\n",
      "26\t4291\t0.3\t1\t4188 103\n",
      "27\t3759\t0.3\t1\t3656 103\n",
      "28\t3365\t0.3\t1\t3277 88\n",
      "29\t2877\t0.3\t1\t2803 74\n",
      "30\t2627\t0.3\t1\t2530 97\n",
      "31\t2104\t0.3\t1\t2052 52\n",
      "32\t1922\t0.3\t1\t1875 47\n",
      "33\t1737\t0.3\t1\t1701 36\n",
      "34\t1635\t0.3\t1\t1588 47\n",
      "35\t1438\t0.3\t1\t1395 43\n",
      "36\t1236\t0.3\t1\t1200 36\n",
      "37\t1177\t0.3\t1\t1134 43\n",
      "38\t982\t0.3\t1\t941 41\n",
      "39\t854\t0.3\t1\t830 24\n",
      "40\t704\t0.3\t1\t676 28\n",
      "41\t595\t0.3\t1\t572 23\n",
      "42\t553\t0.3\t1\t532 21\n",
      "43\t439\t0.3\t1\t424 15\n",
      "44\t404\t0.3\t1\t375 29\n",
      "45\t386\t0.3\t1\t372 14\n",
      "46\t307\t0.3\t1\t296 11\n",
      "47\t288\t0.3\t1\t253 35\n",
      "48\t274\t0.3\t1\t253 21\n",
      "49\t238\t0.3\t1\t222 16\n",
      "50\t184\t0.3\t1\t166 18\n",
      "51\t164\t0.3\t1\t160 4\n",
      "52\t137\t0.3\t1\t121 16\n",
      "53\t109\t0.3\t1\t90 19\n",
      "54\t81\t0.3\t1\t68 13\n",
      "55\t74\t0.3\t1\t66 8\n",
      "56\t95\t0.3\t1\t73 22\n",
      "57\t76\t0.3\t1\t55 21\n",
      "58\t74\t0.3\t1\t52 22\n",
      "59\t65\t0.3\t1\t48 17\n",
      "60\t46\t0.3\t1\t26 20\n",
      "61\t42\t0.3\t1\t32 10\n",
      "62\t37\t0.3\t1\t26 11\n",
      "\n",
      "RUN STATISTICS FOR INPUT FILE: fastqs/ERR899435.fastq.gz\n",
      "=============================================\n",
      "17899859 sequences processed in total\n",
      "Sequences removed because they became shorter than the length cutoff of 20 bp:\t4261 (0.0%)\n",
      "\n",
      "63\t30\t0.3\t1\t20 10\n",
      "64\t33\t0.3\t1\t23 10\n",
      "65\t27\t0.3\t1\t21 6\n",
      "66\t33\t0.3\t1\t13 20\n",
      "67\t43\t0.3\t1\t21 22\n",
      "68\t59\t0.3\t1\t39 20\n",
      "69\t84\t0.3\t1\t41 43\n",
      "70\t155\t0.3\t1\t131 24\n",
      "71\t73\t0.3\t1\t42 31\n",
      "72\t31\t0.3\t1\t17 14\n",
      "73\t19\t0.3\t1\t8 11\n",
      "74\t74\t0.3\t1\t4 70\n",
      "75\t9\t0.3\t1\t2 7\n",
      "76\t11\t0.3\t1\t2 9\n",
      "77\t9\t0.3\t1\t0 9\n",
      "78\t15\t0.3\t1\t1 14\n",
      "79\t8\t0.3\t1\t0 8\n",
      "80\t7\t0.3\t1\t0 7\n",
      "81\t16\t0.3\t1\t1 15\n",
      "82\t25\t0.3\t1\t0 25\n",
      "83\t29\t0.3\t1\t1 28\n",
      "84\t17\t0.3\t1\t2 15\n",
      "85\t11\t0.3\t1\t0 11\n",
      "86\t16\t0.3\t1\t1 15\n",
      "87\t16\t0.3\t1\t0 16\n",
      "88\t13\t0.3\t1\t0 13\n",
      "89\t18\t0.3\t1\t2 16\n",
      "90\t14\t0.3\t1\t1 13\n",
      "91\t19\t0.3\t1\t0 19\n",
      "92\t10\t0.3\t1\t0 10\n",
      "93\t8\t0.3\t1\t1 7\n",
      "94\t10\t0.3\t1\t0 10\n",
      "95\t8\t0.3\t1\t0 8\n",
      "96\t14\t0.3\t1\t1 13\n",
      "97\t9\t0.3\t1\t1 8\n",
      "98\t7\t0.3\t1\t1 6\n",
      "99\t1\t0.3\t1\t0 1\n",
      "100\t9\t0.3\t1\t2 7\n",
      "\n",
      "\n",
      "RUN STATISTICS FOR INPUT FILE: fastqs/ERR899421.fastq.gz\n",
      "=============================================\n",
      "17979693 sequences processed in total\n",
      "Sequences removed because they became shorter than the length cutoff of 20 bp:\t2966 (0.0%)\n",
      "\n",
      "This is cutadapt 1.15 with Python 3.6.4\n",
      "Command line parameters: -f fastq -e 0.1 -q 20 -O 1 -a AGATCGGAAGAGC fastqs/ERR899432.fastq.gz\n",
      "Running on 1 core\n",
      "Trimming 1 adapter with at most 10.0% errors in single-end mode ...\n",
      "Finished in 638.53 s (35 us/read; 1.71 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:              18,211,995\n",
      "Reads with adapters:                 6,152,649 (33.8%)\n",
      "Reads written (passing filters):    18,211,995 (100.0%)\n",
      "\n",
      "Total basepairs processed: 1,821,199,500 bp\n",
      "Quality-trimmed:              18,756,852 bp (1.0%)\n",
      "Total written (filtered):  1,790,818,811 bp (98.3%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: AGATCGGAAGAGC; Type: regular 3'; Length: 13; Trimmed: 6152649 times.\n",
      "\n",
      "No. of allowed errors:\n",
      "0-9 bp: 0; 10-13 bp: 1\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "A: 29.0%\n",
      "C: 33.9%\n",
      "G: 18.9%\n",
      "T: 18.3%\n",
      "none/other: 0.0%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "1\t4169023\t4552998.8\t0\t4169023\n",
      "2\t1285762\t1138249.7\t0\t1285762\n",
      "3\t339912\t284562.4\t0\t339912\n",
      "4\t95916\t71140.6\t0\t95916\n",
      "5\t41612\t17785.2\t0\t41612\n",
      "6\t23375\t4446.3\t0\t23375\n",
      "7\t20091\t1111.6\t0\t20091\n",
      "8\t17379\t277.9\t0\t17379\n",
      "9\t16200\t69.5\t0\t15912 288\n",
      "10\t14337\t17.4\t1\t13693 644\n",
      "11\t13241\t4.3\t1\t12766 475\n",
      "12\t11895\t1.1\t1\t11580 315\n",
      "13\t10630\t0.3\t1\t10408 222\n",
      "14\t9875\t0.3\t1\t9641 234\n",
      "15\t9004\t0.3\t1\t8794 210\n",
      "16\t8207\t0.3\t1\t7997 210\n",
      "17\t7429\t0.3\t1\t7249 180\n",
      "18\t6490\t0.3\t1\t6320 170\n",
      "19\t5244\t0.3\t1\t5103 141\n",
      "20\t4844\t0.3\t1\t4700 144\n",
      "21\t4291\t0.3\t1\t4179 112\n",
      "22\t3910\t0.3\t1\t3821 89\n",
      "23\t3483\t0.3\t1\t3372 111\n",
      "24\t3233\t0.3\t1\t3133 100\n",
      "25\t2944\t0.3\t1\t2839 105\n",
      "26\t2685\t0.3\t1\t2600 85\n",
      "27\t2428\t0.3\t1\t2368 60\n",
      "28\t2036\t0.3\t1\t1977 59\n",
      "29\t1665\t0.3\t1\t1607 58\n",
      "30\t1679\t0.3\t1\t1616 63\n",
      "31\t1266\t0.3\t1\t1229 37\n",
      "32\t1141\t0.3\t1\t1109 32\n",
      "33\t1019\t0.3\t1\t992 27\n",
      "34\t1001\t0.3\t1\t963 38\n",
      "35\t891\t0.3\t1\t855 36\n",
      "36\t885\t0.3\t1\t846 39\n",
      "37\t763\t0.3\t1\t729 34\n",
      "38\t653\t0.3\t1\t621 32\n",
      "39\t553\t0.3\t1\t529 24\n",
      "40\t459\t0.3\t1\t436 23\n",
      "41\t404\t0.3\t1\t383 21\n",
      "42\t345\t0.3\t1\t332 13\n",
      "43\t284\t0.3\t1\t268 16\n",
      "44\t271\t0.3\t1\t251 20\n",
      "45\t254\t0.3\t1\t238 16\n",
      "46\t230\t0.3\t1\t216 14\n",
      "47\t209\t0.3\t1\t173 36\n",
      "48\t203\t0.3\t1\t183 20\n",
      "49\t168\t0.3\t1\t157 11\n",
      "50\t126\t0.3\t1\t113 13\n",
      "51\t113\t0.3\t1\t107 6\n",
      "52\t109\t0.3\t1\t94 15\n",
      "53\t96\t0.3\t1\t80 16\n",
      "54\t90\t0.3\t1\t70 20\n",
      "55\t96\t0.3\t1\t83 13\n",
      "56\t74\t0.3\t1\t47 27\n",
      "57\t72\t0.3\t1\t50 22\n",
      "58\t86\t0.3\t1\t43 43\n",
      "59\t76\t0.3\t1\t48 28\n",
      "60\t67\t0.3\t1\t37 30\n",
      "61\t46\t0.3\t1\t33 13\n",
      "62\t36\t0.3\t1\t25 11\n",
      "63\t40\t0.3\t1\t21 19\n",
      "64\t54\t0.3\t1\t27 27\n",
      "65\t52\t0.3\t1\t25 27\n",
      "66\t49\t0.3\t1\t24 25\n",
      "67\t26\t0.3\t1\t13 13\n",
      "68\t37\t0.3\t1\t13 24\n",
      "69\t56\t0.3\t1\t21 35\n",
      "70\t138\t0.3\t1\t49 89\n",
      "71\t382\t0.3\t1\t73 309\n",
      "72\t270\t0.3\t1\t193 77\n",
      "73\t84\t0.3\t1\t60 24\n",
      "74\t121\t0.3\t1\t31 90\n",
      "75\t31\t0.3\t1\t16 15\n",
      "76\t22\t0.3\t1\t8 14\n",
      "77\t12\t0.3\t1\t1 11\n",
      "78\t27\t0.3\t1\t6 21\n",
      "79\t16\t0.3\t1\t4 12\n",
      "80\t11\t0.3\t1\t0 11\n",
      "81\t26\t0.3\t1\t1 25\n",
      "82\t16\t0.3\t1\t0 16\n",
      "83\t43\t0.3\t1\t1 42\n",
      "84\t20\t0.3\t1\t2 18\n",
      "85\t13\t0.3\t1\t1 12\n",
      "86\t19\t0.3\t1\t1 18\n",
      "87\t11\t0.3\t1\t1 10\n",
      "88\t22\t0.3\t1\t0 22\n",
      "89\t16\t0.3\t1\t1 15\n",
      "90\t11\t0.3\t1\t3 8\n",
      "91\t11\t0.3\t1\t1 10\n",
      "92\t14\t0.3\t1\t1 13\n",
      "93\t3\t0.3\t1\t2 1\n",
      "94\t6\t0.3\t1\t0 6\n",
      "95\t8\t0.3\t1\t1 7\n",
      "96\t9\t0.3\t1\t0 9\n",
      "97\t17\t0.3\t1\t2 15\n",
      "98\t12\t0.3\t1\t0 12\n",
      "99\t9\t0.3\t1\t0 9\n",
      "100\t29\t0.3\t1\t0 29\n",
      "\n",
      "\n",
      "RUN STATISTICS FOR INPUT FILE: fastqs/ERR899432.fastq.gz\n",
      "=============================================\n",
      "18211995 sequences processed in total\n",
      "Sequences removed because they became shorter than the length cutoff of 20 bp:\t3574 (0.0%)\n",
      "\n",
      "This is cutadapt 1.15 with Python 3.6.4\n",
      "Command line parameters: -f fastq -e 0.1 -q 20 -O 1 -a AGATCGGAAGAGC fastqs/ERR899423.fastq.gz\n",
      "Running on 1 core\n",
      "Trimming 1 adapter with at most 10.0% errors in single-end mode ...\n",
      "Finished in 641.94 s (35 us/read; 1.72 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:              18,424,646\n",
      "Reads with adapters:                 6,237,903 (33.9%)\n",
      "Reads written (passing filters):    18,424,646 (100.0%)\n",
      "\n",
      "Total basepairs processed: 1,842,464,600 bp\n",
      "Quality-trimmed:              18,809,376 bp (1.0%)\n",
      "Total written (filtered):  1,810,507,279 bp (98.3%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: AGATCGGAAGAGC; Type: regular 3'; Length: 13; Trimmed: 6237903 times.\n",
      "\n",
      "No. of allowed errors:\n",
      "0-9 bp: 0; 10-13 bp: 1\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "A: 28.0%\n",
      "C: 34.3%\n",
      "G: 19.8%\n",
      "T: 17.8%\n",
      "none/other: 0.0%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "1\t4108616\t4606161.5\t0\t4108616\n",
      "2\t1321696\t1151540.4\t0\t1321696\n",
      "3\t349160\t287885.1\t0\t349160\n",
      "4\t104164\t71971.3\t0\t104164\n",
      "5\t49789\t17992.8\t0\t49789\n",
      "6\t30589\t4498.2\t0\t30589\n",
      "7\t26981\t1124.6\t0\t26981\n",
      "8\t22969\t281.1\t0\t22969\n",
      "9\t21073\t70.3\t0\t20778 295\n",
      "10\t19544\t17.6\t1\t18655 889\n",
      "11\t17137\t4.4\t1\t16435 702\n",
      "12\t16195\t1.1\t1\t15682 513\n",
      "13\t14441\t0.3\t1\t14036 405\n",
      "14\t13566\t0.3\t1\t13155 411\n",
      "15\t12811\t0.3\t1\t12448 363\n",
      "16\t11475\t0.3\t1\t11148 327\n",
      "17\t10644\t0.3\t1\t10311 333\n",
      "18\t9293\t0.3\t1\t9001 292\n",
      "19\t7735\t0.3\t1\t7530 205\n",
      "20\t6886\t0.3\t1\t6630 256\n",
      "21\t6230\t0.3\t1\t6018 212\n",
      "22\t5552\t0.3\t1\t5379 173\n",
      "23\t4997\t0.3\t1\t4851 146\n",
      "24\t4643\t0.3\t1\t4509 134\n",
      "25\t4318\t0.3\t1\t4188 130\n",
      "26\t3894\t0.3\t1\t3778 116\n",
      "27\t3454\t0.3\t1\t3355 99\n",
      "28\t3146\t0.3\t1\t3047 99\n",
      "29\t2706\t0.3\t1\t2600 106\n",
      "30\t2429\t0.3\t1\t2334 95\n",
      "31\t1924\t0.3\t1\t1862 62\n",
      "32\t1706\t0.3\t1\t1629 77\n",
      "33\t1528\t0.3\t1\t1491 37\n",
      "34\t1530\t0.3\t1\t1465 65\n",
      "35\t1432\t0.3\t1\t1372 60\n",
      "36\t1273\t0.3\t1\t1224 49\n",
      "37\t1285\t0.3\t1\t1231 54\n",
      "38\t796\t0.3\t1\t770 26\n",
      "39\t768\t0.3\t1\t741 27\n",
      "40\t710\t0.3\t1\t669 41\n",
      "41\t592\t0.3\t1\t564 28\n",
      "42\t542\t0.3\t1\t522 20\n",
      "43\t482\t0.3\t1\t468 14\n",
      "44\t442\t0.3\t1\t422 20\n",
      "45\t389\t0.3\t1\t361 28\n",
      "46\t390\t0.3\t1\t371 19\n",
      "47\t415\t0.3\t1\t364 51\n",
      "48\t366\t0.3\t1\t345 21\n",
      "49\t288\t0.3\t1\t265 23\n",
      "50\t237\t0.3\t1\t210 27\n",
      "51\t202\t0.3\t1\t194 8\n",
      "52\t177\t0.3\t1\t155 22\n",
      "53\t161\t0.3\t1\t143 18\n",
      "54\t158\t0.3\t1\t133 25\n",
      "55\t142\t0.3\t1\t134 8\n",
      "56\t122\t0.3\t1\t89 33\n",
      "57\t104\t0.3\t1\t93 11\n",
      "58\t145\t0.3\t1\t89 56\n",
      "59\t104\t0.3\t1\t85 19\n",
      "60\t89\t0.3\t1\t64 25\n",
      "61\t66\t0.3\t1\t51 15\n",
      "62\t49\t0.3\t1\t35 14\n",
      "63\t66\t0.3\t1\t39 27\n",
      "64\t113\t0.3\t1\t42 71\n",
      "65\t107\t0.3\t1\t36 71\n",
      "66\t53\t0.3\t1\t27 26\n",
      "67\t64\t0.3\t1\t24 40\n",
      "68\t76\t0.3\t1\t19 57\n",
      "69\t196\t0.3\t1\t33 163\n",
      "70\t265\t0.3\t1\t29 236\n",
      "71\t1035\t0.3\t1\t54 981\n",
      "72\t287\t0.3\t1\t75 212\n",
      "73\t133\t0.3\t1\t34 99\n",
      "74\t105\t0.3\t1\t12 93\n",
      "75\t21\t0.3\t1\t8 13\n",
      "76\t17\t0.3\t1\t2 15\n",
      "77\t31\t0.3\t1\t2 29\n",
      "78\t34\t0.3\t1\t1 33\n",
      "79\t32\t0.3\t1\t2 30\n",
      "80\t14\t0.3\t1\t0 14\n",
      "81\t24\t0.3\t1\t0 24\n",
      "82\t37\t0.3\t1\t1 36\n",
      "83\t40\t0.3\t1\t0 40\n",
      "84\t21\t0.3\t1\t0 21\n",
      "85\t23\t0.3\t1\t0 23\n",
      "86\t19\t0.3\t1\t1 18\n",
      "87\t14\t0.3\t1\t1 13\n",
      "88\t34\t0.3\t1\t0 34\n",
      "89\t37\t0.3\t1\t2 35\n",
      "90\t19\t0.3\t1\t0 19\n",
      "91\t18\t0.3\t1\t0 18\n",
      "92\t16\t0.3\t1\t1 15\n",
      "93\t15\t0.3\t1\t0 15\n",
      "94\t8\t0.3\t1\t0 8\n",
      "95\t14\t0.3\t1\t0 14\n",
      "96\t15\t0.3\t1\t0 15\n",
      "97\t14\t0.3\t1\t1 13\n",
      "98\t18\t0.3\t1\t0 18\n",
      "99\t13\t0.3\t1\t0 13\n",
      "100\t108\t0.3\t1\t0 108\n",
      "\n",
      "\n",
      "RUN STATISTICS FOR INPUT FILE: fastqs/ERR899423.fastq.gz\n",
      "=============================================\n",
      "18424646 sequences processed in total\n",
      "Sequences removed because they became shorter than the length cutoff of 20 bp:\t5188 (0.0%)\n",
      "\n",
      "This is cutadapt 1.15 with Python 3.6.4\n",
      "Command line parameters: -f fastq -e 0.1 -q 20 -O 1 -a AGATCGGAAGAGC fastqs/ERR899436.fastq.gz\n",
      "Running on 1 core\n",
      "Trimming 1 adapter with at most 10.0% errors in single-end mode ...\n",
      "Finished in 642.73 s (35 us/read; 1.70 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:              18,176,427\n",
      "Reads with adapters:                 6,099,215 (33.6%)\n",
      "Reads written (passing filters):    18,176,427 (100.0%)\n",
      "\n",
      "Total basepairs processed: 1,817,642,700 bp\n",
      "Quality-trimmed:              19,027,099 bp (1.0%)\n",
      "Total written (filtered):  1,786,509,822 bp (98.3%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: AGATCGGAAGAGC; Type: regular 3'; Length: 13; Trimmed: 6099215 times.\n",
      "\n",
      "No. of allowed errors:\n",
      "0-9 bp: 0; 10-13 bp: 1\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "A: 28.5%\n",
      "C: 34.1%\n",
      "G: 19.5%\n",
      "T: 18.0%\n",
      "none/other: 0.0%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "1\t4084093\t4544106.8\t0\t4084093\n",
      "2\t1289267\t1136026.7\t0\t1289267\n",
      "3\t338547\t284006.7\t0\t338547\n",
      "4\t96384\t71001.7\t0\t96384\n",
      "5\t42328\t17750.4\t0\t42328\n",
      "6\t23978\t4437.6\t0\t23978\n",
      "7\t20247\t1109.4\t0\t20247\n",
      "8\t18167\t277.4\t0\t18167\n",
      "9\t17303\t69.3\t0\t16964 339\n",
      "10\t16048\t17.3\t1\t14779 1269\n",
      "11\t14138\t4.3\t1\t13186 952\n",
      "12\t13286\t1.1\t1\t12545 741\n",
      "13\t11920\t0.3\t1\t11266 654\n",
      "14\t11271\t0.3\t1\t10670 601\n",
      "15\t10539\t0.3\t1\t10016 523\n",
      "16\t9497\t0.3\t1\t8972 525\n",
      "17\t8785\t0.3\t1\t8311 474\n",
      "18\t7931\t0.3\t1\t7485 446\n",
      "19\t6565\t0.3\t1\t6197 368\n",
      "20\t5820\t0.3\t1\t5501 319\n",
      "21\t5351\t0.3\t1\t5065 286\n",
      "22\t4989\t0.3\t1\t4697 292\n",
      "23\t4350\t0.3\t1\t4094 256\n",
      "24\t4060\t0.3\t1\t3851 209\n",
      "25\t3758\t0.3\t1\t3535 223\n",
      "26\t3283\t0.3\t1\t3117 166\n",
      "27\t2909\t0.3\t1\t2737 172\n",
      "28\t2755\t0.3\t1\t2603 152\n",
      "29\t2207\t0.3\t1\t2071 136\n",
      "30\t2101\t0.3\t1\t1967 134\n",
      "31\t1637\t0.3\t1\t1550 87\n",
      "32\t1517\t0.3\t1\t1406 111\n",
      "33\t1364\t0.3\t1\t1294 70\n",
      "34\t1309\t0.3\t1\t1235 74\n",
      "35\t1182\t0.3\t1\t1098 84\n",
      "36\t1050\t0.3\t1\t978 72\n",
      "37\t951\t0.3\t1\t862 89\n",
      "38\t787\t0.3\t1\t731 56\n",
      "39\t677\t0.3\t1\t636 41\n",
      "40\t641\t0.3\t1\t607 34\n",
      "41\t496\t0.3\t1\t461 35\n",
      "42\t448\t0.3\t1\t409 39\n",
      "43\t371\t0.3\t1\t336 35\n",
      "44\t336\t0.3\t1\t307 29\n",
      "45\t304\t0.3\t1\t285 19\n",
      "46\t299\t0.3\t1\t271 28\n",
      "47\t269\t0.3\t1\t236 33\n",
      "48\t240\t0.3\t1\t206 34\n",
      "49\t220\t0.3\t1\t200 20\n",
      "50\t168\t0.3\t1\t145 23\n",
      "51\t141\t0.3\t1\t131 10\n",
      "52\t140\t0.3\t1\t114 26\n",
      "53\t126\t0.3\t1\t106 20\n",
      "54\t90\t0.3\t1\t75 15\n",
      "55\t89\t0.3\t1\t75 14\n",
      "56\t87\t0.3\t1\t61 26\n",
      "57\t79\t0.3\t1\t66 13\n",
      "58\t96\t0.3\t1\t49 47\n",
      "59\t89\t0.3\t1\t60 29\n",
      "60\t63\t0.3\t1\t42 21\n",
      "61\t42\t0.3\t1\t28 14\n",
      "62\t55\t0.3\t1\t24 31\n",
      "63\t75\t0.3\t1\t45 30\n",
      "64\t60\t0.3\t1\t41 19\n",
      "65\t44\t0.3\t1\t28 16\n",
      "66\t62\t0.3\t1\t26 36\n",
      "67\t90\t0.3\t1\t25 65\n",
      "68\t140\t0.3\t1\t44 96\n",
      "69\t502\t0.3\t1\t61 441\n",
      "70\t301\t0.3\t1\t172 129\n",
      "71\t112\t0.3\t1\t46 66\n",
      "72\t49\t0.3\t1\t23 26\n",
      "73\t29\t0.3\t1\t11 18\n",
      "74\t84\t0.3\t1\t5 79\n",
      "75\t13\t0.3\t1\t4 9\n",
      "76\t12\t0.3\t1\t0 12\n",
      "77\t17\t0.3\t1\t1 16\n",
      "78\t25\t0.3\t1\t2 23\n",
      "79\t14\t0.3\t1\t3 11\n",
      "80\t15\t0.3\t1\t0 15\n",
      "81\t18\t0.3\t1\t1 17\n",
      "82\t21\t0.3\t1\t2 19\n",
      "83\t23\t0.3\t1\t0 23\n",
      "84\t19\t0.3\t1\t0 19\n",
      "85\t7\t0.3\t1\t0 7\n",
      "86\t22\t0.3\t1\t0 22\n",
      "87\t16\t0.3\t1\t0 16\n",
      "88\t26\t0.3\t1\t0 26\n",
      "89\t20\t0.3\t1\t1 19\n",
      "90\t11\t0.3\t1\t0 11\n",
      "91\t13\t0.3\t1\t0 13\n",
      "92\t14\t0.3\t1\t1 13\n",
      "93\t9\t0.3\t1\t1 8\n",
      "94\t9\t0.3\t1\t0 9\n",
      "95\t9\t0.3\t1\t1 8\n",
      "96\t14\t0.3\t1\t0 14\n",
      "97\t11\t0.3\t1\t1 10\n",
      "98\t16\t0.3\t1\t0 16\n",
      "99\t8\t0.3\t1\t0 8\n",
      "100\t45\t0.3\t1\t0 45\n",
      "\n",
      "\n",
      "RUN STATISTICS FOR INPUT FILE: fastqs/ERR899436.fastq.gz\n",
      "=============================================\n",
      "18176427 sequences processed in total\n",
      "Sequences removed because they became shorter than the length cutoff of 20 bp:\t4109 (0.0%)\n",
      "\n",
      "This is cutadapt 1.15 with Python 3.6.4\n",
      "Command line parameters: -f fastq -e 0.1 -q 20 -O 1 -a AGATCGGAAGAGC fastqs/ERR899434.fastq.gz\n",
      "Running on 1 core\n",
      "Trimming 1 adapter with at most 10.0% errors in single-end mode ...\n",
      "Finished in 643.07 s (35 us/read; 1.72 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:              18,421,390\n",
      "Reads with adapters:                 6,169,319 (33.5%)\n",
      "Reads written (passing filters):    18,421,390 (100.0%)\n",
      "\n",
      "Total basepairs processed: 1,842,139,000 bp\n",
      "Quality-trimmed:              18,956,285 bp (1.0%)\n",
      "Total written (filtered):  1,811,305,441 bp (98.3%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: AGATCGGAAGAGC; Type: regular 3'; Length: 13; Trimmed: 6169319 times.\n",
      "\n",
      "No. of allowed errors:\n",
      "0-9 bp: 0; 10-13 bp: 1\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "A: 28.7%\n",
      "C: 33.8%\n",
      "G: 19.5%\n",
      "T: 18.0%\n",
      "none/other: 0.0%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "1\t4162304\t4605347.5\t0\t4162304\n",
      "2\t1304819\t1151336.9\t0\t1304819\n",
      "3\t344221\t287834.2\t0\t344221\n",
      "4\t95414\t71958.6\t0\t95414\n",
      "5\t40982\t17989.6\t0\t40982\n",
      "6\t22096\t4497.4\t0\t22096\n",
      "7\t18714\t1124.4\t0\t18714\n",
      "8\t16541\t281.1\t0\t16541\n",
      "9\t15688\t70.3\t0\t15364 324\n",
      "10\t14101\t17.6\t1\t13104 997\n",
      "11\t12924\t4.4\t1\t12143 781\n",
      "12\t11970\t1.1\t1\t11440 530\n",
      "13\t10467\t0.3\t1\t10026 441\n",
      "14\t9712\t0.3\t1\t9316 396\n",
      "15\t8870\t0.3\t1\t8458 412\n",
      "16\t8273\t0.3\t1\t7920 353\n",
      "17\t7325\t0.3\t1\t7005 320\n",
      "18\t6818\t0.3\t1\t6512 306\n",
      "19\t5359\t0.3\t1\t5137 222\n",
      "20\t5068\t0.3\t1\t4836 232\n",
      "21\t4473\t0.3\t1\t4278 195\n",
      "22\t4082\t0.3\t1\t3915 167\n",
      "23\t3742\t0.3\t1\t3591 151\n",
      "24\t3373\t0.3\t1\t3190 183\n",
      "25\t3273\t0.3\t1\t3125 148\n",
      "26\t2938\t0.3\t1\t2811 127\n",
      "27\t2521\t0.3\t1\t2417 104\n",
      "28\t2390\t0.3\t1\t2276 114\n",
      "29\t1926\t0.3\t1\t1824 102\n",
      "30\t1789\t0.3\t1\t1697 92\n",
      "31\t1418\t0.3\t1\t1349 69\n",
      "32\t1296\t0.3\t1\t1239 57\n",
      "33\t1196\t0.3\t1\t1142 54\n",
      "34\t1108\t0.3\t1\t1047 61\n",
      "35\t1051\t0.3\t1\t993 58\n",
      "36\t841\t0.3\t1\t778 63\n",
      "37\t789\t0.3\t1\t731 58\n",
      "38\t747\t0.3\t1\t711 36\n",
      "39\t563\t0.3\t1\t532 31\n",
      "40\t536\t0.3\t1\t492 44\n",
      "41\t450\t0.3\t1\t427 23\n",
      "42\t414\t0.3\t1\t396 18\n",
      "43\t343\t0.3\t1\t322 21\n",
      "44\t325\t0.3\t1\t307 18\n",
      "45\t296\t0.3\t1\t270 26\n",
      "46\t276\t0.3\t1\t253 23\n",
      "47\t264\t0.3\t1\t230 34\n",
      "48\t267\t0.3\t1\t237 30\n",
      "49\t206\t0.3\t1\t186 20\n",
      "50\t171\t0.3\t1\t157 14\n",
      "51\t140\t0.3\t1\t128 12\n",
      "52\t136\t0.3\t1\t117 19\n",
      "53\t121\t0.3\t1\t105 16\n",
      "54\t122\t0.3\t1\t105 17\n",
      "55\t110\t0.3\t1\t86 24\n",
      "56\t104\t0.3\t1\t63 41\n",
      "57\t84\t0.3\t1\t57 27\n",
      "58\t118\t0.3\t1\t69 49\n",
      "59\t64\t0.3\t1\t42 22\n",
      "60\t87\t0.3\t1\t55 32\n",
      "61\t51\t0.3\t1\t34 17\n",
      "62\t42\t0.3\t1\t23 19\n",
      "63\t61\t0.3\t1\t34 27\n",
      "64\t102\t0.3\t1\t22 80\n",
      "65\t92\t0.3\t1\t24 68\n",
      "66\t76\t0.3\t1\t35 41\n",
      "67\t61\t0.3\t1\t20 41\n",
      "68\t90\t0.3\t1\t18 72\n",
      "69\t232\t0.3\t1\t21 211\n",
      "70\t291\t0.3\t1\t30 261\n",
      "71\t1121\t0.3\t1\t51 1070\n",
      "72\t372\t0.3\t1\t106 266\n",
      "73\t131\t0.3\t1\t36 95\n",
      "74\t164\t0.3\t1\t30 134\n",
      "75\t22\t0.3\t1\t6 16\n",
      "76\t22\t0.3\t1\t4 18\n",
      "77\t41\t0.3\t1\t1 40\n",
      "78\t33\t0.3\t1\t1 32\n",
      "79\t22\t0.3\t1\t1 21\n",
      "80\t15\t0.3\t1\t0 15\n",
      "81\t21\t0.3\t1\t2 19\n",
      "82\t19\t0.3\t1\t2 17\n",
      "83\t59\t0.3\t1\t1 58\n",
      "84\t15\t0.3\t1\t0 15\n",
      "85\t21\t0.3\t1\t0 21\n",
      "86\t28\t0.3\t1\t1 27\n",
      "87\t19\t0.3\t1\t1 18\n",
      "88\t20\t0.3\t1\t0 20\n",
      "89\t19\t0.3\t1\t2 17\n",
      "90\t17\t0.3\t1\t0 17\n",
      "91\t17\t0.3\t1\t0 17\n",
      "92\t16\t0.3\t1\t0 16\n",
      "93\t8\t0.3\t1\t0 8\n",
      "94\t14\t0.3\t1\t1 13\n",
      "95\t13\t0.3\t1\t0 13\n",
      "96\t15\t0.3\t1\t0 15\n",
      "97\t18\t0.3\t1\t2 16\n",
      "98\t12\t0.3\t1\t0 12\n",
      "99\t15\t0.3\t1\t0 15\n",
      "100\t96\t0.3\t1\t1 95\n",
      "\n",
      "\n",
      "RUN STATISTICS FOR INPUT FILE: fastqs/ERR899434.fastq.gz\n",
      "=============================================\n",
      "18421390 sequences processed in total\n",
      "Sequences removed because they became shorter than the length cutoff of 20 bp:\t5326 (0.0%)\n",
      "\n",
      "This is cutadapt 1.15 with Python 3.6.4\n",
      "Command line parameters: -f fastq -e 0.1 -q 20 -O 1 -a AGATCGGAAGAGC fastqs/ERR899424.fastq.gz\n",
      "Running on 1 core\n",
      "Trimming 1 adapter with at most 10.0% errors in single-end mode ...\n",
      "Finished in 650.00 s (35 us/read; 1.71 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:              18,487,849\n",
      "Reads with adapters:                 6,183,132 (33.4%)\n",
      "Reads written (passing filters):    18,487,849 (100.0%)\n",
      "\n",
      "Total basepairs processed: 1,848,784,900 bp\n",
      "Quality-trimmed:              20,303,159 bp (1.1%)\n",
      "Total written (filtered):  1,816,308,624 bp (98.2%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: AGATCGGAAGAGC; Type: regular 3'; Length: 13; Trimmed: 6183132 times.\n",
      "\n",
      "No. of allowed errors:\n",
      "0-9 bp: 0; 10-13 bp: 1\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "A: 28.2%\n",
      "C: 34.3%\n",
      "G: 19.5%\n",
      "T: 18.0%\n",
      "none/other: 0.0%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "1\t4120473\t4621962.2\t0\t4120473\n",
      "2\t1318724\t1155490.6\t0\t1318724\n",
      "3\t341442\t288872.6\t0\t341442\n",
      "4\t100234\t72218.2\t0\t100234\n",
      "5\t45863\t18054.5\t0\t45863\n",
      "6\t27265\t4513.6\t0\t27265\n",
      "7\t23658\t1128.4\t0\t23658\n",
      "8\t19968\t282.1\t0\t19968\n",
      "9\t18856\t70.5\t0\t18560 296\n",
      "10\t17061\t17.6\t1\t15782 1279\n",
      "11\t15523\t4.4\t1\t14509 1014\n",
      "12\t14128\t1.1\t1\t13330 798\n",
      "13\t12607\t0.3\t1\t11982 625\n",
      "14\t11758\t0.3\t1\t11153 605\n",
      "15\t10642\t0.3\t1\t10120 522\n",
      "16\t9643\t0.3\t1\t9139 504\n",
      "17\t8528\t0.3\t1\t8043 485\n",
      "18\t7753\t0.3\t1\t7325 428\n",
      "19\t6200\t0.3\t1\t5877 323\n",
      "20\t5627\t0.3\t1\t5268 359\n",
      "21\t4936\t0.3\t1\t4674 262\n",
      "22\t4514\t0.3\t1\t4276 238\n",
      "23\t3975\t0.3\t1\t3754 221\n",
      "24\t3623\t0.3\t1\t3423 200\n",
      "25\t3381\t0.3\t1\t3198 183\n",
      "26\t3025\t0.3\t1\t2872 153\n",
      "27\t2674\t0.3\t1\t2514 160\n",
      "28\t2340\t0.3\t1\t2203 137\n",
      "29\t1969\t0.3\t1\t1859 110\n",
      "30\t1809\t0.3\t1\t1689 120\n",
      "31\t1398\t0.3\t1\t1322 76\n",
      "32\t1306\t0.3\t1\t1226 80\n",
      "33\t1132\t0.3\t1\t1064 68\n",
      "34\t1068\t0.3\t1\t1010 58\n",
      "35\t981\t0.3\t1\t893 88\n",
      "36\t842\t0.3\t1\t776 66\n",
      "37\t752\t0.3\t1\t680 72\n",
      "38\t734\t0.3\t1\t674 60\n",
      "39\t600\t0.3\t1\t558 42\n",
      "40\t475\t0.3\t1\t440 35\n",
      "41\t405\t0.3\t1\t378 27\n",
      "42\t336\t0.3\t1\t313 23\n",
      "43\t353\t0.3\t1\t322 31\n",
      "44\t324\t0.3\t1\t298 26\n",
      "45\t281\t0.3\t1\t257 24\n",
      "46\t252\t0.3\t1\t229 23\n",
      "47\t242\t0.3\t1\t211 31\n",
      "48\t214\t0.3\t1\t185 29\n",
      "49\t201\t0.3\t1\t182 19\n",
      "50\t176\t0.3\t1\t155 21\n",
      "51\t143\t0.3\t1\t129 14\n",
      "52\t101\t0.3\t1\t77 24\n",
      "53\t118\t0.3\t1\t100 18\n",
      "54\t86\t0.3\t1\t75 11\n",
      "55\t84\t0.3\t1\t66 18\n",
      "56\t90\t0.3\t1\t51 39\n",
      "57\t66\t0.3\t1\t45 21\n",
      "58\t98\t0.3\t1\t50 48\n",
      "59\t52\t0.3\t1\t38 14\n",
      "60\t63\t0.3\t1\t42 21\n",
      "61\t47\t0.3\t1\t29 18\n",
      "62\t31\t0.3\t1\t11 20\n",
      "63\t36\t0.3\t1\t22 14\n",
      "64\t45\t0.3\t1\t28 17\n",
      "65\t56\t0.3\t1\t16 40\n",
      "66\t45\t0.3\t1\t30 15\n",
      "67\t35\t0.3\t1\t17 18\n",
      "68\t47\t0.3\t1\t21 26\n",
      "69\t93\t0.3\t1\t21 72\n",
      "70\t154\t0.3\t1\t35 119\n",
      "71\t453\t0.3\t1\t61 392\n",
      "72\t256\t0.3\t1\t149 107\n",
      "73\t86\t0.3\t1\t54 32\n",
      "74\t120\t0.3\t1\t18 102\n",
      "75\t19\t0.3\t1\t10 9\n",
      "76\t29\t0.3\t1\t11 18\n",
      "77\t23\t0.3\t1\t2 21\n",
      "78\t12\t0.3\t1\t1 11\n",
      "79\t13\t0.3\t1\t4 9\n",
      "80\t19\t0.3\t1\t1 18\n",
      "81\t15\t0.3\t1\t0 15\n",
      "82\t22\t0.3\t1\t1 21\n",
      "83\t31\t0.3\t1\t0 31\n",
      "84\t9\t0.3\t1\t1 8\n",
      "85\t23\t0.3\t1\t0 23\n",
      "86\t21\t0.3\t1\t1 20\n",
      "87\t22\t0.3\t1\t1 21\n",
      "88\t27\t0.3\t1\t1 26\n",
      "89\t19\t0.3\t1\t0 19\n",
      "90\t10\t0.3\t1\t0 10\n",
      "91\t11\t0.3\t1\t0 11\n",
      "92\t18\t0.3\t1\t1 17\n",
      "93\t7\t0.3\t1\t0 7\n",
      "94\t15\t0.3\t1\t2 13\n",
      "95\t8\t0.3\t1\t0 8\n",
      "96\t24\t0.3\t1\t0 24\n",
      "97\t7\t0.3\t1\t2 5\n",
      "98\t13\t0.3\t1\t0 13\n",
      "99\t8\t0.3\t1\t0 8\n",
      "100\t31\t0.3\t1\t0 31\n",
      "\n",
      "\n",
      "RUN STATISTICS FOR INPUT FILE: fastqs/ERR899424.fastq.gz\n",
      "=============================================\n",
      "18487849 sequences processed in total\n",
      "Sequences removed because they became shorter than the length cutoff of 20 bp:\t4147 (0.0%)\n",
      "\n",
      "This is cutadapt 1.15 with Python 3.6.4\n",
      "Command line parameters: -f fastq -e 0.1 -q 20 -O 1 -a AGATCGGAAGAGC fastqs/ERR899422.fastq.gz\n",
      "Running on 1 core\n",
      "Trimming 1 adapter with at most 10.0% errors in single-end mode ...\n",
      "Finished in 651.89 s (35 us/read; 1.73 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:              18,794,828\n",
      "Reads with adapters:                 6,341,051 (33.7%)\n",
      "Reads written (passing filters):    18,794,828 (100.0%)\n",
      "\n",
      "Total basepairs processed: 1,879,482,800 bp\n",
      "Quality-trimmed:              20,052,894 bp (1.1%)\n",
      "Total written (filtered):  1,845,995,108 bp (98.2%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: AGATCGGAAGAGC; Type: regular 3'; Length: 13; Trimmed: 6341051 times.\n",
      "\n",
      "No. of allowed errors:\n",
      "0-9 bp: 0; 10-13 bp: 1\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "A: 28.2%\n",
      "C: 34.2%\n",
      "G: 19.6%\n",
      "T: 18.0%\n",
      "none/other: 0.0%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "1\t4195191\t4698707.0\t0\t4195191\n",
      "2\t1331776\t1174676.8\t0\t1331776\n",
      "3\t353424\t293669.2\t0\t353424\n",
      "4\t103670\t73417.3\t0\t103670\n",
      "5\t49086\t18354.3\t0\t49086\n",
      "6\t29369\t4588.6\t0\t29369\n",
      "7\t25328\t1147.1\t0\t25328\n",
      "8\t22232\t286.8\t0\t22232\n",
      "9\t20621\t71.7\t0\t20323 298\n",
      "10\t19515\t17.9\t1\t18136 1379\n",
      "11\t17596\t4.5\t1\t16447 1149\n",
      "12\t16194\t1.1\t1\t15296 898\n",
      "13\t14614\t0.3\t1\t13835 779\n",
      "14\t13674\t0.3\t1\t12936 738\n",
      "15\t12928\t0.3\t1\t12240 688\n",
      "16\t11923\t0.3\t1\t11263 660\n",
      "17\t10893\t0.3\t1\t10269 624\n",
      "18\t9593\t0.3\t1\t9055 538\n",
      "19\t7772\t0.3\t1\t7338 434\n",
      "20\t7078\t0.3\t1\t6679 399\n",
      "21\t6415\t0.3\t1\t6034 381\n",
      "22\t5857\t0.3\t1\t5519 338\n",
      "23\t5246\t0.3\t1\t4919 327\n",
      "24\t5018\t0.3\t1\t4717 301\n",
      "25\t4588\t0.3\t1\t4334 254\n",
      "26\t4266\t0.3\t1\t4033 233\n",
      "27\t3789\t0.3\t1\t3554 235\n",
      "28\t3343\t0.3\t1\t3161 182\n",
      "29\t2927\t0.3\t1\t2722 205\n",
      "30\t2641\t0.3\t1\t2442 199\n",
      "31\t2104\t0.3\t1\t1980 124\n",
      "32\t1988\t0.3\t1\t1875 113\n",
      "33\t1724\t0.3\t1\t1619 105\n",
      "34\t1677\t0.3\t1\t1584 93\n",
      "35\t1477\t0.3\t1\t1376 101\n",
      "36\t1389\t0.3\t1\t1316 73\n",
      "37\t1358\t0.3\t1\t1274 84\n",
      "38\t1123\t0.3\t1\t1053 70\n",
      "39\t961\t0.3\t1\t907 54\n",
      "40\t829\t0.3\t1\t769 60\n",
      "41\t754\t0.3\t1\t714 40\n",
      "42\t648\t0.3\t1\t595 53\n",
      "43\t568\t0.3\t1\t532 36\n",
      "44\t550\t0.3\t1\t509 41\n",
      "45\t509\t0.3\t1\t471 38\n",
      "46\t474\t0.3\t1\t434 40\n",
      "47\t485\t0.3\t1\t425 60\n",
      "48\t443\t0.3\t1\t412 31\n",
      "49\t368\t0.3\t1\t330 38\n",
      "50\t322\t0.3\t1\t296 26\n",
      "51\t237\t0.3\t1\t221 16\n",
      "52\t224\t0.3\t1\t200 24\n",
      "53\t176\t0.3\t1\t152 24\n",
      "54\t156\t0.3\t1\t139 17\n",
      "55\t165\t0.3\t1\t154 11\n",
      "56\t192\t0.3\t1\t135 57\n",
      "57\t145\t0.3\t1\t117 28\n",
      "58\t198\t0.3\t1\t119 79\n",
      "59\t125\t0.3\t1\t92 33\n",
      "60\t124\t0.3\t1\t97 27\n",
      "61\t81\t0.3\t1\t69 12\n",
      "62\t71\t0.3\t1\t49 22\n",
      "63\t67\t0.3\t1\t42 25\n",
      "64\t84\t0.3\t1\t38 46\n",
      "65\t91\t0.3\t1\t48 43\n",
      "66\t58\t0.3\t1\t31 27\n",
      "67\t67\t0.3\t1\t30 37\n",
      "68\t94\t0.3\t1\t38 56\n",
      "69\t138\t0.3\t1\t32 106\n",
      "70\t282\t0.3\t1\t49 233\n",
      "71\t737\t0.3\t1\t42 695\n",
      "72\t304\t0.3\t1\t125 179\n",
      "73\t112\t0.3\t1\t46 66\n",
      "74\t190\t0.3\t1\t28 162\n",
      "75\t37\t0.3\t1\t18 19\n",
      "76\t24\t0.3\t1\t6 18\n",
      "77\t39\t0.3\t1\t9 30\n",
      "78\t27\t0.3\t1\t1 26\n",
      "79\t20\t0.3\t1\t1 19\n",
      "80\t13\t0.3\t1\t0 13\n",
      "81\t16\t0.3\t1\t0 16\n",
      "82\t30\t0.3\t1\t0 30\n",
      "83\t68\t0.3\t1\t1 67\n",
      "84\t14\t0.3\t1\t0 14\n",
      "85\t19\t0.3\t1\t1 18\n",
      "86\t26\t0.3\t1\t1 25\n",
      "87\t10\t0.3\t1\t0 10\n",
      "88\t29\t0.3\t1\t1 28\n",
      "89\t25\t0.3\t1\t0 25\n",
      "90\t21\t0.3\t1\t0 21\n",
      "91\t15\t0.3\t1\t1 14\n",
      "92\t15\t0.3\t1\t0 15\n",
      "93\t13\t0.3\t1\t1 12\n",
      "94\t9\t0.3\t1\t0 9\n",
      "95\t13\t0.3\t1\t1 12\n",
      "96\t18\t0.3\t1\t0 18\n",
      "97\t17\t0.3\t1\t1 16\n",
      "98\t4\t0.3\t1\t1 3\n",
      "99\t13\t0.3\t1\t0 13\n",
      "100\t80\t0.3\t1\t2 78\n",
      "\n",
      "\n",
      "RUN STATISTICS FOR INPUT FILE: fastqs/ERR899422.fastq.gz\n",
      "=============================================\n",
      "18794828 sequences processed in total\n",
      "Sequences removed because they became shorter than the length cutoff of 20 bp:\t5487 (0.0%)\n",
      "\n",
      "This is cutadapt 1.15 with Python 3.6.4\n",
      "Command line parameters: -f fastq -e 0.1 -q 20 -O 1 -a AGATCGGAAGAGC fastqs/ERR899437.fastq.gz\n",
      "Running on 1 core\n",
      "Trimming 1 adapter with at most 10.0% errors in single-end mode ...\n",
      "Finished in 653.84 s (35 us/read; 1.71 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:              18,655,823\n",
      "Reads with adapters:                 6,254,086 (33.5%)\n",
      "Reads written (passing filters):    18,655,823 (100.0%)\n",
      "\n",
      "Total basepairs processed: 1,865,582,300 bp\n",
      "Quality-trimmed:              20,782,476 bp (1.1%)\n",
      "Total written (filtered):  1,830,909,402 bp (98.1%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: AGATCGGAAGAGC; Type: regular 3'; Length: 13; Trimmed: 6254086 times.\n",
      "\n",
      "No. of allowed errors:\n",
      "0-9 bp: 0; 10-13 bp: 1\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "A: 27.5%\n",
      "C: 34.4%\n",
      "G: 20.0%\n",
      "T: 18.0%\n",
      "none/other: 0.0%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "1\t4093628\t4663955.8\t0\t4093628\n",
      "2\t1317901\t1165988.9\t0\t1317901\n",
      "3\t348596\t291497.2\t0\t348596\n",
      "4\t102889\t72874.3\t0\t102889\n",
      "5\t49133\t18218.6\t0\t49133\n",
      "6\t30215\t4554.6\t0\t30215\n",
      "7\t26566\t1138.7\t0\t26566\n",
      "8\t23772\t284.7\t0\t23772\n",
      "9\t22922\t71.2\t0\t22665 257\n",
      "10\t20763\t17.8\t1\t20040 723\n",
      "11\t19057\t4.4\t1\t18525 532\n",
      "12\t17704\t1.1\t1\t17307 397\n",
      "13\t16030\t0.3\t1\t15759 271\n",
      "14\t15276\t0.3\t1\t14976 300\n",
      "15\t14347\t0.3\t1\t14066 281\n",
      "16\t13373\t0.3\t1\t13097 276\n",
      "17\t12258\t0.3\t1\t12013 245\n",
      "18\t11225\t0.3\t1\t11017 208\n",
      "19\t9373\t0.3\t1\t9218 155\n",
      "20\t8692\t0.3\t1\t8505 187\n",
      "21\t7853\t0.3\t1\t7659 194\n",
      "22\t7153\t0.3\t1\t6996 157\n",
      "23\t6369\t0.3\t1\t6242 127\n",
      "24\t5980\t0.3\t1\t5837 143\n",
      "25\t5591\t0.3\t1\t5461 130\n",
      "26\t5080\t0.3\t1\t4981 99\n",
      "27\t4545\t0.3\t1\t4429 116\n",
      "28\t4243\t0.3\t1\t4143 100\n",
      "29\t3528\t0.3\t1\t3440 88\n",
      "30\t3231\t0.3\t1\t3124 107\n",
      "31\t2562\t0.3\t1\t2510 52\n",
      "32\t2351\t0.3\t1\t2300 51\n",
      "33\t2217\t0.3\t1\t2164 53\n",
      "34\t1935\t0.3\t1\t1879 56\n",
      "35\t1832\t0.3\t1\t1777 55\n",
      "36\t1587\t0.3\t1\t1545 42\n",
      "37\t1514\t0.3\t1\t1458 56\n",
      "38\t1338\t0.3\t1\t1302 36\n",
      "39\t1143\t0.3\t1\t1119 24\n",
      "40\t960\t0.3\t1\t932 28\n",
      "41\t802\t0.3\t1\t776 26\n",
      "42\t678\t0.3\t1\t658 20\n",
      "43\t577\t0.3\t1\t551 26\n",
      "44\t515\t0.3\t1\t496 19\n",
      "45\t456\t0.3\t1\t440 16\n",
      "46\t409\t0.3\t1\t388 21\n",
      "47\t388\t0.3\t1\t359 29\n",
      "48\t369\t0.3\t1\t347 22\n",
      "49\t242\t0.3\t1\t236 6\n",
      "50\t258\t0.3\t1\t233 25\n",
      "51\t219\t0.3\t1\t201 18\n",
      "52\t201\t0.3\t1\t172 29\n",
      "53\t166\t0.3\t1\t152 14\n",
      "54\t122\t0.3\t1\t112 10\n",
      "55\t102\t0.3\t1\t92 10\n",
      "56\t125\t0.3\t1\t83 42\n",
      "57\t100\t0.3\t1\t72 28\n",
      "58\t133\t0.3\t1\t69 64\n",
      "59\t102\t0.3\t1\t71 31\n",
      "60\t85\t0.3\t1\t63 22\n",
      "61\t70\t0.3\t1\t53 17\n",
      "62\t94\t0.3\t1\t31 63\n",
      "63\t116\t0.3\t1\t48 68\n",
      "64\t66\t0.3\t1\t41 25\n",
      "65\t62\t0.3\t1\t27 35\n",
      "66\t92\t0.3\t1\t24 68\n",
      "67\t187\t0.3\t1\t25 162\n",
      "68\t314\t0.3\t1\t55 259\n",
      "69\t902\t0.3\t1\t65 837\n",
      "70\t411\t0.3\t1\t148 263\n",
      "71\t155\t0.3\t1\t60 95\n",
      "72\t70\t0.3\t1\t25 45\n",
      "73\t33\t0.3\t1\t9 24\n",
      "74\t137\t0.3\t1\t6 131\n",
      "75\t19\t0.3\t1\t10 9\n",
      "76\t9\t0.3\t1\t2 7\n",
      "77\t19\t0.3\t1\t3 16\n",
      "78\t23\t0.3\t1\t1 22\n",
      "79\t14\t0.3\t1\t0 14\n",
      "80\t15\t0.3\t1\t1 14\n",
      "81\t23\t0.3\t1\t0 23\n",
      "82\t26\t0.3\t1\t1 25\n",
      "83\t51\t0.3\t1\t0 51\n",
      "84\t13\t0.3\t1\t1 12\n",
      "85\t20\t0.3\t1\t0 20\n",
      "86\t25\t0.3\t1\t1 24\n",
      "87\t10\t0.3\t1\t1 9\n",
      "88\t33\t0.3\t1\t1 32\n",
      "89\t32\t0.3\t1\t0 32\n",
      "90\t24\t0.3\t1\t3 21\n",
      "91\t18\t0.3\t1\t1 17\n",
      "92\t15\t0.3\t1\t0 15\n",
      "93\t11\t0.3\t1\t0 11\n",
      "94\t10\t0.3\t1\t0 10\n",
      "95\t20\t0.3\t1\t4 16\n",
      "96\t20\t0.3\t1\t1 19\n",
      "97\t18\t0.3\t1\t0 18\n",
      "98\t11\t0.3\t1\t0 11\n",
      "99\t16\t0.3\t1\t0 16\n",
      "100\t101\t0.3\t1\t0 101\n",
      "\n",
      "\n",
      "RUN STATISTICS FOR INPUT FILE: fastqs/ERR899437.fastq.gz\n",
      "=============================================\n",
      "18655823 sequences processed in total\n",
      "Sequences removed because they became shorter than the length cutoff of 20 bp:\t6969 (0.0%)\n",
      "\n",
      "This is cutadapt 1.15 with Python 3.6.4\n",
      "Command line parameters: -f fastq -e 0.1 -q 20 -O 1 -a AGATCGGAAGAGC fastqs/ERR899430.fastq.gz\n",
      "Running on 1 core\n",
      "Trimming 1 adapter with at most 10.0% errors in single-end mode ...\n",
      "Finished in 656.66 s (35 us/read; 1.73 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:              18,934,244\n",
      "Reads with adapters:                 6,335,884 (33.5%)\n",
      "Reads written (passing filters):    18,934,244 (100.0%)\n",
      "\n",
      "Total basepairs processed: 1,893,424,400 bp\n",
      "Quality-trimmed:              19,955,657 bp (1.1%)\n",
      "Total written (filtered):  1,861,109,108 bp (98.3%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: AGATCGGAAGAGC; Type: regular 3'; Length: 13; Trimmed: 6335884 times.\n",
      "\n",
      "No. of allowed errors:\n",
      "0-9 bp: 0; 10-13 bp: 1\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "A: 28.5%\n",
      "C: 34.2%\n",
      "G: 19.4%\n",
      "T: 17.9%\n",
      "none/other: 0.0%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "1\t4248667\t4733561.0\t0\t4248667\n",
      "2\t1347888\t1183390.2\t0\t1347888\n",
      "3\t352692\t295847.6\t0\t352692\n",
      "4\t99778\t73961.9\t0\t99778\n",
      "5\t44682\t18490.5\t0\t44682\n",
      "6\t25239\t4622.6\t0\t25239\n",
      "7\t21626\t1155.7\t0\t21626\n",
      "8\t18755\t288.9\t0\t18755\n",
      "9\t17286\t72.2\t0\t16965 321\n",
      "10\t15838\t18.1\t1\t14982 856\n",
      "11\t14023\t4.5\t1\t13349 674\n",
      "12\t12803\t1.1\t1\t12365 438\n",
      "13\t11311\t0.3\t1\t10960 351\n",
      "14\t10773\t0.3\t1\t10441 332\n",
      "15\t9693\t0.3\t1\t9399 294\n",
      "16\t8993\t0.3\t1\t8711 282\n",
      "17\t7729\t0.3\t1\t7469 260\n",
      "18\t7204\t0.3\t1\t6934 270\n",
      "19\t5818\t0.3\t1\t5626 192\n",
      "20\t5260\t0.3\t1\t5049 211\n",
      "21\t4792\t0.3\t1\t4619 173\n",
      "22\t4389\t0.3\t1\t4242 147\n",
      "23\t3776\t0.3\t1\t3646 130\n",
      "24\t3623\t0.3\t1\t3511 112\n",
      "25\t3388\t0.3\t1\t3253 135\n",
      "26\t2934\t0.3\t1\t2834 100\n",
      "27\t2665\t0.3\t1\t2556 109\n",
      "28\t2412\t0.3\t1\t2321 91\n",
      "29\t2066\t0.3\t1\t1982 84\n",
      "30\t1881\t0.3\t1\t1787 94\n",
      "31\t1432\t0.3\t1\t1377 55\n",
      "32\t1282\t0.3\t1\t1242 40\n",
      "33\t1221\t0.3\t1\t1185 36\n",
      "34\t1189\t0.3\t1\t1143 46\n",
      "35\t1046\t0.3\t1\t987 59\n",
      "36\t898\t0.3\t1\t854 44\n",
      "37\t847\t0.3\t1\t795 52\n",
      "38\t805\t0.3\t1\t760 45\n",
      "39\t692\t0.3\t1\t654 38\n",
      "40\t567\t0.3\t1\t525 42\n",
      "41\t504\t0.3\t1\t475 29\n",
      "42\t443\t0.3\t1\t420 23\n",
      "43\t381\t0.3\t1\t353 28\n",
      "44\t383\t0.3\t1\t360 23\n",
      "45\t320\t0.3\t1\t301 19\n",
      "46\t343\t0.3\t1\t311 32\n",
      "47\t298\t0.3\t1\t263 35\n",
      "48\t295\t0.3\t1\t265 30\n",
      "49\t247\t0.3\t1\t221 26\n",
      "50\t203\t0.3\t1\t183 20\n",
      "51\t173\t0.3\t1\t166 7\n",
      "52\t141\t0.3\t1\t120 21\n",
      "53\t125\t0.3\t1\t118 7\n",
      "54\t106\t0.3\t1\t94 12\n",
      "55\t124\t0.3\t1\t109 15\n",
      "56\t132\t0.3\t1\t87 45\n",
      "57\t104\t0.3\t1\t88 16\n",
      "58\t133\t0.3\t1\t77 56\n",
      "59\t92\t0.3\t1\t67 25\n",
      "60\t94\t0.3\t1\t68 26\n",
      "61\t49\t0.3\t1\t34 15\n",
      "62\t56\t0.3\t1\t41 15\n",
      "63\t43\t0.3\t1\t21 22\n",
      "64\t73\t0.3\t1\t23 50\n",
      "65\t75\t0.3\t1\t27 48\n",
      "66\t49\t0.3\t1\t29 20\n",
      "67\t53\t0.3\t1\t20 33\n",
      "68\t74\t0.3\t1\t15 59\n",
      "69\t165\t0.3\t1\t28 137\n",
      "70\t301\t0.3\t1\t44 257\n",
      "71\t938\t0.3\t1\t71 867\n",
      "72\t442\t0.3\t1\t201 241\n",
      "73\t137\t0.3\t1\t63 74\n",
      "74\t182\t0.3\t1\t25 157\n",
      "75\t26\t0.3\t1\t10 16\n",
      "76\t18\t0.3\t1\t5 13\n",
      "77\t36\t0.3\t1\t4 32\n",
      "78\t31\t0.3\t1\t1 30\n",
      "79\t34\t0.3\t1\t5 29\n",
      "80\t26\t0.3\t1\t2 24\n",
      "81\t20\t0.3\t1\t0 20\n",
      "82\t25\t0.3\t1\t1 24\n",
      "83\t50\t0.3\t1\t0 50\n",
      "84\t15\t0.3\t1\t1 14\n",
      "85\t13\t0.3\t1\t1 12\n",
      "86\t29\t0.3\t1\t0 29\n",
      "87\t25\t0.3\t1\t1 24\n",
      "88\t29\t0.3\t1\t1 28\n",
      "89\t27\t0.3\t1\t0 27\n",
      "90\t20\t0.3\t1\t1 19\n",
      "91\t23\t0.3\t1\t1 22\n",
      "92\t20\t0.3\t1\t2 18\n",
      "93\t9\t0.3\t1\t1 8\n",
      "94\t13\t0.3\t1\t1 12\n",
      "95\t13\t0.3\t1\t1 12\n",
      "96\t20\t0.3\t1\t1 19\n",
      "97\t12\t0.3\t1\t2 10\n",
      "98\t15\t0.3\t1\t2 13\n",
      "99\t11\t0.3\t1\t0 11\n",
      "100\t83\t0.3\t1\t0 83\n",
      "\n",
      "\n",
      "RUN STATISTICS FOR INPUT FILE: fastqs/ERR899430.fastq.gz\n",
      "=============================================\n",
      "18934244 sequences processed in total\n",
      "Sequences removed because they became shorter than the length cutoff of 20 bp:\t5394 (0.0%)\n",
      "\n",
      "This is cutadapt 1.15 with Python 3.6.4\n",
      "Command line parameters: -f fastq -e 0.1 -q 20 -O 1 -a AGATCGGAAGAGC fastqs/ERR899426.fastq.gz\n",
      "Running on 1 core\n",
      "Trimming 1 adapter with at most 10.0% errors in single-end mode ...\n",
      "Finished in 668.22 s (34 us/read; 1.74 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:              19,410,496\n",
      "Reads with adapters:                 6,460,097 (33.3%)\n",
      "Reads written (passing filters):    19,410,496 (100.0%)\n",
      "\n",
      "Total basepairs processed: 1,941,049,600 bp\n",
      "Quality-trimmed:              22,887,356 bp (1.2%)\n",
      "Total written (filtered):  1,902,689,259 bp (98.0%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: AGATCGGAAGAGC; Type: regular 3'; Length: 13; Trimmed: 6460097 times.\n",
      "\n",
      "No. of allowed errors:\n",
      "0-9 bp: 0; 10-13 bp: 1\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "A: 26.8%\n",
      "C: 34.1%\n",
      "G: 20.6%\n",
      "T: 18.5%\n",
      "none/other: 0.0%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "1\t4093893\t4852624.0\t0\t4093893\n",
      "2\t1374662\t1213156.0\t0\t1374662\n",
      "3\t375599\t303289.0\t0\t375599\n",
      "4\t115140\t75822.2\t0\t115140\n",
      "5\t60897\t18955.6\t0\t60897\n",
      "6\t40774\t4738.9\t0\t40774\n",
      "7\t36432\t1184.7\t0\t36432\n",
      "8\t32165\t296.2\t0\t32165\n",
      "9\t30655\t74.0\t0\t30451 204\n",
      "10\t27478\t18.5\t1\t26718 760\n",
      "11\t24435\t4.6\t1\t23826 609\n",
      "12\t23047\t1.2\t1\t22610 437\n",
      "13\t21044\t0.3\t1\t20671 373\n",
      "14\t20002\t0.3\t1\t19647 355\n",
      "15\t19024\t0.3\t1\t18694 330\n",
      "16\t17429\t0.3\t1\t17106 323\n",
      "17\t15766\t0.3\t1\t15464 302\n",
      "18\t13903\t0.3\t1\t13658 245\n",
      "19\t11623\t0.3\t1\t11403 220\n",
      "20\t10632\t0.3\t1\t10380 252\n",
      "21\t9501\t0.3\t1\t9315 186\n",
      "22\t9020\t0.3\t1\t8824 196\n",
      "23\t7947\t0.3\t1\t7802 145\n",
      "24\t7393\t0.3\t1\t7242 151\n",
      "25\t6766\t0.3\t1\t6642 124\n",
      "26\t6248\t0.3\t1\t6146 102\n",
      "27\t5410\t0.3\t1\t5274 136\n",
      "28\t4931\t0.3\t1\t4823 108\n",
      "29\t4218\t0.3\t1\t4137 81\n",
      "30\t3926\t0.3\t1\t3812 114\n",
      "31\t3170\t0.3\t1\t3093 77\n",
      "32\t2866\t0.3\t1\t2793 73\n",
      "33\t2589\t0.3\t1\t2541 48\n",
      "34\t2257\t0.3\t1\t2205 52\n",
      "35\t2202\t0.3\t1\t2144 58\n",
      "36\t2050\t0.3\t1\t1997 53\n",
      "37\t1842\t0.3\t1\t1788 54\n",
      "38\t1612\t0.3\t1\t1564 48\n",
      "39\t1331\t0.3\t1\t1307 24\n",
      "40\t1105\t0.3\t1\t1080 25\n",
      "41\t931\t0.3\t1\t901 30\n",
      "42\t826\t0.3\t1\t798 28\n",
      "43\t717\t0.3\t1\t700 17\n",
      "44\t659\t0.3\t1\t625 34\n",
      "45\t578\t0.3\t1\t561 17\n",
      "46\t557\t0.3\t1\t537 20\n",
      "47\t497\t0.3\t1\t461 36\n",
      "48\t430\t0.3\t1\t398 32\n",
      "49\t339\t0.3\t1\t320 19\n",
      "50\t288\t0.3\t1\t275 13\n",
      "51\t227\t0.3\t1\t215 12\n",
      "52\t226\t0.3\t1\t207 19\n",
      "53\t184\t0.3\t1\t168 16\n",
      "54\t127\t0.3\t1\t120 7\n",
      "55\t114\t0.3\t1\t100 14\n",
      "56\t161\t0.3\t1\t125 36\n",
      "57\t122\t0.3\t1\t105 17\n",
      "58\t129\t0.3\t1\t100 29\n",
      "59\t105\t0.3\t1\t88 17\n",
      "60\t74\t0.3\t1\t59 15\n",
      "61\t60\t0.3\t1\t49 11\n",
      "62\t68\t0.3\t1\t44 24\n",
      "63\t65\t0.3\t1\t43 22\n",
      "64\t53\t0.3\t1\t42 11\n",
      "65\t34\t0.3\t1\t14 20\n",
      "66\t51\t0.3\t1\t25 26\n",
      "67\t75\t0.3\t1\t30 45\n",
      "68\t120\t0.3\t1\t44 76\n",
      "69\t337\t0.3\t1\t77 260\n",
      "70\t227\t0.3\t1\t145 82\n",
      "71\t91\t0.3\t1\t44 47\n",
      "72\t44\t0.3\t1\t21 23\n",
      "73\t33\t0.3\t1\t14 19\n",
      "74\t116\t0.3\t1\t6 110\n",
      "75\t15\t0.3\t1\t6 9\n",
      "76\t16\t0.3\t1\t4 12\n",
      "77\t17\t0.3\t1\t2 15\n",
      "78\t17\t0.3\t1\t6 11\n",
      "79\t11\t0.3\t1\t1 10\n",
      "80\t11\t0.3\t1\t0 11\n",
      "81\t12\t0.3\t1\t1 11\n",
      "82\t15\t0.3\t1\t0 15\n",
      "83\t40\t0.3\t1\t0 40\n",
      "84\t10\t0.3\t1\t2 8\n",
      "85\t16\t0.3\t1\t2 14\n",
      "86\t25\t0.3\t1\t2 23\n",
      "87\t12\t0.3\t1\t0 12\n",
      "88\t35\t0.3\t1\t0 35\n",
      "89\t31\t0.3\t1\t1 30\n",
      "90\t11\t0.3\t1\t1 10\n",
      "91\t20\t0.3\t1\t2 18\n",
      "92\t14\t0.3\t1\t1 13\n",
      "93\t8\t0.3\t1\t0 8\n",
      "94\t8\t0.3\t1\t0 8\n",
      "95\t8\t0.3\t1\t0 8\n",
      "96\t24\t0.3\t1\t1 23\n",
      "97\t14\t0.3\t1\t0 14\n",
      "98\t16\t0.3\t1\t1 15\n",
      "99\t2\t0.3\t1\t0 2\n",
      "100\t40\t0.3\t1\t0 40\n",
      "\n",
      "\n",
      "RUN STATISTICS FOR INPUT FILE: fastqs/ERR899426.fastq.gz\n",
      "=============================================\n",
      "19410496 sequences processed in total\n",
      "Sequences removed because they became shorter than the length cutoff of 20 bp:\t4965 (0.0%)\n",
      "\n",
      "This is cutadapt 1.15 with Python 3.6.4\n",
      "Command line parameters: -f fastq -e 0.1 -q 20 -O 1 -a AGATCGGAAGAGC fastqs/ERR899429.fastq.gz\n",
      "Running on 1 core\n",
      "Trimming 1 adapter with at most 10.0% errors in single-end mode ...\n",
      "Finished in 674.21 s (35 us/read; 1.72 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:              19,286,135\n",
      "Reads with adapters:                 6,570,339 (34.1%)\n",
      "Reads written (passing filters):    19,286,135 (100.0%)\n",
      "\n",
      "Total basepairs processed: 1,928,613,500 bp\n",
      "Quality-trimmed:              20,048,306 bp (1.0%)\n",
      "Total written (filtered):  1,893,114,468 bp (98.2%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: AGATCGGAAGAGC; Type: regular 3'; Length: 13; Trimmed: 6570339 times.\n",
      "\n",
      "No. of allowed errors:\n",
      "0-9 bp: 0; 10-13 bp: 1\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "A: 27.5%\n",
      "C: 34.4%\n",
      "G: 20.1%\n",
      "T: 18.1%\n",
      "none/other: 0.0%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "1\t4245820\t4821533.8\t0\t4245820\n",
      "2\t1372805\t1205383.4\t0\t1372805\n",
      "3\t364128\t301345.9\t0\t364128\n",
      "4\t111033\t75336.5\t0\t111033\n",
      "5\t56311\t18834.1\t0\t56311\n",
      "6\t36293\t4708.5\t0\t36293\n",
      "7\t32495\t1177.1\t0\t32495\n",
      "8\t28701\t294.3\t0\t28701\n",
      "9\t27139\t73.6\t0\t26830 309\n",
      "10\t25829\t18.4\t1\t24609 1220\n",
      "11\t23270\t4.6\t1\t22308 962\n",
      "12\t22111\t1.1\t1\t21346 765\n",
      "13\t19969\t0.3\t1\t19297 672\n",
      "14\t18741\t0.3\t1\t18038 703\n",
      "15\t17796\t0.3\t1\t17186 610\n",
      "16\t16538\t0.3\t1\t15944 594\n",
      "17\t15340\t0.3\t1\t14753 587\n",
      "18\t13837\t0.3\t1\t13331 506\n",
      "19\t11485\t0.3\t1\t11043 442\n",
      "20\t10754\t0.3\t1\t10350 404\n",
      "21\t9696\t0.3\t1\t9351 345\n",
      "22\t8901\t0.3\t1\t8554 347\n",
      "23\t8121\t0.3\t1\t7856 265\n",
      "24\t7578\t0.3\t1\t7271 307\n",
      "25\t7133\t0.3\t1\t6866 267\n",
      "26\t6607\t0.3\t1\t6359 248\n",
      "27\t5808\t0.3\t1\t5571 237\n",
      "28\t5192\t0.3\t1\t4984 208\n",
      "29\t4609\t0.3\t1\t4432 177\n",
      "30\t4186\t0.3\t1\t3964 222\n",
      "31\t3507\t0.3\t1\t3358 149\n",
      "32\t3048\t0.3\t1\t2917 131\n",
      "33\t2742\t0.3\t1\t2635 107\n",
      "34\t2577\t0.3\t1\t2463 114\n",
      "35\t2430\t0.3\t1\t2328 102\n",
      "36\t2240\t0.3\t1\t2148 92\n",
      "37\t1910\t0.3\t1\t1821 89\n",
      "38\t1629\t0.3\t1\t1551 78\n",
      "39\t1421\t0.3\t1\t1362 59\n",
      "40\t1170\t0.3\t1\t1116 54\n",
      "41\t1026\t0.3\t1\t985 41\n",
      "42\t880\t0.3\t1\t841 39\n",
      "43\t775\t0.3\t1\t748 27\n",
      "44\t678\t0.3\t1\t629 49\n",
      "45\t658\t0.3\t1\t627 31\n",
      "46\t624\t0.3\t1\t593 31\n",
      "47\t490\t0.3\t1\t438 52\n",
      "48\t458\t0.3\t1\t429 29\n",
      "49\t373\t0.3\t1\t349 24\n",
      "50\t306\t0.3\t1\t286 20\n",
      "51\t277\t0.3\t1\t260 17\n",
      "52\t236\t0.3\t1\t219 17\n",
      "53\t186\t0.3\t1\t159 27\n",
      "54\t152\t0.3\t1\t142 10\n",
      "55\t132\t0.3\t1\t113 19\n",
      "56\t173\t0.3\t1\t117 56\n",
      "57\t137\t0.3\t1\t111 26\n",
      "58\t161\t0.3\t1\t113 48\n",
      "59\t102\t0.3\t1\t73 29\n",
      "60\t80\t0.3\t1\t54 26\n",
      "61\t83\t0.3\t1\t62 21\n",
      "62\t52\t0.3\t1\t32 20\n",
      "63\t65\t0.3\t1\t48 17\n",
      "64\t63\t0.3\t1\t55 8\n",
      "65\t42\t0.3\t1\t24 18\n",
      "66\t42\t0.3\t1\t19 23\n",
      "67\t53\t0.3\t1\t24 29\n",
      "68\t62\t0.3\t1\t35 27\n",
      "69\t127\t0.3\t1\t51 76\n",
      "70\t190\t0.3\t1\t148 42\n",
      "71\t102\t0.3\t1\t63 39\n",
      "72\t36\t0.3\t1\t19 17\n",
      "73\t21\t0.3\t1\t12 9\n",
      "74\t153\t0.3\t1\t16 137\n",
      "75\t16\t0.3\t1\t6 10\n",
      "76\t11\t0.3\t1\t2 9\n",
      "77\t11\t0.3\t1\t0 11\n",
      "78\t18\t0.3\t1\t1 17\n",
      "79\t14\t0.3\t1\t2 12\n",
      "80\t11\t0.3\t1\t0 11\n",
      "81\t10\t0.3\t1\t2 8\n",
      "82\t28\t0.3\t1\t2 26\n",
      "83\t44\t0.3\t1\t2 42\n",
      "84\t11\t0.3\t1\t2 9\n",
      "85\t29\t0.3\t1\t2 27\n",
      "86\t28\t0.3\t1\t2 26\n",
      "87\t15\t0.3\t1\t0 15\n",
      "88\t31\t0.3\t1\t1 30\n",
      "89\t31\t0.3\t1\t3 28\n",
      "90\t17\t0.3\t1\t0 17\n",
      "91\t10\t0.3\t1\t0 10\n",
      "92\t19\t0.3\t1\t1 18\n",
      "93\t6\t0.3\t1\t0 6\n",
      "94\t8\t0.3\t1\t0 8\n",
      "95\t15\t0.3\t1\t1 14\n",
      "96\t17\t0.3\t1\t0 17\n",
      "97\t13\t0.3\t1\t2 11\n",
      "98\t7\t0.3\t1\t2 5\n",
      "99\t3\t0.3\t1\t1 2\n",
      "100\t21\t0.3\t1\t0 21\n",
      "\n",
      "\n",
      "RUN STATISTICS FOR INPUT FILE: fastqs/ERR899429.fastq.gz\n",
      "=============================================\n",
      "19286135 sequences processed in total\n",
      "Sequences removed because they became shorter than the length cutoff of 20 bp:\t3490 (0.0%)\n",
      "\n",
      "This is cutadapt 1.15 with Python 3.6.4\n",
      "Command line parameters: -f fastq -e 0.1 -q 20 -O 1 -a AGATCGGAAGAGC fastqs/ERR899431.fastq.gz\n",
      "Running on 1 core\n",
      "Trimming 1 adapter with at most 10.0% errors in single-end mode ...\n",
      "Finished in 680.47 s (35 us/read; 1.71 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:              19,419,534\n",
      "Reads with adapters:                 6,524,686 (33.6%)\n",
      "Reads written (passing filters):    19,419,534 (100.0%)\n",
      "\n",
      "Total basepairs processed: 1,941,953,400 bp\n",
      "Quality-trimmed:              20,129,271 bp (1.0%)\n",
      "Total written (filtered):  1,908,473,401 bp (98.3%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: AGATCGGAAGAGC; Type: regular 3'; Length: 13; Trimmed: 6524686 times.\n",
      "\n",
      "No. of allowed errors:\n",
      "0-9 bp: 0; 10-13 bp: 1\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "A: 28.3%\n",
      "C: 34.1%\n",
      "G: 19.8%\n",
      "T: 17.8%\n",
      "none/other: 0.0%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "1\t4331832\t4854883.5\t0\t4331832\n",
      "2\t1380178\t1213720.9\t0\t1380178\n",
      "3\t363263\t303430.2\t0\t363263\n",
      "4\t104304\t75857.6\t0\t104304\n",
      "5\t48442\t18964.4\t0\t48442\n",
      "6\t28512\t4741.1\t0\t28512\n",
      "7\t24806\t1185.3\t0\t24806\n",
      "8\t22042\t296.3\t0\t22042\n",
      "9\t20509\t74.1\t0\t20201 308\n",
      "10\t19132\t18.5\t1\t18223 909\n",
      "11\t17136\t4.6\t1\t16392 744\n",
      "12\t16188\t1.2\t1\t15703 485\n",
      "13\t14059\t0.3\t1\t13669 390\n",
      "14\t13349\t0.3\t1\t12961 388\n",
      "15\t12388\t0.3\t1\t12017 371\n",
      "16\t11426\t0.3\t1\t11093 333\n",
      "17\t10204\t0.3\t1\t9884 320\n",
      "18\t9331\t0.3\t1\t9080 251\n",
      "19\t7583\t0.3\t1\t7372 211\n",
      "20\t7318\t0.3\t1\t7040 278\n",
      "21\t6496\t0.3\t1\t6275 221\n",
      "22\t5793\t0.3\t1\t5597 196\n",
      "23\t5161\t0.3\t1\t4992 169\n",
      "24\t4875\t0.3\t1\t4718 157\n",
      "25\t4482\t0.3\t1\t4327 155\n",
      "26\t3972\t0.3\t1\t3839 133\n",
      "27\t3621\t0.3\t1\t3489 132\n",
      "28\t3187\t0.3\t1\t3094 93\n",
      "29\t2760\t0.3\t1\t2672 88\n",
      "30\t2534\t0.3\t1\t2404 130\n",
      "31\t2080\t0.3\t1\t2006 74\n",
      "32\t1915\t0.3\t1\t1843 72\n",
      "33\t1648\t0.3\t1\t1606 42\n",
      "34\t1564\t0.3\t1\t1511 53\n",
      "35\t1386\t0.3\t1\t1336 50\n",
      "36\t1256\t0.3\t1\t1203 53\n",
      "37\t1099\t0.3\t1\t1045 54\n",
      "38\t968\t0.3\t1\t926 42\n",
      "39\t781\t0.3\t1\t747 34\n",
      "40\t734\t0.3\t1\t685 49\n",
      "41\t591\t0.3\t1\t554 37\n",
      "42\t522\t0.3\t1\t502 20\n",
      "43\t450\t0.3\t1\t428 22\n",
      "44\t416\t0.3\t1\t393 23\n",
      "45\t395\t0.3\t1\t364 31\n",
      "46\t351\t0.3\t1\t334 17\n",
      "47\t301\t0.3\t1\t272 29\n",
      "48\t277\t0.3\t1\t261 16\n",
      "49\t267\t0.3\t1\t252 15\n",
      "50\t216\t0.3\t1\t195 21\n",
      "51\t160\t0.3\t1\t146 14\n",
      "52\t161\t0.3\t1\t134 27\n",
      "53\t163\t0.3\t1\t142 21\n",
      "54\t98\t0.3\t1\t91 7\n",
      "55\t101\t0.3\t1\t93 8\n",
      "56\t116\t0.3\t1\t81 35\n",
      "57\t93\t0.3\t1\t74 19\n",
      "58\t114\t0.3\t1\t74 40\n",
      "59\t86\t0.3\t1\t67 19\n",
      "60\t75\t0.3\t1\t51 24\n",
      "61\t52\t0.3\t1\t38 14\n",
      "62\t48\t0.3\t1\t35 13\n",
      "63\t48\t0.3\t1\t38 10\n",
      "64\t48\t0.3\t1\t31 17\n",
      "65\t44\t0.3\t1\t28 16\n",
      "66\t39\t0.3\t1\t18 21\n",
      "67\t59\t0.3\t1\t29 30\n",
      "68\t79\t0.3\t1\t34 45\n",
      "69\t130\t0.3\t1\t51 79\n",
      "70\t232\t0.3\t1\t177 55\n",
      "71\t91\t0.3\t1\t51 40\n",
      "72\t29\t0.3\t1\t19 10\n",
      "73\t16\t0.3\t1\t7 9\n",
      "74\t120\t0.3\t1\t4 116\n",
      "75\t11\t0.3\t1\t3 8\n",
      "76\t13\t0.3\t1\t3 10\n",
      "77\t12\t0.3\t1\t3 9\n",
      "78\t15\t0.3\t1\t0 15\n",
      "79\t11\t0.3\t1\t2 9\n",
      "80\t12\t0.3\t1\t0 12\n",
      "81\t14\t0.3\t1\t1 13\n",
      "82\t19\t0.3\t1\t0 19\n",
      "83\t36\t0.3\t1\t0 36\n",
      "84\t7\t0.3\t1\t1 6\n",
      "85\t15\t0.3\t1\t2 13\n",
      "86\t29\t0.3\t1\t0 29\n",
      "87\t16\t0.3\t1\t0 16\n",
      "88\t25\t0.3\t1\t1 24\n",
      "89\t22\t0.3\t1\t2 20\n",
      "90\t10\t0.3\t1\t0 10\n",
      "91\t22\t0.3\t1\t0 22\n",
      "92\t13\t0.3\t1\t1 12\n",
      "93\t8\t0.3\t1\t2 6\n",
      "94\t9\t0.3\t1\t1 8\n",
      "95\t13\t0.3\t1\t0 13\n",
      "96\t16\t0.3\t1\t3 13\n",
      "97\t8\t0.3\t1\t1 7\n",
      "98\t5\t0.3\t1\t0 5\n",
      "99\t7\t0.3\t1\t1 6\n",
      "100\t16\t0.3\t1\t3 13\n",
      "\n",
      "\n",
      "RUN STATISTICS FOR INPUT FILE: fastqs/ERR899431.fastq.gz\n",
      "=============================================\n",
      "19419534 sequences processed in total\n",
      "Sequences removed because they became shorter than the length cutoff of 20 bp:\t3370 (0.0%)\n",
      "\n",
      "This is cutadapt 1.15 with Python 3.6.4\n",
      "Command line parameters: -f fastq -e 0.1 -q 20 -O 1 -a AGATCGGAAGAGC fastqs/ERR899433.fastq.gz\n",
      "Running on 1 core\n",
      "Trimming 1 adapter with at most 10.0% errors in single-end mode ...\n",
      "Finished in 689.04 s (35 us/read; 1.73 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:              19,836,280\n",
      "Reads with adapters:                 6,642,538 (33.5%)\n",
      "Reads written (passing filters):    19,836,280 (100.0%)\n",
      "\n",
      "Total basepairs processed: 1,983,628,000 bp\n",
      "Quality-trimmed:              21,037,203 bp (1.1%)\n",
      "Total written (filtered):  1,949,461,620 bp (98.3%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: AGATCGGAAGAGC; Type: regular 3'; Length: 13; Trimmed: 6642538 times.\n",
      "\n",
      "No. of allowed errors:\n",
      "0-9 bp: 0; 10-13 bp: 1\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "A: 28.6%\n",
      "C: 34.0%\n",
      "G: 19.5%\n",
      "T: 17.8%\n",
      "none/other: 0.1%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "1\t4455569\t4959070.0\t0\t4455569\n",
      "2\t1408746\t1239767.5\t0\t1408746\n",
      "3\t369105\t309941.9\t0\t369105\n",
      "4\t105047\t77485.5\t0\t105047\n",
      "5\t46755\t19371.4\t0\t46755\n",
      "6\t26929\t4842.8\t0\t26929\n",
      "7\t22422\t1210.7\t0\t22422\n",
      "8\t19471\t302.7\t0\t19471\n",
      "9\t18606\t75.7\t0\t18277 329\n",
      "10\t16845\t18.9\t1\t16140 705\n",
      "11\t14830\t4.7\t1\t14291 539\n",
      "12\t13521\t1.2\t1\t13236 285\n",
      "13\t12158\t0.3\t1\t11911 247\n",
      "14\t11269\t0.3\t1\t11064 205\n",
      "15\t10269\t0.3\t1\t10062 207\n",
      "16\t9389\t0.3\t1\t9188 201\n",
      "17\t8334\t0.3\t1\t8137 197\n",
      "18\t7704\t0.3\t1\t7549 155\n",
      "19\t6046\t0.3\t1\t5928 118\n",
      "20\t5460\t0.3\t1\t5330 130\n",
      "21\t5090\t0.3\t1\t4975 115\n",
      "22\t4253\t0.3\t1\t4128 125\n",
      "23\t4088\t0.3\t1\t3976 112\n",
      "24\t3865\t0.3\t1\t3755 110\n",
      "25\t3314\t0.3\t1\t3227 87\n",
      "26\t3109\t0.3\t1\t3035 74\n",
      "27\t2741\t0.3\t1\t2673 68\n",
      "28\t2457\t0.3\t1\t2394 63\n",
      "29\t2175\t0.3\t1\t2118 57\n",
      "30\t1950\t0.3\t1\t1872 78\n",
      "31\t1543\t0.3\t1\t1507 36\n",
      "32\t1434\t0.3\t1\t1386 48\n",
      "33\t1198\t0.3\t1\t1174 24\n",
      "34\t1145\t0.3\t1\t1106 39\n",
      "35\t1031\t0.3\t1\t988 43\n",
      "36\t989\t0.3\t1\t954 35\n",
      "37\t875\t0.3\t1\t837 38\n",
      "38\t725\t0.3\t1\t694 31\n",
      "39\t699\t0.3\t1\t670 29\n",
      "40\t554\t0.3\t1\t527 27\n",
      "41\t475\t0.3\t1\t454 21\n",
      "42\t422\t0.3\t1\t402 20\n",
      "43\t367\t0.3\t1\t344 23\n",
      "44\t373\t0.3\t1\t334 39\n",
      "45\t343\t0.3\t1\t306 37\n",
      "46\t339\t0.3\t1\t307 32\n",
      "47\t268\t0.3\t1\t220 48\n",
      "48\t229\t0.3\t1\t202 27\n",
      "49\t216\t0.3\t1\t193 23\n",
      "50\t207\t0.3\t1\t173 34\n",
      "51\t154\t0.3\t1\t134 20\n",
      "52\t203\t0.3\t1\t149 54\n",
      "53\t151\t0.3\t1\t129 22\n",
      "54\t109\t0.3\t1\t86 23\n",
      "55\t111\t0.3\t1\t77 34\n",
      "56\t119\t0.3\t1\t58 61\n",
      "57\t126\t0.3\t1\t65 61\n",
      "58\t137\t0.3\t1\t54 83\n",
      "59\t98\t0.3\t1\t57 41\n",
      "60\t96\t0.3\t1\t35 61\n",
      "61\t80\t0.3\t1\t39 41\n",
      "62\t193\t0.3\t1\t26 167\n",
      "63\t186\t0.3\t1\t45 141\n",
      "64\t92\t0.3\t1\t41 51\n",
      "65\t70\t0.3\t1\t16 54\n",
      "66\t160\t0.3\t1\t16 144\n",
      "67\t375\t0.3\t1\t31 344\n",
      "68\t508\t0.3\t1\t45 463\n",
      "69\t2147\t0.3\t1\t81 2066\n",
      "70\t939\t0.3\t1\t166 773\n",
      "71\t388\t0.3\t1\t50 338\n",
      "72\t100\t0.3\t1\t22 78\n",
      "73\t46\t0.3\t1\t12 34\n",
      "74\t133\t0.3\t1\t14 119\n",
      "75\t20\t0.3\t1\t2 18\n",
      "76\t22\t0.3\t1\t0 22\n",
      "77\t24\t0.3\t1\t4 20\n",
      "78\t35\t0.3\t1\t1 34\n",
      "79\t38\t0.3\t1\t2 36\n",
      "80\t18\t0.3\t1\t1 17\n",
      "81\t23\t0.3\t1\t1 22\n",
      "82\t33\t0.3\t1\t0 33\n",
      "83\t39\t0.3\t1\t0 39\n",
      "84\t22\t0.3\t1\t1 21\n",
      "85\t18\t0.3\t1\t0 18\n",
      "86\t26\t0.3\t1\t0 26\n",
      "87\t20\t0.3\t1\t0 20\n",
      "88\t35\t0.3\t1\t1 34\n",
      "89\t23\t0.3\t1\t2 21\n",
      "90\t18\t0.3\t1\t0 18\n",
      "91\t27\t0.3\t1\t4 23\n",
      "92\t18\t0.3\t1\t2 16\n",
      "93\t20\t0.3\t1\t1 19\n",
      "94\t21\t0.3\t1\t0 21\n",
      "95\t20\t0.3\t1\t3 17\n",
      "96\t20\t0.3\t1\t1 19\n",
      "97\t22\t0.3\t1\t1 21\n",
      "98\t34\t0.3\t1\t1 33\n",
      "99\t34\t0.3\t1\t1 33\n",
      "100\t238\t0.3\t1\t0 238\n",
      "\n",
      "\n",
      "RUN STATISTICS FOR INPUT FILE: fastqs/ERR899433.fastq.gz\n",
      "=============================================\n",
      "19836280 sequences processed in total\n",
      "Sequences removed because they became shorter than the length cutoff of 20 bp:\t8673 (0.0%)\n",
      "\n",
      "20000000 sequences processed\n",
      "20000000 sequences processed\n",
      "This is cutadapt 1.15 with Python 3.6.4\n",
      "Command line parameters: -f fastq -e 0.1 -q 20 -O 1 -a AGATCGGAAGAGC fastqs/ERR899425.fastq.gz\n",
      "Running on 1 core\n",
      "Trimming 1 adapter with at most 10.0% errors in single-end mode ...\n",
      "Finished in 712.22 s (34 us/read; 1.74 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:              20,649,441\n",
      "Reads with adapters:                 6,994,289 (33.9%)\n",
      "Reads written (passing filters):    20,649,441 (100.0%)\n",
      "\n",
      "Total basepairs processed: 2,064,944,100 bp\n",
      "Quality-trimmed:              22,041,136 bp (1.1%)\n",
      "Total written (filtered):  2,027,967,524 bp (98.2%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: AGATCGGAAGAGC; Type: regular 3'; Length: 13; Trimmed: 6994289 times.\n",
      "\n",
      "No. of allowed errors:\n",
      "0-9 bp: 0; 10-13 bp: 1\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "A: 27.8%\n",
      "C: 34.6%\n",
      "G: 19.7%\n",
      "T: 17.8%\n",
      "none/other: 0.0%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "1\t4592437\t5162360.2\t0\t4592437\n",
      "2\t1475602\t1290590.1\t0\t1475602\n",
      "3\t391811\t322647.5\t0\t391811\n",
      "4\t118840\t80661.9\t0\t118840\n",
      "5\t57715\t20165.5\t0\t57715\n",
      "6\t36313\t5041.4\t0\t36313\n",
      "7\t31605\t1260.3\t0\t31605\n",
      "8\t27203\t315.1\t0\t27203\n",
      "9\t25316\t78.8\t0\t25004 312\n",
      "10\t23304\t19.7\t1\t22206 1098\n",
      "11\t20846\t4.9\t1\t19907 939\n",
      "12\t19258\t1.2\t1\t18592 666\n",
      "13\t17279\t0.3\t1\t16722 557\n",
      "14\t16303\t0.3\t1\t15769 534\n",
      "15\t14954\t0.3\t1\t14485 469\n",
      "16\t13868\t0.3\t1\t13400 468\n",
      "17\t12071\t0.3\t1\t11681 390\n",
      "18\t10609\t0.3\t1\t10229 380\n",
      "19\t8899\t0.3\t1\t8610 289\n",
      "20\t8063\t0.3\t1\t7765 298\n",
      "21\t7058\t0.3\t1\t6817 241\n",
      "22\t6591\t0.3\t1\t6336 255\n",
      "23\t5778\t0.3\t1\t5566 212\n",
      "24\t5250\t0.3\t1\t5057 193\n",
      "25\t4966\t0.3\t1\t4768 198\n",
      "26\t4328\t0.3\t1\t4182 146\n",
      "27\t3998\t0.3\t1\t3854 144\n",
      "28\t3524\t0.3\t1\t3396 128\n",
      "29\t2996\t0.3\t1\t2888 108\n",
      "30\t2861\t0.3\t1\t2735 126\n",
      "31\t2219\t0.3\t1\t2137 82\n",
      "32\t2022\t0.3\t1\t1935 87\n",
      "33\t1757\t0.3\t1\t1703 54\n",
      "34\t1639\t0.3\t1\t1580 59\n",
      "35\t1546\t0.3\t1\t1463 83\n",
      "36\t1404\t0.3\t1\t1351 53\n",
      "37\t1183\t0.3\t1\t1135 48\n",
      "38\t1124\t0.3\t1\t1068 56\n",
      "39\t940\t0.3\t1\t892 48\n",
      "40\t859\t0.3\t1\t821 38\n",
      "41\t750\t0.3\t1\t715 35\n",
      "42\t610\t0.3\t1\t585 25\n",
      "43\t524\t0.3\t1\t491 33\n",
      "44\t466\t0.3\t1\t438 28\n",
      "45\t498\t0.3\t1\t465 33\n",
      "46\t373\t0.3\t1\t349 24\n",
      "47\t399\t0.3\t1\t352 47\n",
      "48\t360\t0.3\t1\t341 19\n",
      "49\t314\t0.3\t1\t273 41\n",
      "50\t295\t0.3\t1\t252 43\n",
      "51\t213\t0.3\t1\t196 17\n",
      "52\t185\t0.3\t1\t158 27\n",
      "53\t180\t0.3\t1\t164 16\n",
      "54\t143\t0.3\t1\t125 18\n",
      "55\t122\t0.3\t1\t110 12\n",
      "56\t148\t0.3\t1\t88 60\n",
      "57\t124\t0.3\t1\t98 26\n",
      "58\t147\t0.3\t1\t82 65\n",
      "59\t98\t0.3\t1\t75 23\n",
      "60\t109\t0.3\t1\t77 32\n",
      "61\t80\t0.3\t1\t61 19\n",
      "62\t64\t0.3\t1\t40 24\n",
      "63\t63\t0.3\t1\t44 19\n",
      "64\t96\t0.3\t1\t38 58\n",
      "65\t96\t0.3\t1\t48 48\n",
      "66\t78\t0.3\t1\t45 33\n",
      "67\t72\t0.3\t1\t23 49\n",
      "68\t95\t0.3\t1\t27 68\n",
      "69\t195\t0.3\t1\t29 166\n",
      "70\t328\t0.3\t1\t58 270\n",
      "71\t1114\t0.3\t1\t94 1020\n",
      "72\t460\t0.3\t1\t256 204\n",
      "73\t178\t0.3\t1\t79 99\n",
      "74\t196\t0.3\t1\t35 161\n",
      "75\t36\t0.3\t1\t14 22\n",
      "76\t37\t0.3\t1\t17 20\n",
      "77\t33\t0.3\t1\t4 29\n",
      "78\t53\t0.3\t1\t1 52\n",
      "79\t33\t0.3\t1\t3 30\n",
      "80\t23\t0.3\t1\t1 22\n",
      "81\t26\t0.3\t1\t0 26\n",
      "82\t26\t0.3\t1\t2 24\n",
      "83\t54\t0.3\t1\t2 52\n",
      "84\t20\t0.3\t1\t3 17\n",
      "85\t22\t0.3\t1\t0 22\n",
      "86\t38\t0.3\t1\t1 37\n",
      "87\t11\t0.3\t1\t2 9\n",
      "88\t36\t0.3\t1\t0 36\n",
      "89\t32\t0.3\t1\t1 31\n",
      "90\t16\t0.3\t1\t4 12\n",
      "91\t25\t0.3\t1\t1 24\n",
      "92\t19\t0.3\t1\t2 17\n",
      "93\t15\t0.3\t1\t2 13\n",
      "94\t16\t0.3\t1\t2 14\n",
      "95\t21\t0.3\t1\t2 19\n",
      "96\t33\t0.3\t1\t1 32\n",
      "97\t14\t0.3\t1\t1 13\n",
      "98\t9\t0.3\t1\t0 9\n",
      "99\t22\t0.3\t1\t0 22\n",
      "100\t107\t0.3\t1\t1 106\n",
      "\n",
      "\n",
      "RUN STATISTICS FOR INPUT FILE: fastqs/ERR899425.fastq.gz\n",
      "=============================================\n",
      "20649441 sequences processed in total\n",
      "Sequences removed because they became shorter than the length cutoff of 20 bp:\t6168 (0.0%)\n",
      "\n",
      "This is cutadapt 1.15 with Python 3.6.4\n",
      "Command line parameters: -f fastq -e 0.1 -q 20 -O 1 -a AGATCGGAAGAGC fastqs/ERR899438.fastq.gz\n",
      "Running on 1 core\n",
      "Trimming 1 adapter with at most 10.0% errors in single-end mode ...\n",
      "Finished in 728.41 s (34 us/read; 1.75 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:              21,276,620\n",
      "Reads with adapters:                 7,144,739 (33.6%)\n",
      "Reads written (passing filters):    21,276,620 (100.0%)\n",
      "\n",
      "Total basepairs processed: 2,127,662,000 bp\n",
      "Quality-trimmed:              22,734,253 bp (1.1%)\n",
      "Total written (filtered):  2,090,306,686 bp (98.2%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: AGATCGGAAGAGC; Type: regular 3'; Length: 13; Trimmed: 7144739 times.\n",
      "\n",
      "No. of allowed errors:\n",
      "0-9 bp: 0; 10-13 bp: 1\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "A: 28.3%\n",
      "C: 33.8%\n",
      "G: 19.9%\n",
      "T: 18.0%\n",
      "none/other: 0.0%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "1\t4749257\t5319155.0\t0\t4749257\n",
      "2\t1503864\t1329788.8\t0\t1503864\n",
      "3\t400899\t332447.2\t0\t400899\n",
      "4\t117278\t83111.8\t0\t117278\n",
      "5\t53552\t20777.9\t0\t53552\n",
      "6\t32328\t5194.5\t0\t32328\n",
      "7\t28091\t1298.6\t0\t28091\n",
      "8\t24213\t324.7\t0\t24213\n",
      "9\t22601\t81.2\t0\t22256 345\n",
      "10\t20924\t20.3\t1\t20187 737\n",
      "11\t18300\t5.1\t1\t17760 540\n",
      "12\t17025\t1.3\t1\t16710 315\n",
      "13\t15151\t0.3\t1\t14879 272\n",
      "14\t14218\t0.3\t1\t13942 276\n",
      "15\t12870\t0.3\t1\t12636 234\n",
      "16\t11654\t0.3\t1\t11450 204\n",
      "17\t10637\t0.3\t1\t10433 204\n",
      "18\t9495\t0.3\t1\t9322 173\n",
      "19\t7865\t0.3\t1\t7712 153\n",
      "20\t7068\t0.3\t1\t6895 173\n",
      "21\t6457\t0.3\t1\t6334 123\n",
      "22\t5869\t0.3\t1\t5716 153\n",
      "23\t5282\t0.3\t1\t5157 125\n",
      "24\t4862\t0.3\t1\t4774 88\n",
      "25\t4532\t0.3\t1\t4428 104\n",
      "26\t3978\t0.3\t1\t3895 83\n",
      "27\t3581\t0.3\t1\t3493 88\n",
      "28\t3183\t0.3\t1\t3112 71\n",
      "29\t2764\t0.3\t1\t2703 61\n",
      "30\t2679\t0.3\t1\t2573 106\n",
      "31\t2022\t0.3\t1\t1980 42\n",
      "32\t1990\t0.3\t1\t1943 47\n",
      "33\t1792\t0.3\t1\t1758 34\n",
      "34\t1614\t0.3\t1\t1569 45\n",
      "35\t1487\t0.3\t1\t1446 41\n",
      "36\t1435\t0.3\t1\t1381 54\n",
      "37\t1214\t0.3\t1\t1172 42\n",
      "38\t1100\t0.3\t1\t1063 37\n",
      "39\t957\t0.3\t1\t934 23\n",
      "40\t909\t0.3\t1\t875 34\n",
      "41\t743\t0.3\t1\t731 12\n",
      "42\t610\t0.3\t1\t590 20\n",
      "43\t607\t0.3\t1\t582 25\n",
      "44\t517\t0.3\t1\t484 33\n",
      "45\t520\t0.3\t1\t490 30\n",
      "46\t487\t0.3\t1\t465 22\n",
      "47\t438\t0.3\t1\t401 37\n",
      "48\t449\t0.3\t1\t416 33\n",
      "49\t366\t0.3\t1\t336 30\n",
      "50\t290\t0.3\t1\t270 20\n",
      "51\t262\t0.3\t1\t249 13\n",
      "52\t240\t0.3\t1\t223 17\n",
      "53\t241\t0.3\t1\t219 22\n",
      "54\t157\t0.3\t1\t141 16\n",
      "55\t165\t0.3\t1\t157 8\n",
      "56\t166\t0.3\t1\t114 52\n",
      "57\t139\t0.3\t1\t114 25\n",
      "58\t181\t0.3\t1\t133 48\n",
      "59\t143\t0.3\t1\t120 23\n",
      "60\t121\t0.3\t1\t89 32\n",
      "61\t112\t0.3\t1\t94 18\n",
      "62\t113\t0.3\t1\t68 45\n",
      "63\t130\t0.3\t1\t86 44\n",
      "64\t83\t0.3\t1\t64 19\n",
      "65\t52\t0.3\t1\t39 13\n",
      "66\t90\t0.3\t1\t41 49\n",
      "67\t121\t0.3\t1\t39 82\n",
      "68\t201\t0.3\t1\t58 143\n",
      "69\t653\t0.3\t1\t91 562\n",
      "70\t407\t0.3\t1\t224 183\n",
      "71\t142\t0.3\t1\t67 75\n",
      "72\t73\t0.3\t1\t46 27\n",
      "73\t26\t0.3\t1\t13 13\n",
      "74\t121\t0.3\t1\t6 115\n",
      "75\t11\t0.3\t1\t4 7\n",
      "76\t20\t0.3\t1\t4 16\n",
      "77\t25\t0.3\t1\t2 23\n",
      "78\t22\t0.3\t1\t3 19\n",
      "79\t18\t0.3\t1\t1 17\n",
      "80\t20\t0.3\t1\t0 20\n",
      "81\t25\t0.3\t1\t0 25\n",
      "82\t25\t0.3\t1\t0 25\n",
      "83\t46\t0.3\t1\t0 46\n",
      "84\t14\t0.3\t1\t2 12\n",
      "85\t17\t0.3\t1\t1 16\n",
      "86\t45\t0.3\t1\t1 44\n",
      "87\t20\t0.3\t1\t1 19\n",
      "88\t34\t0.3\t1\t1 33\n",
      "89\t23\t0.3\t1\t2 21\n",
      "90\t13\t0.3\t1\t0 13\n",
      "91\t26\t0.3\t1\t1 25\n",
      "92\t20\t0.3\t1\t2 18\n",
      "93\t17\t0.3\t1\t3 14\n",
      "94\t14\t0.3\t1\t0 14\n",
      "95\t9\t0.3\t1\t0 9\n",
      "96\t8\t0.3\t1\t0 8\n",
      "97\t11\t0.3\t1\t0 11\n",
      "98\t13\t0.3\t1\t0 13\n",
      "99\t16\t0.3\t1\t0 16\n",
      "100\t64\t0.3\t1\t0 64\n",
      "\n",
      "\n",
      "RUN STATISTICS FOR INPUT FILE: fastqs/ERR899438.fastq.gz\n",
      "=============================================\n",
      "21276620 sequences processed in total\n",
      "Sequences removed because they became shorter than the length cutoff of 20 bp:\t4782 (0.0%)\n",
      "\n"
     ]
    }
   ],
   "source": [
    "old_cmd='''\n",
    "#Activate the environment where cutadapt is installed.\n",
    "source activate tophat\n",
    "module load fastqc\n",
    "\n",
    "#Quality filtering and adapter trimming\n",
    "PHRED=\"20\"\n",
    "\n",
    "for basename in $(ls fastqs | cut -f1 -d '.'); do\n",
    "    fq1=\".fastq.gz\"\n",
    "    fq1=$FASTQS/$basename$fq1\n",
    "    $TGBINDIR/trim_galore \\\n",
    "        -q $PHRED \\\n",
    "        -o $TFASTQS \\\n",
    "        $fq1\n",
    "done\n",
    "\n",
    "for basename in $(ls paired_fastqs | cut -f1 -d '_' | sort | uniq); do\n",
    "    fq1=\"_1.fastq.gz\"\n",
    "    fq2=\"_2.fastq.gz\"\n",
    "    fq1=$PFASTQS/$basename$fq1\n",
    "    fq2=$PFASTQS/$basename$fq2\n",
    "    fqs=\"$fq1 $fq2\"\n",
    "    $TGBINDIR/trim_galore \\\n",
    "        -q $PHRED \\\n",
    "        --paired \\\n",
    "        -o $TPFASTQS/ \\\n",
    "        $fqs\n",
    "done\n",
    "'''\n",
    "\n",
    "cmds = []\n",
    "fastq_files = glob.glob(os.environ['FASTQS']+\"/*.fastq.gz\") \n",
    "\n",
    "for fastq_file in fastq_files:\n",
    "    cmds.append(\"source activate rnaseq && $TGBINDIR/trim_galore -q 20 -o trimmed_fastqs \"+fastq_file)\n",
    "print(cmds)\n",
    "    \n",
    "\n",
    "pool = Pool(processes=MAXCORES)\n",
    "results = pool.map_async(run_cmd, cmds)\n",
    "pool.close()\n",
    "pool.join()\n",
    "pool.terminate()\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 75,
   "metadata": {
    "collapsed": false,
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "['fastqs/ERR899426.fastq.gz', 'fastqs/ERR899430.fastq.gz', 'fastqs/ERR899435.fastq.gz', 'fastqs/ERR899431.fastq.gz', 'fastqs/ERR899428.fastq.gz', 'fastqs/ERR899436.fastq.gz', 'fastqs/ERR899425.fastq.gz', 'fastqs/ERR899427.fastq.gz', 'fastqs/ERR899433.fastq.gz', 'fastqs/ERR899438.fastq.gz', 'fastqs/ERR899437.fastq.gz', 'fastqs/ERR899422.fastq.gz', 'fastqs/ERR899432.fastq.gz', 'fastqs/ERR899434.fastq.gz', 'fastqs/ERR899421.fastq.gz', 'fastqs/ERR899423.fastq.gz', 'fastqs/ERR899424.fastq.gz', 'fastqs/ERR899429.fastq.gz']\n",
      "Started analysis of ERR899438.fastq.gz\n",
      "Started analysis of ERR899425.fastq.gz\n",
      "Started analysis of ERR899435.fastq.gz\n",
      "Started analysis of ERR899426.fastq.gz\n",
      "Started analysis of ERR899429.fastq.gz\n",
      "Started analysis of ERR899423.fastq.gz\n",
      "Started analysis of ERR899433.fastq.gz\n",
      "Started analysis of ERR899421.fastq.gz\n",
      "Started analysis of ERR899424.fastq.gz\n",
      "Started analysis of ERR899428.fastq.gz\n",
      "Started analysis of ERR899431.fastq.gz\n",
      "Started analysis of ERR899422.fastq.gz\n",
      "Started analysis of ERR899437.fastq.gz\n",
      "Started analysis of ERR899430.fastq.gz\n",
      "Started analysis of ERR899427.fastq.gz\n",
      "Started analysis of ERR899434.fastq.gz\n",
      "Started analysis of ERR899432.fastq.gz\n",
      "Started analysis of ERR899436.fastq.gz\n",
      "Approx 5% complete for ERR899428.fastq.gz\n",
      "Approx 5% complete for ERR899427.fastq.gz\n",
      "Approx 5% complete for ERR899437.fastq.gz\n",
      "Approx 5% complete for ERR899432.fastq.gz\n",
      "Approx 5% complete for ERR899423.fastq.gz\n",
      "Approx 5% complete for ERR899421.fastq.gz\n",
      "Approx 5% complete for ERR899436.fastq.gz\n",
      "Approx 5% complete for ERR899435.fastq.gz\n",
      "Approx 5% complete for ERR899434.fastq.gz\n",
      "Approx 5% complete for ERR899429.fastq.gz\n",
      "Approx 5% complete for ERR899422.fastq.gz\n",
      "Approx 5% complete for ERR899426.fastq.gz\n",
      "Approx 5% complete for ERR899424.fastq.gz\n",
      "Approx 5% complete for ERR899430.fastq.gz\n",
      "Approx 5% complete for ERR899431.fastq.gz\n",
      "Approx 5% complete for ERR899433.fastq.gz\n",
      "Approx 5% complete for ERR899425.fastq.gz\n",
      "Approx 5% complete for ERR899438.fastq.gz\n",
      "Approx 10% complete for ERR899427.fastq.gz\n",
      "Approx 10% complete for ERR899428.fastq.gz\n",
      "Approx 10% complete for ERR899437.fastq.gz\n",
      "Approx 10% complete for ERR899432.fastq.gz\n",
      "Approx 10% complete for ERR899436.fastq.gz\n",
      "Approx 10% complete for ERR899423.fastq.gz\n",
      "Approx 10% complete for ERR899435.fastq.gz\n",
      "Approx 10% complete for ERR899421.fastq.gz\n",
      "Approx 10% complete for ERR899434.fastq.gz\n",
      "Approx 10% complete for ERR899422.fastq.gz\n",
      "Approx 10% complete for ERR899429.fastq.gz\n",
      "Approx 10% complete for ERR899426.fastq.gz\n",
      "Approx 10% complete for ERR899430.fastq.gz\n",
      "Approx 10% complete for ERR899431.fastq.gz\n",
      "Approx 10% complete for ERR899424.fastq.gz\n",
      "Approx 10% complete for ERR899433.fastq.gz\n",
      "Approx 10% complete for ERR899425.fastq.gz\n",
      "Approx 10% complete for ERR899438.fastq.gz\n",
      "Approx 15% complete for ERR899427.fastq.gz\n",
      "Approx 15% complete for ERR899428.fastq.gz\n",
      "Approx 15% complete for ERR899432.fastq.gz\n",
      "Approx 15% complete for ERR899436.fastq.gz\n",
      "Approx 15% complete for ERR899435.fastq.gz\n",
      "Approx 15% complete for ERR899437.fastq.gz\n",
      "Approx 15% complete for ERR899434.fastq.gz\n",
      "Approx 15% complete for ERR899421.fastq.gz\n",
      "Approx 15% complete for ERR899423.fastq.gz\n",
      "Approx 15% complete for ERR899422.fastq.gz\n",
      "Approx 15% complete for ERR899429.fastq.gz\n",
      "Approx 15% complete for ERR899430.fastq.gz\n",
      "Approx 15% complete for ERR899426.fastq.gz\n",
      "Approx 15% complete for ERR899431.fastq.gz\n",
      "Approx 15% complete for ERR899424.fastq.gz\n",
      "Approx 15% complete for ERR899433.fastq.gz\n",
      "Approx 15% complete for ERR899425.fastq.gz\n",
      "Approx 15% complete for ERR899438.fastq.gz\n",
      "Approx 20% complete for ERR899427.fastq.gz\n",
      "Approx 20% complete for ERR899428.fastq.gz\n",
      "Approx 20% complete for ERR899435.fastq.gz\n",
      "Approx 20% complete for ERR899432.fastq.gz\n",
      "Approx 20% complete for ERR899436.fastq.gz\n",
      "Approx 20% complete for ERR899437.fastq.gz\n",
      "Approx 20% complete for ERR899434.fastq.gz\n",
      "Approx 20% complete for ERR899421.fastq.gz\n",
      "Approx 20% complete for ERR899423.fastq.gz\n",
      "Approx 20% complete for ERR899422.fastq.gz\n",
      "Approx 20% complete for ERR899429.fastq.gz\n",
      "Approx 20% complete for ERR899430.fastq.gz\n",
      "Approx 20% complete for ERR899426.fastq.gz\n",
      "Approx 20% complete for ERR899431.fastq.gz\n",
      "Approx 20% complete for ERR899424.fastq.gz\n",
      "Approx 20% complete for ERR899433.fastq.gz\n",
      "Approx 20% complete for ERR899425.fastq.gz\n",
      "Approx 20% complete for ERR899438.fastq.gz\n",
      "Approx 25% complete for ERR899427.fastq.gz\n",
      "Approx 25% complete for ERR899428.fastq.gz\n",
      "Approx 25% complete for ERR899435.fastq.gz\n",
      "Approx 25% complete for ERR899432.fastq.gz\n",
      "Approx 25% complete for ERR899436.fastq.gz\n",
      "Approx 25% complete for ERR899434.fastq.gz\n",
      "Approx 25% complete for ERR899437.fastq.gz\n",
      "Approx 25% complete for ERR899421.fastq.gz\n",
      "Approx 25% complete for ERR899423.fastq.gz\n",
      "Approx 25% complete for ERR899422.fastq.gz\n",
      "Approx 25% complete for ERR899429.fastq.gz\n",
      "Approx 25% complete for ERR899430.fastq.gz\n",
      "Approx 25% complete for ERR899426.fastq.gz\n",
      "Approx 25% complete for ERR899431.fastq.gz\n",
      "Approx 25% complete for ERR899424.fastq.gz\n",
      "Approx 25% complete for ERR899433.fastq.gz\n",
      "Approx 25% complete for ERR899425.fastq.gz\n",
      "Approx 30% complete for ERR899427.fastq.gz\n",
      "Approx 30% complete for ERR899428.fastq.gz\n",
      "Approx 25% complete for ERR899438.fastq.gz\n",
      "Approx 30% complete for ERR899435.fastq.gz\n",
      "Approx 30% complete for ERR899432.fastq.gz\n",
      "Approx 30% complete for ERR899436.fastq.gz\n",
      "Approx 30% complete for ERR899434.fastq.gz\n",
      "Approx 30% complete for ERR899421.fastq.gz\n",
      "Approx 30% complete for ERR899437.fastq.gz\n",
      "Approx 30% complete for ERR899423.fastq.gz\n",
      "Approx 30% complete for ERR899422.fastq.gz\n",
      "Approx 30% complete for ERR899429.fastq.gz\n",
      "Approx 30% complete for ERR899430.fastq.gz\n",
      "Approx 30% complete for ERR899426.fastq.gz\n",
      "Approx 30% complete for ERR899431.fastq.gz\n",
      "Approx 30% complete for ERR899424.fastq.gz\n",
      "Approx 30% complete for ERR899433.fastq.gz\n",
      "Approx 35% complete for ERR899427.fastq.gz\n",
      "Approx 35% complete for ERR899428.fastq.gz\n",
      "Approx 30% complete for ERR899425.fastq.gz\n",
      "Approx 35% complete for ERR899435.fastq.gz\n",
      "Approx 35% complete for ERR899432.fastq.gz\n",
      "Approx 35% complete for ERR899436.fastq.gz\n",
      "Approx 35% complete for ERR899434.fastq.gz\n",
      "Approx 35% complete for ERR899421.fastq.gz\n",
      "Approx 30% complete for ERR899438.fastq.gz\n",
      "Approx 35% complete for ERR899437.fastq.gz\n",
      "Approx 35% complete for ERR899423.fastq.gz\n",
      "Approx 35% complete for ERR899422.fastq.gz\n",
      "Approx 35% complete for ERR899429.fastq.gz\n",
      "Approx 35% complete for ERR899430.fastq.gz\n",
      "Approx 35% complete for ERR899426.fastq.gz\n",
      "Approx 35% complete for ERR899431.fastq.gz\n",
      "Approx 35% complete for ERR899424.fastq.gz\n",
      "Approx 40% complete for ERR899427.fastq.gz\n",
      "Approx 40% complete for ERR899428.fastq.gz\n",
      "Approx 35% complete for ERR899433.fastq.gz\n",
      "Approx 35% complete for ERR899425.fastq.gz\n",
      "Approx 40% complete for ERR899432.fastq.gz\n",
      "Approx 40% complete for ERR899435.fastq.gz\n",
      "Approx 40% complete for ERR899436.fastq.gz\n",
      "Approx 40% complete for ERR899434.fastq.gz\n",
      "Approx 40% complete for ERR899421.fastq.gz\n",
      "Approx 40% complete for ERR899437.fastq.gz\n",
      "Approx 40% complete for ERR899423.fastq.gz\n",
      "Approx 40% complete for ERR899422.fastq.gz\n",
      "Approx 35% complete for ERR899438.fastq.gz\n",
      "Approx 40% complete for ERR899429.fastq.gz\n",
      "Approx 40% complete for ERR899430.fastq.gz\n",
      "Approx 40% complete for ERR899426.fastq.gz\n",
      "Approx 40% complete for ERR899431.fastq.gz\n",
      "Approx 45% complete for ERR899427.fastq.gz\n",
      "Approx 45% complete for ERR899428.fastq.gz\n",
      "Approx 40% complete for ERR899424.fastq.gz\n",
      "Approx 40% complete for ERR899433.fastq.gz\n",
      "Approx 45% complete for ERR899432.fastq.gz\n",
      "Approx 45% complete for ERR899435.fastq.gz\n",
      "Approx 45% complete for ERR899436.fastq.gz\n",
      "Approx 40% complete for ERR899425.fastq.gz\n",
      "Approx 45% complete for ERR899434.fastq.gz\n",
      "Approx 45% complete for ERR899421.fastq.gz\n",
      "Approx 45% complete for ERR899437.fastq.gz\n",
      "Approx 45% complete for ERR899423.fastq.gz\n",
      "Approx 45% complete for ERR899422.fastq.gz\n",
      "Approx 45% complete for ERR899429.fastq.gz\n",
      "Approx 40% complete for ERR899438.fastq.gz\n",
      "Approx 45% complete for ERR899430.fastq.gz\n",
      "Approx 50% complete for ERR899427.fastq.gz\n",
      "Approx 50% complete for ERR899428.fastq.gz\n",
      "Approx 45% complete for ERR899426.fastq.gz\n",
      "Approx 45% complete for ERR899431.fastq.gz\n",
      "Approx 45% complete for ERR899424.fastq.gz\n",
      "Approx 45% complete for ERR899433.fastq.gz\n",
      "Approx 50% complete for ERR899432.fastq.gz\n",
      "Approx 50% complete for ERR899435.fastq.gz\n",
      "Approx 50% complete for ERR899436.fastq.gz\n",
      "Approx 50% complete for ERR899434.fastq.gz\n",
      "Approx 50% complete for ERR899421.fastq.gz\n",
      "Approx 45% complete for ERR899425.fastq.gz\n",
      "Approx 50% complete for ERR899437.fastq.gz\n",
      "Approx 50% complete for ERR899423.fastq.gz\n",
      "Approx 50% complete for ERR899422.fastq.gz\n",
      "Approx 50% complete for ERR899429.fastq.gz\n",
      "Approx 55% complete for ERR899427.fastq.gz\n",
      "Approx 50% complete for ERR899430.fastq.gz\n",
      "Approx 55% complete for ERR899428.fastq.gz\n",
      "Approx 45% complete for ERR899438.fastq.gz\n",
      "Approx 50% complete for ERR899426.fastq.gz\n",
      "Approx 50% complete for ERR899431.fastq.gz\n",
      "Approx 50% complete for ERR899424.fastq.gz\n",
      "Approx 55% complete for ERR899432.fastq.gz\n",
      "Approx 55% complete for ERR899435.fastq.gz\n",
      "Approx 50% complete for ERR899433.fastq.gz\n",
      "Approx 55% complete for ERR899434.fastq.gz\n",
      "Approx 55% complete for ERR899436.fastq.gz\n",
      "Approx 55% complete for ERR899421.fastq.gz\n",
      "Approx 55% complete for ERR899423.fastq.gz\n",
      "Approx 55% complete for ERR899437.fastq.gz\n",
      "Approx 50% complete for ERR899425.fastq.gz\n",
      "Approx 55% complete for ERR899422.fastq.gz\n",
      "Approx 60% complete for ERR899427.fastq.gz\n",
      "Approx 55% complete for ERR899429.fastq.gz\n",
      "Approx 60% complete for ERR899428.fastq.gz\n",
      "Approx 55% complete for ERR899430.fastq.gz\n",
      "Approx 55% complete for ERR899426.fastq.gz\n",
      "Approx 55% complete for ERR899431.fastq.gz\n",
      "Approx 50% complete for ERR899438.fastq.gz\n",
      "Approx 60% complete for ERR899432.fastq.gz\n",
      "Approx 60% complete for ERR899435.fastq.gz\n",
      "Approx 55% complete for ERR899424.fastq.gz\n",
      "Approx 60% complete for ERR899434.fastq.gz\n",
      "Approx 55% complete for ERR899433.fastq.gz\n",
      "Approx 60% complete for ERR899421.fastq.gz\n",
      "Approx 60% complete for ERR899436.fastq.gz\n",
      "Approx 60% complete for ERR899423.fastq.gz\n",
      "Approx 60% complete for ERR899437.fastq.gz\n",
      "Approx 60% complete for ERR899422.fastq.gz\n",
      "Approx 65% complete for ERR899427.fastq.gz\n",
      "Approx 55% complete for ERR899425.fastq.gz\n",
      "Approx 65% complete for ERR899428.fastq.gz\n",
      "Approx 60% complete for ERR899429.fastq.gz\n",
      "Approx 60% complete for ERR899430.fastq.gz\n",
      "Approx 60% complete for ERR899426.fastq.gz\n",
      "Approx 60% complete for ERR899431.fastq.gz\n",
      "Approx 65% complete for ERR899432.fastq.gz\n",
      "Approx 65% complete for ERR899435.fastq.gz\n",
      "Approx 55% complete for ERR899438.fastq.gz\n",
      "Approx 65% complete for ERR899434.fastq.gz\n",
      "Approx 65% complete for ERR899421.fastq.gz\n",
      "Approx 60% complete for ERR899424.fastq.gz\n",
      "Approx 65% complete for ERR899436.fastq.gz\n",
      "Approx 60% complete for ERR899433.fastq.gz\n",
      "Approx 65% complete for ERR899423.fastq.gz\n",
      "Approx 65% complete for ERR899437.fastq.gz\n",
      "Approx 65% complete for ERR899422.fastq.gz\n",
      "Approx 70% complete for ERR899427.fastq.gz\n",
      "Approx 70% complete for ERR899428.fastq.gz\n",
      "Approx 60% complete for ERR899425.fastq.gz\n",
      "Approx 65% complete for ERR899429.fastq.gz\n",
      "Approx 65% complete for ERR899430.fastq.gz\n",
      "Approx 70% complete for ERR899432.fastq.gz\n",
      "Approx 70% complete for ERR899435.fastq.gz\n",
      "Approx 65% complete for ERR899426.fastq.gz\n",
      "Approx 65% complete for ERR899431.fastq.gz\n",
      "Approx 70% complete for ERR899421.fastq.gz\n",
      "Approx 70% complete for ERR899434.fastq.gz\n",
      "Approx 70% complete for ERR899436.fastq.gz\n",
      "Approx 60% complete for ERR899438.fastq.gz\n",
      "Approx 65% complete for ERR899424.fastq.gz\n",
      "Approx 70% complete for ERR899423.fastq.gz\n",
      "Approx 70% complete for ERR899437.fastq.gz\n",
      "Approx 65% complete for ERR899433.fastq.gz\n",
      "Approx 75% complete for ERR899427.fastq.gz\n",
      "Approx 70% complete for ERR899422.fastq.gz\n",
      "Approx 75% complete for ERR899428.fastq.gz\n",
      "Approx 70% complete for ERR899429.fastq.gz\n",
      "Approx 65% complete for ERR899425.fastq.gz\n",
      "Approx 70% complete for ERR899430.fastq.gz\n",
      "Approx 75% complete for ERR899435.fastq.gz\n",
      "Approx 75% complete for ERR899432.fastq.gz\n",
      "Approx 70% complete for ERR899426.fastq.gz\n",
      "Approx 70% complete for ERR899431.fastq.gz\n",
      "Approx 75% complete for ERR899421.fastq.gz\n",
      "Approx 75% complete for ERR899434.fastq.gz\n",
      "Approx 75% complete for ERR899436.fastq.gz\n",
      "Approx 70% complete for ERR899424.fastq.gz\n",
      "Approx 75% complete for ERR899423.fastq.gz\n",
      "Approx 80% complete for ERR899427.fastq.gz\n",
      "Approx 75% complete for ERR899437.fastq.gz\n",
      "Approx 65% complete for ERR899438.fastq.gz\n",
      "Approx 75% complete for ERR899422.fastq.gz\n",
      "Approx 80% complete for ERR899428.fastq.gz\n",
      "Approx 70% complete for ERR899433.fastq.gz\n",
      "Approx 75% complete for ERR899429.fastq.gz\n",
      "Approx 70% complete for ERR899425.fastq.gz\n",
      "Approx 80% complete for ERR899435.fastq.gz\n",
      "Approx 80% complete for ERR899432.fastq.gz\n",
      "Approx 75% complete for ERR899430.fastq.gz\n",
      "Approx 75% complete for ERR899426.fastq.gz\n",
      "Approx 75% complete for ERR899431.fastq.gz\n",
      "Approx 80% complete for ERR899421.fastq.gz\n",
      "Approx 80% complete for ERR899434.fastq.gz\n",
      "Approx 80% complete for ERR899436.fastq.gz\n",
      "Approx 80% complete for ERR899423.fastq.gz\n",
      "Approx 85% complete for ERR899427.fastq.gz\n",
      "Approx 75% complete for ERR899424.fastq.gz\n",
      "Approx 80% complete for ERR899437.fastq.gz\n",
      "Approx 85% complete for ERR899428.fastq.gz\n",
      "Approx 80% complete for ERR899422.fastq.gz\n",
      "Approx 70% complete for ERR899438.fastq.gz\n",
      "Approx 75% complete for ERR899433.fastq.gz\n",
      "Approx 80% complete for ERR899429.fastq.gz\n",
      "Approx 85% complete for ERR899435.fastq.gz\n",
      "Approx 85% complete for ERR899432.fastq.gz\n",
      "Approx 75% complete for ERR899425.fastq.gz\n",
      "Approx 80% complete for ERR899430.fastq.gz\n",
      "Approx 85% complete for ERR899421.fastq.gz\n",
      "Approx 85% complete for ERR899434.fastq.gz\n",
      "Approx 80% complete for ERR899426.fastq.gz\n",
      "Approx 80% complete for ERR899431.fastq.gz\n",
      "Approx 85% complete for ERR899436.fastq.gz\n",
      "Approx 90% complete for ERR899427.fastq.gz\n",
      "Approx 85% complete for ERR899423.fastq.gz\n",
      "Approx 80% complete for ERR899424.fastq.gz\n",
      "Approx 90% complete for ERR899428.fastq.gz\n",
      "Approx 85% complete for ERR899437.fastq.gz\n",
      "Approx 85% complete for ERR899422.fastq.gz\n",
      "Approx 80% complete for ERR899433.fastq.gz\n",
      "Approx 75% complete for ERR899438.fastq.gz\n",
      "Approx 90% complete for ERR899435.fastq.gz\n",
      "Approx 85% complete for ERR899429.fastq.gz\n",
      "Approx 90% complete for ERR899432.fastq.gz\n",
      "Approx 90% complete for ERR899421.fastq.gz\n",
      "Approx 85% complete for ERR899430.fastq.gz\n",
      "Approx 90% complete for ERR899434.fastq.gz\n",
      "Approx 80% complete for ERR899425.fastq.gz\n",
      "Approx 90% complete for ERR899436.fastq.gz\n",
      "Approx 85% complete for ERR899426.fastq.gz\n",
      "Approx 95% complete for ERR899427.fastq.gz\n",
      "Approx 85% complete for ERR899431.fastq.gz\n",
      "Approx 90% complete for ERR899423.fastq.gz\n",
      "Approx 95% complete for ERR899428.fastq.gz\n",
      "Approx 85% complete for ERR899424.fastq.gz\n",
      "Approx 90% complete for ERR899437.fastq.gz\n",
      "Approx 90% complete for ERR899422.fastq.gz\n",
      "Approx 85% complete for ERR899433.fastq.gz\n",
      "Approx 80% complete for ERR899438.fastq.gz\n",
      "Approx 95% complete for ERR899435.fastq.gz\n",
      "Approx 95% complete for ERR899432.fastq.gz\n",
      "Approx 90% complete for ERR899429.fastq.gz\n",
      "Approx 95% complete for ERR899421.fastq.gz\n",
      "Approx 95% complete for ERR899434.fastq.gz\n",
      "Analysis complete for ERR899427.fastq.gz\n",
      "Approx 90% complete for ERR899430.fastq.gz\n",
      "Approx 95% complete for ERR899436.fastq.gz\n",
      "Approx 90% complete for ERR899426.fastq.gz\n",
      "Approx 85% complete for ERR899425.fastq.gz\n",
      "Approx 90% complete for ERR899431.fastq.gz\n",
      "Analysis complete for ERR899428.fastq.gz\n",
      "Approx 95% complete for ERR899423.fastq.gz\n",
      "Approx 95% complete for ERR899437.fastq.gz\n",
      "Approx 90% complete for ERR899424.fastq.gz\n",
      "Approx 95% complete for ERR899422.fastq.gz\n",
      "Approx 90% complete for ERR899433.fastq.gz\n",
      "Analysis complete for ERR899435.fastq.gz\n",
      "Analysis complete for ERR899432.fastq.gz\n",
      "Approx 85% complete for ERR899438.fastq.gz\n",
      "Approx 95% complete for ERR899429.fastq.gz\n",
      "Analysis complete for ERR899421.fastq.gz\n",
      "Analysis complete for ERR899434.fastq.gz\n",
      "Analysis complete for ERR899436.fastq.gz\n",
      "Approx 95% complete for ERR899430.fastq.gz\n",
      "Analysis complete for ERR899423.fastq.gz\n",
      "Approx 95% complete for ERR899431.fastq.gz\n",
      "Approx 95% complete for ERR899426.fastq.gz\n",
      "Approx 90% complete for ERR899425.fastq.gz\n",
      "Analysis complete for ERR899437.fastq.gz\n",
      "Approx 95% complete for ERR899424.fastq.gz\n",
      "Analysis complete for ERR899422.fastq.gz\n",
      "Approx 95% complete for ERR899433.fastq.gz\n",
      "Analysis complete for ERR899429.fastq.gz\n",
      "Approx 90% complete for ERR899438.fastq.gz\n",
      "Analysis complete for ERR899430.fastq.gz\n",
      "Analysis complete for ERR899431.fastq.gz\n",
      "Analysis complete for ERR899426.fastq.gz\n",
      "Approx 95% complete for ERR899425.fastq.gz\n",
      "Analysis complete for ERR899424.fastq.gz\n",
      "Analysis complete for ERR899433.fastq.gz\n",
      "Approx 95% complete for ERR899438.fastq.gz\n",
      "Analysis complete for ERR899425.fastq.gz\n",
      "Analysis complete for ERR899438.fastq.gz\n"
     ]
    }
   ],
   "source": [
    "#fastqc of sequencing reads\n",
    "\n",
    "old_cmd = '''\n",
    "#load fastqc\n",
    "module load fastqc\n",
    "\n",
    "fqs=\"\"\n",
    "\n",
    "for basename in $(ls $FASTQS | cut -f1 -d '.'); do\n",
    "    #basename=$(echo $fq | cut -f1 -d '.')\n",
    "    fqd=$FASTQS\n",
    "    fq1=\".fastq.gz\"\n",
    "    fq1=$fqd/$basename$fq1\n",
    "    fqs=\"$fqs $fq1\"\n",
    "    \n",
    "done\n",
    "\n",
    "for basename in $(ls $PFASTQS | cut -f1 -d '_' | sort | uniq); do\n",
    "    echo $basename\n",
    "    fqd=$PFASTQS\n",
    "    fq1=\"_1.fastq.gz\"\n",
    "    fq2=\"_2.fastq.gz\"\n",
    "    fq1=$fqd/$basename$fq1\n",
    "    fq2=$fqd/$basename$fq2\n",
    "    fqs=\"$fqs $fq1 $fq2\"\n",
    "done\n",
    "\n",
    "echo \"Performing FastQC for $fqs\"\n",
    "fastqc --threads $MAXCORES $fqs -o $FASTQCOUT\n",
    "'''\n",
    "#out = run_cmd(cmd)\n",
    "\n",
    "fastq_files = glob.glob(FASTQS+\"/*.fastq.gz\") \n",
    "print(fastq_files)\n",
    "cmds = []\n",
    "for fastq_file in fastq_files:\n",
    "    cmds.append(ENVCMD + \" && \" +\n",
    "                \"module load fastqc && \" +\n",
    "                \"fastqc \" + \n",
    "                fastq_file +\n",
    "                \" -o \" + FASTQCOUT)\n",
    "\n",
    "pool = Pool(processes=MAXCORES)\n",
    "results = pool.map_async(run_cmd, cmds)\n",
    "pool.close()\n",
    "pool.join()\n",
    "pool.terminate()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 70,
   "metadata": {
    "collapsed": false,
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Started analysis of ERR899435_trimmed.fq.gz\n",
      "Started analysis of ERR899425_trimmed.fq.gz\n",
      "Started analysis of ERR899431_trimmed.fq.gz\n",
      "Started analysis of ERR899434_trimmed.fq.gz\n",
      "Started analysis of ERR899432_trimmed.fq.gz\n",
      "Started analysis of ERR899423_trimmed.fq.gz\n",
      "Started analysis of ERR899429_trimmed.fq.gz\n",
      "Started analysis of ERR899424_trimmed.fq.gz\n",
      "Started analysis of ERR899428_trimmed.fq.gz\n",
      "Started analysis of ERR899436_trimmed.fq.gz\n",
      "Started analysis of ERR899422_trimmed.fq.gz\n",
      "Started analysis of ERR899427_trimmed.fq.gz\n",
      "Started analysis of ERR899421_trimmed.fq.gz\n",
      "Started analysis of ERR899437_trimmed.fq.gz\n",
      "Started analysis of ERR899433_trimmed.fq.gz\n",
      "Started analysis of ERR899430_trimmed.fq.gz\n",
      "Started analysis of ERR899438_trimmed.fq.gz\n",
      "Started analysis of ERR899426_trimmed.fq.gz\n",
      "Approx 5% complete for ERR899427_trimmed.fq.gz\n",
      "Approx 5% complete for ERR899428_trimmed.fq.gz\n",
      "Approx 5% complete for ERR899435_trimmed.fq.gz\n",
      "Approx 5% complete for ERR899421_trimmed.fq.gz\n",
      "Approx 5% complete for ERR899432_trimmed.fq.gz\n",
      "Approx 5% complete for ERR899437_trimmed.fq.gz\n",
      "Approx 5% complete for ERR899434_trimmed.fq.gz\n",
      "Approx 5% complete for ERR899436_trimmed.fq.gz\n",
      "Approx 5% complete for ERR899423_trimmed.fq.gz\n",
      "Approx 5% complete for ERR899424_trimmed.fq.gz\n",
      "Approx 5% complete for ERR899422_trimmed.fq.gz\n",
      "Approx 5% complete for ERR899429_trimmed.fq.gz\n",
      "Approx 5% complete for ERR899431_trimmed.fq.gz\n",
      "Approx 5% complete for ERR899433_trimmed.fq.gz\n",
      "Approx 5% complete for ERR899426_trimmed.fq.gz\n",
      "Approx 5% complete for ERR899430_trimmed.fq.gz\n",
      "Approx 5% complete for ERR899425_trimmed.fq.gz\n",
      "Approx 5% complete for ERR899438_trimmed.fq.gz\n",
      "Approx 10% complete for ERR899427_trimmed.fq.gz\n",
      "Approx 10% complete for ERR899428_trimmed.fq.gz\n",
      "Approx 10% complete for ERR899435_trimmed.fq.gz\n",
      "Approx 10% complete for ERR899421_trimmed.fq.gz\n",
      "Approx 10% complete for ERR899432_trimmed.fq.gz\n",
      "Approx 10% complete for ERR899436_trimmed.fq.gz\n",
      "Approx 10% complete for ERR899437_trimmed.fq.gz\n",
      "Approx 10% complete for ERR899434_trimmed.fq.gz\n",
      "Approx 10% complete for ERR899423_trimmed.fq.gz\n",
      "Approx 10% complete for ERR899424_trimmed.fq.gz\n",
      "Approx 10% complete for ERR899422_trimmed.fq.gz\n",
      "Approx 10% complete for ERR899429_trimmed.fq.gz\n",
      "Approx 10% complete for ERR899430_trimmed.fq.gz\n",
      "Approx 10% complete for ERR899431_trimmed.fq.gz\n",
      "Approx 10% complete for ERR899426_trimmed.fq.gz\n",
      "Approx 10% complete for ERR899433_trimmed.fq.gz\n",
      "Approx 10% complete for ERR899425_trimmed.fq.gz\n",
      "Approx 10% complete for ERR899438_trimmed.fq.gz\n",
      "Approx 15% complete for ERR899427_trimmed.fq.gz\n",
      "Approx 15% complete for ERR899428_trimmed.fq.gz\n",
      "Approx 15% complete for ERR899435_trimmed.fq.gz\n",
      "Approx 15% complete for ERR899421_trimmed.fq.gz\n",
      "Approx 15% complete for ERR899432_trimmed.fq.gz\n",
      "Approx 15% complete for ERR899436_trimmed.fq.gz\n",
      "Approx 15% complete for ERR899434_trimmed.fq.gz\n",
      "Approx 15% complete for ERR899423_trimmed.fq.gz\n",
      "Approx 15% complete for ERR899437_trimmed.fq.gz\n",
      "Approx 15% complete for ERR899424_trimmed.fq.gz\n",
      "Approx 15% complete for ERR899422_trimmed.fq.gz\n",
      "Approx 15% complete for ERR899429_trimmed.fq.gz\n",
      "Approx 15% complete for ERR899426_trimmed.fq.gz\n",
      "Approx 15% complete for ERR899430_trimmed.fq.gz\n",
      "Approx 15% complete for ERR899431_trimmed.fq.gz\n",
      "Approx 15% complete for ERR899433_trimmed.fq.gz\n",
      "Approx 15% complete for ERR899425_trimmed.fq.gz\n",
      "Approx 15% complete for ERR899438_trimmed.fq.gz\n",
      "Approx 20% complete for ERR899427_trimmed.fq.gz\n",
      "Approx 20% complete for ERR899428_trimmed.fq.gz\n",
      "Approx 20% complete for ERR899435_trimmed.fq.gz\n",
      "Approx 20% complete for ERR899421_trimmed.fq.gz\n",
      "Approx 20% complete for ERR899432_trimmed.fq.gz\n",
      "Approx 20% complete for ERR899436_trimmed.fq.gz\n",
      "Approx 20% complete for ERR899434_trimmed.fq.gz\n",
      "Approx 20% complete for ERR899423_trimmed.fq.gz\n",
      "Approx 20% complete for ERR899437_trimmed.fq.gz\n",
      "Approx 20% complete for ERR899424_trimmed.fq.gz\n",
      "Approx 20% complete for ERR899422_trimmed.fq.gz\n",
      "Approx 20% complete for ERR899429_trimmed.fq.gz\n",
      "Approx 20% complete for ERR899426_trimmed.fq.gz\n",
      "Approx 20% complete for ERR899430_trimmed.fq.gz\n",
      "Approx 20% complete for ERR899431_trimmed.fq.gz\n",
      "Approx 20% complete for ERR899433_trimmed.fq.gz\n",
      "Approx 20% complete for ERR899425_trimmed.fq.gz\n",
      "Approx 25% complete for ERR899427_trimmed.fq.gz\n",
      "Approx 20% complete for ERR899438_trimmed.fq.gz\n",
      "Approx 25% complete for ERR899428_trimmed.fq.gz\n",
      "Approx 25% complete for ERR899435_trimmed.fq.gz\n",
      "Approx 25% complete for ERR899421_trimmed.fq.gz\n",
      "Approx 25% complete for ERR899432_trimmed.fq.gz\n",
      "Approx 25% complete for ERR899436_trimmed.fq.gz\n",
      "Approx 25% complete for ERR899423_trimmed.fq.gz\n",
      "Approx 25% complete for ERR899434_trimmed.fq.gz\n",
      "Approx 25% complete for ERR899424_trimmed.fq.gz\n",
      "Approx 25% complete for ERR899437_trimmed.fq.gz\n",
      "Approx 25% complete for ERR899422_trimmed.fq.gz\n",
      "Approx 25% complete for ERR899430_trimmed.fq.gz\n",
      "Approx 25% complete for ERR899431_trimmed.fq.gz\n",
      "Approx 25% complete for ERR899429_trimmed.fq.gz\n",
      "Approx 25% complete for ERR899426_trimmed.fq.gz\n",
      "Approx 25% complete for ERR899433_trimmed.fq.gz\n",
      "Approx 25% complete for ERR899425_trimmed.fq.gz\n",
      "Approx 30% complete for ERR899427_trimmed.fq.gz\n",
      "Approx 30% complete for ERR899428_trimmed.fq.gz\n",
      "Approx 30% complete for ERR899435_trimmed.fq.gz\n",
      "Approx 25% complete for ERR899438_trimmed.fq.gz\n",
      "Approx 30% complete for ERR899421_trimmed.fq.gz\n",
      "Approx 30% complete for ERR899432_trimmed.fq.gz\n",
      "Approx 30% complete for ERR899436_trimmed.fq.gz\n",
      "Approx 30% complete for ERR899434_trimmed.fq.gz\n",
      "Approx 30% complete for ERR899423_trimmed.fq.gz\n",
      "Approx 30% complete for ERR899437_trimmed.fq.gz\n",
      "Approx 30% complete for ERR899424_trimmed.fq.gz\n",
      "Approx 30% complete for ERR899422_trimmed.fq.gz\n",
      "Approx 30% complete for ERR899430_trimmed.fq.gz\n",
      "Approx 30% complete for ERR899426_trimmed.fq.gz\n",
      "Approx 30% complete for ERR899431_trimmed.fq.gz\n",
      "Approx 30% complete for ERR899429_trimmed.fq.gz\n",
      "Approx 35% complete for ERR899427_trimmed.fq.gz\n",
      "Approx 30% complete for ERR899433_trimmed.fq.gz\n",
      "Approx 35% complete for ERR899428_trimmed.fq.gz\n",
      "Approx 30% complete for ERR899425_trimmed.fq.gz\n",
      "Approx 35% complete for ERR899435_trimmed.fq.gz\n",
      "Approx 35% complete for ERR899421_trimmed.fq.gz\n",
      "Approx 35% complete for ERR899432_trimmed.fq.gz\n",
      "Approx 35% complete for ERR899436_trimmed.fq.gz\n",
      "Approx 35% complete for ERR899434_trimmed.fq.gz\n",
      "Approx 35% complete for ERR899423_trimmed.fq.gz\n",
      "Approx 30% complete for ERR899438_trimmed.fq.gz\n",
      "Approx 35% complete for ERR899437_trimmed.fq.gz\n",
      "Approx 35% complete for ERR899424_trimmed.fq.gz\n",
      "Approx 35% complete for ERR899422_trimmed.fq.gz\n",
      "Approx 35% complete for ERR899430_trimmed.fq.gz\n",
      "Approx 35% complete for ERR899426_trimmed.fq.gz\n",
      "Approx 35% complete for ERR899431_trimmed.fq.gz\n",
      "Approx 35% complete for ERR899429_trimmed.fq.gz\n",
      "Approx 40% complete for ERR899427_trimmed.fq.gz\n",
      "Approx 40% complete for ERR899428_trimmed.fq.gz\n",
      "Approx 35% complete for ERR899433_trimmed.fq.gz\n",
      "Approx 40% complete for ERR899435_trimmed.fq.gz\n",
      "Approx 35% complete for ERR899425_trimmed.fq.gz\n",
      "Approx 40% complete for ERR899421_trimmed.fq.gz\n",
      "Approx 40% complete for ERR899432_trimmed.fq.gz\n",
      "Approx 40% complete for ERR899436_trimmed.fq.gz\n",
      "Approx 40% complete for ERR899434_trimmed.fq.gz\n",
      "Approx 40% complete for ERR899423_trimmed.fq.gz\n",
      "Approx 40% complete for ERR899437_trimmed.fq.gz\n",
      "Approx 40% complete for ERR899424_trimmed.fq.gz\n",
      "Approx 40% complete for ERR899422_trimmed.fq.gz\n",
      "Approx 35% complete for ERR899438_trimmed.fq.gz\n",
      "Approx 40% complete for ERR899430_trimmed.fq.gz\n",
      "Approx 45% complete for ERR899427_trimmed.fq.gz\n",
      "Approx 40% complete for ERR899426_trimmed.fq.gz\n",
      "Approx 40% complete for ERR899431_trimmed.fq.gz\n",
      "Approx 40% complete for ERR899429_trimmed.fq.gz\n",
      "Approx 45% complete for ERR899428_trimmed.fq.gz\n",
      "Approx 40% complete for ERR899433_trimmed.fq.gz\n",
      "Approx 45% complete for ERR899435_trimmed.fq.gz\n",
      "Approx 45% complete for ERR899421_trimmed.fq.gz\n",
      "Approx 45% complete for ERR899432_trimmed.fq.gz\n",
      "Approx 40% complete for ERR899425_trimmed.fq.gz\n",
      "Approx 45% complete for ERR899436_trimmed.fq.gz\n",
      "Approx 45% complete for ERR899434_trimmed.fq.gz\n",
      "Approx 45% complete for ERR899423_trimmed.fq.gz\n",
      "Approx 45% complete for ERR899437_trimmed.fq.gz\n",
      "Approx 45% complete for ERR899424_trimmed.fq.gz\n",
      "Approx 45% complete for ERR899422_trimmed.fq.gz\n",
      "Approx 50% complete for ERR899427_trimmed.fq.gz\n",
      "Approx 40% complete for ERR899438_trimmed.fq.gz\n",
      "Approx 45% complete for ERR899430_trimmed.fq.gz\n",
      "Approx 45% complete for ERR899431_trimmed.fq.gz\n",
      "Approx 50% complete for ERR899428_trimmed.fq.gz\n",
      "Approx 45% complete for ERR899426_trimmed.fq.gz\n",
      "Approx 45% complete for ERR899429_trimmed.fq.gz\n",
      "Approx 45% complete for ERR899433_trimmed.fq.gz\n",
      "Approx 50% complete for ERR899435_trimmed.fq.gz\n",
      "Approx 50% complete for ERR899421_trimmed.fq.gz\n",
      "Approx 50% complete for ERR899432_trimmed.fq.gz\n",
      "Approx 50% complete for ERR899436_trimmed.fq.gz\n",
      "Approx 50% complete for ERR899434_trimmed.fq.gz\n",
      "Approx 50% complete for ERR899423_trimmed.fq.gz\n",
      "Approx 45% complete for ERR899425_trimmed.fq.gz\n",
      "Approx 50% complete for ERR899437_trimmed.fq.gz\n",
      "Approx 50% complete for ERR899424_trimmed.fq.gz\n",
      "Approx 50% complete for ERR899422_trimmed.fq.gz\n",
      "Approx 55% complete for ERR899427_trimmed.fq.gz\n",
      "Approx 55% complete for ERR899428_trimmed.fq.gz\n",
      "Approx 50% complete for ERR899430_trimmed.fq.gz\n",
      "Approx 45% complete for ERR899438_trimmed.fq.gz\n",
      "Approx 50% complete for ERR899431_trimmed.fq.gz\n",
      "Approx 50% complete for ERR899426_trimmed.fq.gz\n",
      "Approx 50% complete for ERR899429_trimmed.fq.gz\n",
      "Approx 55% complete for ERR899435_trimmed.fq.gz\n",
      "Approx 50% complete for ERR899433_trimmed.fq.gz\n",
      "Approx 55% complete for ERR899421_trimmed.fq.gz\n",
      "Approx 55% complete for ERR899432_trimmed.fq.gz\n",
      "Approx 55% complete for ERR899436_trimmed.fq.gz\n",
      "Approx 55% complete for ERR899434_trimmed.fq.gz\n",
      "Approx 55% complete for ERR899423_trimmed.fq.gz\n",
      "Approx 55% complete for ERR899437_trimmed.fq.gz\n",
      "Approx 55% complete for ERR899424_trimmed.fq.gz\n",
      "Approx 60% complete for ERR899427_trimmed.fq.gz\n",
      "Approx 50% complete for ERR899425_trimmed.fq.gz\n",
      "Approx 55% complete for ERR899422_trimmed.fq.gz\n",
      "Approx 60% complete for ERR899428_trimmed.fq.gz\n",
      "Approx 55% complete for ERR899430_trimmed.fq.gz\n",
      "Approx 55% complete for ERR899431_trimmed.fq.gz\n",
      "Approx 55% complete for ERR899426_trimmed.fq.gz\n",
      "Approx 55% complete for ERR899429_trimmed.fq.gz\n",
      "Approx 50% complete for ERR899438_trimmed.fq.gz\n",
      "Approx 60% complete for ERR899435_trimmed.fq.gz\n",
      "Approx 60% complete for ERR899421_trimmed.fq.gz\n",
      "Approx 60% complete for ERR899432_trimmed.fq.gz\n",
      "Approx 60% complete for ERR899436_trimmed.fq.gz\n",
      "Approx 55% complete for ERR899433_trimmed.fq.gz\n",
      "Approx 60% complete for ERR899434_trimmed.fq.gz\n",
      "Approx 60% complete for ERR899423_trimmed.fq.gz\n",
      "Approx 65% complete for ERR899427_trimmed.fq.gz\n",
      "Approx 60% complete for ERR899424_trimmed.fq.gz\n",
      "Approx 60% complete for ERR899437_trimmed.fq.gz\n",
      "Approx 60% complete for ERR899422_trimmed.fq.gz\n",
      "Approx 55% complete for ERR899425_trimmed.fq.gz\n",
      "Approx 65% complete for ERR899428_trimmed.fq.gz\n",
      "Approx 60% complete for ERR899430_trimmed.fq.gz\n",
      "Approx 60% complete for ERR899426_trimmed.fq.gz\n",
      "Approx 60% complete for ERR899431_trimmed.fq.gz\n",
      "Approx 60% complete for ERR899429_trimmed.fq.gz\n",
      "Approx 65% complete for ERR899435_trimmed.fq.gz\n",
      "Approx 65% complete for ERR899421_trimmed.fq.gz\n",
      "Approx 55% complete for ERR899438_trimmed.fq.gz\n",
      "Approx 65% complete for ERR899436_trimmed.fq.gz\n",
      "Approx 65% complete for ERR899432_trimmed.fq.gz\n",
      "Approx 60% complete for ERR899433_trimmed.fq.gz\n",
      "Approx 65% complete for ERR899434_trimmed.fq.gz\n",
      "Approx 70% complete for ERR899427_trimmed.fq.gz\n",
      "Approx 65% complete for ERR899423_trimmed.fq.gz\n",
      "Approx 65% complete for ERR899437_trimmed.fq.gz\n",
      "Approx 65% complete for ERR899424_trimmed.fq.gz\n",
      "Approx 70% complete for ERR899428_trimmed.fq.gz\n",
      "Approx 65% complete for ERR899422_trimmed.fq.gz\n",
      "Approx 60% complete for ERR899425_trimmed.fq.gz\n",
      "Approx 65% complete for ERR899430_trimmed.fq.gz\n",
      "Approx 65% complete for ERR899426_trimmed.fq.gz\n",
      "Approx 70% complete for ERR899435_trimmed.fq.gz\n",
      "Approx 65% complete for ERR899431_trimmed.fq.gz\n",
      "Approx 65% complete for ERR899429_trimmed.fq.gz\n",
      "Approx 70% complete for ERR899421_trimmed.fq.gz\n",
      "Approx 70% complete for ERR899432_trimmed.fq.gz\n",
      "Approx 70% complete for ERR899436_trimmed.fq.gz\n",
      "Approx 60% complete for ERR899438_trimmed.fq.gz\n",
      "Approx 75% complete for ERR899427_trimmed.fq.gz\n",
      "Approx 70% complete for ERR899434_trimmed.fq.gz\n",
      "Approx 65% complete for ERR899433_trimmed.fq.gz\n",
      "Approx 70% complete for ERR899423_trimmed.fq.gz\n",
      "Approx 70% complete for ERR899437_trimmed.fq.gz\n",
      "Approx 70% complete for ERR899424_trimmed.fq.gz\n",
      "Approx 75% complete for ERR899428_trimmed.fq.gz\n",
      "Approx 70% complete for ERR899422_trimmed.fq.gz\n",
      "Approx 70% complete for ERR899430_trimmed.fq.gz\n",
      "Approx 65% complete for ERR899425_trimmed.fq.gz\n",
      "Approx 75% complete for ERR899435_trimmed.fq.gz\n",
      "Approx 70% complete for ERR899426_trimmed.fq.gz\n",
      "Approx 70% complete for ERR899431_trimmed.fq.gz\n",
      "Approx 70% complete for ERR899429_trimmed.fq.gz\n",
      "Approx 75% complete for ERR899421_trimmed.fq.gz\n",
      "Approx 80% complete for ERR899427_trimmed.fq.gz\n",
      "Approx 75% complete for ERR899436_trimmed.fq.gz\n",
      "Approx 75% complete for ERR899432_trimmed.fq.gz\n",
      "Approx 75% complete for ERR899434_trimmed.fq.gz\n",
      "Approx 65% complete for ERR899438_trimmed.fq.gz\n",
      "Approx 75% complete for ERR899423_trimmed.fq.gz\n",
      "Approx 70% complete for ERR899433_trimmed.fq.gz\n",
      "Approx 75% complete for ERR899437_trimmed.fq.gz\n",
      "Approx 75% complete for ERR899424_trimmed.fq.gz\n",
      "Approx 80% complete for ERR899428_trimmed.fq.gz\n",
      "Approx 75% complete for ERR899422_trimmed.fq.gz\n",
      "Approx 75% complete for ERR899430_trimmed.fq.gz\n",
      "Approx 80% complete for ERR899435_trimmed.fq.gz\n",
      "Approx 70% complete for ERR899425_trimmed.fq.gz\n",
      "Approx 75% complete for ERR899426_trimmed.fq.gz\n",
      "Approx 80% complete for ERR899421_trimmed.fq.gz\n",
      "Approx 85% complete for ERR899427_trimmed.fq.gz\n",
      "Approx 75% complete for ERR899431_trimmed.fq.gz\n",
      "Approx 75% complete for ERR899429_trimmed.fq.gz\n",
      "Approx 80% complete for ERR899436_trimmed.fq.gz\n",
      "Approx 80% complete for ERR899432_trimmed.fq.gz\n",
      "Approx 80% complete for ERR899434_trimmed.fq.gz\n",
      "Approx 80% complete for ERR899423_trimmed.fq.gz\n",
      "Approx 85% complete for ERR899428_trimmed.fq.gz\n",
      "Approx 80% complete for ERR899424_trimmed.fq.gz\n",
      "Approx 80% complete for ERR899437_trimmed.fq.gz\n",
      "Approx 75% complete for ERR899433_trimmed.fq.gz\n",
      "Approx 70% complete for ERR899438_trimmed.fq.gz\n",
      "Approx 80% complete for ERR899422_trimmed.fq.gz\n",
      "Approx 80% complete for ERR899430_trimmed.fq.gz\n",
      "Approx 85% complete for ERR899435_trimmed.fq.gz\n",
      "Approx 90% complete for ERR899427_trimmed.fq.gz\n",
      "Approx 85% complete for ERR899421_trimmed.fq.gz\n",
      "Approx 80% complete for ERR899426_trimmed.fq.gz\n",
      "Approx 75% complete for ERR899425_trimmed.fq.gz\n",
      "Approx 85% complete for ERR899436_trimmed.fq.gz\n",
      "Approx 85% complete for ERR899432_trimmed.fq.gz\n",
      "Approx 80% complete for ERR899429_trimmed.fq.gz\n",
      "Approx 80% complete for ERR899431_trimmed.fq.gz\n",
      "Approx 85% complete for ERR899434_trimmed.fq.gz\n",
      "Approx 85% complete for ERR899423_trimmed.fq.gz\n",
      "Approx 90% complete for ERR899428_trimmed.fq.gz\n",
      "Approx 85% complete for ERR899424_trimmed.fq.gz\n",
      "Approx 85% complete for ERR899437_trimmed.fq.gz\n",
      "Approx 80% complete for ERR899433_trimmed.fq.gz\n",
      "Approx 85% complete for ERR899422_trimmed.fq.gz\n",
      "Approx 75% complete for ERR899438_trimmed.fq.gz\n",
      "Approx 90% complete for ERR899435_trimmed.fq.gz\n",
      "Approx 85% complete for ERR899430_trimmed.fq.gz\n",
      "Approx 95% complete for ERR899427_trimmed.fq.gz\n",
      "Approx 90% complete for ERR899421_trimmed.fq.gz\n",
      "Approx 85% complete for ERR899426_trimmed.fq.gz\n",
      "Approx 90% complete for ERR899436_trimmed.fq.gz\n",
      "Approx 90% complete for ERR899432_trimmed.fq.gz\n",
      "Approx 85% complete for ERR899429_trimmed.fq.gz\n",
      "Approx 90% complete for ERR899434_trimmed.fq.gz\n",
      "Approx 80% complete for ERR899425_trimmed.fq.gz\n",
      "Approx 85% complete for ERR899431_trimmed.fq.gz\n",
      "Approx 95% complete for ERR899428_trimmed.fq.gz\n",
      "Approx 90% complete for ERR899423_trimmed.fq.gz\n",
      "Approx 90% complete for ERR899424_trimmed.fq.gz\n",
      "Approx 90% complete for ERR899437_trimmed.fq.gz\n",
      "Approx 85% complete for ERR899433_trimmed.fq.gz\n",
      "Approx 90% complete for ERR899422_trimmed.fq.gz\n",
      "Approx 80% complete for ERR899438_trimmed.fq.gz\n",
      "Approx 95% complete for ERR899435_trimmed.fq.gz\n",
      "Analysis complete for ERR899427_trimmed.fq.gz\n",
      "Approx 90% complete for ERR899430_trimmed.fq.gz\n",
      "Approx 95% complete for ERR899421_trimmed.fq.gz\n",
      "Approx 95% complete for ERR899436_trimmed.fq.gz\n",
      "Approx 95% complete for ERR899432_trimmed.fq.gz\n",
      "Approx 90% complete for ERR899426_trimmed.fq.gz\n",
      "Approx 95% complete for ERR899434_trimmed.fq.gz\n",
      "Approx 90% complete for ERR899429_trimmed.fq.gz\n",
      "Approx 90% complete for ERR899431_trimmed.fq.gz\n",
      "Analysis complete for ERR899428_trimmed.fq.gz\n",
      "Approx 85% complete for ERR899425_trimmed.fq.gz\n",
      "Approx 95% complete for ERR899423_trimmed.fq.gz\n",
      "Approx 95% complete for ERR899424_trimmed.fq.gz\n",
      "Approx 95% complete for ERR899437_trimmed.fq.gz\n",
      "Approx 90% complete for ERR899433_trimmed.fq.gz\n",
      "Approx 95% complete for ERR899422_trimmed.fq.gz\n",
      "Analysis complete for ERR899435_trimmed.fq.gz\n",
      "Approx 95% complete for ERR899430_trimmed.fq.gz\n",
      "Approx 85% complete for ERR899438_trimmed.fq.gz\n",
      "Analysis complete for ERR899421_trimmed.fq.gz\n",
      "Analysis complete for ERR899432_trimmed.fq.gz\n",
      "Analysis complete for ERR899436_trimmed.fq.gz\n",
      "Approx 95% complete for ERR899426_trimmed.fq.gz\n",
      "Analysis complete for ERR899434_trimmed.fq.gz\n",
      "Approx 95% complete for ERR899431_trimmed.fq.gz\n",
      "Approx 95% complete for ERR899429_trimmed.fq.gz\n",
      "Analysis complete for ERR899423_trimmed.fq.gz\n",
      "Approx 90% complete for ERR899425_trimmed.fq.gz\n",
      "Analysis complete for ERR899424_trimmed.fq.gz\n",
      "Analysis complete for ERR899437_trimmed.fq.gz\n",
      "Analysis complete for ERR899422_trimmed.fq.gz\n",
      "Approx 95% complete for ERR899433_trimmed.fq.gz\n",
      "Analysis complete for ERR899430_trimmed.fq.gz\n",
      "Approx 90% complete for ERR899438_trimmed.fq.gz\n",
      "Analysis complete for ERR899426_trimmed.fq.gz\n",
      "Analysis complete for ERR899429_trimmed.fq.gz\n",
      "Analysis complete for ERR899431_trimmed.fq.gz\n",
      "Approx 95% complete for ERR899425_trimmed.fq.gz\n",
      "Analysis complete for ERR899433_trimmed.fq.gz\n",
      "Approx 95% complete for ERR899438_trimmed.fq.gz\n",
      "Analysis complete for ERR899425_trimmed.fq.gz\n",
      "Analysis complete for ERR899438_trimmed.fq.gz\n"
     ]
    }
   ],
   "source": [
    "old_cmd = '''\n",
    "#fastqc of trimmed_sequencing reads\n",
    "\n",
    "#load fastqc\n",
    "module load fastqc\n",
    "\n",
    "fqs=\"\"\n",
    "\n",
    "for basename in $(ls $FASTQS | cut -f1 -d '.'); do\n",
    "    #basename=$(echo $fq | cut -f1 -d '.')\n",
    "    fqd=$TFASTQS\n",
    "    fq1=\".fastq.gz\"\n",
    "    fq1=$fqd/$basename$fq1\n",
    "    fqs=\"$fqs $fq1\"\n",
    "    \n",
    "done\n",
    "\n",
    "for basename in $(ls $P$FASTQS | cut -f1 -d '_' | sort | uniq); do\n",
    "    echo $basename\n",
    "    fqd=$PTFASTQS\n",
    "    fq1=\"_trimmed_1.fastq.gz\"\n",
    "    fq2=\"_trimmed_2.fastq.gz\"\n",
    "    fq1=$fqd/$basename$fq1\n",
    "    fq2=$fqd/$basename$fq2\n",
    "    fqs=\"$fqs $fq1 $fq2\"\n",
    "done\n",
    "\n",
    "echo \"Performing FastQC for $fqs\"\n",
    "fastqc --threads $MAXCORES $fqs -o $TFASTQCOUT\n",
    "'''\n",
    "\n",
    "#out=run_cmd(cmd)\n",
    "\n",
    "tfastq_files = glob.glob(TFASTQS+\"/*.fq.gz\") \n",
    "\n",
    "cmds = []\n",
    "for tfastq_file in tfastq_files:\n",
    "    cmds.append(ENVCMD + \" && \" +\n",
    "                \"module load fastqc && \" +\n",
    "                \"fastqc \" + \n",
    "                tfastq_file +\n",
    "                \" -o \" + TFASTQCOUT)\n",
    "\n",
    "pool = Pool(processes=MAXCORES)\n",
    "results = pool.map_async(run_cmd, cmds)\n",
    "pool.close()\n",
    "pool.join()\n",
    "pool.terminate()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 48,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Aligning reads from ERR899421 to the reference genome\n",
      "Jan 19 12:19:38 ..... started STAR run\n",
      "Jan 19 12:19:38 ..... loading genome\n",
      "Jan 19 12:19:52 ..... processing annotations GTF\n",
      "Jan 19 12:20:25 ..... inserting junctions into the genome indices\n",
      "Jan 19 12:21:33 ..... started mapping\n",
      "Jan 19 12:23:17 ..... finished successfully\n",
      "Aligning reads from ERR899422 to the reference genome\n",
      "Jan 19 12:23:43 ..... started STAR run\n",
      "Jan 19 12:23:44 ..... loading genome\n",
      "Jan 19 12:23:57 ..... processing annotations GTF\n",
      "Jan 19 12:24:06 ..... inserting junctions into the genome indices\n",
      "Jan 19 12:25:13 ..... started mapping\n",
      "Jan 19 12:26:56 ..... finished successfully\n",
      "Aligning reads from ERR899423 to the reference genome\n",
      "Jan 19 12:27:25 ..... started STAR run\n",
      "Jan 19 12:27:25 ..... loading genome\n",
      "Jan 19 12:27:38 ..... processing annotations GTF\n",
      "Jan 19 12:28:12 ..... inserting junctions into the genome indices\n",
      "Jan 19 12:29:19 ..... started mapping\n",
      "Jan 19 12:30:59 ..... finished successfully\n",
      "Aligning reads from ERR899424 to the reference genome\n",
      "Jan 19 12:31:29 ..... started STAR run\n",
      "Jan 19 12:31:29 ..... loading genome\n",
      "Jan 19 12:31:41 ..... processing annotations GTF\n",
      "Jan 19 12:31:50 ..... inserting junctions into the genome indices\n",
      "Jan 19 12:32:57 ..... started mapping\n",
      "Jan 19 12:34:39 ..... finished successfully\n",
      "Aligning reads from ERR899425 to the reference genome\n",
      "Jan 19 12:35:07 ..... started STAR run\n",
      "Jan 19 12:35:08 ..... loading genome\n",
      "Jan 19 12:35:21 ..... processing annotations GTF\n",
      "Jan 19 12:35:56 ..... inserting junctions into the genome indices\n",
      "Jan 19 12:37:02 ..... started mapping\n",
      "Jan 19 12:38:55 ..... finished successfully\n",
      "Aligning reads from ERR899426 to the reference genome\n",
      "Jan 19 12:39:24 ..... started STAR run\n",
      "Jan 19 12:39:24 ..... loading genome\n",
      "Jan 19 12:39:37 ..... processing annotations GTF\n",
      "Jan 19 12:39:45 ..... inserting junctions into the genome indices\n",
      "Jan 19 12:40:52 ..... started mapping\n",
      "Jan 19 12:42:41 ..... finished successfully\n",
      "Aligning reads from ERR899427 to the reference genome\n",
      "Jan 19 12:43:06 ..... started STAR run\n",
      "Jan 19 12:43:07 ..... loading genome\n",
      "Jan 19 12:43:19 ..... processing annotations GTF\n",
      "Jan 19 12:43:53 ..... inserting junctions into the genome indices\n",
      "Jan 19 12:45:01 ..... started mapping\n",
      "Jan 19 12:46:34 ..... finished successfully\n",
      "Aligning reads from ERR899428 to the reference genome\n",
      "Jan 19 12:47:00 ..... started STAR run\n",
      "Jan 19 12:47:00 ..... loading genome\n",
      "Jan 19 12:47:13 ..... processing annotations GTF\n",
      "Jan 19 12:47:22 ..... inserting junctions into the genome indices\n",
      "Jan 19 12:48:28 ..... started mapping\n",
      "Jan 19 12:50:09 ..... finished successfully\n",
      "Aligning reads from ERR899429 to the reference genome\n",
      "Jan 19 12:50:36 ..... started STAR run\n",
      "Jan 19 12:50:36 ..... loading genome\n",
      "Jan 19 12:50:50 ..... processing annotations GTF\n",
      "Jan 19 12:51:38 ..... inserting junctions into the genome indices\n",
      "Jan 19 12:52:44 ..... started mapping\n",
      "Jan 19 12:54:30 ..... finished successfully\n",
      "Aligning reads from ERR899430 to the reference genome\n",
      "Jan 19 12:55:05 ..... started STAR run\n",
      "Jan 19 12:55:05 ..... loading genome\n",
      "Jan 19 12:56:40 ..... processing annotations GTF\n",
      "Jan 19 12:56:50 ..... inserting junctions into the genome indices\n",
      "Jan 19 12:57:55 ..... started mapping\n",
      "Jan 19 12:59:42 ..... finished successfully\n",
      "Aligning reads from ERR899431 to the reference genome\n",
      "Jan 19 13:00:11 ..... started STAR run\n",
      "Jan 19 13:00:11 ..... loading genome\n",
      "Jan 19 13:00:25 ..... processing annotations GTF\n",
      "Jan 19 13:00:34 ..... inserting junctions into the genome indices\n",
      "Jan 19 13:01:40 ..... started mapping\n",
      "Jan 19 13:03:25 ..... finished successfully\n",
      "Aligning reads from ERR899432 to the reference genome\n",
      "Jan 19 13:03:55 ..... started STAR run\n",
      "Jan 19 13:03:55 ..... loading genome\n",
      "Jan 19 13:04:09 ..... processing annotations GTF\n",
      "Jan 19 13:04:19 ..... inserting junctions into the genome indices\n",
      "Jan 19 13:05:25 ..... started mapping\n",
      "Jan 19 13:07:08 ..... finished successfully\n",
      "Aligning reads from ERR899433 to the reference genome\n",
      "Jan 19 13:07:35 ..... started STAR run\n",
      "Jan 19 13:07:35 ..... loading genome\n",
      "Jan 19 13:08:00 ..... processing annotations GTF\n",
      "Jan 19 13:08:09 ..... inserting junctions into the genome indices\n",
      "Jan 19 13:09:14 ..... started mapping\n",
      "Jan 19 13:11:03 ..... finished successfully\n",
      "Aligning reads from ERR899434 to the reference genome\n",
      "Jan 19 13:11:32 ..... started STAR run\n",
      "Jan 19 13:11:32 ..... loading genome\n",
      "Jan 19 13:11:47 ..... processing annotations GTF\n",
      "Jan 19 13:11:56 ..... inserting junctions into the genome indices\n",
      "Jan 19 13:13:01 ..... started mapping\n",
      "Jan 19 13:14:43 ..... finished successfully\n",
      "Aligning reads from ERR899435 to the reference genome\n",
      "Jan 19 13:15:11 ..... started STAR run\n",
      "Jan 19 13:15:11 ..... loading genome\n",
      "Jan 19 13:15:26 ..... processing annotations GTF\n",
      "Jan 19 13:15:37 ..... inserting junctions into the genome indices\n",
      "Jan 19 13:16:42 ..... started mapping\n",
      "Jan 19 13:18:26 ..... finished successfully\n",
      "Aligning reads from ERR899436 to the reference genome\n",
      "Jan 19 13:18:53 ..... started STAR run\n",
      "Jan 19 13:18:53 ..... loading genome\n",
      "Jan 19 13:19:07 ..... processing annotations GTF\n",
      "Jan 19 13:19:16 ..... inserting junctions into the genome indices\n",
      "Jan 19 13:20:21 ..... started mapping\n",
      "Jan 19 13:22:04 ..... finished successfully\n",
      "Aligning reads from ERR899437 to the reference genome\n",
      "Jan 19 13:22:32 ..... started STAR run\n",
      "Jan 19 13:22:32 ..... loading genome\n",
      "Jan 19 13:22:46 ..... processing annotations GTF\n",
      "Jan 19 13:22:55 ..... inserting junctions into the genome indices\n",
      "Jan 19 13:24:01 ..... started mapping\n",
      "Jan 19 13:25:45 ..... finished successfully\n",
      "Aligning reads from ERR899438 to the reference genome\n",
      "Jan 19 13:26:13 ..... started STAR run\n",
      "Jan 19 13:26:13 ..... loading genome\n",
      "Jan 19 13:26:27 ..... processing annotations GTF\n",
      "Jan 19 13:26:37 ..... inserting junctions into the genome indices\n",
      "Jan 19 13:27:41 ..... started mapping\n",
      "Jan 19 13:29:38 ..... finished successfully\n"
     ]
    }
   ],
   "source": [
    "cmd='''\n",
    "#module load star\n",
    "\n",
    "for basename in $(ls $FASTQC | cut -f1 -d '.'); do\n",
    "    fqd=$TFASTQS\n",
    "    fq1=\".fastq.gz\"\n",
    "    fq1=$fqd/$basename$fq1\n",
    "    \n",
    "    echo \"Aligning reads from $basename to the reference genome\"\n",
    "    $BINDIR/STAR \\\n",
    "        --genomeDir $STARGENOMEDIR \\\n",
    "        --sjdbGTFfile $GENEANNOTATIONDIR/genes.gtf \\\n",
    "        --runThreadN $MAXCORES \\\n",
    "        --outSAMstrandField intronMotif \\\n",
    "        --outFilterIntronMotifs RemoveNoncanonical \\\n",
    "        --outFileNamePrefix $STARDIR/$basename  \\\n",
    "        --readFilesIn $fq1 \\\n",
    "        --readFilesCommand zcat \\\n",
    "        --outSAMtype BAM Unsorted \\\n",
    "        --outReadsUnmapped Fastx \\\n",
    "        --outSAMmode Full\n",
    "done\n",
    "\n",
    "\n",
    "for basename in $(ls $PFASTQC | cut -f1 -d '_' | sort | uniq); do\n",
    "    fqd=$TPFASTQS\n",
    "    fq1=\"trimmed_1.fastq.gz\"\n",
    "    fq2=\"trimmed_2.fastq.gz\"\n",
    "    fq1=$fqd/$basename$fq1\n",
    "    fq2=$fqd/$basename$fq2\n",
    "    fqs=\"$fqs $fq1 $fq2\"\n",
    "    \n",
    "    \n",
    "    echo \"Aligning reads from $basename to the reference genome\"\n",
    "    $BINDIR/STAR \\\n",
    "        --genomeDir $STARGENOMEDIR \\\n",
    "        --sjdbGTFfile $GENEANNOTATIONDIR/genes.gtf \\\n",
    "        --runThreadN $MAXCORES \\\n",
    "        --outSAMstrandField intronMotif \\\n",
    "        --outFilterIntronMotifs RemoveNoncanonical \\\n",
    "        --outFileNamePrefix $STARDIR/$basename \\\n",
    "        --readFilesIn $fq1 $fq2 \\\n",
    "        --readFilesCommand zcat \\\n",
    "        --outSAMtype BAM Unsorted \\\n",
    "        --outReadsUnmapped Fastx \\\n",
    "        --outSAMmode Full\n",
    "done\n",
    "'''\n",
    "#out=run_cmd(cmd)\n",
    "\n",
    "\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 54,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "        ==========     _____ _    _ ____  _____  ______          _____  \n",
      "        =====         / ____| |  | |  _ \\|  __ \\|  ____|   /\\   |  __ \\ \n",
      "          =====      | (___ | |  | | |_) | |__) | |__     /  \\  | |  | |\n",
      "            ====      \\___ \\| |  | |  _ <|  _  /|  __|   / /\\ \\ | |  | |\n",
      "              ====    ____) | |__| | |_) | | \\ \\| |____ / ____ \\| |__| |\n",
      "        ==========   |_____/ \\____/|____/|_|  \\_\\______/_/    \\_\\_____/\n",
      "\t  v1.6.0\n",
      "\n",
      "//========================== featureCounts setting ===========================\\\\\n",
      "||                                                                            ||\n",
      "||             Input files : 1 BAM file                                       ||\n",
      "||                           S ./star_output/ERR899421Aligned.out.bam         ||\n",
      "||                                                                            ||\n",
      "||             Output file : ./featureCount_output/ERR899421Aligned.count ... ||\n",
      "||                 Summary : ./featureCount_output/ERR899421Aligned.count ... ||\n",
      "||              Annotation : /fs/project/PAS1307/genomes/Mus_musculus/Ens ... ||\n",
      "||      Dir for temp files : ./featureCount_output                            ||\n",
      "||                                                                            ||\n",
      "||                 Threads : 24                                               ||\n",
      "||                   Level : meta-feature level                               ||\n",
      "||              Paired-end : no                                               ||\n",
      "||         Strand specific : no                                               ||\n",
      "||      Multimapping reads : not counted                                      ||\n",
      "|| Multi-overlapping reads : not counted                                      ||\n",
      "||   Min overlapping bases : 1                                                ||\n",
      "||                                                                            ||\n",
      "\\\\===================== http://subread.sourceforge.net/ ======================//\n",
      "\n",
      "//================================= Running ==================================\\\\\n",
      "||                                                                            ||\n",
      "|| Load annotation file /fs/project/PAS1307/genomes/Mus_musculus/Ensembl/ ... ||\n",
      "||    Features : 690647                                                       ||\n",
      "||    Meta-features : 46078                                                   ||\n",
      "||    Chromosomes/contigs : 73                                                ||\n",
      "||                                                                            ||\n",
      "|| Process BAM file ./star_output/ERR899421Aligned.out.bam...                 ||\n",
      "||    Single-end reads are included.                                          ||\n",
      "||    Assign reads to features...                                             ||\n",
      "||    Total reads : 7101659                                                   ||\n",
      "||    Successfully assigned reads : 4086182 (57.5%)                           ||\n",
      "||    Running time : 0.32 minutes                                             ||\n",
      "||                                                                            ||\n",
      "||                         Read assignment finished.                          ||\n",
      "||                                                                            ||\n",
      "|| Summary of counting results can be found in file \"./featureCount_output/E  ||\n",
      "|| RR899421Aligned.count.txt.summary\"                                         ||\n",
      "||                                                                            ||\n",
      "\\\\===================== http://subread.sourceforge.net/ ======================//\n",
      "\n",
      "\n",
      "        ==========     _____ _    _ ____  _____  ______          _____  \n",
      "        =====         / ____| |  | |  _ \\|  __ \\|  ____|   /\\   |  __ \\ \n",
      "          =====      | (___ | |  | | |_) | |__) | |__     /  \\  | |  | |\n",
      "            ====      \\___ \\| |  | |  _ <|  _  /|  __|   / /\\ \\ | |  | |\n",
      "              ====    ____) | |__| | |_) | | \\ \\| |____ / ____ \\| |__| |\n",
      "        ==========   |_____/ \\____/|____/|_|  \\_\\______/_/    \\_\\_____/\n",
      "\t  v1.6.0\n",
      "\n",
      "//========================== featureCounts setting ===========================\\\\\n",
      "||                                                                            ||\n",
      "||             Input files : 1 BAM file                                       ||\n",
      "||                           S ./star_output/ERR899422Aligned.out.bam         ||\n",
      "||                                                                            ||\n",
      "||             Output file : ./featureCount_output/ERR899422Aligned.count ... ||\n",
      "||                 Summary : ./featureCount_output/ERR899422Aligned.count ... ||\n",
      "||              Annotation : /fs/project/PAS1307/genomes/Mus_musculus/Ens ... ||\n",
      "||      Dir for temp files : ./featureCount_output                            ||\n",
      "||                                                                            ||\n",
      "||                 Threads : 24                                               ||\n",
      "||                   Level : meta-feature level                               ||\n",
      "||              Paired-end : no                                               ||\n",
      "||         Strand specific : no                                               ||\n",
      "||      Multimapping reads : not counted                                      ||\n",
      "|| Multi-overlapping reads : not counted                                      ||\n",
      "||   Min overlapping bases : 1                                                ||\n",
      "||                                                                            ||\n",
      "\\\\===================== http://subread.sourceforge.net/ ======================//\n",
      "\n",
      "//================================= Running ==================================\\\\\n",
      "||                                                                            ||\n",
      "|| Load annotation file /fs/project/PAS1307/genomes/Mus_musculus/Ensembl/ ... ||\n",
      "||    Features : 690647                                                       ||\n",
      "||    Meta-features : 46078                                                   ||\n",
      "||    Chromosomes/contigs : 73                                                ||\n",
      "||                                                                            ||\n",
      "|| Process BAM file ./star_output/ERR899422Aligned.out.bam...                 ||\n",
      "||    Single-end reads are included.                                          ||\n",
      "||    Assign reads to features...                                             ||\n",
      "||    Total reads : 8079369                                                   ||\n",
      "||    Successfully assigned reads : 4246258 (52.6%)                           ||\n",
      "||    Running time : 0.35 minutes                                             ||\n",
      "||                                                                            ||\n",
      "||                         Read assignment finished.                          ||\n",
      "||                                                                            ||\n",
      "|| Summary of counting results can be found in file \"./featureCount_output/E  ||\n",
      "|| RR899422Aligned.count.txt.summary\"                                         ||\n",
      "||                                                                            ||\n",
      "\\\\===================== http://subread.sourceforge.net/ ======================//\n",
      "\n",
      "\n",
      "        ==========     _____ _    _ ____  _____  ______          _____  \n",
      "        =====         / ____| |  | |  _ \\|  __ \\|  ____|   /\\   |  __ \\ \n",
      "          =====      | (___ | |  | | |_) | |__) | |__     /  \\  | |  | |\n",
      "            ====      \\___ \\| |  | |  _ <|  _  /|  __|   / /\\ \\ | |  | |\n",
      "              ====    ____) | |__| | |_) | | \\ \\| |____ / ____ \\| |__| |\n",
      "        ==========   |_____/ \\____/|____/|_|  \\_\\______/_/    \\_\\_____/\n",
      "\t  v1.6.0\n",
      "\n",
      "//========================== featureCounts setting ===========================\\\\\n",
      "||                                                                            ||\n",
      "||             Input files : 1 BAM file                                       ||\n",
      "||                           S ./star_output/ERR899423Aligned.out.bam         ||\n",
      "||                                                                            ||\n",
      "||             Output file : ./featureCount_output/ERR899423Aligned.count ... ||\n",
      "||                 Summary : ./featureCount_output/ERR899423Aligned.count ... ||\n",
      "||              Annotation : /fs/project/PAS1307/genomes/Mus_musculus/Ens ... ||\n",
      "||      Dir for temp files : ./featureCount_output                            ||\n",
      "||                                                                            ||\n",
      "||                 Threads : 24                                               ||\n",
      "||                   Level : meta-feature level                               ||\n",
      "||              Paired-end : no                                               ||\n",
      "||         Strand specific : no                                               ||\n",
      "||      Multimapping reads : not counted                                      ||\n",
      "|| Multi-overlapping reads : not counted                                      ||\n",
      "||   Min overlapping bases : 1                                                ||\n",
      "||                                                                            ||\n",
      "\\\\===================== http://subread.sourceforge.net/ ======================//\n",
      "\n",
      "//================================= Running ==================================\\\\\n",
      "||                                                                            ||\n",
      "|| Load annotation file /fs/project/PAS1307/genomes/Mus_musculus/Ensembl/ ... ||\n",
      "||    Features : 690647                                                       ||\n",
      "||    Meta-features : 46078                                                   ||\n",
      "||    Chromosomes/contigs : 73                                                ||\n",
      "||                                                                            ||\n",
      "|| Process BAM file ./star_output/ERR899423Aligned.out.bam...                 ||\n",
      "||    Single-end reads are included.                                          ||\n",
      "||    Assign reads to features...                                             ||\n",
      "||    Total reads : 7567977                                                   ||\n",
      "||    Successfully assigned reads : 4140773 (54.7%)                           ||\n",
      "||    Running time : 0.33 minutes                                             ||\n",
      "||                                                                            ||\n",
      "||                         Read assignment finished.                          ||\n",
      "||                                                                            ||\n",
      "|| Summary of counting results can be found in file \"./featureCount_output/E  ||\n",
      "|| RR899423Aligned.count.txt.summary\"                                         ||\n",
      "||                                                                            ||\n",
      "\\\\===================== http://subread.sourceforge.net/ ======================//\n",
      "\n",
      "\n",
      "        ==========     _____ _    _ ____  _____  ______          _____  \n",
      "        =====         / ____| |  | |  _ \\|  __ \\|  ____|   /\\   |  __ \\ \n",
      "          =====      | (___ | |  | | |_) | |__) | |__     /  \\  | |  | |\n",
      "            ====      \\___ \\| |  | |  _ <|  _  /|  __|   / /\\ \\ | |  | |\n",
      "              ====    ____) | |__| | |_) | | \\ \\| |____ / ____ \\| |__| |\n",
      "        ==========   |_____/ \\____/|____/|_|  \\_\\______/_/    \\_\\_____/\n",
      "\t  v1.6.0\n",
      "\n",
      "//========================== featureCounts setting ===========================\\\\\n",
      "||                                                                            ||\n",
      "||             Input files : 1 BAM file                                       ||\n",
      "||                           S ./star_output/ERR899424Aligned.out.bam         ||\n",
      "||                                                                            ||\n",
      "||             Output file : ./featureCount_output/ERR899424Aligned.count ... ||\n",
      "||                 Summary : ./featureCount_output/ERR899424Aligned.count ... ||\n",
      "||              Annotation : /fs/project/PAS1307/genomes/Mus_musculus/Ens ... ||\n",
      "||      Dir for temp files : ./featureCount_output                            ||\n",
      "||                                                                            ||\n",
      "||                 Threads : 24                                               ||\n",
      "||                   Level : meta-feature level                               ||\n",
      "||              Paired-end : no                                               ||\n",
      "||         Strand specific : no                                               ||\n",
      "||      Multimapping reads : not counted                                      ||\n",
      "|| Multi-overlapping reads : not counted                                      ||\n",
      "||   Min overlapping bases : 1                                                ||\n",
      "||                                                                            ||\n",
      "\\\\===================== http://subread.sourceforge.net/ ======================//\n",
      "\n",
      "//================================= Running ==================================\\\\\n",
      "||                                                                            ||\n",
      "|| Load annotation file /fs/project/PAS1307/genomes/Mus_musculus/Ensembl/ ... ||\n",
      "||    Features : 690647                                                       ||\n",
      "||    Meta-features : 46078                                                   ||\n",
      "||    Chromosomes/contigs : 73                                                ||\n",
      "||                                                                            ||\n",
      "|| Process BAM file ./star_output/ERR899424Aligned.out.bam...                 ||\n",
      "||    Single-end reads are included.                                          ||\n",
      "||    Assign reads to features...                                             ||\n",
      "||    Total reads : 8060973                                                   ||\n",
      "||    Successfully assigned reads : 4075483 (50.6%)                           ||\n",
      "||    Running time : 0.37 minutes                                             ||\n",
      "||                                                                            ||\n",
      "||                         Read assignment finished.                          ||\n",
      "||                                                                            ||\n",
      "|| Summary of counting results can be found in file \"./featureCount_output/E  ||\n",
      "|| RR899424Aligned.count.txt.summary\"                                         ||\n",
      "||                                                                            ||\n",
      "\\\\===================== http://subread.sourceforge.net/ ======================//\n",
      "\n",
      "\n",
      "        ==========     _____ _    _ ____  _____  ______          _____  \n",
      "        =====         / ____| |  | |  _ \\|  __ \\|  ____|   /\\   |  __ \\ \n",
      "          =====      | (___ | |  | | |_) | |__) | |__     /  \\  | |  | |\n",
      "            ====      \\___ \\| |  | |  _ <|  _  /|  __|   / /\\ \\ | |  | |\n",
      "              ====    ____) | |__| | |_) | | \\ \\| |____ / ____ \\| |__| |\n",
      "        ==========   |_____/ \\____/|____/|_|  \\_\\______/_/    \\_\\_____/\n",
      "\t  v1.6.0\n",
      "\n",
      "//========================== featureCounts setting ===========================\\\\\n",
      "||                                                                            ||\n",
      "||             Input files : 1 BAM file                                       ||\n",
      "||                           S ./star_output/ERR899425Aligned.out.bam         ||\n",
      "||                                                                            ||\n",
      "||             Output file : ./featureCount_output/ERR899425Aligned.count ... ||\n",
      "||                 Summary : ./featureCount_output/ERR899425Aligned.count ... ||\n",
      "||              Annotation : /fs/project/PAS1307/genomes/Mus_musculus/Ens ... ||\n",
      "||      Dir for temp files : ./featureCount_output                            ||\n",
      "||                                                                            ||\n",
      "||                 Threads : 24                                               ||\n",
      "||                   Level : meta-feature level                               ||\n",
      "||              Paired-end : no                                               ||\n",
      "||         Strand specific : no                                               ||\n",
      "||      Multimapping reads : not counted                                      ||\n",
      "|| Multi-overlapping reads : not counted                                      ||\n",
      "||   Min overlapping bases : 1                                                ||\n",
      "||                                                                            ||\n",
      "\\\\===================== http://subread.sourceforge.net/ ======================//\n",
      "\n",
      "//================================= Running ==================================\\\\\n",
      "||                                                                            ||\n",
      "|| Load annotation file /fs/project/PAS1307/genomes/Mus_musculus/Ensembl/ ... ||\n",
      "||    Features : 690647                                                       ||\n",
      "||    Meta-features : 46078                                                   ||\n",
      "||    Chromosomes/contigs : 73                                                ||\n",
      "||                                                                            ||\n",
      "|| Process BAM file ./star_output/ERR899425Aligned.out.bam...                 ||\n",
      "||    Single-end reads are included.                                          ||\n",
      "||    Assign reads to features...                                             ||\n",
      "||    Total reads : 8654636                                                   ||\n",
      "||    Successfully assigned reads : 4655774 (53.8%)                           ||\n",
      "||    Running time : 0.38 minutes                                             ||\n",
      "||                                                                            ||\n",
      "||                         Read assignment finished.                          ||\n",
      "||                                                                            ||\n",
      "|| Summary of counting results can be found in file \"./featureCount_output/E  ||\n",
      "|| RR899425Aligned.count.txt.summary\"                                         ||\n",
      "||                                                                            ||\n",
      "\\\\===================== http://subread.sourceforge.net/ ======================//\n",
      "\n",
      "\n",
      "        ==========     _____ _    _ ____  _____  ______          _____  \n",
      "        =====         / ____| |  | |  _ \\|  __ \\|  ____|   /\\   |  __ \\ \n",
      "          =====      | (___ | |  | | |_) | |__) | |__     /  \\  | |  | |\n",
      "            ====      \\___ \\| |  | |  _ <|  _  /|  __|   / /\\ \\ | |  | |\n",
      "              ====    ____) | |__| | |_) | | \\ \\| |____ / ____ \\| |__| |\n",
      "        ==========   |_____/ \\____/|____/|_|  \\_\\______/_/    \\_\\_____/\n",
      "\t  v1.6.0\n",
      "\n",
      "//========================== featureCounts setting ===========================\\\\\n",
      "||                                                                            ||\n",
      "||             Input files : 1 BAM file                                       ||\n",
      "||                           S ./star_output/ERR899426Aligned.out.bam         ||\n",
      "||                                                                            ||\n",
      "||             Output file : ./featureCount_output/ERR899426Aligned.count ... ||\n",
      "||                 Summary : ./featureCount_output/ERR899426Aligned.count ... ||\n",
      "||              Annotation : /fs/project/PAS1307/genomes/Mus_musculus/Ens ... ||\n",
      "||      Dir for temp files : ./featureCount_output                            ||\n",
      "||                                                                            ||\n",
      "||                 Threads : 24                                               ||\n",
      "||                   Level : meta-feature level                               ||\n",
      "||              Paired-end : no                                               ||\n",
      "||         Strand specific : no                                               ||\n",
      "||      Multimapping reads : not counted                                      ||\n",
      "|| Multi-overlapping reads : not counted                                      ||\n",
      "||   Min overlapping bases : 1                                                ||\n",
      "||                                                                            ||\n",
      "\\\\===================== http://subread.sourceforge.net/ ======================//\n",
      "\n",
      "//================================= Running ==================================\\\\\n",
      "||                                                                            ||\n",
      "|| Load annotation file /fs/project/PAS1307/genomes/Mus_musculus/Ensembl/ ... ||\n",
      "||    Features : 690647                                                       ||\n",
      "||    Meta-features : 46078                                                   ||\n",
      "||    Chromosomes/contigs : 73                                                ||\n",
      "||                                                                            ||\n",
      "|| Process BAM file ./star_output/ERR899426Aligned.out.bam...                 ||\n",
      "||    Single-end reads are included.                                          ||\n",
      "||    Assign reads to features...                                             ||\n",
      "||    Total reads : 10659421                                                  ||\n",
      "||    Successfully assigned reads : 4123752 (38.7%)                           ||\n",
      "||    Running time : 0.44 minutes                                             ||\n",
      "||                                                                            ||\n",
      "||                         Read assignment finished.                          ||\n",
      "||                                                                            ||\n",
      "|| Summary of counting results can be found in file \"./featureCount_output/E  ||\n",
      "|| RR899426Aligned.count.txt.summary\"                                         ||\n",
      "||                                                                            ||\n",
      "\\\\===================== http://subread.sourceforge.net/ ======================//\n",
      "\n",
      "\n",
      "        ==========     _____ _    _ ____  _____  ______          _____  \n",
      "        =====         / ____| |  | |  _ \\|  __ \\|  ____|   /\\   |  __ \\ \n",
      "          =====      | (___ | |  | | |_) | |__) | |__     /  \\  | |  | |\n",
      "            ====      \\___ \\| |  | |  _ <|  _  /|  __|   / /\\ \\ | |  | |\n",
      "              ====    ____) | |__| | |_) | | \\ \\| |____ / ____ \\| |__| |\n",
      "        ==========   |_____/ \\____/|____/|_|  \\_\\______/_/    \\_\\_____/\n",
      "\t  v1.6.0\n",
      "\n",
      "//========================== featureCounts setting ===========================\\\\\n",
      "||                                                                            ||\n",
      "||             Input files : 1 BAM file                                       ||\n",
      "||                           S ./star_output/ERR899427Aligned.out.bam         ||\n",
      "||                                                                            ||\n",
      "||             Output file : ./featureCount_output/ERR899427Aligned.count ... ||\n",
      "||                 Summary : ./featureCount_output/ERR899427Aligned.count ... ||\n",
      "||              Annotation : /fs/project/PAS1307/genomes/Mus_musculus/Ens ... ||\n",
      "||      Dir for temp files : ./featureCount_output                            ||\n",
      "||                                                                            ||\n",
      "||                 Threads : 24                                               ||\n",
      "||                   Level : meta-feature level                               ||\n",
      "||              Paired-end : no                                               ||\n",
      "||         Strand specific : no                                               ||\n",
      "||      Multimapping reads : not counted                                      ||\n",
      "|| Multi-overlapping reads : not counted                                      ||\n",
      "||   Min overlapping bases : 1                                                ||\n",
      "||                                                                            ||\n",
      "\\\\===================== http://subread.sourceforge.net/ ======================//\n",
      "\n",
      "//================================= Running ==================================\\\\\n",
      "||                                                                            ||\n",
      "|| Load annotation file /fs/project/PAS1307/genomes/Mus_musculus/Ensembl/ ... ||\n",
      "||    Features : 690647                                                       ||\n",
      "||    Meta-features : 46078                                                   ||\n",
      "||    Chromosomes/contigs : 73                                                ||\n",
      "||                                                                            ||\n",
      "|| Process BAM file ./star_output/ERR899427Aligned.out.bam...                 ||\n",
      "||    Single-end reads are included.                                          ||\n",
      "||    Assign reads to features...                                             ||\n",
      "||    Total reads : 7615582                                                   ||\n",
      "||    Successfully assigned reads : 3890283 (51.1%)                           ||\n",
      "||    Running time : 0.35 minutes                                             ||\n",
      "||                                                                            ||\n",
      "||                         Read assignment finished.                          ||\n",
      "||                                                                            ||\n",
      "|| Summary of counting results can be found in file \"./featureCount_output/E  ||\n",
      "|| RR899427Aligned.count.txt.summary\"                                         ||\n",
      "||                                                                            ||\n",
      "\\\\===================== http://subread.sourceforge.net/ ======================//\n",
      "\n",
      "\n",
      "        ==========     _____ _    _ ____  _____  ______          _____  \n",
      "        =====         / ____| |  | |  _ \\|  __ \\|  ____|   /\\   |  __ \\ \n",
      "          =====      | (___ | |  | | |_) | |__) | |__     /  \\  | |  | |\n",
      "            ====      \\___ \\| |  | |  _ <|  _  /|  __|   / /\\ \\ | |  | |\n",
      "              ====    ____) | |__| | |_) | | \\ \\| |____ / ____ \\| |__| |\n",
      "        ==========   |_____/ \\____/|____/|_|  \\_\\______/_/    \\_\\_____/\n",
      "\t  v1.6.0\n",
      "\n",
      "//========================== featureCounts setting ===========================\\\\\n",
      "||                                                                            ||\n",
      "||             Input files : 1 BAM file                                       ||\n",
      "||                           S ./star_output/ERR899428Aligned.out.bam         ||\n",
      "||                                                                            ||\n",
      "||             Output file : ./featureCount_output/ERR899428Aligned.count ... ||\n",
      "||                 Summary : ./featureCount_output/ERR899428Aligned.count ... ||\n",
      "||              Annotation : /fs/project/PAS1307/genomes/Mus_musculus/Ens ... ||\n",
      "||      Dir for temp files : ./featureCount_output                            ||\n",
      "||                                                                            ||\n",
      "||                 Threads : 24                                               ||\n",
      "||                   Level : meta-feature level                               ||\n",
      "||              Paired-end : no                                               ||\n",
      "||         Strand specific : no                                               ||\n",
      "||      Multimapping reads : not counted                                      ||\n",
      "|| Multi-overlapping reads : not counted                                      ||\n",
      "||   Min overlapping bases : 1                                                ||\n",
      "||                                                                            ||\n",
      "\\\\===================== http://subread.sourceforge.net/ ======================//\n",
      "\n",
      "//================================= Running ==================================\\\\\n",
      "||                                                                            ||\n",
      "|| Load annotation file /fs/project/PAS1307/genomes/Mus_musculus/Ensembl/ ... ||\n",
      "||    Features : 690647                                                       ||\n",
      "||    Meta-features : 46078                                                   ||\n",
      "||    Chromosomes/contigs : 73                                                ||\n",
      "||                                                                            ||\n",
      "|| Process BAM file ./star_output/ERR899428Aligned.out.bam...                 ||\n",
      "||    Single-end reads are included.                                          ||\n",
      "||    Assign reads to features...                                             ||\n",
      "||    Total reads : 7216520                                                   ||\n",
      "||    Successfully assigned reads : 3742170 (51.9%)                           ||\n",
      "||    Running time : 0.34 minutes                                             ||\n",
      "||                                                                            ||\n",
      "||                         Read assignment finished.                          ||\n",
      "||                                                                            ||\n",
      "|| Summary of counting results can be found in file \"./featureCount_output/E  ||\n",
      "|| RR899428Aligned.count.txt.summary\"                                         ||\n",
      "||                                                                            ||\n",
      "\\\\===================== http://subread.sourceforge.net/ ======================//\n",
      "\n",
      "\n",
      "        ==========     _____ _    _ ____  _____  ______          _____  \n",
      "        =====         / ____| |  | |  _ \\|  __ \\|  ____|   /\\   |  __ \\ \n",
      "          =====      | (___ | |  | | |_) | |__) | |__     /  \\  | |  | |\n",
      "            ====      \\___ \\| |  | |  _ <|  _  /|  __|   / /\\ \\ | |  | |\n",
      "              ====    ____) | |__| | |_) | | \\ \\| |____ / ____ \\| |__| |\n",
      "        ==========   |_____/ \\____/|____/|_|  \\_\\______/_/    \\_\\_____/\n",
      "\t  v1.6.0\n",
      "\n",
      "//========================== featureCounts setting ===========================\\\\\n",
      "||                                                                            ||\n",
      "||             Input files : 1 BAM file                                       ||\n",
      "||                           S ./star_output/ERR899429Aligned.out.bam         ||\n",
      "||                                                                            ||\n",
      "||             Output file : ./featureCount_output/ERR899429Aligned.count ... ||\n",
      "||                 Summary : ./featureCount_output/ERR899429Aligned.count ... ||\n",
      "||              Annotation : /fs/project/PAS1307/genomes/Mus_musculus/Ens ... ||\n",
      "||      Dir for temp files : ./featureCount_output                            ||\n",
      "||                                                                            ||\n",
      "||                 Threads : 24                                               ||\n",
      "||                   Level : meta-feature level                               ||\n",
      "||              Paired-end : no                                               ||\n",
      "||         Strand specific : no                                               ||\n",
      "||      Multimapping reads : not counted                                      ||\n",
      "|| Multi-overlapping reads : not counted                                      ||\n",
      "||   Min overlapping bases : 1                                                ||\n",
      "||                                                                            ||\n",
      "\\\\===================== http://subread.sourceforge.net/ ======================//\n",
      "\n",
      "//================================= Running ==================================\\\\\n",
      "||                                                                            ||\n",
      "|| Load annotation file /fs/project/PAS1307/genomes/Mus_musculus/Ensembl/ ... ||\n",
      "||    Features : 690647                                                       ||\n",
      "||    Meta-features : 46078                                                   ||\n",
      "||    Chromosomes/contigs : 73                                                ||\n",
      "||                                                                            ||\n",
      "|| Process BAM file ./star_output/ERR899429Aligned.out.bam...                 ||\n",
      "||    Single-end reads are included.                                          ||\n",
      "||    Assign reads to features...                                             ||\n",
      "||    Total reads : 7814484                                                   ||\n",
      "||    Successfully assigned reads : 4384281 (56.1%)                           ||\n",
      "||    Running time : 0.34 minutes                                             ||\n",
      "||                                                                            ||\n",
      "||                         Read assignment finished.                          ||\n",
      "||                                                                            ||\n",
      "|| Summary of counting results can be found in file \"./featureCount_output/E  ||\n",
      "|| RR899429Aligned.count.txt.summary\"                                         ||\n",
      "||                                                                            ||\n",
      "\\\\===================== http://subread.sourceforge.net/ ======================//\n",
      "\n",
      "\n",
      "        ==========     _____ _    _ ____  _____  ______          _____  \n",
      "        =====         / ____| |  | |  _ \\|  __ \\|  ____|   /\\   |  __ \\ \n",
      "          =====      | (___ | |  | | |_) | |__) | |__     /  \\  | |  | |\n",
      "            ====      \\___ \\| |  | |  _ <|  _  /|  __|   / /\\ \\ | |  | |\n",
      "              ====    ____) | |__| | |_) | | \\ \\| |____ / ____ \\| |__| |\n",
      "        ==========   |_____/ \\____/|____/|_|  \\_\\______/_/    \\_\\_____/\n",
      "\t  v1.6.0\n",
      "\n",
      "//========================== featureCounts setting ===========================\\\\\n",
      "||                                                                            ||\n",
      "||             Input files : 1 BAM file                                       ||\n",
      "||                           S ./star_output/ERR899430Aligned.out.bam         ||\n",
      "||                                                                            ||\n",
      "||             Output file : ./featureCount_output/ERR899430Aligned.count ... ||\n",
      "||                 Summary : ./featureCount_output/ERR899430Aligned.count ... ||\n",
      "||              Annotation : /fs/project/PAS1307/genomes/Mus_musculus/Ens ... ||\n",
      "||      Dir for temp files : ./featureCount_output                            ||\n",
      "||                                                                            ||\n",
      "||                 Threads : 24                                               ||\n",
      "||                   Level : meta-feature level                               ||\n",
      "||              Paired-end : no                                               ||\n",
      "||         Strand specific : no                                               ||\n",
      "||      Multimapping reads : not counted                                      ||\n",
      "|| Multi-overlapping reads : not counted                                      ||\n",
      "||   Min overlapping bases : 1                                                ||\n",
      "||                                                                            ||\n",
      "\\\\===================== http://subread.sourceforge.net/ ======================//\n",
      "\n",
      "//================================= Running ==================================\\\\\n",
      "||                                                                            ||\n",
      "|| Load annotation file /fs/project/PAS1307/genomes/Mus_musculus/Ensembl/ ... ||\n",
      "||    Features : 690647                                                       ||\n",
      "||    Meta-features : 46078                                                   ||\n",
      "||    Chromosomes/contigs : 73                                                ||\n",
      "||                                                                            ||\n",
      "|| Process BAM file ./star_output/ERR899430Aligned.out.bam...                 ||\n",
      "||    Single-end reads are included.                                          ||\n",
      "||    Assign reads to features...                                             ||\n",
      "||    Total reads : 7894117                                                   ||\n",
      "||    Successfully assigned reads : 4250499 (53.8%)                           ||\n",
      "||    Running time : 0.34 minutes                                             ||\n",
      "||                                                                            ||\n",
      "||                         Read assignment finished.                          ||\n",
      "||                                                                            ||\n",
      "|| Summary of counting results can be found in file \"./featureCount_output/E  ||\n",
      "|| RR899430Aligned.count.txt.summary\"                                         ||\n",
      "||                                                                            ||\n",
      "\\\\===================== http://subread.sourceforge.net/ ======================//\n",
      "\n",
      "\n",
      "        ==========     _____ _    _ ____  _____  ______          _____  \n",
      "        =====         / ____| |  | |  _ \\|  __ \\|  ____|   /\\   |  __ \\ \n",
      "          =====      | (___ | |  | | |_) | |__) | |__     /  \\  | |  | |\n",
      "            ====      \\___ \\| |  | |  _ <|  _  /|  __|   / /\\ \\ | |  | |\n",
      "              ====    ____) | |__| | |_) | | \\ \\| |____ / ____ \\| |__| |\n",
      "        ==========   |_____/ \\____/|____/|_|  \\_\\______/_/    \\_\\_____/\n",
      "\t  v1.6.0\n",
      "\n",
      "//========================== featureCounts setting ===========================\\\\\n",
      "||                                                                            ||\n",
      "||             Input files : 1 BAM file                                       ||\n",
      "||                           S ./star_output/ERR899431Aligned.out.bam         ||\n",
      "||                                                                            ||\n",
      "||             Output file : ./featureCount_output/ERR899431Aligned.count ... ||\n",
      "||                 Summary : ./featureCount_output/ERR899431Aligned.count ... ||\n",
      "||              Annotation : /fs/project/PAS1307/genomes/Mus_musculus/Ens ... ||\n",
      "||      Dir for temp files : ./featureCount_output                            ||\n",
      "||                                                                            ||\n",
      "||                 Threads : 24                                               ||\n",
      "||                   Level : meta-feature level                               ||\n",
      "||              Paired-end : no                                               ||\n",
      "||         Strand specific : no                                               ||\n",
      "||      Multimapping reads : not counted                                      ||\n",
      "|| Multi-overlapping reads : not counted                                      ||\n",
      "||   Min overlapping bases : 1                                                ||\n",
      "||                                                                            ||\n",
      "\\\\===================== http://subread.sourceforge.net/ ======================//\n",
      "\n",
      "//================================= Running ==================================\\\\\n",
      "||                                                                            ||\n",
      "|| Load annotation file /fs/project/PAS1307/genomes/Mus_musculus/Ensembl/ ... ||\n",
      "||    Features : 690647                                                       ||\n",
      "||    Meta-features : 46078                                                   ||\n",
      "||    Chromosomes/contigs : 73                                                ||\n",
      "||                                                                            ||\n",
      "|| Process BAM file ./star_output/ERR899431Aligned.out.bam...                 ||\n",
      "||    Single-end reads are included.                                          ||\n",
      "||    Assign reads to features...                                             ||\n",
      "||    Total reads : 7844861                                                   ||\n",
      "||    Successfully assigned reads : 4532529 (57.8%)                           ||\n",
      "||    Running time : 0.34 minutes                                             ||\n",
      "||                                                                            ||\n",
      "||                         Read assignment finished.                          ||\n",
      "||                                                                            ||\n",
      "|| Summary of counting results can be found in file \"./featureCount_output/E  ||\n",
      "|| RR899431Aligned.count.txt.summary\"                                         ||\n",
      "||                                                                            ||\n",
      "\\\\===================== http://subread.sourceforge.net/ ======================//\n",
      "\n",
      "\n",
      "        ==========     _____ _    _ ____  _____  ______          _____  \n",
      "        =====         / ____| |  | |  _ \\|  __ \\|  ____|   /\\   |  __ \\ \n",
      "          =====      | (___ | |  | | |_) | |__) | |__     /  \\  | |  | |\n",
      "            ====      \\___ \\| |  | |  _ <|  _  /|  __|   / /\\ \\ | |  | |\n",
      "              ====    ____) | |__| | |_) | | \\ \\| |____ / ____ \\| |__| |\n",
      "        ==========   |_____/ \\____/|____/|_|  \\_\\______/_/    \\_\\_____/\n",
      "\t  v1.6.0\n",
      "\n",
      "//========================== featureCounts setting ===========================\\\\\n",
      "||                                                                            ||\n",
      "||             Input files : 1 BAM file                                       ||\n",
      "||                           S ./star_output/ERR899432Aligned.out.bam         ||\n",
      "||                                                                            ||\n",
      "||             Output file : ./featureCount_output/ERR899432Aligned.count ... ||\n",
      "||                 Summary : ./featureCount_output/ERR899432Aligned.count ... ||\n",
      "||              Annotation : /fs/project/PAS1307/genomes/Mus_musculus/Ens ... ||\n",
      "||      Dir for temp files : ./featureCount_output                            ||\n",
      "||                                                                            ||\n",
      "||                 Threads : 24                                               ||\n",
      "||                   Level : meta-feature level                               ||\n",
      "||              Paired-end : no                                               ||\n",
      "||         Strand specific : no                                               ||\n",
      "||      Multimapping reads : not counted                                      ||\n",
      "|| Multi-overlapping reads : not counted                                      ||\n",
      "||   Min overlapping bases : 1                                                ||\n",
      "||                                                                            ||\n",
      "\\\\===================== http://subread.sourceforge.net/ ======================//\n",
      "\n",
      "//================================= Running ==================================\\\\\n",
      "||                                                                            ||\n",
      "|| Load annotation file /fs/project/PAS1307/genomes/Mus_musculus/Ensembl/ ... ||\n",
      "||    Features : 690647                                                       ||\n",
      "||    Meta-features : 46078                                                   ||\n",
      "||    Chromosomes/contigs : 73                                                ||\n",
      "||                                                                            ||\n",
      "|| Process BAM file ./star_output/ERR899432Aligned.out.bam...                 ||\n",
      "||    Single-end reads are included.                                          ||\n",
      "||    Assign reads to features...                                             ||\n",
      "||    Total reads : 7213605                                                   ||\n",
      "||    Successfully assigned reads : 4243939 (58.8%)                           ||\n",
      "||    Running time : 0.31 minutes                                             ||\n",
      "||                                                                            ||\n",
      "||                         Read assignment finished.                          ||\n",
      "||                                                                            ||\n",
      "|| Summary of counting results can be found in file \"./featureCount_output/E  ||\n",
      "|| RR899432Aligned.count.txt.summary\"                                         ||\n",
      "||                                                                            ||\n",
      "\\\\===================== http://subread.sourceforge.net/ ======================//\n",
      "\n",
      "\n",
      "        ==========     _____ _    _ ____  _____  ______          _____  \n",
      "        =====         / ____| |  | |  _ \\|  __ \\|  ____|   /\\   |  __ \\ \n",
      "          =====      | (___ | |  | | |_) | |__) | |__     /  \\  | |  | |\n",
      "            ====      \\___ \\| |  | |  _ <|  _  /|  __|   / /\\ \\ | |  | |\n",
      "              ====    ____) | |__| | |_) | | \\ \\| |____ / ____ \\| |__| |\n",
      "        ==========   |_____/ \\____/|____/|_|  \\_\\______/_/    \\_\\_____/\n",
      "\t  v1.6.0\n",
      "\n",
      "//========================== featureCounts setting ===========================\\\\\n",
      "||                                                                            ||\n",
      "||             Input files : 1 BAM file                                       ||\n",
      "||                           S ./star_output/ERR899433Aligned.out.bam         ||\n",
      "||                                                                            ||\n",
      "||             Output file : ./featureCount_output/ERR899433Aligned.count ... ||\n",
      "||                 Summary : ./featureCount_output/ERR899433Aligned.count ... ||\n",
      "||              Annotation : /fs/project/PAS1307/genomes/Mus_musculus/Ens ... ||\n",
      "||      Dir for temp files : ./featureCount_output                            ||\n",
      "||                                                                            ||\n",
      "||                 Threads : 24                                               ||\n",
      "||                   Level : meta-feature level                               ||\n",
      "||              Paired-end : no                                               ||\n",
      "||         Strand specific : no                                               ||\n",
      "||      Multimapping reads : not counted                                      ||\n",
      "|| Multi-overlapping reads : not counted                                      ||\n",
      "||   Min overlapping bases : 1                                                ||\n",
      "||                                                                            ||\n",
      "\\\\===================== http://subread.sourceforge.net/ ======================//\n",
      "\n",
      "//================================= Running ==================================\\\\\n",
      "||                                                                            ||\n",
      "|| Load annotation file /fs/project/PAS1307/genomes/Mus_musculus/Ensembl/ ... ||\n",
      "||    Features : 690647                                                       ||\n",
      "||    Meta-features : 46078                                                   ||\n",
      "||    Chromosomes/contigs : 73                                                ||\n",
      "||                                                                            ||\n",
      "|| Process BAM file ./star_output/ERR899433Aligned.out.bam...                 ||\n",
      "||    Single-end reads are included.                                          ||\n",
      "||    Assign reads to features...                                             ||\n",
      "||    Total reads : 8098199                                                   ||\n",
      "||    Successfully assigned reads : 4533982 (56.0%)                           ||\n",
      "||    Running time : 0.35 minutes                                             ||\n",
      "||                                                                            ||\n",
      "||                         Read assignment finished.                          ||\n",
      "||                                                                            ||\n",
      "|| Summary of counting results can be found in file \"./featureCount_output/E  ||\n",
      "|| RR899433Aligned.count.txt.summary\"                                         ||\n",
      "||                                                                            ||\n",
      "\\\\===================== http://subread.sourceforge.net/ ======================//\n",
      "\n",
      "\n",
      "        ==========     _____ _    _ ____  _____  ______          _____  \n",
      "        =====         / ____| |  | |  _ \\|  __ \\|  ____|   /\\   |  __ \\ \n",
      "          =====      | (___ | |  | | |_) | |__) | |__     /  \\  | |  | |\n",
      "            ====      \\___ \\| |  | |  _ <|  _  /|  __|   / /\\ \\ | |  | |\n",
      "              ====    ____) | |__| | |_) | | \\ \\| |____ / ____ \\| |__| |\n",
      "        ==========   |_____/ \\____/|____/|_|  \\_\\______/_/    \\_\\_____/\n",
      "\t  v1.6.0\n",
      "\n",
      "//========================== featureCounts setting ===========================\\\\\n",
      "||                                                                            ||\n",
      "||             Input files : 1 BAM file                                       ||\n",
      "||                           S ./star_output/ERR899434Aligned.out.bam         ||\n",
      "||                                                                            ||\n",
      "||             Output file : ./featureCount_output/ERR899434Aligned.count ... ||\n",
      "||                 Summary : ./featureCount_output/ERR899434Aligned.count ... ||\n",
      "||              Annotation : /fs/project/PAS1307/genomes/Mus_musculus/Ens ... ||\n",
      "||      Dir for temp files : ./featureCount_output                            ||\n",
      "||                                                                            ||\n",
      "||                 Threads : 24                                               ||\n",
      "||                   Level : meta-feature level                               ||\n",
      "||              Paired-end : no                                               ||\n",
      "||         Strand specific : no                                               ||\n",
      "||      Multimapping reads : not counted                                      ||\n",
      "|| Multi-overlapping reads : not counted                                      ||\n",
      "||   Min overlapping bases : 1                                                ||\n",
      "||                                                                            ||\n",
      "\\\\===================== http://subread.sourceforge.net/ ======================//\n",
      "\n",
      "//================================= Running ==================================\\\\\n",
      "||                                                                            ||\n",
      "|| Load annotation file /fs/project/PAS1307/genomes/Mus_musculus/Ensembl/ ... ||\n",
      "||    Features : 690647                                                       ||\n",
      "||    Meta-features : 46078                                                   ||\n",
      "||    Chromosomes/contigs : 73                                                ||\n",
      "||                                                                            ||\n",
      "|| Process BAM file ./star_output/ERR899434Aligned.out.bam...                 ||\n",
      "||    Single-end reads are included.                                          ||\n",
      "||    Assign reads to features...                                             ||\n",
      "||    Total reads : 7816005                                                   ||\n",
      "||    Successfully assigned reads : 4288921 (54.9%)                           ||\n",
      "||    Running time : 0.34 minutes                                             ||\n",
      "||                                                                            ||\n",
      "||                         Read assignment finished.                          ||\n",
      "||                                                                            ||\n",
      "|| Summary of counting results can be found in file \"./featureCount_output/E  ||\n",
      "|| RR899434Aligned.count.txt.summary\"                                         ||\n",
      "||                                                                            ||\n",
      "\\\\===================== http://subread.sourceforge.net/ ======================//\n",
      "\n",
      "\n",
      "        ==========     _____ _    _ ____  _____  ______          _____  \n",
      "        =====         / ____| |  | |  _ \\|  __ \\|  ____|   /\\   |  __ \\ \n",
      "          =====      | (___ | |  | | |_) | |__) | |__     /  \\  | |  | |\n",
      "            ====      \\___ \\| |  | |  _ <|  _  /|  __|   / /\\ \\ | |  | |\n",
      "              ====    ____) | |__| | |_) | | \\ \\| |____ / ____ \\| |__| |\n",
      "        ==========   |_____/ \\____/|____/|_|  \\_\\______/_/    \\_\\_____/\n",
      "\t  v1.6.0\n",
      "\n",
      "//========================== featureCounts setting ===========================\\\\\n",
      "||                                                                            ||\n",
      "||             Input files : 1 BAM file                                       ||\n",
      "||                           S ./star_output/ERR899435Aligned.out.bam         ||\n",
      "||                                                                            ||\n",
      "||             Output file : ./featureCount_output/ERR899435Aligned.count ... ||\n",
      "||                 Summary : ./featureCount_output/ERR899435Aligned.count ... ||\n",
      "||              Annotation : /fs/project/PAS1307/genomes/Mus_musculus/Ens ... ||\n",
      "||      Dir for temp files : ./featureCount_output                            ||\n",
      "||                                                                            ||\n",
      "||                 Threads : 24                                               ||\n",
      "||                   Level : meta-feature level                               ||\n",
      "||              Paired-end : no                                               ||\n",
      "||         Strand specific : no                                               ||\n",
      "||      Multimapping reads : not counted                                      ||\n",
      "|| Multi-overlapping reads : not counted                                      ||\n",
      "||   Min overlapping bases : 1                                                ||\n",
      "||                                                                            ||\n",
      "\\\\===================== http://subread.sourceforge.net/ ======================//\n",
      "\n",
      "//================================= Running ==================================\\\\\n",
      "||                                                                            ||\n",
      "|| Load annotation file /fs/project/PAS1307/genomes/Mus_musculus/Ensembl/ ... ||\n",
      "||    Features : 690647                                                       ||\n",
      "||    Meta-features : 46078                                                   ||\n",
      "||    Chromosomes/contigs : 73                                                ||\n",
      "||                                                                            ||\n",
      "|| Process BAM file ./star_output/ERR899435Aligned.out.bam...                 ||\n",
      "||    Single-end reads are included.                                          ||\n",
      "||    Assign reads to features...                                             ||\n",
      "||    Total reads : 7319465                                                   ||\n",
      "||    Successfully assigned reads : 4091465 (55.9%)                           ||\n",
      "||    Running time : 0.35 minutes                                             ||\n",
      "||                                                                            ||\n",
      "||                         Read assignment finished.                          ||\n",
      "||                                                                            ||\n",
      "|| Summary of counting results can be found in file \"./featureCount_output/E  ||\n",
      "|| RR899435Aligned.count.txt.summary\"                                         ||\n",
      "||                                                                            ||\n",
      "\\\\===================== http://subread.sourceforge.net/ ======================//\n",
      "\n",
      "\n",
      "        ==========     _____ _    _ ____  _____  ______          _____  \n",
      "        =====         / ____| |  | |  _ \\|  __ \\|  ____|   /\\   |  __ \\ \n",
      "          =====      | (___ | |  | | |_) | |__) | |__     /  \\  | |  | |\n",
      "            ====      \\___ \\| |  | |  _ <|  _  /|  __|   / /\\ \\ | |  | |\n",
      "              ====    ____) | |__| | |_) | | \\ \\| |____ / ____ \\| |__| |\n",
      "        ==========   |_____/ \\____/|____/|_|  \\_\\______/_/    \\_\\_____/\n",
      "\t  v1.6.0\n",
      "\n",
      "//========================== featureCounts setting ===========================\\\\\n",
      "||                                                                            ||\n",
      "||             Input files : 1 BAM file                                       ||\n",
      "||                           S ./star_output/ERR899436Aligned.out.bam         ||\n",
      "||                                                                            ||\n",
      "||             Output file : ./featureCount_output/ERR899436Aligned.count ... ||\n",
      "||                 Summary : ./featureCount_output/ERR899436Aligned.count ... ||\n",
      "||              Annotation : /fs/project/PAS1307/genomes/Mus_musculus/Ens ... ||\n",
      "||      Dir for temp files : ./featureCount_output                            ||\n",
      "||                                                                            ||\n",
      "||                 Threads : 24                                               ||\n",
      "||                   Level : meta-feature level                               ||\n",
      "||              Paired-end : no                                               ||\n",
      "||         Strand specific : no                                               ||\n",
      "||      Multimapping reads : not counted                                      ||\n",
      "|| Multi-overlapping reads : not counted                                      ||\n",
      "||   Min overlapping bases : 1                                                ||\n",
      "||                                                                            ||\n",
      "\\\\===================== http://subread.sourceforge.net/ ======================//\n",
      "\n",
      "//================================= Running ==================================\\\\\n",
      "||                                                                            ||\n",
      "|| Load annotation file /fs/project/PAS1307/genomes/Mus_musculus/Ensembl/ ... ||\n",
      "||    Features : 690647                                                       ||\n",
      "||    Meta-features : 46078                                                   ||\n",
      "||    Chromosomes/contigs : 73                                                ||\n",
      "||                                                                            ||\n",
      "|| Process BAM file ./star_output/ERR899436Aligned.out.bam...                 ||\n",
      "||    Single-end reads are included.                                          ||\n",
      "||    Assign reads to features...                                             ||\n",
      "||    Total reads : 7357420                                                   ||\n",
      "||    Successfully assigned reads : 4157877 (56.5%)                           ||\n",
      "||    Running time : 0.32 minutes                                             ||\n",
      "||                                                                            ||\n",
      "||                         Read assignment finished.                          ||\n",
      "||                                                                            ||\n",
      "|| Summary of counting results can be found in file \"./featureCount_output/E  ||\n",
      "|| RR899436Aligned.count.txt.summary\"                                         ||\n",
      "||                                                                            ||\n",
      "\\\\===================== http://subread.sourceforge.net/ ======================//\n",
      "\n",
      "\n",
      "        ==========     _____ _    _ ____  _____  ______          _____  \n",
      "        =====         / ____| |  | |  _ \\|  __ \\|  ____|   /\\   |  __ \\ \n",
      "          =====      | (___ | |  | | |_) | |__) | |__     /  \\  | |  | |\n",
      "            ====      \\___ \\| |  | |  _ <|  _  /|  __|   / /\\ \\ | |  | |\n",
      "              ====    ____) | |__| | |_) | | \\ \\| |____ / ____ \\| |__| |\n",
      "        ==========   |_____/ \\____/|____/|_|  \\_\\______/_/    \\_\\_____/\n",
      "\t  v1.6.0\n",
      "\n",
      "//========================== featureCounts setting ===========================\\\\\n",
      "||                                                                            ||\n",
      "||             Input files : 1 BAM file                                       ||\n",
      "||                           S ./star_output/ERR899437Aligned.out.bam         ||\n",
      "||                                                                            ||\n",
      "||             Output file : ./featureCount_output/ERR899437Aligned.count ... ||\n",
      "||                 Summary : ./featureCount_output/ERR899437Aligned.count ... ||\n",
      "||              Annotation : /fs/project/PAS1307/genomes/Mus_musculus/Ens ... ||\n",
      "||      Dir for temp files : ./featureCount_output                            ||\n",
      "||                                                                            ||\n",
      "||                 Threads : 24                                               ||\n",
      "||                   Level : meta-feature level                               ||\n",
      "||              Paired-end : no                                               ||\n",
      "||         Strand specific : no                                               ||\n",
      "||      Multimapping reads : not counted                                      ||\n",
      "|| Multi-overlapping reads : not counted                                      ||\n",
      "||   Min overlapping bases : 1                                                ||\n",
      "||                                                                            ||\n",
      "\\\\===================== http://subread.sourceforge.net/ ======================//\n",
      "\n",
      "//================================= Running ==================================\\\\\n",
      "||                                                                            ||\n",
      "|| Load annotation file /fs/project/PAS1307/genomes/Mus_musculus/Ensembl/ ... ||\n",
      "||    Features : 690647                                                       ||\n",
      "||    Meta-features : 46078                                                   ||\n",
      "||    Chromosomes/contigs : 73                                                ||\n",
      "||                                                                            ||\n",
      "|| Process BAM file ./star_output/ERR899437Aligned.out.bam...                 ||\n",
      "||    Single-end reads are included.                                          ||\n",
      "||    Assign reads to features...                                             ||\n",
      "||    Total reads : 8847290                                                   ||\n",
      "||    Successfully assigned reads : 4213290 (47.6%)                           ||\n",
      "||    Running time : 0.38 minutes                                             ||\n",
      "||                                                                            ||\n",
      "||                         Read assignment finished.                          ||\n",
      "||                                                                            ||\n",
      "|| Summary of counting results can be found in file \"./featureCount_output/E  ||\n",
      "|| RR899437Aligned.count.txt.summary\"                                         ||\n",
      "||                                                                            ||\n",
      "\\\\===================== http://subread.sourceforge.net/ ======================//\n",
      "\n",
      "\n",
      "        ==========     _____ _    _ ____  _____  ______          _____  \n",
      "        =====         / ____| |  | |  _ \\|  __ \\|  ____|   /\\   |  __ \\ \n",
      "          =====      | (___ | |  | | |_) | |__) | |__     /  \\  | |  | |\n",
      "            ====      \\___ \\| |  | |  _ <|  _  /|  __|   / /\\ \\ | |  | |\n",
      "              ====    ____) | |__| | |_) | | \\ \\| |____ / ____ \\| |__| |\n",
      "        ==========   |_____/ \\____/|____/|_|  \\_\\______/_/    \\_\\_____/\n",
      "\t  v1.6.0\n",
      "\n",
      "//========================== featureCounts setting ===========================\\\\\n",
      "||                                                                            ||\n",
      "||             Input files : 1 BAM file                                       ||\n",
      "||                           S ./star_output/ERR899438Aligned.out.bam         ||\n",
      "||                                                                            ||\n",
      "||             Output file : ./featureCount_output/ERR899438Aligned.count ... ||\n",
      "||                 Summary : ./featureCount_output/ERR899438Aligned.count ... ||\n",
      "||              Annotation : /fs/project/PAS1307/genomes/Mus_musculus/Ens ... ||\n",
      "||      Dir for temp files : ./featureCount_output                            ||\n",
      "||                                                                            ||\n",
      "||                 Threads : 24                                               ||\n",
      "||                   Level : meta-feature level                               ||\n",
      "||              Paired-end : no                                               ||\n",
      "||         Strand specific : no                                               ||\n",
      "||      Multimapping reads : not counted                                      ||\n",
      "|| Multi-overlapping reads : not counted                                      ||\n",
      "||   Min overlapping bases : 1                                                ||\n",
      "||                                                                            ||\n",
      "\\\\===================== http://subread.sourceforge.net/ ======================//\n",
      "\n",
      "//================================= Running ==================================\\\\\n",
      "||                                                                            ||\n",
      "|| Load annotation file /fs/project/PAS1307/genomes/Mus_musculus/Ensembl/ ... ||\n",
      "||    Features : 690647                                                       ||\n",
      "||    Meta-features : 46078                                                   ||\n",
      "||    Chromosomes/contigs : 73                                                ||\n",
      "||                                                                            ||\n",
      "|| Process BAM file ./star_output/ERR899438Aligned.out.bam...                 ||\n",
      "||    Single-end reads are included.                                          ||\n",
      "||    Assign reads to features...                                             ||\n",
      "||    Total reads : 9009449                                                   ||\n",
      "||    Successfully assigned reads : 4858518 (53.9%)                           ||\n",
      "||    Running time : 0.39 minutes                                             ||\n",
      "||                                                                            ||\n",
      "||                         Read assignment finished.                          ||\n",
      "||                                                                            ||\n",
      "|| Summary of counting results can be found in file \"./featureCount_output/E  ||\n",
      "|| RR899438Aligned.count.txt.summary\"                                         ||\n",
      "||                                                                            ||\n",
      "\\\\===================== http://subread.sourceforge.net/ ======================//\n",
      "\n",
      "\n",
      "ERROR: invalid parameter: './star_output/ERR899421Aligned.out.bam.out.bam'\n",
      "\n"
     ]
    }
   ],
   "source": [
    "cmd='''\n",
    "for basename in $(find star_output/ -name '*Aligned.out.bam' -exec basename {} \\; | cut -f1 -d '.' ); do\n",
    "    suffix=\".out.bam\"\n",
    "    outname=\"$FCDIR/$basename.count.txt\"\n",
    "    bam=\"$STARDIR/$basename$suffix\"    \n",
    "    $BINDIR/featureCounts \\\n",
    "        -T $MAXCORES \\\n",
    "        -t exon \\\n",
    "        -g gene_id \\\n",
    "        -a  $GENEANNOTATIONDIR/genes.gtf \\\n",
    "        -o $outname \\\n",
    "        $bam \n",
    "done\n",
    "'''\n",
    "out=run_cmd(cmd)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "cmd='''\n",
    "bams=\"\"\n",
    "for basename in $(find star_output/ -name '*Aligned.out.bam' -exec basename {} \\; | cut -f1 -d '.' ); do\n",
    "    suffix=\".out.bam\"\n",
    "    outname=\"combined.counts.txt\"\n",
    "    bam=\"$STARDIR/$basename$suffix\"    \n",
    "    bams=\"$bams $bam\"\n",
    "done\n",
    "\n",
    "$BINDIR/featureCounts \\\n",
    "        -T $MAXCORES \\\n",
    "        -t exon \\\n",
    "        -g gene_id \\\n",
    "        -a  $GENEANNOTATIONDIR/genes.gtf \\\n",
    "        -o $outname \\\n",
    "        $bams\n",
    "'''\n",
    "out=run_cmd(cmd)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "cmd='''\n",
    "#multiqc of sequencing reads\n",
    "\n",
    "source activate tophat\n",
    "multiqc .\n",
    "'''\n",
    "out=run_cmd(cmd)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 76,
   "metadata": {
    "collapsed": false,
    "scrolled": false
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "/users/PAS1307/osu8805/rnaseq_2/fastQC_output/<br>\n",
       "&nbsp;&nbsp;<a href='/users/PAS1307/osu8805/rnaseq_2/fastQC_output/ERR899423_fastqc.html' target='_blank'>ERR899423_fastqc.html</a><br>\n",
       "&nbsp;&nbsp;<a href='/users/PAS1307/osu8805/rnaseq_2/fastQC_output/ERR899424_fastqc.html' target='_blank'>ERR899424_fastqc.html</a><br>\n",
       "&nbsp;&nbsp;<a href='/users/PAS1307/osu8805/rnaseq_2/fastQC_output/ERR899428_fastqc.html' target='_blank'>ERR899428_fastqc.html</a><br>\n",
       "&nbsp;&nbsp;<a href='/users/PAS1307/osu8805/rnaseq_2/fastQC_output/ERR899427_fastqc.html' target='_blank'>ERR899427_fastqc.html</a><br>\n",
       "&nbsp;&nbsp;<a href='/users/PAS1307/osu8805/rnaseq_2/fastQC_output/ERR899421_fastqc.html' target='_blank'>ERR899421_fastqc.html</a><br>\n",
       "&nbsp;&nbsp;<a href='/users/PAS1307/osu8805/rnaseq_2/fastQC_output/ERR899435_fastqc.html' target='_blank'>ERR899435_fastqc.html</a><br>\n",
       "&nbsp;&nbsp;<a href='/users/PAS1307/osu8805/rnaseq_2/fastQC_output/ERR899432_fastqc.html' target='_blank'>ERR899432_fastqc.html</a><br>\n",
       "&nbsp;&nbsp;<a href='/users/PAS1307/osu8805/rnaseq_2/fastQC_output/ERR899422_fastqc.html' target='_blank'>ERR899422_fastqc.html</a><br>\n",
       "&nbsp;&nbsp;<a href='/users/PAS1307/osu8805/rnaseq_2/fastQC_output/ERR899436_fastqc.html' target='_blank'>ERR899436_fastqc.html</a><br>\n",
       "&nbsp;&nbsp;<a href='/users/PAS1307/osu8805/rnaseq_2/fastQC_output/ERR899434_fastqc.html' target='_blank'>ERR899434_fastqc.html</a><br>\n",
       "&nbsp;&nbsp;<a href='/users/PAS1307/osu8805/rnaseq_2/fastQC_output/ERR899430_fastqc.html' target='_blank'>ERR899430_fastqc.html</a><br>\n",
       "&nbsp;&nbsp;<a href='/users/PAS1307/osu8805/rnaseq_2/fastQC_output/ERR899437_fastqc.html' target='_blank'>ERR899437_fastqc.html</a><br>\n",
       "&nbsp;&nbsp;<a href='/users/PAS1307/osu8805/rnaseq_2/fastQC_output/ERR899426_fastqc.html' target='_blank'>ERR899426_fastqc.html</a><br>\n",
       "&nbsp;&nbsp;<a href='/users/PAS1307/osu8805/rnaseq_2/fastQC_output/ERR899429_fastqc.html' target='_blank'>ERR899429_fastqc.html</a><br>\n",
       "&nbsp;&nbsp;<a href='/users/PAS1307/osu8805/rnaseq_2/fastQC_output/ERR899431_fastqc.html' target='_blank'>ERR899431_fastqc.html</a><br>\n",
       "&nbsp;&nbsp;<a href='/users/PAS1307/osu8805/rnaseq_2/fastQC_output/ERR899433_fastqc.html' target='_blank'>ERR899433_fastqc.html</a><br>\n",
       "&nbsp;&nbsp;<a href='/users/PAS1307/osu8805/rnaseq_2/fastQC_output/ERR899425_fastqc.html' target='_blank'>ERR899425_fastqc.html</a><br>\n",
       "&nbsp;&nbsp;<a href='/users/PAS1307/osu8805/rnaseq_2/fastQC_output/ERR899438_fastqc.html' target='_blank'>ERR899438_fastqc.html</a><br>\n",
       "/users/PAS1307/osu8805/rnaseq_2/trimmed_fastQC_output/<br>\n",
       "&nbsp;&nbsp;<a href='/users/PAS1307/osu8805/rnaseq_2/trimmed_fastQC_output/ERR899427_fastqc.html' target='_blank'>ERR899427_fastqc.html</a><br>\n",
       "&nbsp;&nbsp;<a href='/users/PAS1307/osu8805/rnaseq_2/trimmed_fastQC_output/ERR899428_fastqc.html' target='_blank'>ERR899428_fastqc.html</a><br>\n",
       "&nbsp;&nbsp;<a href='/users/PAS1307/osu8805/rnaseq_2/trimmed_fastQC_output/ERR899430_fastqc.html' target='_blank'>ERR899430_fastqc.html</a><br>\n",
       "&nbsp;&nbsp;<a href='/users/PAS1307/osu8805/rnaseq_2/trimmed_fastQC_output/ERR899421_fastqc.html' target='_blank'>ERR899421_fastqc.html</a><br>\n",
       "&nbsp;&nbsp;<a href='/users/PAS1307/osu8805/rnaseq_2/trimmed_fastQC_output/ERR899423_fastqc.html' target='_blank'>ERR899423_fastqc.html</a><br>\n",
       "&nbsp;&nbsp;<a href='/users/PAS1307/osu8805/rnaseq_2/trimmed_fastQC_output/ERR899432_fastqc.html' target='_blank'>ERR899432_fastqc.html</a><br>\n",
       "&nbsp;&nbsp;<a href='/users/PAS1307/osu8805/rnaseq_2/trimmed_fastQC_output/ERR899422_fastqc.html' target='_blank'>ERR899422_fastqc.html</a><br>\n",
       "&nbsp;&nbsp;<a href='/users/PAS1307/osu8805/rnaseq_2/trimmed_fastQC_output/ERR899435_fastqc.html' target='_blank'>ERR899435_fastqc.html</a><br>\n",
       "&nbsp;&nbsp;<a href='/users/PAS1307/osu8805/rnaseq_2/trimmed_fastQC_output/ERR899424_fastqc.html' target='_blank'>ERR899424_fastqc.html</a><br>\n",
       "&nbsp;&nbsp;<a href='/users/PAS1307/osu8805/rnaseq_2/trimmed_fastQC_output/ERR899436_fastqc.html' target='_blank'>ERR899436_fastqc.html</a><br>\n",
       "&nbsp;&nbsp;<a href='/users/PAS1307/osu8805/rnaseq_2/trimmed_fastQC_output/ERR899434_fastqc.html' target='_blank'>ERR899434_fastqc.html</a><br>\n",
       "&nbsp;&nbsp;<a href='/users/PAS1307/osu8805/rnaseq_2/trimmed_fastQC_output/ERR899437_fastqc.html' target='_blank'>ERR899437_fastqc.html</a><br>\n",
       "&nbsp;&nbsp;<a href='/users/PAS1307/osu8805/rnaseq_2/trimmed_fastQC_output/ERR899429_fastqc.html' target='_blank'>ERR899429_fastqc.html</a><br>\n",
       "&nbsp;&nbsp;<a href='/users/PAS1307/osu8805/rnaseq_2/trimmed_fastQC_output/ERR899431_fastqc.html' target='_blank'>ERR899431_fastqc.html</a><br>\n",
       "&nbsp;&nbsp;<a href='/users/PAS1307/osu8805/rnaseq_2/trimmed_fastQC_output/ERR899426_fastqc.html' target='_blank'>ERR899426_fastqc.html</a><br>\n",
       "&nbsp;&nbsp;<a href='/users/PAS1307/osu8805/rnaseq_2/trimmed_fastQC_output/ERR899433_fastqc.html' target='_blank'>ERR899433_fastqc.html</a><br>\n",
       "&nbsp;&nbsp;<a href='/users/PAS1307/osu8805/rnaseq_2/trimmed_fastQC_output/ERR899425_fastqc.html' target='_blank'>ERR899425_fastqc.html</a><br>\n",
       "&nbsp;&nbsp;<a href='/users/PAS1307/osu8805/rnaseq_2/trimmed_fastQC_output/ERR899438_fastqc.html' target='_blank'>ERR899438_fastqc.html</a><br>\n",
       "&nbsp;&nbsp;<a href='/users/PAS1307/osu8805/rnaseq_2/trimmed_fastQC_output/ERR899428_trimmed_fastqc.html' target='_blank'>ERR899428_trimmed_fastqc.html</a><br>\n",
       "&nbsp;&nbsp;<a href='/users/PAS1307/osu8805/rnaseq_2/trimmed_fastQC_output/ERR899427_trimmed_fastqc.html' target='_blank'>ERR899427_trimmed_fastqc.html</a><br>\n",
       "&nbsp;&nbsp;<a href='/users/PAS1307/osu8805/rnaseq_2/trimmed_fastQC_output/ERR899435_trimmed_fastqc.html' target='_blank'>ERR899435_trimmed_fastqc.html</a><br>\n",
       "&nbsp;&nbsp;<a href='/users/PAS1307/osu8805/rnaseq_2/trimmed_fastQC_output/ERR899421_trimmed_fastqc.html' target='_blank'>ERR899421_trimmed_fastqc.html</a><br>\n",
       "&nbsp;&nbsp;<a href='/users/PAS1307/osu8805/rnaseq_2/trimmed_fastQC_output/ERR899432_trimmed_fastqc.html' target='_blank'>ERR899432_trimmed_fastqc.html</a><br>\n",
       "&nbsp;&nbsp;<a href='/users/PAS1307/osu8805/rnaseq_2/trimmed_fastQC_output/ERR899436_trimmed_fastqc.html' target='_blank'>ERR899436_trimmed_fastqc.html</a><br>\n",
       "&nbsp;&nbsp;<a href='/users/PAS1307/osu8805/rnaseq_2/trimmed_fastQC_output/ERR899434_trimmed_fastqc.html' target='_blank'>ERR899434_trimmed_fastqc.html</a><br>\n",
       "&nbsp;&nbsp;<a href='/users/PAS1307/osu8805/rnaseq_2/trimmed_fastQC_output/ERR899423_trimmed_fastqc.html' target='_blank'>ERR899423_trimmed_fastqc.html</a><br>\n",
       "&nbsp;&nbsp;<a href='/users/PAS1307/osu8805/rnaseq_2/trimmed_fastQC_output/ERR899424_trimmed_fastqc.html' target='_blank'>ERR899424_trimmed_fastqc.html</a><br>\n",
       "&nbsp;&nbsp;<a href='/users/PAS1307/osu8805/rnaseq_2/trimmed_fastQC_output/ERR899437_trimmed_fastqc.html' target='_blank'>ERR899437_trimmed_fastqc.html</a><br>\n",
       "&nbsp;&nbsp;<a href='/users/PAS1307/osu8805/rnaseq_2/trimmed_fastQC_output/ERR899422_trimmed_fastqc.html' target='_blank'>ERR899422_trimmed_fastqc.html</a><br>\n",
       "&nbsp;&nbsp;<a href='/users/PAS1307/osu8805/rnaseq_2/trimmed_fastQC_output/ERR899430_trimmed_fastqc.html' target='_blank'>ERR899430_trimmed_fastqc.html</a><br>\n",
       "&nbsp;&nbsp;<a href='/users/PAS1307/osu8805/rnaseq_2/trimmed_fastQC_output/ERR899426_trimmed_fastqc.html' target='_blank'>ERR899426_trimmed_fastqc.html</a><br>\n",
       "&nbsp;&nbsp;<a href='/users/PAS1307/osu8805/rnaseq_2/trimmed_fastQC_output/ERR899429_trimmed_fastqc.html' target='_blank'>ERR899429_trimmed_fastqc.html</a><br>\n",
       "&nbsp;&nbsp;<a href='/users/PAS1307/osu8805/rnaseq_2/trimmed_fastQC_output/ERR899431_trimmed_fastqc.html' target='_blank'>ERR899431_trimmed_fastqc.html</a><br>\n",
       "&nbsp;&nbsp;<a href='/users/PAS1307/osu8805/rnaseq_2/trimmed_fastQC_output/ERR899433_trimmed_fastqc.html' target='_blank'>ERR899433_trimmed_fastqc.html</a><br>\n",
       "&nbsp;&nbsp;<a href='/users/PAS1307/osu8805/rnaseq_2/trimmed_fastQC_output/ERR899425_trimmed_fastqc.html' target='_blank'>ERR899425_trimmed_fastqc.html</a><br>\n",
       "&nbsp;&nbsp;<a href='/users/PAS1307/osu8805/rnaseq_2/trimmed_fastQC_output/ERR899438_trimmed_fastqc.html' target='_blank'>ERR899438_trimmed_fastqc.html</a><br>"
      ],
      "text/plain": [
       "/users/PAS1307/osu8805/rnaseq_2/fastQC_output/\n",
       "  ERR899423_fastqc.html\n",
       "  ERR899424_fastqc.html\n",
       "  ERR899428_fastqc.html\n",
       "  ERR899427_fastqc.html\n",
       "  ERR899421_fastqc.html\n",
       "  ERR899435_fastqc.html\n",
       "  ERR899432_fastqc.html\n",
       "  ERR899422_fastqc.html\n",
       "  ERR899436_fastqc.html\n",
       "  ERR899434_fastqc.html\n",
       "  ERR899430_fastqc.html\n",
       "  ERR899437_fastqc.html\n",
       "  ERR899426_fastqc.html\n",
       "  ERR899429_fastqc.html\n",
       "  ERR899431_fastqc.html\n",
       "  ERR899433_fastqc.html\n",
       "  ERR899425_fastqc.html\n",
       "  ERR899438_fastqc.html\n",
       "/users/PAS1307/osu8805/rnaseq_2/trimmed_fastQC_output/\n",
       "  ERR899427_fastqc.html\n",
       "  ERR899428_fastqc.html\n",
       "  ERR899430_fastqc.html\n",
       "  ERR899421_fastqc.html\n",
       "  ERR899423_fastqc.html\n",
       "  ERR899432_fastqc.html\n",
       "  ERR899422_fastqc.html\n",
       "  ERR899435_fastqc.html\n",
       "  ERR899424_fastqc.html\n",
       "  ERR899436_fastqc.html\n",
       "  ERR899434_fastqc.html\n",
       "  ERR899437_fastqc.html\n",
       "  ERR899429_fastqc.html\n",
       "  ERR899431_fastqc.html\n",
       "  ERR899426_fastqc.html\n",
       "  ERR899433_fastqc.html\n",
       "  ERR899425_fastqc.html\n",
       "  ERR899438_fastqc.html\n",
       "  ERR899428_trimmed_fastqc.html\n",
       "  ERR899427_trimmed_fastqc.html\n",
       "  ERR899435_trimmed_fastqc.html\n",
       "  ERR899421_trimmed_fastqc.html\n",
       "  ERR899432_trimmed_fastqc.html\n",
       "  ERR899436_trimmed_fastqc.html\n",
       "  ERR899434_trimmed_fastqc.html\n",
       "  ERR899423_trimmed_fastqc.html\n",
       "  ERR899424_trimmed_fastqc.html\n",
       "  ERR899437_trimmed_fastqc.html\n",
       "  ERR899422_trimmed_fastqc.html\n",
       "  ERR899430_trimmed_fastqc.html\n",
       "  ERR899426_trimmed_fastqc.html\n",
       "  ERR899429_trimmed_fastqc.html\n",
       "  ERR899431_trimmed_fastqc.html\n",
       "  ERR899433_trimmed_fastqc.html\n",
       "  ERR899425_trimmed_fastqc.html\n",
       "  ERR899438_trimmed_fastqc.html"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "/users/PAS1307/osu8805/rnaseq_2/fastQC_output/<br>\n",
       "&nbsp;&nbsp;<a href='/users/PAS1307/osu8805/rnaseq_2/fastQC_output/ERR899423_fastqc.html' target='_blank'>ERR899423_fastqc.html</a><br>\n",
       "&nbsp;&nbsp;<a href='/users/PAS1307/osu8805/rnaseq_2/fastQC_output/ERR899424_fastqc.html' target='_blank'>ERR899424_fastqc.html</a><br>\n",
       "&nbsp;&nbsp;<a href='/users/PAS1307/osu8805/rnaseq_2/fastQC_output/ERR899428_fastqc.html' target='_blank'>ERR899428_fastqc.html</a><br>\n",
       "&nbsp;&nbsp;<a href='/users/PAS1307/osu8805/rnaseq_2/fastQC_output/ERR899427_fastqc.html' target='_blank'>ERR899427_fastqc.html</a><br>\n",
       "&nbsp;&nbsp;<a href='/users/PAS1307/osu8805/rnaseq_2/fastQC_output/ERR899421_fastqc.html' target='_blank'>ERR899421_fastqc.html</a><br>\n",
       "&nbsp;&nbsp;<a href='/users/PAS1307/osu8805/rnaseq_2/fastQC_output/ERR899435_fastqc.html' target='_blank'>ERR899435_fastqc.html</a><br>\n",
       "&nbsp;&nbsp;<a href='/users/PAS1307/osu8805/rnaseq_2/fastQC_output/ERR899432_fastqc.html' target='_blank'>ERR899432_fastqc.html</a><br>\n",
       "&nbsp;&nbsp;<a href='/users/PAS1307/osu8805/rnaseq_2/fastQC_output/ERR899422_fastqc.html' target='_blank'>ERR899422_fastqc.html</a><br>\n",
       "&nbsp;&nbsp;<a href='/users/PAS1307/osu8805/rnaseq_2/fastQC_output/ERR899436_fastqc.html' target='_blank'>ERR899436_fastqc.html</a><br>\n",
       "&nbsp;&nbsp;<a href='/users/PAS1307/osu8805/rnaseq_2/fastQC_output/ERR899434_fastqc.html' target='_blank'>ERR899434_fastqc.html</a><br>\n",
       "&nbsp;&nbsp;<a href='/users/PAS1307/osu8805/rnaseq_2/fastQC_output/ERR899430_fastqc.html' target='_blank'>ERR899430_fastqc.html</a><br>\n",
       "&nbsp;&nbsp;<a href='/users/PAS1307/osu8805/rnaseq_2/fastQC_output/ERR899437_fastqc.html' target='_blank'>ERR899437_fastqc.html</a><br>\n",
       "&nbsp;&nbsp;<a href='/users/PAS1307/osu8805/rnaseq_2/fastQC_output/ERR899426_fastqc.html' target='_blank'>ERR899426_fastqc.html</a><br>\n",
       "&nbsp;&nbsp;<a href='/users/PAS1307/osu8805/rnaseq_2/fastQC_output/ERR899429_fastqc.html' target='_blank'>ERR899429_fastqc.html</a><br>\n",
       "&nbsp;&nbsp;<a href='/users/PAS1307/osu8805/rnaseq_2/fastQC_output/ERR899431_fastqc.html' target='_blank'>ERR899431_fastqc.html</a><br>\n",
       "&nbsp;&nbsp;<a href='/users/PAS1307/osu8805/rnaseq_2/fastQC_output/ERR899433_fastqc.html' target='_blank'>ERR899433_fastqc.html</a><br>\n",
       "&nbsp;&nbsp;<a href='/users/PAS1307/osu8805/rnaseq_2/fastQC_output/ERR899425_fastqc.html' target='_blank'>ERR899425_fastqc.html</a><br>\n",
       "&nbsp;&nbsp;<a href='/users/PAS1307/osu8805/rnaseq_2/fastQC_output/ERR899438_fastqc.html' target='_blank'>ERR899438_fastqc.html</a><br>"
      ],
      "text/plain": [
       "/users/PAS1307/osu8805/rnaseq_2/fastQC_output/\n",
       "  ERR899423_fastqc.html\n",
       "  ERR899424_fastqc.html\n",
       "  ERR899428_fastqc.html\n",
       "  ERR899427_fastqc.html\n",
       "  ERR899421_fastqc.html\n",
       "  ERR899435_fastqc.html\n",
       "  ERR899432_fastqc.html\n",
       "  ERR899422_fastqc.html\n",
       "  ERR899436_fastqc.html\n",
       "  ERR899434_fastqc.html\n",
       "  ERR899430_fastqc.html\n",
       "  ERR899437_fastqc.html\n",
       "  ERR899426_fastqc.html\n",
       "  ERR899429_fastqc.html\n",
       "  ERR899431_fastqc.html\n",
       "  ERR899433_fastqc.html\n",
       "  ERR899425_fastqc.html\n",
       "  ERR899438_fastqc.html"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "/users/PAS1307/osu8805/rnaseq_2/trimmed_fastQC_output/<br>\n",
       "&nbsp;&nbsp;<a href='/users/PAS1307/osu8805/rnaseq_2/trimmed_fastQC_output/ERR899427_fastqc.html' target='_blank'>ERR899427_fastqc.html</a><br>\n",
       "&nbsp;&nbsp;<a href='/users/PAS1307/osu8805/rnaseq_2/trimmed_fastQC_output/ERR899428_fastqc.html' target='_blank'>ERR899428_fastqc.html</a><br>\n",
       "&nbsp;&nbsp;<a href='/users/PAS1307/osu8805/rnaseq_2/trimmed_fastQC_output/ERR899430_fastqc.html' target='_blank'>ERR899430_fastqc.html</a><br>\n",
       "&nbsp;&nbsp;<a href='/users/PAS1307/osu8805/rnaseq_2/trimmed_fastQC_output/ERR899421_fastqc.html' target='_blank'>ERR899421_fastqc.html</a><br>\n",
       "&nbsp;&nbsp;<a href='/users/PAS1307/osu8805/rnaseq_2/trimmed_fastQC_output/ERR899423_fastqc.html' target='_blank'>ERR899423_fastqc.html</a><br>\n",
       "&nbsp;&nbsp;<a href='/users/PAS1307/osu8805/rnaseq_2/trimmed_fastQC_output/ERR899432_fastqc.html' target='_blank'>ERR899432_fastqc.html</a><br>\n",
       "&nbsp;&nbsp;<a href='/users/PAS1307/osu8805/rnaseq_2/trimmed_fastQC_output/ERR899422_fastqc.html' target='_blank'>ERR899422_fastqc.html</a><br>\n",
       "&nbsp;&nbsp;<a href='/users/PAS1307/osu8805/rnaseq_2/trimmed_fastQC_output/ERR899435_fastqc.html' target='_blank'>ERR899435_fastqc.html</a><br>\n",
       "&nbsp;&nbsp;<a href='/users/PAS1307/osu8805/rnaseq_2/trimmed_fastQC_output/ERR899424_fastqc.html' target='_blank'>ERR899424_fastqc.html</a><br>\n",
       "&nbsp;&nbsp;<a href='/users/PAS1307/osu8805/rnaseq_2/trimmed_fastQC_output/ERR899436_fastqc.html' target='_blank'>ERR899436_fastqc.html</a><br>\n",
       "&nbsp;&nbsp;<a href='/users/PAS1307/osu8805/rnaseq_2/trimmed_fastQC_output/ERR899434_fastqc.html' target='_blank'>ERR899434_fastqc.html</a><br>\n",
       "&nbsp;&nbsp;<a href='/users/PAS1307/osu8805/rnaseq_2/trimmed_fastQC_output/ERR899437_fastqc.html' target='_blank'>ERR899437_fastqc.html</a><br>\n",
       "&nbsp;&nbsp;<a href='/users/PAS1307/osu8805/rnaseq_2/trimmed_fastQC_output/ERR899429_fastqc.html' target='_blank'>ERR899429_fastqc.html</a><br>\n",
       "&nbsp;&nbsp;<a href='/users/PAS1307/osu8805/rnaseq_2/trimmed_fastQC_output/ERR899431_fastqc.html' target='_blank'>ERR899431_fastqc.html</a><br>\n",
       "&nbsp;&nbsp;<a href='/users/PAS1307/osu8805/rnaseq_2/trimmed_fastQC_output/ERR899426_fastqc.html' target='_blank'>ERR899426_fastqc.html</a><br>\n",
       "&nbsp;&nbsp;<a href='/users/PAS1307/osu8805/rnaseq_2/trimmed_fastQC_output/ERR899433_fastqc.html' target='_blank'>ERR899433_fastqc.html</a><br>\n",
       "&nbsp;&nbsp;<a href='/users/PAS1307/osu8805/rnaseq_2/trimmed_fastQC_output/ERR899425_fastqc.html' target='_blank'>ERR899425_fastqc.html</a><br>\n",
       "&nbsp;&nbsp;<a href='/users/PAS1307/osu8805/rnaseq_2/trimmed_fastQC_output/ERR899438_fastqc.html' target='_blank'>ERR899438_fastqc.html</a><br>\n",
       "&nbsp;&nbsp;<a href='/users/PAS1307/osu8805/rnaseq_2/trimmed_fastQC_output/ERR899428_trimmed_fastqc.html' target='_blank'>ERR899428_trimmed_fastqc.html</a><br>\n",
       "&nbsp;&nbsp;<a href='/users/PAS1307/osu8805/rnaseq_2/trimmed_fastQC_output/ERR899427_trimmed_fastqc.html' target='_blank'>ERR899427_trimmed_fastqc.html</a><br>\n",
       "&nbsp;&nbsp;<a href='/users/PAS1307/osu8805/rnaseq_2/trimmed_fastQC_output/ERR899435_trimmed_fastqc.html' target='_blank'>ERR899435_trimmed_fastqc.html</a><br>\n",
       "&nbsp;&nbsp;<a href='/users/PAS1307/osu8805/rnaseq_2/trimmed_fastQC_output/ERR899421_trimmed_fastqc.html' target='_blank'>ERR899421_trimmed_fastqc.html</a><br>\n",
       "&nbsp;&nbsp;<a href='/users/PAS1307/osu8805/rnaseq_2/trimmed_fastQC_output/ERR899432_trimmed_fastqc.html' target='_blank'>ERR899432_trimmed_fastqc.html</a><br>\n",
       "&nbsp;&nbsp;<a href='/users/PAS1307/osu8805/rnaseq_2/trimmed_fastQC_output/ERR899436_trimmed_fastqc.html' target='_blank'>ERR899436_trimmed_fastqc.html</a><br>\n",
       "&nbsp;&nbsp;<a href='/users/PAS1307/osu8805/rnaseq_2/trimmed_fastQC_output/ERR899434_trimmed_fastqc.html' target='_blank'>ERR899434_trimmed_fastqc.html</a><br>\n",
       "&nbsp;&nbsp;<a href='/users/PAS1307/osu8805/rnaseq_2/trimmed_fastQC_output/ERR899423_trimmed_fastqc.html' target='_blank'>ERR899423_trimmed_fastqc.html</a><br>\n",
       "&nbsp;&nbsp;<a href='/users/PAS1307/osu8805/rnaseq_2/trimmed_fastQC_output/ERR899424_trimmed_fastqc.html' target='_blank'>ERR899424_trimmed_fastqc.html</a><br>\n",
       "&nbsp;&nbsp;<a href='/users/PAS1307/osu8805/rnaseq_2/trimmed_fastQC_output/ERR899437_trimmed_fastqc.html' target='_blank'>ERR899437_trimmed_fastqc.html</a><br>\n",
       "&nbsp;&nbsp;<a href='/users/PAS1307/osu8805/rnaseq_2/trimmed_fastQC_output/ERR899422_trimmed_fastqc.html' target='_blank'>ERR899422_trimmed_fastqc.html</a><br>\n",
       "&nbsp;&nbsp;<a href='/users/PAS1307/osu8805/rnaseq_2/trimmed_fastQC_output/ERR899430_trimmed_fastqc.html' target='_blank'>ERR899430_trimmed_fastqc.html</a><br>\n",
       "&nbsp;&nbsp;<a href='/users/PAS1307/osu8805/rnaseq_2/trimmed_fastQC_output/ERR899426_trimmed_fastqc.html' target='_blank'>ERR899426_trimmed_fastqc.html</a><br>\n",
       "&nbsp;&nbsp;<a href='/users/PAS1307/osu8805/rnaseq_2/trimmed_fastQC_output/ERR899429_trimmed_fastqc.html' target='_blank'>ERR899429_trimmed_fastqc.html</a><br>\n",
       "&nbsp;&nbsp;<a href='/users/PAS1307/osu8805/rnaseq_2/trimmed_fastQC_output/ERR899431_trimmed_fastqc.html' target='_blank'>ERR899431_trimmed_fastqc.html</a><br>\n",
       "&nbsp;&nbsp;<a href='/users/PAS1307/osu8805/rnaseq_2/trimmed_fastQC_output/ERR899433_trimmed_fastqc.html' target='_blank'>ERR899433_trimmed_fastqc.html</a><br>\n",
       "&nbsp;&nbsp;<a href='/users/PAS1307/osu8805/rnaseq_2/trimmed_fastQC_output/ERR899425_trimmed_fastqc.html' target='_blank'>ERR899425_trimmed_fastqc.html</a><br>\n",
       "&nbsp;&nbsp;<a href='/users/PAS1307/osu8805/rnaseq_2/trimmed_fastQC_output/ERR899438_trimmed_fastqc.html' target='_blank'>ERR899438_trimmed_fastqc.html</a><br>"
      ],
      "text/plain": [
       "/users/PAS1307/osu8805/rnaseq_2/trimmed_fastQC_output/\n",
       "  ERR899427_fastqc.html\n",
       "  ERR899428_fastqc.html\n",
       "  ERR899430_fastqc.html\n",
       "  ERR899421_fastqc.html\n",
       "  ERR899423_fastqc.html\n",
       "  ERR899432_fastqc.html\n",
       "  ERR899422_fastqc.html\n",
       "  ERR899435_fastqc.html\n",
       "  ERR899424_fastqc.html\n",
       "  ERR899436_fastqc.html\n",
       "  ERR899434_fastqc.html\n",
       "  ERR899437_fastqc.html\n",
       "  ERR899429_fastqc.html\n",
       "  ERR899431_fastqc.html\n",
       "  ERR899426_fastqc.html\n",
       "  ERR899433_fastqc.html\n",
       "  ERR899425_fastqc.html\n",
       "  ERR899438_fastqc.html\n",
       "  ERR899428_trimmed_fastqc.html\n",
       "  ERR899427_trimmed_fastqc.html\n",
       "  ERR899435_trimmed_fastqc.html\n",
       "  ERR899421_trimmed_fastqc.html\n",
       "  ERR899432_trimmed_fastqc.html\n",
       "  ERR899436_trimmed_fastqc.html\n",
       "  ERR899434_trimmed_fastqc.html\n",
       "  ERR899423_trimmed_fastqc.html\n",
       "  ERR899424_trimmed_fastqc.html\n",
       "  ERR899437_trimmed_fastqc.html\n",
       "  ERR899422_trimmed_fastqc.html\n",
       "  ERR899430_trimmed_fastqc.html\n",
       "  ERR899426_trimmed_fastqc.html\n",
       "  ERR899429_trimmed_fastqc.html\n",
       "  ERR899431_trimmed_fastqc.html\n",
       "  ERR899433_trimmed_fastqc.html\n",
       "  ERR899425_trimmed_fastqc.html\n",
       "  ERR899438_trimmed_fastqc.html"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "from IPython.display import FileLinks, display\n",
    "\n",
    "d = FileLinks(os.environ['WORKDIR'], included_suffixes=['.html'])\n",
    "e = FileLinks(os.path.join(os.environ['WORKDIR'], os.environ['FASTQCOUT']), included_suffixes=['.html'])\n",
    "f = FileLinks(os.path.join(os.environ['WORKDIR'], os.environ['TFASTQCOUT']), included_suffixes=['.html'])\n",
    "\n",
    "display(d)\n",
    "display(e)\n",
    "display(f)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python [conda env:rnaseq]",
   "language": "python",
   "name": "conda-env-rnaseq-py"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.6.4"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
