{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## OSU DS4S Data Science Club\n",
    "### RNASeq Workflow\n",
    "\n",
    "1. Setup Environment\n",
    "1. Download SRA Files\n",
    "1. Convert SRA to fastq\n",
    "1. Index Genome\n",
    "1. Run fastQC\n",
    "1. Align Reads\n",
    "1. Run FeatureCounts"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "import os\n",
    "\n",
    "# MAX CPUS For Owens should be 24 (28 possible but reserve a few for interactivity)\n",
    "os.environ['MAXCORES'] = '24'\n",
    "\n",
    "# Set Working Directory\n",
    "os.environ['WORKDIR'] = os.getcwd()\n",
    "\n",
    "# Set sra_toolkit cfg location.  Needed to overide default download directory\n",
    "# Will default to data directory in the current working directory.\n",
    "os.environ['VDB_CONFIG'] = os.path.expandvars(\"$WORKDIR/sra_toolkit.kfg\")\n",
    "\n",
    "# The data download directories\n",
    "os.environ['DATA'] =  os.path.expandvars(\"$WORKDIR/data/sra\")\n",
    "\n",
    "# The results directories\n",
    "os.environ['FASTQS'] = 'fastqs'\n",
    "os.environ['TFASTQS'] = 'trimmed_fastqs'\n",
    "os.environ['PFASTQS'] = 'paired_fastqs'\n",
    "os.environ['TPFASTQS'] = 'trimmed_paired_fastqs'\n",
    "os.environ['FASTQCOUT'] = 'fastQC_output'\n",
    "os.environ['TFASTQCOUT'] = 'trimmed_fastQC_output'\n",
    "os.environ['STARDIR'] = './star_output'\n",
    "os.environ['FCDIR'] = './featureCount_output'\n",
    "\n",
    "# The binary directories\n",
    "os.environ['BASEDIR'] = '/fs/project/PAS1307'\n",
    "os.environ['BINDIR'] = os.path.expandvars(\"$BASEDIR/bin\")\n",
    "os.environ['SRABINDIR'] = os.path.expandvars(\"$BASEDIR/bin/sratoolkit.2.8.2-1-ubuntu64/bin\")\n",
    "os.environ['TGBINDIR'] = os.path.expandvars(\"$BASEDIR/bin/trim_galore\")\n",
    "\n",
    "# The Genome directories\n",
    "os.environ['GENOMEDIR'] = os.path.expandvars(\"$BASEDIR/genomes\")\n",
    "#os.environ['GENOME'] = 'Drosophila_melanogaster/Ensembl/BDGP5.25/'         \n",
    "os.environ['GENOME'] = 'Mus_musculus/Ensembl/GRCm38' \n",
    "os.environ['WHOLEGENOMEDIR'] = os.path.expandvars(\"$GENOMEDIR/$GENOME/Sequence/WholeGenomeFasta\")\n",
    "os.environ['GENEANNOTATIONDIR'] = os.path.expandvars(\"$GENOMEDIR/$GENOME/Annotation/Genes\")\n",
    "os.environ['STARGENOMEDIR'] = os.path.expandvars(\"$GENOMEDIR/$GENOME/Sequence/StarIndex\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "%%bash\n",
    "\n",
    "mkdir -p $FASTQS\n",
    "mkdir -p $PFASTQS\n",
    "mkdir -p $FASTQCOUT\n",
    "mkdir -p $STARDIR\n",
    "mkdir -p $FCDIR\n",
    "mkdir -p $TFASTQS\n",
    "mkdir -p $TPFASTQS\n",
    "mkdir -p $TFASTQCOUT"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {
    "collapsed": false,
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "2018-01-24T17:16:25 prefetch.2.8.2: 1) 'ERR899429' is found locally\n",
      "\n",
      "2018-01-24T17:16:27 prefetch.2.8.2: 2) 'ERR899433' is found locally\n",
      "\n",
      "2018-01-24T17:16:27 prefetch.2.8.2: 3) 'ERR899427' is found locally\n",
      "\n",
      "2018-01-24T17:16:27 prefetch.2.8.2: 4) 'ERR899421' is found locally\n",
      "\n",
      "2018-01-24T17:16:27 prefetch.2.8.2: 5) 'ERR899425' is found locally\n",
      "\n",
      "2018-01-24T17:16:28 prefetch.2.8.2: 6) 'ERR899435' is found locally\n",
      "\n",
      "2018-01-24T17:16:28 prefetch.2.8.2: 7) 'ERR899430' is found locally\n",
      "\n",
      "2018-01-24T17:16:28 prefetch.2.8.2: 8) 'ERR899432' is found locally\n",
      "\n",
      "2018-01-24T17:16:28 prefetch.2.8.2: 9) 'ERR899431' is found locally\n",
      "\n",
      "2018-01-24T17:16:28 prefetch.2.8.2: 10) 'ERR899434' is found locally\n",
      "\n",
      "2018-01-24T17:16:29 prefetch.2.8.2: 11) 'ERR899438' is found locally\n",
      "\n",
      "2018-01-24T17:16:29 prefetch.2.8.2: 12) 'ERR899422' is found locally\n",
      "\n",
      "2018-01-24T17:16:29 prefetch.2.8.2: 13) 'ERR899423' is found locally\n",
      "\n",
      "2018-01-24T17:16:29 prefetch.2.8.2: 14) 'ERR899424' is found locally\n",
      "\n",
      "2018-01-24T17:16:29 prefetch.2.8.2: 15) Downloading 'ERR899428'...\n",
      "2018-01-24T17:16:29 prefetch.2.8.2:  Downloading via https...\n",
      "2018-01-24T17:16:45 prefetch.2.8.2: 15) 'ERR899428' was downloaded successfully\n",
      "\n",
      "2018-01-24T17:16:46 prefetch.2.8.2: 16) Downloading 'ERR899426'...\n",
      "2018-01-24T17:16:46 prefetch.2.8.2:  Downloading via https...\n",
      "2018-01-24T17:17:04 prefetch.2.8.2: 16) 'ERR899426' was downloaded successfully\n",
      "\n",
      "2018-01-24T17:17:05 prefetch.2.8.2: 17) Downloading 'ERR899437'...\n",
      "2018-01-24T17:17:05 prefetch.2.8.2:  Downloading via https...\n",
      "2018-01-24T17:17:22 prefetch.2.8.2: 17) 'ERR899437' was downloaded successfully\n",
      "\n",
      "2018-01-24T17:17:23 prefetch.2.8.2: 18) Downloading 'ERR899436'...\n",
      "2018-01-24T17:17:23 prefetch.2.8.2:  Downloading via https...\n",
      "2018-01-24T17:17:38 prefetch.2.8.2: 18) 'ERR899436' was downloaded successfully\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "\n",
    "$SRABINDIR/prefetch --option-file $WORKDIR/SraAccList.txt"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "%%bash\n",
    "\n",
    "## Make STAR index if not exists\n",
    "if [ ! -d \"$STARGENOMEDIR/\" ]; then\n",
    "    echo \"STAR index does not exist, building STAR index\"\n",
    "\n",
    "    mkdir $STARGENOMEDIR\n",
    "\n",
    "    $BINDIR/STAR \\\n",
    "        --runThreadN $MAXCORES \\\n",
    "        --runMode genomeGenerate \\\n",
    "        --genomeDir $STARGENOMEDIR \\\n",
    "        --genomeFastaFiles $WHOLEGENOMEDIR/genome.fa \\\n",
    "        --sjdbGTFfile $GENEANNOTATIONDIR/genes.gtf \\\n",
    "        --sjdbOverhang 100\n",
    "fi"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {
    "collapsed": false,
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Dumping .sra files to .fastq.gz files\n",
      "/users/PAS1307/osu8805/rnaseq_2/data/sra/ERR899421.sra\n",
      "/fs/project/PAS1307/bin/sratoolkit.2.8.2-1-ubuntu64/bin\n",
      "/users/PAS1307/osu8805/rnaseq_2/data/sra/ERR899422.sra\n",
      "/fs/project/PAS1307/bin/sratoolkit.2.8.2-1-ubuntu64/bin\n",
      "/users/PAS1307/osu8805/rnaseq_2/data/sra/ERR899423.sra\n",
      "/fs/project/PAS1307/bin/sratoolkit.2.8.2-1-ubuntu64/bin\n",
      "/users/PAS1307/osu8805/rnaseq_2/data/sra/ERR899424.sra\n",
      "/fs/project/PAS1307/bin/sratoolkit.2.8.2-1-ubuntu64/bin\n",
      "/users/PAS1307/osu8805/rnaseq_2/data/sra/ERR899425.sra\n",
      "/fs/project/PAS1307/bin/sratoolkit.2.8.2-1-ubuntu64/bin\n",
      "/users/PAS1307/osu8805/rnaseq_2/data/sra/ERR899426.sra\n",
      "/fs/project/PAS1307/bin/sratoolkit.2.8.2-1-ubuntu64/bin\n",
      "/users/PAS1307/osu8805/rnaseq_2/data/sra/ERR899427.sra\n",
      "/fs/project/PAS1307/bin/sratoolkit.2.8.2-1-ubuntu64/bin\n",
      "/users/PAS1307/osu8805/rnaseq_2/data/sra/ERR899428.sra\n",
      "/fs/project/PAS1307/bin/sratoolkit.2.8.2-1-ubuntu64/bin\n",
      "/users/PAS1307/osu8805/rnaseq_2/data/sra/ERR899429.sra\n",
      "/fs/project/PAS1307/bin/sratoolkit.2.8.2-1-ubuntu64/bin\n",
      "/users/PAS1307/osu8805/rnaseq_2/data/sra/ERR899430.sra\n",
      "/fs/project/PAS1307/bin/sratoolkit.2.8.2-1-ubuntu64/bin\n",
      "/users/PAS1307/osu8805/rnaseq_2/data/sra/ERR899431.sra\n",
      "/fs/project/PAS1307/bin/sratoolkit.2.8.2-1-ubuntu64/bin\n",
      "/users/PAS1307/osu8805/rnaseq_2/data/sra/ERR899432.sra\n",
      "/fs/project/PAS1307/bin/sratoolkit.2.8.2-1-ubuntu64/bin\n",
      "/users/PAS1307/osu8805/rnaseq_2/data/sra/ERR899433.sra\n",
      "/fs/project/PAS1307/bin/sratoolkit.2.8.2-1-ubuntu64/bin\n",
      "/users/PAS1307/osu8805/rnaseq_2/data/sra/ERR899434.sra\n",
      "/fs/project/PAS1307/bin/sratoolkit.2.8.2-1-ubuntu64/bin\n",
      "/users/PAS1307/osu8805/rnaseq_2/data/sra/ERR899435.sra\n",
      "/fs/project/PAS1307/bin/sratoolkit.2.8.2-1-ubuntu64/bin\n",
      "/users/PAS1307/osu8805/rnaseq_2/data/sra/ERR899436.sra\n",
      "/fs/project/PAS1307/bin/sratoolkit.2.8.2-1-ubuntu64/bin\n",
      "/users/PAS1307/osu8805/rnaseq_2/data/sra/ERR899437.sra\n",
      "/fs/project/PAS1307/bin/sratoolkit.2.8.2-1-ubuntu64/bin\n",
      "/users/PAS1307/osu8805/rnaseq_2/data/sra/ERR899438.sra\n",
      "/fs/project/PAS1307/bin/sratoolkit.2.8.2-1-ubuntu64/bin\n",
      "/users/PAS1307/osu8805/rnaseq_2/data/sra/ERR899426.sra is detected as single-end sequencing reads\n",
      "/users/PAS1307/osu8805/rnaseq_2/data/sra/ERR899427.sra is detected as single-end sequencing reads\n",
      "/users/PAS1307/osu8805/rnaseq_2/data/sra/ERR899438.sra is detected as single-end sequencing reads\n",
      "/users/PAS1307/osu8805/rnaseq_2/data/sra/ERR899437.sra is detected as single-end sequencing reads\n",
      "/users/PAS1307/osu8805/rnaseq_2/data/sra/ERR899421.sra is detected as single-end sequencing reads\n",
      "/users/PAS1307/osu8805/rnaseq_2/data/sra/ERR899428.sra is detected as single-end sequencing reads\n",
      "/users/PAS1307/osu8805/rnaseq_2/data/sra/ERR899434.sra is detected as single-end sequencing reads\n",
      "/users/PAS1307/osu8805/rnaseq_2/data/sra/ERR899430.sra is detected as single-end sequencing reads\n",
      "/users/PAS1307/osu8805/rnaseq_2/data/sra/ERR899433.sra is detected as single-end sequencing reads\n",
      "/users/PAS1307/osu8805/rnaseq_2/data/sra/ERR899425.sra is detected as single-end sequencing reads\n",
      "/users/PAS1307/osu8805/rnaseq_2/data/sra/ERR899422.sra is detected as single-end sequencing reads\n",
      "/users/PAS1307/osu8805/rnaseq_2/data/sra/ERR899435.sra is detected as single-end sequencing reads\n",
      "/users/PAS1307/osu8805/rnaseq_2/data/sra/ERR899436.sra is detected as single-end sequencing reads\n",
      "/users/PAS1307/osu8805/rnaseq_2/data/sra/ERR899432.sra is detected as single-end sequencing reads\n",
      "/users/PAS1307/osu8805/rnaseq_2/data/sra/ERR899431.sra is detected as single-end sequencing reads\n",
      "/users/PAS1307/osu8805/rnaseq_2/data/sra/ERR899423.sra is detected as single-end sequencing reads\n",
      "/users/PAS1307/osu8805/rnaseq_2/data/sra/ERR899424.sra is detected as single-end sequencing reads\n",
      "/users/PAS1307/osu8805/rnaseq_2/data/sra/ERR899429.sra is detected as single-end sequencing reads\n",
      "Read 17183069 spots for /users/PAS1307/osu8805/rnaseq_2/data/sra/ERR899427.sra\n",
      "Written 17183069 spots for /users/PAS1307/osu8805/rnaseq_2/data/sra/ERR899427.sra\n",
      "Read 17513754 spots for /users/PAS1307/osu8805/rnaseq_2/data/sra/ERR899428.sra\n",
      "Written 17513754 spots for /users/PAS1307/osu8805/rnaseq_2/data/sra/ERR899428.sra\n",
      "Read 17899859 spots for /users/PAS1307/osu8805/rnaseq_2/data/sra/ERR899435.sra\n",
      "Written 17899859 spots for /users/PAS1307/osu8805/rnaseq_2/data/sra/ERR899435.sra\n",
      "Read 18934244 spots for /users/PAS1307/osu8805/rnaseq_2/data/sra/ERR899430.sra\n",
      "Written 18934244 spots for /users/PAS1307/osu8805/rnaseq_2/data/sra/ERR899430.sra\n",
      "Read 17979693 spots for /users/PAS1307/osu8805/rnaseq_2/data/sra/ERR899421.sra\n",
      "Written 17979693 spots for /users/PAS1307/osu8805/rnaseq_2/data/sra/ERR899421.sra\n",
      "Read 18211995 spots for /users/PAS1307/osu8805/rnaseq_2/data/sra/ERR899432.sra\n",
      "Written 18211995 spots for /users/PAS1307/osu8805/rnaseq_2/data/sra/ERR899432.sra\n",
      "Read 18176427 spots for /users/PAS1307/osu8805/rnaseq_2/data/sra/ERR899436.sra\n",
      "Written 18176427 spots for /users/PAS1307/osu8805/rnaseq_2/data/sra/ERR899436.sra\n",
      "Read 18421390 spots for /users/PAS1307/osu8805/rnaseq_2/data/sra/ERR899434.sra\n",
      "Written 18421390 spots for /users/PAS1307/osu8805/rnaseq_2/data/sra/ERR899434.sra\n",
      "Read 18424646 spots for /users/PAS1307/osu8805/rnaseq_2/data/sra/ERR899423.sra\n",
      "Written 18424646 spots for /users/PAS1307/osu8805/rnaseq_2/data/sra/ERR899423.sra\n",
      "Read 18655823 spots for /users/PAS1307/osu8805/rnaseq_2/data/sra/ERR899437.sra\n",
      "Written 18655823 spots for /users/PAS1307/osu8805/rnaseq_2/data/sra/ERR899437.sra\n",
      "Read 18487849 spots for /users/PAS1307/osu8805/rnaseq_2/data/sra/ERR899424.sra\n",
      "Written 18487849 spots for /users/PAS1307/osu8805/rnaseq_2/data/sra/ERR899424.sra\n",
      "Read 19410496 spots for /users/PAS1307/osu8805/rnaseq_2/data/sra/ERR899426.sra\n",
      "Written 19410496 spots for /users/PAS1307/osu8805/rnaseq_2/data/sra/ERR899426.sra\n",
      "Read 18794828 spots for /users/PAS1307/osu8805/rnaseq_2/data/sra/ERR899422.sra\n",
      "Written 18794828 spots for /users/PAS1307/osu8805/rnaseq_2/data/sra/ERR899422.sra\n",
      "Read 19419534 spots for /users/PAS1307/osu8805/rnaseq_2/data/sra/ERR899431.sra\n",
      "Written 19419534 spots for /users/PAS1307/osu8805/rnaseq_2/data/sra/ERR899431.sra\n",
      "Read 19286135 spots for /users/PAS1307/osu8805/rnaseq_2/data/sra/ERR899429.sra\n",
      "Written 19286135 spots for /users/PAS1307/osu8805/rnaseq_2/data/sra/ERR899429.sra\n",
      "Read 19836280 spots for /users/PAS1307/osu8805/rnaseq_2/data/sra/ERR899433.sra\n",
      "Written 19836280 spots for /users/PAS1307/osu8805/rnaseq_2/data/sra/ERR899433.sra\n",
      "Read 20649441 spots for /users/PAS1307/osu8805/rnaseq_2/data/sra/ERR899425.sra\n",
      "Written 20649441 spots for /users/PAS1307/osu8805/rnaseq_2/data/sra/ERR899425.sra\n",
      "Read 21276620 spots for /users/PAS1307/osu8805/rnaseq_2/data/sra/ERR899438.sra\n",
      "Written 21276620 spots for /users/PAS1307/osu8805/rnaseq_2/data/sra/ERR899438.sra\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "\n",
    "is_paired() {\n",
    "    # function to examine whether a .sra file is paired-end sequencing\n",
    "    # ref: https://www.biostars.org/p/139422/\n",
    "    local SRA=\"$1\"\n",
    "    local x=$(\n",
    "        $SRABINDIR/fastq-dump -I -X 1 -Z --split-spot \"$SRA\" 2>/dev/null \\\n",
    "        | awk '{if(NR % 2 == 1) print substr($1,length($1),1)}' \\\n",
    "        | uniq \\\n",
    "        | wc -l\n",
    "    )\n",
    "    # echo $SRA $x\n",
    "    # $x should be 2 if paired-end, 1 if single-end\n",
    "    if [ $x == 2 ]; then\n",
    "        return 0 # true\n",
    "    else\n",
    "        return 1 # false\n",
    "    fi\n",
    "}\n",
    "\n",
    "dump_sra() {\n",
    "    # function to dump sra to fastq file with auto-detecting\n",
    "    # whether reads are from single-end or paired-end\n",
    "    local sra=\"$1\"\n",
    "    echo $sra\n",
    "    echo $SRABINDIR\n",
    "    if is_paired $sra; then\n",
    "        echo \"$sra is detected as paired-end sequencing reads\"\n",
    "        # Note that paired-end sequencing reads should be dumped into two fastq files\n",
    "        $SRABINDIR/fastq-dump --gzip -I --split-files -O paired_fastqs \"$sra\"\n",
    "    else\n",
    "        echo \"$sra is detected as single-end sequencing reads\"\n",
    "        $SRABINDIR/fastq-dump --gzip -O fastqs \"$sra\"\n",
    "    fi\n",
    "}\n",
    "\n",
    "\n",
    "## Dump .sra to .fastq\n",
    "echo \"Dumping .sra files to .fastq.gz files\"\n",
    "\n",
    "## Dump in parallel using xargs\n",
    "export -f is_paired\n",
    "export -f dump_sra\n",
    "\n",
    "find $DATA/*.sra | xargs --max-args=1 --max-procs=$MAXCORES -I {} bash -c 'dump_sra \"$@\"' _ {}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {
    "collapsed": false,
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Process is terminated.\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "#fastqc of sequencing reads\n",
    "\n",
    "#load fastqc\n",
    "module load fastqc\n",
    "\n",
    "fqs=\"\"\n",
    "\n",
    "for basename in $(ls fastqs | cut -f1 -d '.'); do\n",
    "    #basename=$(echo $fq | cut -f1 -d '.')\n",
    "    fqd=$FASTQS\n",
    "    fq1=\".fastq.gz\"\n",
    "    fq1=$fqd/$basename$fq1\n",
    "    fqs=\"$fqs $fq1\"\n",
    "    \n",
    "done\n",
    "\n",
    "for basename in $(ls paired_fastqs | cut -f1 -d '_' | sort | uniq); do\n",
    "    echo $basename\n",
    "    fqd=$PFASTQS\n",
    "    fq1=\"_1.fastq.gz\"\n",
    "    fq2=\"_2.fastq.gz\"\n",
    "    fq1=$fqd/$basename$fq1\n",
    "    fq2=$fqd/$basename$fq2\n",
    "    fqs=\"$fqs $fq1 $fq2\"\n",
    "done\n",
    "\n",
    "echo \"Performing FastQC for $fqs\"\n",
    "fastqc --threads $MAXCORES $fqs -o $FASTQCOUT\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "%%bash\n",
    "#Activate the environment where cutadapt is installed.\n",
    "source activate tophat\n",
    "module load fastqc\n",
    "\n",
    "#Quality filtering and adapter trimming\n",
    "PHRED=\"20\"\n",
    "\n",
    "for basename in $(ls fastqs | cut -f1 -d '.'); do\n",
    "    fq1=\".fastq.gz\"\n",
    "    fq1=$FASTQS/$basename$fq1\n",
    "    $TGBINDIR/trim_galore \\\n",
    "        -q $PHRED \\\n",
    "        --fastqc \\\n",
    "        --fastqc_args '\"-o $TFASTQCOUT\"' \\\n",
    "        -o $TFASTQS \\\n",
    "        $fq1\n",
    "done\n",
    "\n",
    "for basename in $(ls paired_fastqs | cut -f1 -d '_' | sort | uniq); do\n",
    "    fq1=\"_1.fastq.gz\"\n",
    "    fq2=\"_2.fastq.gz\"\n",
    "    fq1=$PFASTQS/$basename$fq1\n",
    "    fq2=$PFASTQS/$basename$fq2\n",
    "    fqs=\"$fq1 $fq2\"\n",
    "    $TGBINDIR/trim_galore \\\n",
    "        -q $PHRED \\\n",
    "        --fastqc \\\n",
    "        --fastqc_args '\"-o $TFASTQCOUT\"' \\\n",
    "        --paired \\\n",
    "        -o $TPFASTQS/ \\\n",
    "        $fqs\n",
    "done"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 48,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Aligning reads from ERR899421 to the reference genome\n",
      "Jan 19 12:19:38 ..... started STAR run\n",
      "Jan 19 12:19:38 ..... loading genome\n",
      "Jan 19 12:19:52 ..... processing annotations GTF\n",
      "Jan 19 12:20:25 ..... inserting junctions into the genome indices\n",
      "Jan 19 12:21:33 ..... started mapping\n",
      "Jan 19 12:23:17 ..... finished successfully\n",
      "Aligning reads from ERR899422 to the reference genome\n",
      "Jan 19 12:23:43 ..... started STAR run\n",
      "Jan 19 12:23:44 ..... loading genome\n",
      "Jan 19 12:23:57 ..... processing annotations GTF\n",
      "Jan 19 12:24:06 ..... inserting junctions into the genome indices\n",
      "Jan 19 12:25:13 ..... started mapping\n",
      "Jan 19 12:26:56 ..... finished successfully\n",
      "Aligning reads from ERR899423 to the reference genome\n",
      "Jan 19 12:27:25 ..... started STAR run\n",
      "Jan 19 12:27:25 ..... loading genome\n",
      "Jan 19 12:27:38 ..... processing annotations GTF\n",
      "Jan 19 12:28:12 ..... inserting junctions into the genome indices\n",
      "Jan 19 12:29:19 ..... started mapping\n",
      "Jan 19 12:30:59 ..... finished successfully\n",
      "Aligning reads from ERR899424 to the reference genome\n",
      "Jan 19 12:31:29 ..... started STAR run\n",
      "Jan 19 12:31:29 ..... loading genome\n",
      "Jan 19 12:31:41 ..... processing annotations GTF\n",
      "Jan 19 12:31:50 ..... inserting junctions into the genome indices\n",
      "Jan 19 12:32:57 ..... started mapping\n",
      "Jan 19 12:34:39 ..... finished successfully\n",
      "Aligning reads from ERR899425 to the reference genome\n",
      "Jan 19 12:35:07 ..... started STAR run\n",
      "Jan 19 12:35:08 ..... loading genome\n",
      "Jan 19 12:35:21 ..... processing annotations GTF\n",
      "Jan 19 12:35:56 ..... inserting junctions into the genome indices\n",
      "Jan 19 12:37:02 ..... started mapping\n",
      "Jan 19 12:38:55 ..... finished successfully\n",
      "Aligning reads from ERR899426 to the reference genome\n",
      "Jan 19 12:39:24 ..... started STAR run\n",
      "Jan 19 12:39:24 ..... loading genome\n",
      "Jan 19 12:39:37 ..... processing annotations GTF\n",
      "Jan 19 12:39:45 ..... inserting junctions into the genome indices\n",
      "Jan 19 12:40:52 ..... started mapping\n",
      "Jan 19 12:42:41 ..... finished successfully\n",
      "Aligning reads from ERR899427 to the reference genome\n",
      "Jan 19 12:43:06 ..... started STAR run\n",
      "Jan 19 12:43:07 ..... loading genome\n",
      "Jan 19 12:43:19 ..... processing annotations GTF\n",
      "Jan 19 12:43:53 ..... inserting junctions into the genome indices\n",
      "Jan 19 12:45:01 ..... started mapping\n",
      "Jan 19 12:46:34 ..... finished successfully\n",
      "Aligning reads from ERR899428 to the reference genome\n",
      "Jan 19 12:47:00 ..... started STAR run\n",
      "Jan 19 12:47:00 ..... loading genome\n",
      "Jan 19 12:47:13 ..... processing annotations GTF\n",
      "Jan 19 12:47:22 ..... inserting junctions into the genome indices\n",
      "Jan 19 12:48:28 ..... started mapping\n",
      "Jan 19 12:50:09 ..... finished successfully\n",
      "Aligning reads from ERR899429 to the reference genome\n",
      "Jan 19 12:50:36 ..... started STAR run\n",
      "Jan 19 12:50:36 ..... loading genome\n",
      "Jan 19 12:50:50 ..... processing annotations GTF\n",
      "Jan 19 12:51:38 ..... inserting junctions into the genome indices\n",
      "Jan 19 12:52:44 ..... started mapping\n",
      "Jan 19 12:54:30 ..... finished successfully\n",
      "Aligning reads from ERR899430 to the reference genome\n",
      "Jan 19 12:55:05 ..... started STAR run\n",
      "Jan 19 12:55:05 ..... loading genome\n",
      "Jan 19 12:56:40 ..... processing annotations GTF\n",
      "Jan 19 12:56:50 ..... inserting junctions into the genome indices\n",
      "Jan 19 12:57:55 ..... started mapping\n",
      "Jan 19 12:59:42 ..... finished successfully\n",
      "Aligning reads from ERR899431 to the reference genome\n",
      "Jan 19 13:00:11 ..... started STAR run\n",
      "Jan 19 13:00:11 ..... loading genome\n",
      "Jan 19 13:00:25 ..... processing annotations GTF\n",
      "Jan 19 13:00:34 ..... inserting junctions into the genome indices\n",
      "Jan 19 13:01:40 ..... started mapping\n",
      "Jan 19 13:03:25 ..... finished successfully\n",
      "Aligning reads from ERR899432 to the reference genome\n",
      "Jan 19 13:03:55 ..... started STAR run\n",
      "Jan 19 13:03:55 ..... loading genome\n",
      "Jan 19 13:04:09 ..... processing annotations GTF\n",
      "Jan 19 13:04:19 ..... inserting junctions into the genome indices\n",
      "Jan 19 13:05:25 ..... started mapping\n",
      "Jan 19 13:07:08 ..... finished successfully\n",
      "Aligning reads from ERR899433 to the reference genome\n",
      "Jan 19 13:07:35 ..... started STAR run\n",
      "Jan 19 13:07:35 ..... loading genome\n",
      "Jan 19 13:08:00 ..... processing annotations GTF\n",
      "Jan 19 13:08:09 ..... inserting junctions into the genome indices\n",
      "Jan 19 13:09:14 ..... started mapping\n",
      "Jan 19 13:11:03 ..... finished successfully\n",
      "Aligning reads from ERR899434 to the reference genome\n",
      "Jan 19 13:11:32 ..... started STAR run\n",
      "Jan 19 13:11:32 ..... loading genome\n",
      "Jan 19 13:11:47 ..... processing annotations GTF\n",
      "Jan 19 13:11:56 ..... inserting junctions into the genome indices\n",
      "Jan 19 13:13:01 ..... started mapping\n",
      "Jan 19 13:14:43 ..... finished successfully\n",
      "Aligning reads from ERR899435 to the reference genome\n",
      "Jan 19 13:15:11 ..... started STAR run\n",
      "Jan 19 13:15:11 ..... loading genome\n",
      "Jan 19 13:15:26 ..... processing annotations GTF\n",
      "Jan 19 13:15:37 ..... inserting junctions into the genome indices\n",
      "Jan 19 13:16:42 ..... started mapping\n",
      "Jan 19 13:18:26 ..... finished successfully\n",
      "Aligning reads from ERR899436 to the reference genome\n",
      "Jan 19 13:18:53 ..... started STAR run\n",
      "Jan 19 13:18:53 ..... loading genome\n",
      "Jan 19 13:19:07 ..... processing annotations GTF\n",
      "Jan 19 13:19:16 ..... inserting junctions into the genome indices\n",
      "Jan 19 13:20:21 ..... started mapping\n",
      "Jan 19 13:22:04 ..... finished successfully\n",
      "Aligning reads from ERR899437 to the reference genome\n",
      "Jan 19 13:22:32 ..... started STAR run\n",
      "Jan 19 13:22:32 ..... loading genome\n",
      "Jan 19 13:22:46 ..... processing annotations GTF\n",
      "Jan 19 13:22:55 ..... inserting junctions into the genome indices\n",
      "Jan 19 13:24:01 ..... started mapping\n",
      "Jan 19 13:25:45 ..... finished successfully\n",
      "Aligning reads from ERR899438 to the reference genome\n",
      "Jan 19 13:26:13 ..... started STAR run\n",
      "Jan 19 13:26:13 ..... loading genome\n",
      "Jan 19 13:26:27 ..... processing annotations GTF\n",
      "Jan 19 13:26:37 ..... inserting junctions into the genome indices\n",
      "Jan 19 13:27:41 ..... started mapping\n",
      "Jan 19 13:29:38 ..... finished successfully\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "\n",
    "#module load star\n",
    "\n",
    "for basename in $(ls fastqs | cut -f1 -d '.'); do\n",
    "    fqd=$TFASTQS\n",
    "    fq1=\".fastq.gz\"\n",
    "    fq1=$fqd/$basename$fq1\n",
    "    \n",
    "    echo \"Aligning reads from $basename to the reference genome\"\n",
    "    $BINDIR/STAR \\\n",
    "        --genomeDir $STARGENOMEDIR \\\n",
    "        --sjdbGTFfile $GENEANNOTATIONDIR/genes.gtf \\\n",
    "        --runThreadN $MAXCORES \\\n",
    "        --outSAMstrandField intronMotif \\\n",
    "        --outFilterIntronMotifs RemoveNoncanonical \\\n",
    "        --outFileNamePrefix $STARDIR/$basename  \\\n",
    "        --readFilesIn $fq1 \\\n",
    "        --readFilesCommand zcat \\\n",
    "        --outSAMtype BAM Unsorted \\\n",
    "        --outReadsUnmapped Fastx \\\n",
    "        --outSAMmode Full\n",
    "done\n",
    "\n",
    "\n",
    "for basename in $(ls paired_fastqs | cut -f1 -d '_' | sort | uniq); do\n",
    "    fqd=$TPFASTQS\n",
    "    fq1=\"_1.fastq.gz\"\n",
    "    fq2=\"_2.fastq.gz\"\n",
    "    fq1=$fqd/$basename$fq1\n",
    "    fq2=$fqd/$basename$fq2\n",
    "    fqs=\"$fqs $fq1 $fq2\"\n",
    "    \n",
    "    \n",
    "    echo \"Aligning reads from $basename to the reference genome\"\n",
    "    $BINDIR/STAR \\\n",
    "        --genomeDir $STARGENOMEDIR \\\n",
    "        --sjdbGTFfile $GENEANNOTATIONDIR/genes.gtf \\\n",
    "        --runThreadN $MAXCORES \\\n",
    "        --outSAMstrandField intronMotif \\\n",
    "        --outFilterIntronMotifs RemoveNoncanonical \\\n",
    "        --outFileNamePrefix $STARDIR/$basename \\\n",
    "        --readFilesIn $fq1 $fq2 \\\n",
    "        --readFilesCommand zcat \\\n",
    "        --outSAMtype BAM Unsorted \\\n",
    "        --outReadsUnmapped Fastx \\\n",
    "        --outSAMmode Full\n",
    "done"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 54,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "        ==========     _____ _    _ ____  _____  ______          _____  \n",
      "        =====         / ____| |  | |  _ \\|  __ \\|  ____|   /\\   |  __ \\ \n",
      "          =====      | (___ | |  | | |_) | |__) | |__     /  \\  | |  | |\n",
      "            ====      \\___ \\| |  | |  _ <|  _  /|  __|   / /\\ \\ | |  | |\n",
      "              ====    ____) | |__| | |_) | | \\ \\| |____ / ____ \\| |__| |\n",
      "        ==========   |_____/ \\____/|____/|_|  \\_\\______/_/    \\_\\_____/\n",
      "\t  v1.6.0\n",
      "\n",
      "//========================== featureCounts setting ===========================\\\\\n",
      "||                                                                            ||\n",
      "||             Input files : 1 BAM file                                       ||\n",
      "||                           S ./star_output/ERR899421Aligned.out.bam         ||\n",
      "||                                                                            ||\n",
      "||             Output file : ./featureCount_output/ERR899421Aligned.count ... ||\n",
      "||                 Summary : ./featureCount_output/ERR899421Aligned.count ... ||\n",
      "||              Annotation : /fs/project/PAS1307/genomes/Mus_musculus/Ens ... ||\n",
      "||      Dir for temp files : ./featureCount_output                            ||\n",
      "||                                                                            ||\n",
      "||                 Threads : 24                                               ||\n",
      "||                   Level : meta-feature level                               ||\n",
      "||              Paired-end : no                                               ||\n",
      "||         Strand specific : no                                               ||\n",
      "||      Multimapping reads : not counted                                      ||\n",
      "|| Multi-overlapping reads : not counted                                      ||\n",
      "||   Min overlapping bases : 1                                                ||\n",
      "||                                                                            ||\n",
      "\\\\===================== http://subread.sourceforge.net/ ======================//\n",
      "\n",
      "//================================= Running ==================================\\\\\n",
      "||                                                                            ||\n",
      "|| Load annotation file /fs/project/PAS1307/genomes/Mus_musculus/Ensembl/ ... ||\n",
      "||    Features : 690647                                                       ||\n",
      "||    Meta-features : 46078                                                   ||\n",
      "||    Chromosomes/contigs : 73                                                ||\n",
      "||                                                                            ||\n",
      "|| Process BAM file ./star_output/ERR899421Aligned.out.bam...                 ||\n",
      "||    Single-end reads are included.                                          ||\n",
      "||    Assign reads to features...                                             ||\n",
      "||    Total reads : 7101659                                                   ||\n",
      "||    Successfully assigned reads : 4086182 (57.5%)                           ||\n",
      "||    Running time : 0.32 minutes                                             ||\n",
      "||                                                                            ||\n",
      "||                         Read assignment finished.                          ||\n",
      "||                                                                            ||\n",
      "|| Summary of counting results can be found in file \"./featureCount_output/E  ||\n",
      "|| RR899421Aligned.count.txt.summary\"                                         ||\n",
      "||                                                                            ||\n",
      "\\\\===================== http://subread.sourceforge.net/ ======================//\n",
      "\n",
      "\n",
      "        ==========     _____ _    _ ____  _____  ______          _____  \n",
      "        =====         / ____| |  | |  _ \\|  __ \\|  ____|   /\\   |  __ \\ \n",
      "          =====      | (___ | |  | | |_) | |__) | |__     /  \\  | |  | |\n",
      "            ====      \\___ \\| |  | |  _ <|  _  /|  __|   / /\\ \\ | |  | |\n",
      "              ====    ____) | |__| | |_) | | \\ \\| |____ / ____ \\| |__| |\n",
      "        ==========   |_____/ \\____/|____/|_|  \\_\\______/_/    \\_\\_____/\n",
      "\t  v1.6.0\n",
      "\n",
      "//========================== featureCounts setting ===========================\\\\\n",
      "||                                                                            ||\n",
      "||             Input files : 1 BAM file                                       ||\n",
      "||                           S ./star_output/ERR899422Aligned.out.bam         ||\n",
      "||                                                                            ||\n",
      "||             Output file : ./featureCount_output/ERR899422Aligned.count ... ||\n",
      "||                 Summary : ./featureCount_output/ERR899422Aligned.count ... ||\n",
      "||              Annotation : /fs/project/PAS1307/genomes/Mus_musculus/Ens ... ||\n",
      "||      Dir for temp files : ./featureCount_output                            ||\n",
      "||                                                                            ||\n",
      "||                 Threads : 24                                               ||\n",
      "||                   Level : meta-feature level                               ||\n",
      "||              Paired-end : no                                               ||\n",
      "||         Strand specific : no                                               ||\n",
      "||      Multimapping reads : not counted                                      ||\n",
      "|| Multi-overlapping reads : not counted                                      ||\n",
      "||   Min overlapping bases : 1                                                ||\n",
      "||                                                                            ||\n",
      "\\\\===================== http://subread.sourceforge.net/ ======================//\n",
      "\n",
      "//================================= Running ==================================\\\\\n",
      "||                                                                            ||\n",
      "|| Load annotation file /fs/project/PAS1307/genomes/Mus_musculus/Ensembl/ ... ||\n",
      "||    Features : 690647                                                       ||\n",
      "||    Meta-features : 46078                                                   ||\n",
      "||    Chromosomes/contigs : 73                                                ||\n",
      "||                                                                            ||\n",
      "|| Process BAM file ./star_output/ERR899422Aligned.out.bam...                 ||\n",
      "||    Single-end reads are included.                                          ||\n",
      "||    Assign reads to features...                                             ||\n",
      "||    Total reads : 8079369                                                   ||\n",
      "||    Successfully assigned reads : 4246258 (52.6%)                           ||\n",
      "||    Running time : 0.35 minutes                                             ||\n",
      "||                                                                            ||\n",
      "||                         Read assignment finished.                          ||\n",
      "||                                                                            ||\n",
      "|| Summary of counting results can be found in file \"./featureCount_output/E  ||\n",
      "|| RR899422Aligned.count.txt.summary\"                                         ||\n",
      "||                                                                            ||\n",
      "\\\\===================== http://subread.sourceforge.net/ ======================//\n",
      "\n",
      "\n",
      "        ==========     _____ _    _ ____  _____  ______          _____  \n",
      "        =====         / ____| |  | |  _ \\|  __ \\|  ____|   /\\   |  __ \\ \n",
      "          =====      | (___ | |  | | |_) | |__) | |__     /  \\  | |  | |\n",
      "            ====      \\___ \\| |  | |  _ <|  _  /|  __|   / /\\ \\ | |  | |\n",
      "              ====    ____) | |__| | |_) | | \\ \\| |____ / ____ \\| |__| |\n",
      "        ==========   |_____/ \\____/|____/|_|  \\_\\______/_/    \\_\\_____/\n",
      "\t  v1.6.0\n",
      "\n",
      "//========================== featureCounts setting ===========================\\\\\n",
      "||                                                                            ||\n",
      "||             Input files : 1 BAM file                                       ||\n",
      "||                           S ./star_output/ERR899423Aligned.out.bam         ||\n",
      "||                                                                            ||\n",
      "||             Output file : ./featureCount_output/ERR899423Aligned.count ... ||\n",
      "||                 Summary : ./featureCount_output/ERR899423Aligned.count ... ||\n",
      "||              Annotation : /fs/project/PAS1307/genomes/Mus_musculus/Ens ... ||\n",
      "||      Dir for temp files : ./featureCount_output                            ||\n",
      "||                                                                            ||\n",
      "||                 Threads : 24                                               ||\n",
      "||                   Level : meta-feature level                               ||\n",
      "||              Paired-end : no                                               ||\n",
      "||         Strand specific : no                                               ||\n",
      "||      Multimapping reads : not counted                                      ||\n",
      "|| Multi-overlapping reads : not counted                                      ||\n",
      "||   Min overlapping bases : 1                                                ||\n",
      "||                                                                            ||\n",
      "\\\\===================== http://subread.sourceforge.net/ ======================//\n",
      "\n",
      "//================================= Running ==================================\\\\\n",
      "||                                                                            ||\n",
      "|| Load annotation file /fs/project/PAS1307/genomes/Mus_musculus/Ensembl/ ... ||\n",
      "||    Features : 690647                                                       ||\n",
      "||    Meta-features : 46078                                                   ||\n",
      "||    Chromosomes/contigs : 73                                                ||\n",
      "||                                                                            ||\n",
      "|| Process BAM file ./star_output/ERR899423Aligned.out.bam...                 ||\n",
      "||    Single-end reads are included.                                          ||\n",
      "||    Assign reads to features...                                             ||\n",
      "||    Total reads : 7567977                                                   ||\n",
      "||    Successfully assigned reads : 4140773 (54.7%)                           ||\n",
      "||    Running time : 0.33 minutes                                             ||\n",
      "||                                                                            ||\n",
      "||                         Read assignment finished.                          ||\n",
      "||                                                                            ||\n",
      "|| Summary of counting results can be found in file \"./featureCount_output/E  ||\n",
      "|| RR899423Aligned.count.txt.summary\"                                         ||\n",
      "||                                                                            ||\n",
      "\\\\===================== http://subread.sourceforge.net/ ======================//\n",
      "\n",
      "\n",
      "        ==========     _____ _    _ ____  _____  ______          _____  \n",
      "        =====         / ____| |  | |  _ \\|  __ \\|  ____|   /\\   |  __ \\ \n",
      "          =====      | (___ | |  | | |_) | |__) | |__     /  \\  | |  | |\n",
      "            ====      \\___ \\| |  | |  _ <|  _  /|  __|   / /\\ \\ | |  | |\n",
      "              ====    ____) | |__| | |_) | | \\ \\| |____ / ____ \\| |__| |\n",
      "        ==========   |_____/ \\____/|____/|_|  \\_\\______/_/    \\_\\_____/\n",
      "\t  v1.6.0\n",
      "\n",
      "//========================== featureCounts setting ===========================\\\\\n",
      "||                                                                            ||\n",
      "||             Input files : 1 BAM file                                       ||\n",
      "||                           S ./star_output/ERR899424Aligned.out.bam         ||\n",
      "||                                                                            ||\n",
      "||             Output file : ./featureCount_output/ERR899424Aligned.count ... ||\n",
      "||                 Summary : ./featureCount_output/ERR899424Aligned.count ... ||\n",
      "||              Annotation : /fs/project/PAS1307/genomes/Mus_musculus/Ens ... ||\n",
      "||      Dir for temp files : ./featureCount_output                            ||\n",
      "||                                                                            ||\n",
      "||                 Threads : 24                                               ||\n",
      "||                   Level : meta-feature level                               ||\n",
      "||              Paired-end : no                                               ||\n",
      "||         Strand specific : no                                               ||\n",
      "||      Multimapping reads : not counted                                      ||\n",
      "|| Multi-overlapping reads : not counted                                      ||\n",
      "||   Min overlapping bases : 1                                                ||\n",
      "||                                                                            ||\n",
      "\\\\===================== http://subread.sourceforge.net/ ======================//\n",
      "\n",
      "//================================= Running ==================================\\\\\n",
      "||                                                                            ||\n",
      "|| Load annotation file /fs/project/PAS1307/genomes/Mus_musculus/Ensembl/ ... ||\n",
      "||    Features : 690647                                                       ||\n",
      "||    Meta-features : 46078                                                   ||\n",
      "||    Chromosomes/contigs : 73                                                ||\n",
      "||                                                                            ||\n",
      "|| Process BAM file ./star_output/ERR899424Aligned.out.bam...                 ||\n",
      "||    Single-end reads are included.                                          ||\n",
      "||    Assign reads to features...                                             ||\n",
      "||    Total reads : 8060973                                                   ||\n",
      "||    Successfully assigned reads : 4075483 (50.6%)                           ||\n",
      "||    Running time : 0.37 minutes                                             ||\n",
      "||                                                                            ||\n",
      "||                         Read assignment finished.                          ||\n",
      "||                                                                            ||\n",
      "|| Summary of counting results can be found in file \"./featureCount_output/E  ||\n",
      "|| RR899424Aligned.count.txt.summary\"                                         ||\n",
      "||                                                                            ||\n",
      "\\\\===================== http://subread.sourceforge.net/ ======================//\n",
      "\n",
      "\n",
      "        ==========     _____ _    _ ____  _____  ______          _____  \n",
      "        =====         / ____| |  | |  _ \\|  __ \\|  ____|   /\\   |  __ \\ \n",
      "          =====      | (___ | |  | | |_) | |__) | |__     /  \\  | |  | |\n",
      "            ====      \\___ \\| |  | |  _ <|  _  /|  __|   / /\\ \\ | |  | |\n",
      "              ====    ____) | |__| | |_) | | \\ \\| |____ / ____ \\| |__| |\n",
      "        ==========   |_____/ \\____/|____/|_|  \\_\\______/_/    \\_\\_____/\n",
      "\t  v1.6.0\n",
      "\n",
      "//========================== featureCounts setting ===========================\\\\\n",
      "||                                                                            ||\n",
      "||             Input files : 1 BAM file                                       ||\n",
      "||                           S ./star_output/ERR899425Aligned.out.bam         ||\n",
      "||                                                                            ||\n",
      "||             Output file : ./featureCount_output/ERR899425Aligned.count ... ||\n",
      "||                 Summary : ./featureCount_output/ERR899425Aligned.count ... ||\n",
      "||              Annotation : /fs/project/PAS1307/genomes/Mus_musculus/Ens ... ||\n",
      "||      Dir for temp files : ./featureCount_output                            ||\n",
      "||                                                                            ||\n",
      "||                 Threads : 24                                               ||\n",
      "||                   Level : meta-feature level                               ||\n",
      "||              Paired-end : no                                               ||\n",
      "||         Strand specific : no                                               ||\n",
      "||      Multimapping reads : not counted                                      ||\n",
      "|| Multi-overlapping reads : not counted                                      ||\n",
      "||   Min overlapping bases : 1                                                ||\n",
      "||                                                                            ||\n",
      "\\\\===================== http://subread.sourceforge.net/ ======================//\n",
      "\n",
      "//================================= Running ==================================\\\\\n",
      "||                                                                            ||\n",
      "|| Load annotation file /fs/project/PAS1307/genomes/Mus_musculus/Ensembl/ ... ||\n",
      "||    Features : 690647                                                       ||\n",
      "||    Meta-features : 46078                                                   ||\n",
      "||    Chromosomes/contigs : 73                                                ||\n",
      "||                                                                            ||\n",
      "|| Process BAM file ./star_output/ERR899425Aligned.out.bam...                 ||\n",
      "||    Single-end reads are included.                                          ||\n",
      "||    Assign reads to features...                                             ||\n",
      "||    Total reads : 8654636                                                   ||\n",
      "||    Successfully assigned reads : 4655774 (53.8%)                           ||\n",
      "||    Running time : 0.38 minutes                                             ||\n",
      "||                                                                            ||\n",
      "||                         Read assignment finished.                          ||\n",
      "||                                                                            ||\n",
      "|| Summary of counting results can be found in file \"./featureCount_output/E  ||\n",
      "|| RR899425Aligned.count.txt.summary\"                                         ||\n",
      "||                                                                            ||\n",
      "\\\\===================== http://subread.sourceforge.net/ ======================//\n",
      "\n",
      "\n",
      "        ==========     _____ _    _ ____  _____  ______          _____  \n",
      "        =====         / ____| |  | |  _ \\|  __ \\|  ____|   /\\   |  __ \\ \n",
      "          =====      | (___ | |  | | |_) | |__) | |__     /  \\  | |  | |\n",
      "            ====      \\___ \\| |  | |  _ <|  _  /|  __|   / /\\ \\ | |  | |\n",
      "              ====    ____) | |__| | |_) | | \\ \\| |____ / ____ \\| |__| |\n",
      "        ==========   |_____/ \\____/|____/|_|  \\_\\______/_/    \\_\\_____/\n",
      "\t  v1.6.0\n",
      "\n",
      "//========================== featureCounts setting ===========================\\\\\n",
      "||                                                                            ||\n",
      "||             Input files : 1 BAM file                                       ||\n",
      "||                           S ./star_output/ERR899426Aligned.out.bam         ||\n",
      "||                                                                            ||\n",
      "||             Output file : ./featureCount_output/ERR899426Aligned.count ... ||\n",
      "||                 Summary : ./featureCount_output/ERR899426Aligned.count ... ||\n",
      "||              Annotation : /fs/project/PAS1307/genomes/Mus_musculus/Ens ... ||\n",
      "||      Dir for temp files : ./featureCount_output                            ||\n",
      "||                                                                            ||\n",
      "||                 Threads : 24                                               ||\n",
      "||                   Level : meta-feature level                               ||\n",
      "||              Paired-end : no                                               ||\n",
      "||         Strand specific : no                                               ||\n",
      "||      Multimapping reads : not counted                                      ||\n",
      "|| Multi-overlapping reads : not counted                                      ||\n",
      "||   Min overlapping bases : 1                                                ||\n",
      "||                                                                            ||\n",
      "\\\\===================== http://subread.sourceforge.net/ ======================//\n",
      "\n",
      "//================================= Running ==================================\\\\\n",
      "||                                                                            ||\n",
      "|| Load annotation file /fs/project/PAS1307/genomes/Mus_musculus/Ensembl/ ... ||\n",
      "||    Features : 690647                                                       ||\n",
      "||    Meta-features : 46078                                                   ||\n",
      "||    Chromosomes/contigs : 73                                                ||\n",
      "||                                                                            ||\n",
      "|| Process BAM file ./star_output/ERR899426Aligned.out.bam...                 ||\n",
      "||    Single-end reads are included.                                          ||\n",
      "||    Assign reads to features...                                             ||\n",
      "||    Total reads : 10659421                                                  ||\n",
      "||    Successfully assigned reads : 4123752 (38.7%)                           ||\n",
      "||    Running time : 0.44 minutes                                             ||\n",
      "||                                                                            ||\n",
      "||                         Read assignment finished.                          ||\n",
      "||                                                                            ||\n",
      "|| Summary of counting results can be found in file \"./featureCount_output/E  ||\n",
      "|| RR899426Aligned.count.txt.summary\"                                         ||\n",
      "||                                                                            ||\n",
      "\\\\===================== http://subread.sourceforge.net/ ======================//\n",
      "\n",
      "\n",
      "        ==========     _____ _    _ ____  _____  ______          _____  \n",
      "        =====         / ____| |  | |  _ \\|  __ \\|  ____|   /\\   |  __ \\ \n",
      "          =====      | (___ | |  | | |_) | |__) | |__     /  \\  | |  | |\n",
      "            ====      \\___ \\| |  | |  _ <|  _  /|  __|   / /\\ \\ | |  | |\n",
      "              ====    ____) | |__| | |_) | | \\ \\| |____ / ____ \\| |__| |\n",
      "        ==========   |_____/ \\____/|____/|_|  \\_\\______/_/    \\_\\_____/\n",
      "\t  v1.6.0\n",
      "\n",
      "//========================== featureCounts setting ===========================\\\\\n",
      "||                                                                            ||\n",
      "||             Input files : 1 BAM file                                       ||\n",
      "||                           S ./star_output/ERR899427Aligned.out.bam         ||\n",
      "||                                                                            ||\n",
      "||             Output file : ./featureCount_output/ERR899427Aligned.count ... ||\n",
      "||                 Summary : ./featureCount_output/ERR899427Aligned.count ... ||\n",
      "||              Annotation : /fs/project/PAS1307/genomes/Mus_musculus/Ens ... ||\n",
      "||      Dir for temp files : ./featureCount_output                            ||\n",
      "||                                                                            ||\n",
      "||                 Threads : 24                                               ||\n",
      "||                   Level : meta-feature level                               ||\n",
      "||              Paired-end : no                                               ||\n",
      "||         Strand specific : no                                               ||\n",
      "||      Multimapping reads : not counted                                      ||\n",
      "|| Multi-overlapping reads : not counted                                      ||\n",
      "||   Min overlapping bases : 1                                                ||\n",
      "||                                                                            ||\n",
      "\\\\===================== http://subread.sourceforge.net/ ======================//\n",
      "\n",
      "//================================= Running ==================================\\\\\n",
      "||                                                                            ||\n",
      "|| Load annotation file /fs/project/PAS1307/genomes/Mus_musculus/Ensembl/ ... ||\n",
      "||    Features : 690647                                                       ||\n",
      "||    Meta-features : 46078                                                   ||\n",
      "||    Chromosomes/contigs : 73                                                ||\n",
      "||                                                                            ||\n",
      "|| Process BAM file ./star_output/ERR899427Aligned.out.bam...                 ||\n",
      "||    Single-end reads are included.                                          ||\n",
      "||    Assign reads to features...                                             ||\n",
      "||    Total reads : 7615582                                                   ||\n",
      "||    Successfully assigned reads : 3890283 (51.1%)                           ||\n",
      "||    Running time : 0.35 minutes                                             ||\n",
      "||                                                                            ||\n",
      "||                         Read assignment finished.                          ||\n",
      "||                                                                            ||\n",
      "|| Summary of counting results can be found in file \"./featureCount_output/E  ||\n",
      "|| RR899427Aligned.count.txt.summary\"                                         ||\n",
      "||                                                                            ||\n",
      "\\\\===================== http://subread.sourceforge.net/ ======================//\n",
      "\n",
      "\n",
      "        ==========     _____ _    _ ____  _____  ______          _____  \n",
      "        =====         / ____| |  | |  _ \\|  __ \\|  ____|   /\\   |  __ \\ \n",
      "          =====      | (___ | |  | | |_) | |__) | |__     /  \\  | |  | |\n",
      "            ====      \\___ \\| |  | |  _ <|  _  /|  __|   / /\\ \\ | |  | |\n",
      "              ====    ____) | |__| | |_) | | \\ \\| |____ / ____ \\| |__| |\n",
      "        ==========   |_____/ \\____/|____/|_|  \\_\\______/_/    \\_\\_____/\n",
      "\t  v1.6.0\n",
      "\n",
      "//========================== featureCounts setting ===========================\\\\\n",
      "||                                                                            ||\n",
      "||             Input files : 1 BAM file                                       ||\n",
      "||                           S ./star_output/ERR899428Aligned.out.bam         ||\n",
      "||                                                                            ||\n",
      "||             Output file : ./featureCount_output/ERR899428Aligned.count ... ||\n",
      "||                 Summary : ./featureCount_output/ERR899428Aligned.count ... ||\n",
      "||              Annotation : /fs/project/PAS1307/genomes/Mus_musculus/Ens ... ||\n",
      "||      Dir for temp files : ./featureCount_output                            ||\n",
      "||                                                                            ||\n",
      "||                 Threads : 24                                               ||\n",
      "||                   Level : meta-feature level                               ||\n",
      "||              Paired-end : no                                               ||\n",
      "||         Strand specific : no                                               ||\n",
      "||      Multimapping reads : not counted                                      ||\n",
      "|| Multi-overlapping reads : not counted                                      ||\n",
      "||   Min overlapping bases : 1                                                ||\n",
      "||                                                                            ||\n",
      "\\\\===================== http://subread.sourceforge.net/ ======================//\n",
      "\n",
      "//================================= Running ==================================\\\\\n",
      "||                                                                            ||\n",
      "|| Load annotation file /fs/project/PAS1307/genomes/Mus_musculus/Ensembl/ ... ||\n",
      "||    Features : 690647                                                       ||\n",
      "||    Meta-features : 46078                                                   ||\n",
      "||    Chromosomes/contigs : 73                                                ||\n",
      "||                                                                            ||\n",
      "|| Process BAM file ./star_output/ERR899428Aligned.out.bam...                 ||\n",
      "||    Single-end reads are included.                                          ||\n",
      "||    Assign reads to features...                                             ||\n",
      "||    Total reads : 7216520                                                   ||\n",
      "||    Successfully assigned reads : 3742170 (51.9%)                           ||\n",
      "||    Running time : 0.34 minutes                                             ||\n",
      "||                                                                            ||\n",
      "||                         Read assignment finished.                          ||\n",
      "||                                                                            ||\n",
      "|| Summary of counting results can be found in file \"./featureCount_output/E  ||\n",
      "|| RR899428Aligned.count.txt.summary\"                                         ||\n",
      "||                                                                            ||\n",
      "\\\\===================== http://subread.sourceforge.net/ ======================//\n",
      "\n",
      "\n",
      "        ==========     _____ _    _ ____  _____  ______          _____  \n",
      "        =====         / ____| |  | |  _ \\|  __ \\|  ____|   /\\   |  __ \\ \n",
      "          =====      | (___ | |  | | |_) | |__) | |__     /  \\  | |  | |\n",
      "            ====      \\___ \\| |  | |  _ <|  _  /|  __|   / /\\ \\ | |  | |\n",
      "              ====    ____) | |__| | |_) | | \\ \\| |____ / ____ \\| |__| |\n",
      "        ==========   |_____/ \\____/|____/|_|  \\_\\______/_/    \\_\\_____/\n",
      "\t  v1.6.0\n",
      "\n",
      "//========================== featureCounts setting ===========================\\\\\n",
      "||                                                                            ||\n",
      "||             Input files : 1 BAM file                                       ||\n",
      "||                           S ./star_output/ERR899429Aligned.out.bam         ||\n",
      "||                                                                            ||\n",
      "||             Output file : ./featureCount_output/ERR899429Aligned.count ... ||\n",
      "||                 Summary : ./featureCount_output/ERR899429Aligned.count ... ||\n",
      "||              Annotation : /fs/project/PAS1307/genomes/Mus_musculus/Ens ... ||\n",
      "||      Dir for temp files : ./featureCount_output                            ||\n",
      "||                                                                            ||\n",
      "||                 Threads : 24                                               ||\n",
      "||                   Level : meta-feature level                               ||\n",
      "||              Paired-end : no                                               ||\n",
      "||         Strand specific : no                                               ||\n",
      "||      Multimapping reads : not counted                                      ||\n",
      "|| Multi-overlapping reads : not counted                                      ||\n",
      "||   Min overlapping bases : 1                                                ||\n",
      "||                                                                            ||\n",
      "\\\\===================== http://subread.sourceforge.net/ ======================//\n",
      "\n",
      "//================================= Running ==================================\\\\\n",
      "||                                                                            ||\n",
      "|| Load annotation file /fs/project/PAS1307/genomes/Mus_musculus/Ensembl/ ... ||\n",
      "||    Features : 690647                                                       ||\n",
      "||    Meta-features : 46078                                                   ||\n",
      "||    Chromosomes/contigs : 73                                                ||\n",
      "||                                                                            ||\n",
      "|| Process BAM file ./star_output/ERR899429Aligned.out.bam...                 ||\n",
      "||    Single-end reads are included.                                          ||\n",
      "||    Assign reads to features...                                             ||\n",
      "||    Total reads : 7814484                                                   ||\n",
      "||    Successfully assigned reads : 4384281 (56.1%)                           ||\n",
      "||    Running time : 0.34 minutes                                             ||\n",
      "||                                                                            ||\n",
      "||                         Read assignment finished.                          ||\n",
      "||                                                                            ||\n",
      "|| Summary of counting results can be found in file \"./featureCount_output/E  ||\n",
      "|| RR899429Aligned.count.txt.summary\"                                         ||\n",
      "||                                                                            ||\n",
      "\\\\===================== http://subread.sourceforge.net/ ======================//\n",
      "\n",
      "\n",
      "        ==========     _____ _    _ ____  _____  ______          _____  \n",
      "        =====         / ____| |  | |  _ \\|  __ \\|  ____|   /\\   |  __ \\ \n",
      "          =====      | (___ | |  | | |_) | |__) | |__     /  \\  | |  | |\n",
      "            ====      \\___ \\| |  | |  _ <|  _  /|  __|   / /\\ \\ | |  | |\n",
      "              ====    ____) | |__| | |_) | | \\ \\| |____ / ____ \\| |__| |\n",
      "        ==========   |_____/ \\____/|____/|_|  \\_\\______/_/    \\_\\_____/\n",
      "\t  v1.6.0\n",
      "\n",
      "//========================== featureCounts setting ===========================\\\\\n",
      "||                                                                            ||\n",
      "||             Input files : 1 BAM file                                       ||\n",
      "||                           S ./star_output/ERR899430Aligned.out.bam         ||\n",
      "||                                                                            ||\n",
      "||             Output file : ./featureCount_output/ERR899430Aligned.count ... ||\n",
      "||                 Summary : ./featureCount_output/ERR899430Aligned.count ... ||\n",
      "||              Annotation : /fs/project/PAS1307/genomes/Mus_musculus/Ens ... ||\n",
      "||      Dir for temp files : ./featureCount_output                            ||\n",
      "||                                                                            ||\n",
      "||                 Threads : 24                                               ||\n",
      "||                   Level : meta-feature level                               ||\n",
      "||              Paired-end : no                                               ||\n",
      "||         Strand specific : no                                               ||\n",
      "||      Multimapping reads : not counted                                      ||\n",
      "|| Multi-overlapping reads : not counted                                      ||\n",
      "||   Min overlapping bases : 1                                                ||\n",
      "||                                                                            ||\n",
      "\\\\===================== http://subread.sourceforge.net/ ======================//\n",
      "\n",
      "//================================= Running ==================================\\\\\n",
      "||                                                                            ||\n",
      "|| Load annotation file /fs/project/PAS1307/genomes/Mus_musculus/Ensembl/ ... ||\n",
      "||    Features : 690647                                                       ||\n",
      "||    Meta-features : 46078                                                   ||\n",
      "||    Chromosomes/contigs : 73                                                ||\n",
      "||                                                                            ||\n",
      "|| Process BAM file ./star_output/ERR899430Aligned.out.bam...                 ||\n",
      "||    Single-end reads are included.                                          ||\n",
      "||    Assign reads to features...                                             ||\n",
      "||    Total reads : 7894117                                                   ||\n",
      "||    Successfully assigned reads : 4250499 (53.8%)                           ||\n",
      "||    Running time : 0.34 minutes                                             ||\n",
      "||                                                                            ||\n",
      "||                         Read assignment finished.                          ||\n",
      "||                                                                            ||\n",
      "|| Summary of counting results can be found in file \"./featureCount_output/E  ||\n",
      "|| RR899430Aligned.count.txt.summary\"                                         ||\n",
      "||                                                                            ||\n",
      "\\\\===================== http://subread.sourceforge.net/ ======================//\n",
      "\n",
      "\n",
      "        ==========     _____ _    _ ____  _____  ______          _____  \n",
      "        =====         / ____| |  | |  _ \\|  __ \\|  ____|   /\\   |  __ \\ \n",
      "          =====      | (___ | |  | | |_) | |__) | |__     /  \\  | |  | |\n",
      "            ====      \\___ \\| |  | |  _ <|  _  /|  __|   / /\\ \\ | |  | |\n",
      "              ====    ____) | |__| | |_) | | \\ \\| |____ / ____ \\| |__| |\n",
      "        ==========   |_____/ \\____/|____/|_|  \\_\\______/_/    \\_\\_____/\n",
      "\t  v1.6.0\n",
      "\n",
      "//========================== featureCounts setting ===========================\\\\\n",
      "||                                                                            ||\n",
      "||             Input files : 1 BAM file                                       ||\n",
      "||                           S ./star_output/ERR899431Aligned.out.bam         ||\n",
      "||                                                                            ||\n",
      "||             Output file : ./featureCount_output/ERR899431Aligned.count ... ||\n",
      "||                 Summary : ./featureCount_output/ERR899431Aligned.count ... ||\n",
      "||              Annotation : /fs/project/PAS1307/genomes/Mus_musculus/Ens ... ||\n",
      "||      Dir for temp files : ./featureCount_output                            ||\n",
      "||                                                                            ||\n",
      "||                 Threads : 24                                               ||\n",
      "||                   Level : meta-feature level                               ||\n",
      "||              Paired-end : no                                               ||\n",
      "||         Strand specific : no                                               ||\n",
      "||      Multimapping reads : not counted                                      ||\n",
      "|| Multi-overlapping reads : not counted                                      ||\n",
      "||   Min overlapping bases : 1                                                ||\n",
      "||                                                                            ||\n",
      "\\\\===================== http://subread.sourceforge.net/ ======================//\n",
      "\n",
      "//================================= Running ==================================\\\\\n",
      "||                                                                            ||\n",
      "|| Load annotation file /fs/project/PAS1307/genomes/Mus_musculus/Ensembl/ ... ||\n",
      "||    Features : 690647                                                       ||\n",
      "||    Meta-features : 46078                                                   ||\n",
      "||    Chromosomes/contigs : 73                                                ||\n",
      "||                                                                            ||\n",
      "|| Process BAM file ./star_output/ERR899431Aligned.out.bam...                 ||\n",
      "||    Single-end reads are included.                                          ||\n",
      "||    Assign reads to features...                                             ||\n",
      "||    Total reads : 7844861                                                   ||\n",
      "||    Successfully assigned reads : 4532529 (57.8%)                           ||\n",
      "||    Running time : 0.34 minutes                                             ||\n",
      "||                                                                            ||\n",
      "||                         Read assignment finished.                          ||\n",
      "||                                                                            ||\n",
      "|| Summary of counting results can be found in file \"./featureCount_output/E  ||\n",
      "|| RR899431Aligned.count.txt.summary\"                                         ||\n",
      "||                                                                            ||\n",
      "\\\\===================== http://subread.sourceforge.net/ ======================//\n",
      "\n",
      "\n",
      "        ==========     _____ _    _ ____  _____  ______          _____  \n",
      "        =====         / ____| |  | |  _ \\|  __ \\|  ____|   /\\   |  __ \\ \n",
      "          =====      | (___ | |  | | |_) | |__) | |__     /  \\  | |  | |\n",
      "            ====      \\___ \\| |  | |  _ <|  _  /|  __|   / /\\ \\ | |  | |\n",
      "              ====    ____) | |__| | |_) | | \\ \\| |____ / ____ \\| |__| |\n",
      "        ==========   |_____/ \\____/|____/|_|  \\_\\______/_/    \\_\\_____/\n",
      "\t  v1.6.0\n",
      "\n",
      "//========================== featureCounts setting ===========================\\\\\n",
      "||                                                                            ||\n",
      "||             Input files : 1 BAM file                                       ||\n",
      "||                           S ./star_output/ERR899432Aligned.out.bam         ||\n",
      "||                                                                            ||\n",
      "||             Output file : ./featureCount_output/ERR899432Aligned.count ... ||\n",
      "||                 Summary : ./featureCount_output/ERR899432Aligned.count ... ||\n",
      "||              Annotation : /fs/project/PAS1307/genomes/Mus_musculus/Ens ... ||\n",
      "||      Dir for temp files : ./featureCount_output                            ||\n",
      "||                                                                            ||\n",
      "||                 Threads : 24                                               ||\n",
      "||                   Level : meta-feature level                               ||\n",
      "||              Paired-end : no                                               ||\n",
      "||         Strand specific : no                                               ||\n",
      "||      Multimapping reads : not counted                                      ||\n",
      "|| Multi-overlapping reads : not counted                                      ||\n",
      "||   Min overlapping bases : 1                                                ||\n",
      "||                                                                            ||\n",
      "\\\\===================== http://subread.sourceforge.net/ ======================//\n",
      "\n",
      "//================================= Running ==================================\\\\\n",
      "||                                                                            ||\n",
      "|| Load annotation file /fs/project/PAS1307/genomes/Mus_musculus/Ensembl/ ... ||\n",
      "||    Features : 690647                                                       ||\n",
      "||    Meta-features : 46078                                                   ||\n",
      "||    Chromosomes/contigs : 73                                                ||\n",
      "||                                                                            ||\n",
      "|| Process BAM file ./star_output/ERR899432Aligned.out.bam...                 ||\n",
      "||    Single-end reads are included.                                          ||\n",
      "||    Assign reads to features...                                             ||\n",
      "||    Total reads : 7213605                                                   ||\n",
      "||    Successfully assigned reads : 4243939 (58.8%)                           ||\n",
      "||    Running time : 0.31 minutes                                             ||\n",
      "||                                                                            ||\n",
      "||                         Read assignment finished.                          ||\n",
      "||                                                                            ||\n",
      "|| Summary of counting results can be found in file \"./featureCount_output/E  ||\n",
      "|| RR899432Aligned.count.txt.summary\"                                         ||\n",
      "||                                                                            ||\n",
      "\\\\===================== http://subread.sourceforge.net/ ======================//\n",
      "\n",
      "\n",
      "        ==========     _____ _    _ ____  _____  ______          _____  \n",
      "        =====         / ____| |  | |  _ \\|  __ \\|  ____|   /\\   |  __ \\ \n",
      "          =====      | (___ | |  | | |_) | |__) | |__     /  \\  | |  | |\n",
      "            ====      \\___ \\| |  | |  _ <|  _  /|  __|   / /\\ \\ | |  | |\n",
      "              ====    ____) | |__| | |_) | | \\ \\| |____ / ____ \\| |__| |\n",
      "        ==========   |_____/ \\____/|____/|_|  \\_\\______/_/    \\_\\_____/\n",
      "\t  v1.6.0\n",
      "\n",
      "//========================== featureCounts setting ===========================\\\\\n",
      "||                                                                            ||\n",
      "||             Input files : 1 BAM file                                       ||\n",
      "||                           S ./star_output/ERR899433Aligned.out.bam         ||\n",
      "||                                                                            ||\n",
      "||             Output file : ./featureCount_output/ERR899433Aligned.count ... ||\n",
      "||                 Summary : ./featureCount_output/ERR899433Aligned.count ... ||\n",
      "||              Annotation : /fs/project/PAS1307/genomes/Mus_musculus/Ens ... ||\n",
      "||      Dir for temp files : ./featureCount_output                            ||\n",
      "||                                                                            ||\n",
      "||                 Threads : 24                                               ||\n",
      "||                   Level : meta-feature level                               ||\n",
      "||              Paired-end : no                                               ||\n",
      "||         Strand specific : no                                               ||\n",
      "||      Multimapping reads : not counted                                      ||\n",
      "|| Multi-overlapping reads : not counted                                      ||\n",
      "||   Min overlapping bases : 1                                                ||\n",
      "||                                                                            ||\n",
      "\\\\===================== http://subread.sourceforge.net/ ======================//\n",
      "\n",
      "//================================= Running ==================================\\\\\n",
      "||                                                                            ||\n",
      "|| Load annotation file /fs/project/PAS1307/genomes/Mus_musculus/Ensembl/ ... ||\n",
      "||    Features : 690647                                                       ||\n",
      "||    Meta-features : 46078                                                   ||\n",
      "||    Chromosomes/contigs : 73                                                ||\n",
      "||                                                                            ||\n",
      "|| Process BAM file ./star_output/ERR899433Aligned.out.bam...                 ||\n",
      "||    Single-end reads are included.                                          ||\n",
      "||    Assign reads to features...                                             ||\n",
      "||    Total reads : 8098199                                                   ||\n",
      "||    Successfully assigned reads : 4533982 (56.0%)                           ||\n",
      "||    Running time : 0.35 minutes                                             ||\n",
      "||                                                                            ||\n",
      "||                         Read assignment finished.                          ||\n",
      "||                                                                            ||\n",
      "|| Summary of counting results can be found in file \"./featureCount_output/E  ||\n",
      "|| RR899433Aligned.count.txt.summary\"                                         ||\n",
      "||                                                                            ||\n",
      "\\\\===================== http://subread.sourceforge.net/ ======================//\n",
      "\n",
      "\n",
      "        ==========     _____ _    _ ____  _____  ______          _____  \n",
      "        =====         / ____| |  | |  _ \\|  __ \\|  ____|   /\\   |  __ \\ \n",
      "          =====      | (___ | |  | | |_) | |__) | |__     /  \\  | |  | |\n",
      "            ====      \\___ \\| |  | |  _ <|  _  /|  __|   / /\\ \\ | |  | |\n",
      "              ====    ____) | |__| | |_) | | \\ \\| |____ / ____ \\| |__| |\n",
      "        ==========   |_____/ \\____/|____/|_|  \\_\\______/_/    \\_\\_____/\n",
      "\t  v1.6.0\n",
      "\n",
      "//========================== featureCounts setting ===========================\\\\\n",
      "||                                                                            ||\n",
      "||             Input files : 1 BAM file                                       ||\n",
      "||                           S ./star_output/ERR899434Aligned.out.bam         ||\n",
      "||                                                                            ||\n",
      "||             Output file : ./featureCount_output/ERR899434Aligned.count ... ||\n",
      "||                 Summary : ./featureCount_output/ERR899434Aligned.count ... ||\n",
      "||              Annotation : /fs/project/PAS1307/genomes/Mus_musculus/Ens ... ||\n",
      "||      Dir for temp files : ./featureCount_output                            ||\n",
      "||                                                                            ||\n",
      "||                 Threads : 24                                               ||\n",
      "||                   Level : meta-feature level                               ||\n",
      "||              Paired-end : no                                               ||\n",
      "||         Strand specific : no                                               ||\n",
      "||      Multimapping reads : not counted                                      ||\n",
      "|| Multi-overlapping reads : not counted                                      ||\n",
      "||   Min overlapping bases : 1                                                ||\n",
      "||                                                                            ||\n",
      "\\\\===================== http://subread.sourceforge.net/ ======================//\n",
      "\n",
      "//================================= Running ==================================\\\\\n",
      "||                                                                            ||\n",
      "|| Load annotation file /fs/project/PAS1307/genomes/Mus_musculus/Ensembl/ ... ||\n",
      "||    Features : 690647                                                       ||\n",
      "||    Meta-features : 46078                                                   ||\n",
      "||    Chromosomes/contigs : 73                                                ||\n",
      "||                                                                            ||\n",
      "|| Process BAM file ./star_output/ERR899434Aligned.out.bam...                 ||\n",
      "||    Single-end reads are included.                                          ||\n",
      "||    Assign reads to features...                                             ||\n",
      "||    Total reads : 7816005                                                   ||\n",
      "||    Successfully assigned reads : 4288921 (54.9%)                           ||\n",
      "||    Running time : 0.34 minutes                                             ||\n",
      "||                                                                            ||\n",
      "||                         Read assignment finished.                          ||\n",
      "||                                                                            ||\n",
      "|| Summary of counting results can be found in file \"./featureCount_output/E  ||\n",
      "|| RR899434Aligned.count.txt.summary\"                                         ||\n",
      "||                                                                            ||\n",
      "\\\\===================== http://subread.sourceforge.net/ ======================//\n",
      "\n",
      "\n",
      "        ==========     _____ _    _ ____  _____  ______          _____  \n",
      "        =====         / ____| |  | |  _ \\|  __ \\|  ____|   /\\   |  __ \\ \n",
      "          =====      | (___ | |  | | |_) | |__) | |__     /  \\  | |  | |\n",
      "            ====      \\___ \\| |  | |  _ <|  _  /|  __|   / /\\ \\ | |  | |\n",
      "              ====    ____) | |__| | |_) | | \\ \\| |____ / ____ \\| |__| |\n",
      "        ==========   |_____/ \\____/|____/|_|  \\_\\______/_/    \\_\\_____/\n",
      "\t  v1.6.0\n",
      "\n",
      "//========================== featureCounts setting ===========================\\\\\n",
      "||                                                                            ||\n",
      "||             Input files : 1 BAM file                                       ||\n",
      "||                           S ./star_output/ERR899435Aligned.out.bam         ||\n",
      "||                                                                            ||\n",
      "||             Output file : ./featureCount_output/ERR899435Aligned.count ... ||\n",
      "||                 Summary : ./featureCount_output/ERR899435Aligned.count ... ||\n",
      "||              Annotation : /fs/project/PAS1307/genomes/Mus_musculus/Ens ... ||\n",
      "||      Dir for temp files : ./featureCount_output                            ||\n",
      "||                                                                            ||\n",
      "||                 Threads : 24                                               ||\n",
      "||                   Level : meta-feature level                               ||\n",
      "||              Paired-end : no                                               ||\n",
      "||         Strand specific : no                                               ||\n",
      "||      Multimapping reads : not counted                                      ||\n",
      "|| Multi-overlapping reads : not counted                                      ||\n",
      "||   Min overlapping bases : 1                                                ||\n",
      "||                                                                            ||\n",
      "\\\\===================== http://subread.sourceforge.net/ ======================//\n",
      "\n",
      "//================================= Running ==================================\\\\\n",
      "||                                                                            ||\n",
      "|| Load annotation file /fs/project/PAS1307/genomes/Mus_musculus/Ensembl/ ... ||\n",
      "||    Features : 690647                                                       ||\n",
      "||    Meta-features : 46078                                                   ||\n",
      "||    Chromosomes/contigs : 73                                                ||\n",
      "||                                                                            ||\n",
      "|| Process BAM file ./star_output/ERR899435Aligned.out.bam...                 ||\n",
      "||    Single-end reads are included.                                          ||\n",
      "||    Assign reads to features...                                             ||\n",
      "||    Total reads : 7319465                                                   ||\n",
      "||    Successfully assigned reads : 4091465 (55.9%)                           ||\n",
      "||    Running time : 0.35 minutes                                             ||\n",
      "||                                                                            ||\n",
      "||                         Read assignment finished.                          ||\n",
      "||                                                                            ||\n",
      "|| Summary of counting results can be found in file \"./featureCount_output/E  ||\n",
      "|| RR899435Aligned.count.txt.summary\"                                         ||\n",
      "||                                                                            ||\n",
      "\\\\===================== http://subread.sourceforge.net/ ======================//\n",
      "\n",
      "\n",
      "        ==========     _____ _    _ ____  _____  ______          _____  \n",
      "        =====         / ____| |  | |  _ \\|  __ \\|  ____|   /\\   |  __ \\ \n",
      "          =====      | (___ | |  | | |_) | |__) | |__     /  \\  | |  | |\n",
      "            ====      \\___ \\| |  | |  _ <|  _  /|  __|   / /\\ \\ | |  | |\n",
      "              ====    ____) | |__| | |_) | | \\ \\| |____ / ____ \\| |__| |\n",
      "        ==========   |_____/ \\____/|____/|_|  \\_\\______/_/    \\_\\_____/\n",
      "\t  v1.6.0\n",
      "\n",
      "//========================== featureCounts setting ===========================\\\\\n",
      "||                                                                            ||\n",
      "||             Input files : 1 BAM file                                       ||\n",
      "||                           S ./star_output/ERR899436Aligned.out.bam         ||\n",
      "||                                                                            ||\n",
      "||             Output file : ./featureCount_output/ERR899436Aligned.count ... ||\n",
      "||                 Summary : ./featureCount_output/ERR899436Aligned.count ... ||\n",
      "||              Annotation : /fs/project/PAS1307/genomes/Mus_musculus/Ens ... ||\n",
      "||      Dir for temp files : ./featureCount_output                            ||\n",
      "||                                                                            ||\n",
      "||                 Threads : 24                                               ||\n",
      "||                   Level : meta-feature level                               ||\n",
      "||              Paired-end : no                                               ||\n",
      "||         Strand specific : no                                               ||\n",
      "||      Multimapping reads : not counted                                      ||\n",
      "|| Multi-overlapping reads : not counted                                      ||\n",
      "||   Min overlapping bases : 1                                                ||\n",
      "||                                                                            ||\n",
      "\\\\===================== http://subread.sourceforge.net/ ======================//\n",
      "\n",
      "//================================= Running ==================================\\\\\n",
      "||                                                                            ||\n",
      "|| Load annotation file /fs/project/PAS1307/genomes/Mus_musculus/Ensembl/ ... ||\n",
      "||    Features : 690647                                                       ||\n",
      "||    Meta-features : 46078                                                   ||\n",
      "||    Chromosomes/contigs : 73                                                ||\n",
      "||                                                                            ||\n",
      "|| Process BAM file ./star_output/ERR899436Aligned.out.bam...                 ||\n",
      "||    Single-end reads are included.                                          ||\n",
      "||    Assign reads to features...                                             ||\n",
      "||    Total reads : 7357420                                                   ||\n",
      "||    Successfully assigned reads : 4157877 (56.5%)                           ||\n",
      "||    Running time : 0.32 minutes                                             ||\n",
      "||                                                                            ||\n",
      "||                         Read assignment finished.                          ||\n",
      "||                                                                            ||\n",
      "|| Summary of counting results can be found in file \"./featureCount_output/E  ||\n",
      "|| RR899436Aligned.count.txt.summary\"                                         ||\n",
      "||                                                                            ||\n",
      "\\\\===================== http://subread.sourceforge.net/ ======================//\n",
      "\n",
      "\n",
      "        ==========     _____ _    _ ____  _____  ______          _____  \n",
      "        =====         / ____| |  | |  _ \\|  __ \\|  ____|   /\\   |  __ \\ \n",
      "          =====      | (___ | |  | | |_) | |__) | |__     /  \\  | |  | |\n",
      "            ====      \\___ \\| |  | |  _ <|  _  /|  __|   / /\\ \\ | |  | |\n",
      "              ====    ____) | |__| | |_) | | \\ \\| |____ / ____ \\| |__| |\n",
      "        ==========   |_____/ \\____/|____/|_|  \\_\\______/_/    \\_\\_____/\n",
      "\t  v1.6.0\n",
      "\n",
      "//========================== featureCounts setting ===========================\\\\\n",
      "||                                                                            ||\n",
      "||             Input files : 1 BAM file                                       ||\n",
      "||                           S ./star_output/ERR899437Aligned.out.bam         ||\n",
      "||                                                                            ||\n",
      "||             Output file : ./featureCount_output/ERR899437Aligned.count ... ||\n",
      "||                 Summary : ./featureCount_output/ERR899437Aligned.count ... ||\n",
      "||              Annotation : /fs/project/PAS1307/genomes/Mus_musculus/Ens ... ||\n",
      "||      Dir for temp files : ./featureCount_output                            ||\n",
      "||                                                                            ||\n",
      "||                 Threads : 24                                               ||\n",
      "||                   Level : meta-feature level                               ||\n",
      "||              Paired-end : no                                               ||\n",
      "||         Strand specific : no                                               ||\n",
      "||      Multimapping reads : not counted                                      ||\n",
      "|| Multi-overlapping reads : not counted                                      ||\n",
      "||   Min overlapping bases : 1                                                ||\n",
      "||                                                                            ||\n",
      "\\\\===================== http://subread.sourceforge.net/ ======================//\n",
      "\n",
      "//================================= Running ==================================\\\\\n",
      "||                                                                            ||\n",
      "|| Load annotation file /fs/project/PAS1307/genomes/Mus_musculus/Ensembl/ ... ||\n",
      "||    Features : 690647                                                       ||\n",
      "||    Meta-features : 46078                                                   ||\n",
      "||    Chromosomes/contigs : 73                                                ||\n",
      "||                                                                            ||\n",
      "|| Process BAM file ./star_output/ERR899437Aligned.out.bam...                 ||\n",
      "||    Single-end reads are included.                                          ||\n",
      "||    Assign reads to features...                                             ||\n",
      "||    Total reads : 8847290                                                   ||\n",
      "||    Successfully assigned reads : 4213290 (47.6%)                           ||\n",
      "||    Running time : 0.38 minutes                                             ||\n",
      "||                                                                            ||\n",
      "||                         Read assignment finished.                          ||\n",
      "||                                                                            ||\n",
      "|| Summary of counting results can be found in file \"./featureCount_output/E  ||\n",
      "|| RR899437Aligned.count.txt.summary\"                                         ||\n",
      "||                                                                            ||\n",
      "\\\\===================== http://subread.sourceforge.net/ ======================//\n",
      "\n",
      "\n",
      "        ==========     _____ _    _ ____  _____  ______          _____  \n",
      "        =====         / ____| |  | |  _ \\|  __ \\|  ____|   /\\   |  __ \\ \n",
      "          =====      | (___ | |  | | |_) | |__) | |__     /  \\  | |  | |\n",
      "            ====      \\___ \\| |  | |  _ <|  _  /|  __|   / /\\ \\ | |  | |\n",
      "              ====    ____) | |__| | |_) | | \\ \\| |____ / ____ \\| |__| |\n",
      "        ==========   |_____/ \\____/|____/|_|  \\_\\______/_/    \\_\\_____/\n",
      "\t  v1.6.0\n",
      "\n",
      "//========================== featureCounts setting ===========================\\\\\n",
      "||                                                                            ||\n",
      "||             Input files : 1 BAM file                                       ||\n",
      "||                           S ./star_output/ERR899438Aligned.out.bam         ||\n",
      "||                                                                            ||\n",
      "||             Output file : ./featureCount_output/ERR899438Aligned.count ... ||\n",
      "||                 Summary : ./featureCount_output/ERR899438Aligned.count ... ||\n",
      "||              Annotation : /fs/project/PAS1307/genomes/Mus_musculus/Ens ... ||\n",
      "||      Dir for temp files : ./featureCount_output                            ||\n",
      "||                                                                            ||\n",
      "||                 Threads : 24                                               ||\n",
      "||                   Level : meta-feature level                               ||\n",
      "||              Paired-end : no                                               ||\n",
      "||         Strand specific : no                                               ||\n",
      "||      Multimapping reads : not counted                                      ||\n",
      "|| Multi-overlapping reads : not counted                                      ||\n",
      "||   Min overlapping bases : 1                                                ||\n",
      "||                                                                            ||\n",
      "\\\\===================== http://subread.sourceforge.net/ ======================//\n",
      "\n",
      "//================================= Running ==================================\\\\\n",
      "||                                                                            ||\n",
      "|| Load annotation file /fs/project/PAS1307/genomes/Mus_musculus/Ensembl/ ... ||\n",
      "||    Features : 690647                                                       ||\n",
      "||    Meta-features : 46078                                                   ||\n",
      "||    Chromosomes/contigs : 73                                                ||\n",
      "||                                                                            ||\n",
      "|| Process BAM file ./star_output/ERR899438Aligned.out.bam...                 ||\n",
      "||    Single-end reads are included.                                          ||\n",
      "||    Assign reads to features...                                             ||\n",
      "||    Total reads : 9009449                                                   ||\n",
      "||    Successfully assigned reads : 4858518 (53.9%)                           ||\n",
      "||    Running time : 0.39 minutes                                             ||\n",
      "||                                                                            ||\n",
      "||                         Read assignment finished.                          ||\n",
      "||                                                                            ||\n",
      "|| Summary of counting results can be found in file \"./featureCount_output/E  ||\n",
      "|| RR899438Aligned.count.txt.summary\"                                         ||\n",
      "||                                                                            ||\n",
      "\\\\===================== http://subread.sourceforge.net/ ======================//\n",
      "\n",
      "\n",
      "ERROR: invalid parameter: './star_output/ERR899421Aligned.out.bam.out.bam'\n",
      "\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "\n",
    "for basename in $(find star_output/ -name '*Aligned.out.bam' -exec basename {} \\; | cut -f1 -d '.' ); do\n",
    "    suffix=\".out.bam\"\n",
    "    outname=\"$FCDIR/$basename.count.txt\"\n",
    "    bam=\"$STARDIR/$basename$suffix\"    \n",
    "    $BINDIR/featureCounts \\\n",
    "        -T $MAXCORES \\\n",
    "        -t exon \\\n",
    "        -g gene_id \\\n",
    "        -a  $GENEANNOTATIONDIR/genes.gtf \\\n",
    "        -o $outname \\\n",
    "        $bam \n",
    "done"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "%%bash\n",
    "\n",
    "bams=\"\"\n",
    "for basename in $(find star_output/ -name '*Aligned.out.bam' -exec basename {} \\; | cut -f1 -d '.' ); do\n",
    "    suffix=\".out.bam\"\n",
    "    outname=\"combined.counts.txt\"\n",
    "    bam=\"$STARDIR/$basename$suffix\"    \n",
    "    bams=\"$bams $bam\"\n",
    "done\n",
    "\n",
    "$BINDIR/featureCounts \\\n",
    "        -T $MAXCORES \\\n",
    "        -t exon \\\n",
    "        -g gene_id \\\n",
    "        -a  $GENEANNOTATIONDIR/genes.gtf \\\n",
    "        -o $outname \\\n",
    "        $bams"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "%%bash\n",
    "#multiqc of sequencing reads\n",
    "\n",
    "source activate tophat\n",
    "multiqc ."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "from IPython.display import FileLinks, display\n",
    "\n",
    "d = FileLinks(os.environ['WORKDIR'], included_suffixes=['.html'])\n",
    "e = FileLinks(os.path.join(os.environ['WORKDIR'], os.environ['FASTQCOUT']), included_suffixes=['.html'])\n",
    "f = FileLinks(os.path.join(os.environ['WORKDIR'], os.environ['TFASTQCOUT']), included_suffixes=['.html'])\n",
    "\n",
    "display(d)\n",
    "display(e)\n",
    "display(f)"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python [conda env:tophat]",
   "language": "python",
   "name": "conda-env-tophat-py"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 2
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython2",
   "version": "2.7.14"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
